---
title: "Noshay_et_al_Manuscript_2018_updated"
output:
  html_document:
    code_folding: hide
    fig_retina: 2
    highlight: pygments
    author: "Jaclyn_Noshay"
    number_sections: no
    theme: cosmo
    toc: yes
    toc_depth: 4
    toc_float: yes
---


# Final Annotation Files (20 December 2018)
https://docs.google.com/spreadsheets/d/1luUQvw9VjrGWY7qLa3qFIi3X4rmBPTzuBUGHixh6Szg/edit?ts=5c1bd754#gid=0

### B73
Full TE: /home/springer/sna/B73/B73.2018.12.19.allTE.corrected.gff3
Repeat Masker: /home/springer/sna/resources/B73_chr1-10.bed

Filtered TE: /home/springer/sna/B73/B73.structuralTEv2.2018.12.20.filteredTE.gff3
Filtered TE Disjoined: /home/springer/sna/B73/B73.structuralTEv2.2018.12.20.filteredTE.disjoined.gff3
Nested: /home/springer/sna/B73/B73_nested_filteredTE_20Dec18.txt

/home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3
/home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3
/home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.noctg.gff3


### Mo17
Full TE: /home/springer/sna/Mo17/Mo17.structuralTEv2.fulllength.2018-12-11.gff3

### W22
Full TE: /home/springer/sna/W22/W22.structuralTEv2.2018-09-12.allTE.gff3
Repeat Masker: /home/springer/sna/resources/W22_chr1-10.bed

Filtered TE: /home/springer/sna/W22/W22.structuralTEv2.10.08.2018.filteredTE.gff3
Filtered TE Disjoined: /home/springer/sna/W22/W22.structuralTEv2.10.08.2018.filteredTE.disjoined.gff3

/home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.LTR.gff3
/home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.TIR.gff3

### PH207
Full TE: /home/springer/sna/PH207/PH207.2018-10-15.allTE.gff3
Repeat Masker: /home/springer/sna/resources/PH207_chr1-10.bed

Filtered TE: /home/springer/sna/PH207/PH207.structuralTEv2.2018.10.30.filteredTE.gff3
Filtered TE Disjoined: /home/springer/sna/PH207/PH207.structuralTEv2.10.30.2018.filteredTE.disjoined.gff3




# Table S1: TEs and TE family counts 
```{r eval=F}
# Note insertions based on B72 to W22 or PH207 or Mo17 while W22 to B73 and PH207 to B73

B73 <- c(1297, 170, 23430, 132, 30623, "x", 140809, "x")
W22 <- c(1183, 339, 23312, 194, 156447, 1620, 137471, 4874)
PH207 <- c("x", "x", "x", "x", "x", "x", "x", "x")
Mo17 <- c("x", "x", "x", "x", "x", "x", "x", "x")
df <- rbind(B73, W22, PH207, Mo17)
colnames(df) <- c("TIR_fam_genome", "TIR_fam_non-nested", "LTR_fam_genome", "LTR_fam_non-nested", "TIR_genome", "TIR_insertion", "LTR_genome", "LTR_insertion")
df
```

# Table S2: WGBS datasets 
```{r eval=F}
B73_shoot <- c("sra", 1, 636294670, 0.65, 14.55, 0.95, 0.81, "Classify")
B73_root <- c("sra", 1, 230772157, 0.65, 7.10, 0.94, 0.79, "NA")
B73_leaf_S1 <- c("sra", 6, 928696652, 0.62, 4.93, 0.96, 0.76, "NA")
B73_leaf_S2 <- c("sra", 1, 138604020, 0.58, "NA", 0.87, 0.56, "NA")
B73_leaf_S3 <- c("sra", 1, 165993413, 0.67, "NA", 0.79, 0.47, "NA")
Mo17_leaf <- c("sra", 1, 141635754, 0.50, "NA", 0.78, 0.46, "Validate")
PH207_shoot <- c("sra", 1, 430997520, 0.47, 11.34, 0.83, 0.74, "Validate")
PH207_root <- c("sra", 1, 215582045, 0.48, 6.44, 0.84, 0.71, "NA")
PH207_leaf <- c("sra", 1, 381565504, 0.46, 15.22, 0.77, 0.46, "NA")
W22_leaf_S1 <- c("sra", 2, 333172392, 0.46, 8.33, 0.87, 0.52, "Validate")
W22_leaf_S2 <- c("sra", 1, 183670406, 0.46, 7.36, "NA", "NA", "NA")
df <- rbind(B73_shoot, B73_root, B73_leaf_S1, B73_leaf_S2, B73_leaf_S3, Mo17_leaf, PH207_shoot, PH207_root, PH207_leaf, W22_leaf_S1, W22_leaf_S2)
colnames(df) <- c("SRA", "Samples", "Reads", "Mapping_Rate", "Avg_Coverage", "Tiles_with_Data", "Tiles_with_Coverage", "Use_in_Paper")
df
```

# Figure S1: Prop of tiles with coverage across regions and genotypes
- The level of DNA methylation in B73_Shoot (left), PH207_Shoot (middle), and W22_leaf (right) was determined by WGBS.  Bar graphs are shown for the proportion of 100bp tiles with >2x coverage for all regions (pink), as well as 100bp tiles annotated as genes (green) and TEs (blue) for each genotype

- bar plot showing % coverage (> 2x) for all tiles, TE tiles, and gene tiles for each genotype
```{r eval=F}
cd /home/springer/nosha003/wgbs_schmitz/tiles_output

## Gene
bedtools intersect -b /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.32.gene.sort.gff3 -a /home/springer/nosha003/database/B73v4/B73v4_100bp_bins.gff -wo > /home/springer/nosha003/database/B73v4/B73v4_100bp_bins_overlap_gene_20Dec2018.bed

awk '{print $1"\t""PH207""\t""bin""\t"$2+1"\t"$3"\t"".""\t"".""\t"".""\t""ID=bin"$1"-"$2}' PH207_100bp_bins.txt | grep -v 'UNKNOWN' | grep -v 'Pt' | grep -v 'Mt' | sed 's/chr0//g'  | sed 's/chr//g' > PH207_100bp_bins.gff
awk '{print $1"\t""Mo17""\t""bin""\t"$2+1"\t"$3"\t"".""\t"".""\t"".""\t""ID=bin"$1"-"$2}' Mo17_100bp_bins.txt | grep -v 'M99' | sed 's/M0//g'  | sed 's/M//g' > Mo17_100bp_bins.gff
awk '{print $1"\t""W22""\t""bin""\t"$2"\t"$3"\t"".""\t"".""\t"".""\t""ID=bin"$1"-"$2}' W22_100bp_bins.txt | grep -v 'unmapped' | sed 's/chr0//g'  | sed 's/chr//g' > W22_100bp_bins.gff
sed 's/chr0//g' /home/maize/shared/databases/genomes/Zea_mays/PH207/ZmaysPH207_443_v1.1.gene.gff3 | sed 's/chr//g' > /home/maize/shared/databases/genomes/Zea_mays/PH207/ZmaysPH207_443_v1.1.gene_chr.gff3
sed 's/chr0//g' /home/maize/shared/databases/genomes/Zea_mays/W22/zea_maysw22_gene.gff | sed 's/chr//g' > /home/maize/shared/databases/genomes/Zea_mays/W22/zea_maysw22_gene_chr.gff
sed 's/M0//g' /home/springer/nosha003/database/Mo17/GCA_003185045.1_Zm-Mo17-REFERENCE-CAU-1.0_gene.gff | sed 's/M//g' > /home/springer/nosha003/database/Mo17/GCA_003185045.1_Zm-Mo17-REFERENCE-CAU-1.0_gene_chr.gff

bedtools intersect -b /home/maize/shared/databases/genomes/Zea_mays/PH207/ZmaysPH207_443_v1.1.gene_chr.gff3 -a /home/springer/nosha003/database/PH207/PH207_100bp_bins.gff -wo > /home/springer/nosha003/database/PH207/PH207_100bp_bins_overlap_gene_6Dec2018.bed
bedtools intersect -b /home/maize/shared/databases/genomes/Zea_mays/W22/zea_maysw22_gene_chr.gff -a /home/springer/nosha003/database/W22/W22_100bp_bins.gff -wo > /home/springer/nosha003/database/W22/W22_100bp_bins_overlap_gene_6Dec2018.bed
bedtools intersect -b /home/springer/nosha003/database/Mo17/GCA_003185045.1_Zm-Mo17-REFERENCE-CAU-1.0_gene_chr.gff -a /home/springer/nosha003/database/Mo17/Mo17_100bp_bins.gff -wo > /home/springer/nosha003/database/Mo17/Mo17_100bp_bins_overlap_gene_6Dec2018.bed

## TE
# calculate complete TE bins in genome
bedtools intersect -u -b /home/maize/shared/databases/genomes/Zea_mays/B73/B73.2018.09.03.allTE.gff3 -a /home/springer/nosha003/database/B73v4/B73v4_100bp_bins.gff > /home/springer/nosha003/database/B73v4/B73v4_100bp_bins_overlap_te_20Dec2018.bed

sed 's/chr0//g' /home/maize/shared/databases/genomes/Zea_mays/PH207/PH207.2018-10-15.allTE.gff3 | sed 's/chr//g' | grep -v 'scaffold' > /home/maize/shared/databases/genomes/Zea_mays/PH207/PH207.2018-10-15.allTE_chr.gff3
sed 's/chr0//g' /home/maize/shared/databases/genomes/Zea_mays/W22/W22.structuralTEv2.2018-09-12.allTE.gff3 | sed 's/chr//g' > /home/maize/shared/databases/genomes/Zea_mays/W22/W22.structuralTEv2.2018-09-12.allTE_chr.gff3

bedtools intersect -u -b /home/maize/shared/databases/genomes/Zea_mays/PH207/PH207.2018-10-15.allTE_chr.gff3 -a /home/springer/nosha003/database/PH207/PH207_100bp_bins.gff > /home/springer/nosha003/database/PH207/PH207_100bp_bins_overlap_te_6Dec2018.bed
bedtools intersect -u -b /home/maize/shared/databases/genomes/Zea_mays/W22/W22.structuralTEv2.2018-09-12.allTE_chr.gff3 -a /home/springer/nosha003/database/W22/W22_100bp_bins.gff > /home/springer/nosha003/database/W22/W22_100bp_bins_overlap_te_6Dec2018.bed
bedtools intersect -u -b /home/springer/sna/Mo17/Mo17.structuralTEv2.fulllength.2018-12-11.gff3 -a /home/springer/nosha003/database/Mo17/Mo17_100bp_bins.gff > /home/springer/nosha003/database/Mo17/Mo17_100bp_bins_overlap_te_6Dec2018.bed
```


## Mapped to corresponding genotypes
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

library(dplyr)
library(ggplot2)

#### B73 schmitz root data
setwd("/home/springer/nosha003/database/B73v4")
gene <- read.delim("B73v4_100bp_bins_overlap_gene_20Dec2018.bed", header=F, sep="\t")
te <- read.delim("B73v4_100bp_bins_overlap_te_20Dec2018.bed", header=F, sep="\t")
gene_bin <- gene[,c(1,4:5)]
te_bin <- te[,c(1,4:5)]
colnames(gene_bin) <- c("chr", "start", "end")
colnames(te_bin) <- c("chr", "start", "end")

nrow(gene_bin)
# 1645626
nrow(te_bin)
# 14774283

setwd("/home/springer/nosha003/wgbs_schmitz/tiles_output")
Broot <- read.delim("B73_V2_100bp_cov.txt", header=T, sep="\t")
nrow(Broot)
# 21063387
Broot_cov <- subset(Broot, Broot$CHH_cov > 2)
nrow(Broot_cov)
# 19322740

Broot_cov_gene <- inner_join(Broot_cov, gene_bin, by=c("chr", "start", "end"))
nrow(Broot_cov_gene)
# 1577436 - shoot
Broot_cov_te <- inner_join(Broot_cov, te_bin, by=c("chr", "start", "end"))
nrow(Broot_cov_te)
# 13444365 - shoot
te_gene_overlap <- inner_join(Broot_cov_gene, Broot_cov_te, by=c("chr", "start", "end"))
nrow(te_gene_overlap)
# 400152 - shoot

#### PH207 schmitz V2 data
setwd("/home/springer/nosha003/database/PH207")
gene <- read.delim("PH207_100bp_bins_overlap_gene_6Dec2018.bed", header=F, sep="\t")
te <- read.delim("PH207_100bp_bins_overlap_te_6Dec2018.bed", header=F, sep="\t")
gene_bin <- gene[,c(1,4:5)]
te_bin <- te[,c(1,4:5)]
colnames(gene_bin) <- c("chr", "start", "end")
colnames(te_bin) <- c("chr", "start", "end")

nrow(gene_bin)
# 2449554
nrow(te_bin)
# 12298235

# sed 's/chr0//' PH207_V2_100bp_cov.txt | grep -v 'scaffold' | grep -v 'chrUNKNOWN' | sed 's/chr10/10/g' > PH207_V2_100bp_cov_chr.txt
setwd("/home/springer/nosha003/wgbs_schmitz/PH207")
Proot <- read.delim("PH207_V2_100bp_cov_chr.txt", header=T, sep="\t", stringsAsFactors = F)
nrow(Proot)
# 16578578
Proot_cov <- subset(Proot, Proot$CHH_cov > 2)
nrow(Proot_cov)
# 16128174

Proot_cov$chr <- as.numeric(Proot_cov$chr)
Proot_cov_gene <- inner_join(Proot_cov, gene_bin, by=c("chr", "start", "end"))
nrow(Proot_cov_gene)
# 1759541
Proot_cov_te <- inner_join(Proot_cov, te_bin, by=c("chr", "start", "end"))
nrow(Proot_cov_te)
# 9878782
te_gene_overlap <- inner_join(Proot_cov_gene, Proot_cov_te, by=c("chr", "start", "end"))
nrow(te_gene_overlap)
# 985949


#### W22 merged swift data
setwd("/home/springer/nosha003/database/W22")
gene <- read.delim("W22_100bp_bins_overlap_gene_6Dec2018.bed", header=F, sep="\t")
te <- read.delim("W22_100bp_bins_overlap_te_6Dec2018.bed", header=F, sep="\t")
gene_bin <- gene[,c(1,4:5)]
te_bin <- te[,c(1,4:5)]
colnames(gene_bin) <- c("chr", "start", "end")
colnames(te_bin) <- c("chr", "start", "end")

nrow(gene_bin)
# 1800296
nrow(te_bin)
# 15067084

setwd("/home/springer/nosha003/wgbs/dec2017/w22/")
W22 <- read.delim("W22_toW22_merge_100bp_cov.txt", header=T, sep="\t", stringsAsFactors = F)
W22$chr <- sub('chr', '', W22$chr)
nrow(W22)
# 19714065
W22_cov <- subset(W22, W22$CHH_cov > 2 | W22$CHH_cov == NA)
nrow(W22_cov)
# 16373935

W22_cov$chr <- as.numeric(W22_cov$chr)
W22_cov_gene <- inner_join(W22_cov, gene_bin, by=c("chr", "start", "end"))
nrow(W22_cov_gene)
# 1587714
W22_cov_te <- inner_join(W22_cov, te_bin, by=c("chr", "start", "end"))
nrow(W22_cov_te)
# 10857969
te_gene_overlap <- inner_join(W22_cov_gene, W22_cov_te, by=c("chr", "start", "end"))
nrow(te_gene_overlap)
# 416513


#### Mo17 data
setwd("/home/springer/nosha003/database/Mo17")
gene <- read.delim("Mo17_100bp_bins_overlap_gene_6Dec2018.bed", header=F, sep="\t")
te <- read.delim("Mo17_100bp_bins_overlap_te_6Dec2018.bed", header=F, sep="\t")
gene_bin <- gene[,c(1,4:5)]
te_bin <- te[,c(1,4:5)]
colnames(gene_bin) <- c("chr", "start", "end")
colnames(te_bin) <- c("chr", "start", "end")

nrow(gene_bin)
# 1585525
nrow(te_bin)
# 14738151

# sed 's/M0//' Mo17_leaf_100bp_cov.txt | grep -v 'scaffold' | sed 's/M10/10/g' > Mo17_leaf_100bp_cov_chr.txt
setwd("/home/springer/nosha003/wgbs_schmitz/Mo17")
Proot <- read.delim("Mo17_leaf_100bp_cov_chr.txt", header=T, sep="\t")
nrow(Proot)
# 19615547
Proot_cov <- subset(Proot, Proot$CHH_cov > 2)
nrow(Proot_cov)
# 11642266

Proot_cov$chr <- as.numeric(Proot_cov$chr)
Proot_cov_gene <- inner_join(Proot_cov, gene_bin, by=c("chr", "start", "end"))
nrow(Proot_cov_gene)
# 821167
Proot_cov_te <- inner_join(Proot_cov, te_bin, by=c("chr", "start", "end"))
nrow(Proot_cov_te)
# 7606988
te_gene_overlap <- inner_join(Proot_cov_gene, Proot_cov_te, by=c("chr", "start", "end"))
nrow(te_gene_overlap)
# 217542


# table of values
library(reshape2)
library(RColorBrewer)
library(ggplot2)

## schmitz V2 data
b73 <- c(19322740/21063387, 1577436/1645626, (13444365-400152)/14774283)
w22 <- c(16373935/19811123, 1587714/1800296, (10857969-416513)/15067084)
ph207 <- c(16128174/16578578, 1759541/2449554, (9878782-985949)/12298235)
mo17 <- c(11642266/19615547, 821167/1645626, (7606988-217542)/14738151)
df <- rbind(b73, ph207, w22, mo17)
colnames(df) <- c("prop_tiles", "prop_gene_tiles", "prop_te_tiles")
df_melt <- melt(df)
colnames(df_melt) <- c("genotype", "tile_type", "prop")

setwd("/home/springer/nosha003/methylation_spreading")
pdf("mapping_coverage_barplot_23Nov2018.pdf")
ggplot(data=df_melt) + geom_bar(aes(x=tile_type, y=prop, fill=genotype), stat="identity") + facet_grid(. ~ genotype) + theme_classic() + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + scale_fill_brewer(palette="Greys")
dev.off()
pdf("mapping_coverage_barplot2_23Nov2018.pdf")
ggplot(data=df_melt) + geom_bar(aes(x=tile_type, y=prop, fill=tile_type), stat="identity") + facet_grid(. ~ genotype) + theme_classic() + theme(axis.text.x = element_text(angle = 90, hjust = 1)) + scale_fill_brewer(palette="Greys")
dev.off()

```



# Data
- The level of DNA methylation in B73 Root tissue was determined by WGBS.  Line plots were generated for CG (A), CHG (B), and CHH (C) methylation ratio over the 1kb flanking regions of annotated genes (red), Terminal Inverted Repeats (black), and Long Terminal Retrotransposons (blue) for B73 (solid lines) and W22 (dashed lines)

## bedtools
### B73
```{r eval=F}
#20Dec2018
#b73_V2_metaplot_updated.sh
#generate gene, tir, ltr based metaplot data b73 methylation data
	
#!/bin/bash
#PBS -l walltime=08:00:00,nodes=1:ppn=1,mem=10gb
#PBS -o /scratch.global/nosha003/e_o/methyl_o
#PBS -e /scratch.global/nosha003/e_o/methyl_e
#PBS -N methyl
#PBS -V
#PBS -r n

sed '1d' /home/springer/nosha003/methylation_B73v4/schmitz/B73_V2_100bp_cov.txt | cut -f 1-4 | sed 's/chr//g' | sort -k 1,1 -k 2,2n > /scratch.global/nosha003/methylation_spreading/b73_v2_CG.txt
sed '1d' /home/springer/nosha003/methylation_B73v4/schmitz/B73_V2_100bp_cov.txt | cut -f 1-3,9 | sed 's/chr//g' | sort -k 1,1 -k 2,2n > /scratch.global/nosha003/methylation_spreading/b73_v2_CHG.txt
sed '1d' /home/springer/nosha003/methylation_B73v4/schmitz/B73_V2_100bp_cov.txt | cut -f 1-3,14 | sed 's/chr//g' | sort -k 1,1 -k 2,2n > /scratch.global/nosha003/methylation_spreading/b73_v2_CHH.txt

cd /scratch.global/nosha003/methylation_spreading

# gene
module load bedtools
bedtools closest -D a -t first -a b73_v2_CG.txt -b /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.32.gene.sort2.gff3 > b73_v2_CG_gene.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i b73_v2_CG_gene.txt -o b73_v2_CG_gene_metaplot.txt
bedtools closest -D a -t first -a b73_v2_CHG.txt -b /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.32.gene.sort2.gff3 > b73_v2_CHG_gene.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i b73_v2_CHG_gene.txt -o b73_v2_CHG_gene_metaplot.txt
bedtools closest -D a -t first -a b73_v2_CHH.txt -b /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.32.gene.sort2.gff3 > b73_v2_CHH_gene.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i b73_v2_CHH_gene.txt -o b73_v2_CHH_gene_metaplot.txt

# tir
bedtools closest -D a -t first -a b73_v2_CG.txt -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 > b73_v2_CG_TIR.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i b73_v2_CG_TIR.txt -o b73_v2_CG_TIR_metaplot.txt
bedtools closest -D a -t first -a b73_v2_CHG.txt -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 > b73_v2_CHG_TIR.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i b73_v2_CHG_TIR.txt -o b73_v2_CHG_TIR_metaplot.txt
bedtools closest -D a -t first -a b73_v2_CHH.txt -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 > b73_v2_CHH_TIR.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i b73_v2_CHH_TIR.txt -o b73_v2_CHH_TIR_metaplot.txt

#ltr
bedtools closest -D a -t first -a b73_v2_CG.txt -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 > b73_v2_CG_LTR.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i b73_v2_CG_LTR.txt -o b73_v2_CG_LTR_metaplot.txt
bedtools closest -D a -t first -a b73_v2_CHG.txt -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 > b73_v2_CHG_LTR.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i b73_v2_CHG_LTR.txt -o b73_v2_CHG_LTR_metaplot.txt
bedtools closest -D a -t first -a b73_v2_CHH.txt -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 > b73_v2_CHH_LTR.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i b73_v2_CHH_LTR.txt -o b73_v2_CHH_LTR_metaplot.txt

# qsub /home/springer/nosha003/scripts/b73_V2_metaplot.sh
# qsub /home/springer/nosha003/scripts/b73_V2_metaplot_updated.sh
```

### W22
```{r eval=F}
#20Dec2018
#w22_swift_metaplot_updated
#generate gene based metaplot data for mnase data
	
#!/bin/bash
#PBS -l walltime=08:00:00,nodes=1:ppn=1,mem=10gb
#PBS -o /scratch.global/nosha003/e_o/methyl_o
#PBS -e /scratch.global/nosha003/e_o/methyl_e
#PBS -N methyl
#PBS -V
#PBS -r n

sed '1d' /home/springer/nosha003/wgbs/dec2017/bins/W22_merged_100bp_cov.txt | cut -f 1-4 | sed 's/chr//g' | sort -k 1,1 -k 2,2n > /scratch.global/nosha003/methylation_spreading/w22_v2_CG.txt
sed '1d' /home/springer/nosha003/wgbs/dec2017/bins/W22_merged_100bp_cov.txt | cut -f 1-3,9 | sed 's/chr//g' | sort -k 1,1 -k 2,2n > /scratch.global/nosha003/methylation_spreading/w22_v2_CHG.txt
sed '1d' /home/springer/nosha003/wgbs/dec2017/bins/W22_merged_100bp_cov.txt | cut -f 1-3,14 | sed 's/chr//g' | sort -k 1,1 -k 2,2n > /scratch.global/nosha003/methylation_spreading/w22_v2_CHH.txt

cd /scratch.global/nosha003/methylation_spreading

module load bedtools
bedtools closest -D a -t first -a w22_v2_CG.txt -b /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.32.gene.sort2.gff3 > w22_v2_CG_gene.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i w22_v2_CG_gene.txt -o w22_v2_CG_gene_metaplot.txt
bedtools closest -D a -t first -a w22_v2_CHG.txt -b /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.32.gene.sort2.gff3 > w22_v2_CHG_gene.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i w22_v2_CHG_gene.txt -o w22_v2_CHG_gene_metaplot.txt
bedtools closest -D a -t first -a w22_v2_CHH.txt -b /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.32.gene.sort2.gff3 > w22_v2_CHH_gene.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i w22_v2_CHH_gene.txt -o w22_v2_CHH_gene_metaplot.txt

bedtools closest -D a -t first -a w22_v2_CG.txt -b /home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.TIR.gff3 > w22_v2_CG_TIR.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i w22_v2_CG_TIR.txt -o w22_v2_CG_TIR_metaplot.txt
bedtools closest -D a -t first -a w22_v2_CHG.txt -b /home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.TIR.gff3 > w22_v2_CHG_TIR.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i w22_v2_CHG_TIR.txt -o w22_v2_CHG_TIR_metaplot.txt
bedtools closest -D a -t first -a w22_v2_CHH.txt -b /home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.TIR.gff3 > w22_v2_CHH_TIR.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i w22_v2_CHH_TIR.txt -o w22_v2_CHH_TIR_metaplot.txt

bedtools closest -D a -t first -a w22_v2_CG.txt -b /home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.LTR.gff3 > w22_v2_CG_LTR.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i w22_v2_CG_LTR.txt -o w22_v2_CG_LTR_metaplot.txt
bedtools closest -D a -t first -a w22_v2_CHG.txt -b /home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.LTR.gff3 > w22_v2_CHG_LTR.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i w22_v2_CHG_LTR.txt -o w22_v2_CHG_LTR_metaplot.txt
bedtools closest -D a -t first -a w22_v2_CHH.txt -b /home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.LTR.gff3 > w22_v2_CHH_LTR.txt
perl /home/springer/nosha003/W22/scripts/metaplot_data.pl -i w22_v2_CHH_LTR.txt -o w22_v2_CHH_LTR_metaplot.txt

# qsub /home/springer/nosha003/scripts/w22_swift_metaplot.sh
# qsub /home/springer/nosha003/scripts/w22_swift_metaplot_updated.sh
```


## closest TE to bin
- Each bin can only be assigned to one TE
- Individual and Family basis

Nested TE file: 

```{r, eval=F}
module load bedtools

grep -v 'ctg' /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 | sort -k 1,1 -k 4,4n > /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3
grep -v 'ctg' /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 | sort -k 1,1 -k 4,4n > /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3

# B73 
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_20Dec2018.txt
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_20Dec2018.txt

## families that have at least non-nested 20 members
# R
setwd("/home/springer/sna/B73")
nested <- read.delim("B73_nested_filteredTE_20Dec18.txt", header=T)
setwd("/home/springer/nosha003/database/B73v4/te/")
TIR <- read.delim("B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3", header=F, sep="\t")
TIR$id <- substr(TIR$V9, 4, 24)
LTR <- read.delim("B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3", header=F, sep="\t")
LTR$id <- substr(LTR$V9, 4, 24)

library(dplyr)
TIR_nonnested <- anti_join(TIR, nested, by = c("id" = "nested_te"))
LTR_nonnested <- anti_join(LTR, nested, by = c("id" = "nested_te"))
write.table(TIR_nonnested[,1:9], "B73.structuralTEv2.2018.12.20.filteredTE.TIRnonnested.gff3", quote=F, row.names=F, sep="\t")
write.table(LTR_nonnested[,1:9], "B73.structuralTEv2.2018.12.20.filteredTE.LTRnonnested.gff3", quote=F, row.names=F, sep="\t")


# unix
cut -f 9 /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.noctg.gff3  | sed 's/ID=//g' | cut -c1-8 | sort | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.FamCount.txt

cut -f 9 /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/B73_tir_fam_count_nonnested20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/B73_tir_fam_count_nonnested20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/B73_tir_fam_count20_nonnested_20Dec2018.txt 
cut -f 9 /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/B73_ltr_fam_count_nonnested_20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/B73_ltr_fam_count_nonnested_20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/B73_ltr_fam_count20_nonnested_20Dec2018.txt

cp /home/springer/nosha003/database/B73v4/te/B73*_fam_count20_nonnested_20Dec2018.txt /home/springer/nosha003/methylation_spreading/.


##### B73 tissues

# Root
## LTR
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_Root_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_Root_ltr_closestTEtoBin_20Dec2018.txt
cut -f 9 /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/B73_Root_ltr_fam_count_nonnested20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/B73_Root_ltr_fam_count_nonnested20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/B73_Root_ltr_fam_count20_nonnested_20Dec2018.txt 
## TIR
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_Root_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_Root_all_tir_closestTEtoBin_20Dec2018.txt
cut -f 9 /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/B73_Root_tir_fam_count_nonnested20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/B73_Root_tir_fam_count_nonnested20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/B73_Root_tir_fam_count20_nonnested_20Dec2018.txt 

# Leaf
## LTR
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/wgbs/dec2017/bins/B73_merged_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_Leaf_ltr_closestTEtoBin_20Dec2018.txt
cut -f 9 /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/B73_Leaf_ltr_fam_count_nonnested20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/B73_Leaf_ltr_fam_count_nonnested20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/B73_Leaf_ltr_fam_count20_nonnested_20Dec2018.txt 
## TIR
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/wgbs/dec2017/bins/B73_merged_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_Leaf_all_tir_closestTEtoBin_20Dec2018.txt
cut -f 9 /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/B73_Leaf_tir_fam_count_nonnested20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/B73_Leaf_tir_fam_count_nonnested20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/B73_Leaf_tir_fam_count20_nonnested_20Dec2018.txt 
```



## orientation
```{r eval=F}
library(tidyverse)
setwd("/home/springer/nosha003/methylation_spreading/")

# read in desired columns only
#TE_distance <- read_tsv("B73_ltr_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))
TE_distance <- read_tsv("B73_tir_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))
#TE_distance <- read_tsv("B73_Root_ltr_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))
#TE_distance <- read_tsv("B73_Leaf_ltr_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))

# calculate which TE to flip
correction_values <- TE_distance %>%
  mutate(flank = ifelse(distance < 0, "L", ifelse(distance > 0, "R", ifelse(distance == 0, "TE_body", "what_the")))) %>%
  group_by(TE, flank) %>%
  select(TE, CG, distance, flank) %>%
  summarise(mean_CG = mean(CG, na.rm = T)) %>%
  group_by(TE) %>%
  spread(key = flank, value = mean_CG) %>%
  mutate(correction = ifelse(L > R, 1, ifelse(R > L, -1, 0))) %>%
  select(TE, correction)

# join on flip values
TE_distance <- TE_distance %>%
  left_join(correction_values, by = "TE")  

# flip values
TE_distance <- TE_distance %>%
  mutate(distance_2 = distance * correction)

#write.table(TE_distance, "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)
write.table(TE_distance, "B73_tir_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)
#write.table(TE_distance, "B73_Root_ltr_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)
#write.table(TE_distance, "B73_Leaf_ltr_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)
```

```{r eval=F}
#20Dec2018
#te_mC_orientation.sh
#TIR and LTR orientation based on higher methylation trend on the left 1kb than the right
	
#!/bin/bash
#PBS -l walltime=24:00:00,nodes=1:ppn=6,mem=60gb
#PBS -o /scratch.global/nosha003/e_o/orientation_o
#PBS -e /scratch.global/nosha003/e_o/orientation_e
#PBS -N orientation
#PBS -V
#PBS -r n

cd /home/springer/nosha003/scripts/
module load R
R CMD BATCH te_mC_orientation.R
wait

# qsub /home/springer/nosha003/scripts/te_mC_orientation.sh
```


# Figure S2: Proportion of bins methylated / unmethylated
- Proportion of bins categorized as methylated (>40%), unmethylated (<5%) or intermediate(5-40%) across genomic regions.  A WGBS dataset for maize shoot was mapped to the B73v4 genome.  The level of DNA methylation in each sequence context (CG, CHG, CHH) was determined for each 100bp tile.  Regions of the genome were classified as TE, Exon, Intron, or Intergenic based on B73v4 annotations.  Each 100bp tile within these regions were classified into one of three groups.  mC is defined by a methylation level > 40%, C is defined by a methylation level < 5%, and intermediate is defined by methylation levels >= 5% and <= 40%.  The proportion of 100bp tiles classified as mC, C, or intermediate in each region was determined.  

- barplots of genome mC distribution (methylated, unmethylated, and intermediate)
```{r eval=F}
cd /home/springer/nosha003/wgbs_schmitz/tiles_output

sed 1d /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_cov.txt | sort -k 1,1 -k 2,2n > /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp.tab

grep -v 'CDS' /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.37.sort.gff3 | grep -v 'RNase_MRP_RNA' | grep -v 'SRP_RNA' | grep -v 'chromosome' | grep -v 'contig' | grep -v 'five_prime_UTR' | grep -v 'lnc_RNA' | grep -v 'mRNA' | grep -v 'miRNA' | grep -v 'ncRNA' | grep -v 'ncRNA_gene' | grep -v 'pre_miRNA' | grep -v 'rRNA' | grep -v 'snRNA' | grep -v 'snoRNA' | grep -v 'tRNA' | grep -v 'three_prime_UTR' > /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.37.sort.gene.exon.gff3

bedtools closest -a B73_V2_100bp.tab -b /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.37.sort.gene.exon.gff3 -d > B73_V2_100bp_geneIntersect.tab
bedtools closest -a B73_V2_100bp_geneIntersect.tab -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.noctg.gff3 -d > B73_V2_100bp_geneTEIntersect_20Dec2018.tab
```

```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G
# module load R/3.3.1
# R

library(tidyverse)
library(ggplot2)
library(reshape2)

setwd("/home/springer/nosha003/wgbs_schmitz/tiles_output")
closest <- read.delim("B73_V2_100bp_geneTEIntersect_20Dec2018.tab", header=F, sep="\t")

# V28 = closest gene gff3 
# V21 = gene feature (chr, gene, exon, mRNA, 5'UTR, 3'UTR, etc.)
# V38 = closest TE
# V37 = TE id

TE <- subset(closest, closest$V38 == 0)
nonTE <- subset(closest, closest$V38 != 0)
TE_gene <- subset(TE, TE$V21 == 'gene')
TE_gene2 <- subset(TE_gene, TE_gene$V28 == 0)
TE_nongene <- anti_join(TE, TE_gene2, by=c("V1", "V2", "V3"))

nonTE_geneGFF <- subset(nonTE, nonTE$V28 == 0)
nonTE_gene <- subset(nonTE_geneGFF, nonTE_geneGFF$V21 == 'gene')
nonTE_exon <- subset(nonTE_geneGFF, nonTE_geneGFF$V21 == 'exon')
nonTE_intron <- anti_join(nonTE_gene, nonTE_exon, by=c("V1", "V2", "V3"))

intergenic <- anti_join(nonTE, nonTE_gene, by=c("V1", "V2", "V3"))

TE_gene_mC <- TE_gene2[,c(1:4,9,14)]
TE_nongene_mC <- TE_nongene[,c(1:4,9,14)]
nonTE_intron_mC <- nonTE_intron[,c(1:4,9,14)]
nonTE_exon_mC <- nonTE_exon[,c(1:4,9,14)]
intergenic_mC <- intergenic[,c(1:4,9,14)]

TE_gene_mC["type"] <- "TE_gene"
TE_nongene_mC["type"] <- "TE_nongene"
nonTE_intron_mC["type"] <- "nonTE_intron"
nonTE_exon_mC["type"] <- "nonTE_exon"
intergenic_mC["type"] <- "intergenic"

mC <- rbind(TE_gene_mC, TE_nongene_mC, nonTE_intron_mC, nonTE_exon_mC, intergenic_mC)
colnames(mC) <- c("chr", "start", "end", "CG", "CHG", "CHH", "type")
write.table(mC, "genome_mC_distribution_20Dec2018.txt", quote=F, row.names=F)
```

## Stacked Barplot
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

setwd("/home/springer/nosha003/wgbs_schmitz/tiles_output")
mC <- read.delim("genome_mC_distribution_20Dec2018.txt", header=T, sep=" ")

library(reshape2)
library(ggplot2)

mC_melt <- melt(mC[,c(4:7)])
colnames(mC_melt) <- c("type", "context", "mC_ratio")
mC_melt_na <- na.omit(mC_melt)

library(dplyr)
mC_region <- mC_melt_na %>% group_by(context, type) %>% summarise(region_count = n()) 
mC_count <- mC_melt_na %>% group_by(context, type) %>% mutate(mC_state = ifelse(context == "CG", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "CHG", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "CHH", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(context, type, mC_state) 
mC_count_na <- na.omit(mC_count)
mC_count_na2 <- mC_count_na %>% summarise(count = n())
mC_region_na <- na.omit(mC_region)

mC_state <- inner_join(mC_region_na, mC_count_na2, by=c("context", "type"))
mC_prop <- mC_state %>% mutate(prop = count/region_count)

#mC_prop %>% print(n = Inf)

library(ggplot2)
library(RColorBrewer)
setwd("/home/springer/nosha003/methylation_spreading")
mC_prop_nogene <- subset(mC_prop, mC_prop$type != "TE_gene")
mC_prop_na <- na.omit(mC_prop_nogene)
mC_prop_na$type <- factor(mC_prop_na$type, levels = c("nonTE_exon","nonTE_intron","intergenic", "TE_nongene"))
mC_prop_cgchg <- subset(mC_prop_na, mC_prop_na$context != "CHH")

pdf("genome_mC_barplot_20Dec2018.pdf")
ggplot(data=mC_prop_cgchg) + geom_bar(aes(x=type, y=prop, fill=factor(mC_state, levels=c("mC", "intermediate", "C"))), stat="identity") + facet_grid(. ~ context) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()
```


## Histogram / Density plot of methylation across genome
** Don't need to re-run **
- B73, W22, PH207, and Mo17

```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=1,mem=40G

setwd("/home/springer/nosha003/wgbs_schmitz/tiles_output")
b73 <- read.delim("B73_V2_100bp_ratio.txt", header=F, sep="\t")
setwd("/home/springer/nosha003/wgbs_schmitz/PH207")
ph207 <- read.delim("PH207_100bp_ratio.txt", header=F, sep="\t")
setwd("/home/springer/nosha003/wgbs_schmitz/Mo17")
mo17 <- read.delim("Mo17_100bp_ratio.txt", header=F, sep="\t")
setwd("/home/springer/nosha003/wgbs/dec2017/w22")
w22 <- read.delim("W22_100bp_ratio.txt", header=F, sep="\t")

library(reshape2)
b_melt <- melt(b73, id=c("V1", "V2", "V3"))
b_melt$type <- "B73"
b_cgchg <- subset(b_melt, b_melt$variable != "V6")
p_melt <- melt(ph207, id=c("V1", "V2", "V3"))
p_melt$type <- "PH207"
p_cgchg <- subset(p_melt, p_melt$variable != "V6")
w_melt <- melt(w22, id=c("V1", "V2", "V3"))
w_melt$type <- "W22"
w_cgchg <- subset(w_melt, w_melt$variable != "V6")
m_melt <- melt(mo17, id=c("V1", "V2", "V3"))
m_melt$type <- "Mo17"
m_cgchg <- subset(m_melt, m_melt$variable != "V6")

cg_chg <- rbind(b_cgchg, p_cgchg, w_cgchg, m_cgchg)

setwd("/scratch.global/nosha003/")
write.table(cg_chg, "bpwm_cg_chg_bins.txt", quote=F, row.names=F, sep="\t")

library(ggplot2)
pdf("b73_cg_histogram.pdf")
ggplot(b73, aes(V4)) + geom_histogram(binwidth = 0.1) + theme_classic() 
dev.off()
pdf("b73_cgchg_density.pdf")
ggplot(b_cgchg) + geom_density(aes(x=value, color=variable), alpha = 1) + theme_classic() 
dev.off()

pdf("bpwm_cgchg_density.pdf")
ggplot(cg_chg) + geom_density(aes(x=value, color=variable), alpha=1) + theme_classic() + facet_grid(type ~ variable)
dev.off()
pdf("bpwm_cgchg_histogram.pdf")
ggplot(cg_chg) + geom_histogram(aes(x=value), alpha=1) + theme_classic() + facet_grid(type ~ variable)
dev.off()

library(RColorBrewer)
pdf("bpwm_cgchg_histogram_box.pdf")
ggplot(cg_chg) + geom_histogram(aes(x=value), alpha=1) + theme_classic() + facet_grid(type ~ variable) + geom_rect(xmin = 0, xmax = 0.2, fill = "lightslateblue", ymin = -Inf, ymax = Inf) + geom_rect(xmin = 0.2, xmax = 0.4, fill = "sienna1", ymin = -Inf, ymax = Inf) + geom_rect(xmin = 0.4, xmax = 1.0, fill = "mediumseagreen", ymin = -Inf, ymax = Inf)
dev.off()
```



# Figure 1: TE schematic
- A) A region on chromosome 1 of the B73v4 genome from 225,749,884bp to 225,830,165bp is displayed as a schematic of genes (rectangles) and TEs (triangles).   All TEs are labeled with their TE family name and size with red text indicating small families (< 20 members) and blue text indicating large families (>= 20 members).  Nested (within another TE) and non-nested elements are shown in orange and grey respectively.  Flanking regions are identified by color based on their ability to be assessed (blue) or lack of incorporation due to confounding TE regions (red).  B) A table listing the number of TE families / elements for TIRs and LTRs in the B73 (left) and W22 (right) genomes included in downstream analyses based on the criteria of a non-nested TE with a large family.  

- show nested and non-nested TEs with large and small copy number to indicate which TEs can be assessed in these analyses

# Figure 2: Methylation heatmap and CG/CHG metaplots surrounding TEs
- 100bp bins were assigned to a single annotated TE in the B73v4 genome.  For each TE, the CG and CHG methylation data was compiled within 1kb of the TE (excluding all bins that overlapped the annotated TE coordinates). TE families were identified and chosen for those that included >= 20 non-nested annotated members. All TEs belonging to one of these TE families were oriented with higher methylation on the left. All TEs belonging to the same TE family were averaged for the flanking bins and plotted as a single row in the heatmap. Order of the families listed was determined by k-means clustering into 3 distinct groups for LTR (A) and TIR (B) annotated as cluster A (blue), B (green) and C (pink) on the left of each heatmap.  Metaplots of the levels of CG methylation (left) and CHG methylation (right) within LTR (C) and TIR (D) elements and the 1kb flanking regions for cluster A (blue), B (green) and C (pink).  Patterns observed in B73 and W22 are shown in solid and dashed lines respectively.

## Heatmap
- orientation of bins (upstream or dowstream of TE) based on the highest average methylation being on the left side
- Direction based on 5’ or 3’ having higher methylation… Force average higher on left side of TE

```{r eval=F}
#3Jan2019
#heatmaps.sh
#make tir and ltr clustered heatmaps, intermal mC boxplot and CG/CHG metaplots
	
#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/heatmap_o
#PBS -e /scratch.global/nosha003/e_o/heatmap_e
#PBS -N metaplot
#PBS -V
#PBS -r n

cd /home/springer/nosha003/scripts/
module load R
R CMD BATCH ltr_heatmap.R
R CMD BATCH ltr_boxplot.R
R CMD BATCH tir_heatmap.R
R CMD BATCH tir_boxplot.R
R CMD BATCH cgchg_metaplot.R
wait

# qsub /home/springer/nosha003/scripts/heatmaps.sh
```

### LTR
- create heatmap with corrected orientation 
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=1,mem=40G

library(fields)
library(tidyverse)
library(dplyr)
setwd("/home/springer/nosha003/methylation_spreading/")
b_ltr_fam <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", sep="\t", header=F)
b_ltr <- read.delim("B73_ltr_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
#b_ltr <- read.delim("B73_Root_ltr_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
#b_ltr <- read.delim("B73_Leaf_ltr_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)

b_ltr$TEfam <- substr(b_ltr$TE, 4, 11)
b_ltr2 <- subset(b_ltr, b_ltr$TEfam %in% b_ltr_fam$V2)
write.table(b_ltr2[,],"B73_ltr_cluster_heatmap_TEs_20Dec2018.txt", row.names=F, quote=F, sep="\t")

dist = subset(b_ltr2, b_ltr2$distance_2 > -2000 & b_ltr2$distance_2 <= 2000)
b_ltr <- unique(b_ltr2$TEfam)

b_ltr <- unique(b_ltr2$TEfam)
output <- data.frame(matrix(NA, nrow=length(b_ltr), ncol=40))
rownames(output) <- b_ltr
colnames(output) <- seq(-1950,1950, 100)
for(i in 1:length(b_ltr)){
	data.sub=subset(dist, dist$TEfam == b_ltr[i])
	stats.test=stats.bin(data.sub$distance_2,data.sub$CG,N=40,breaks=seq(-2000,2000, 100))
	output[i,] = stats.test$stats["mean",]
}
write.table(output, "b_ltr_output_cg_meandirection_20Dec.txt", quote=FALSE, sep="\t")
p.1 <- output
for(i in 1:length(b_ltr)){
	data.sub=subset(dist, dist$TEfam == b_ltr[i])
	stats.test=stats.bin(data.sub$distance_2,data.sub$CHG,N=40,breaks=seq(-2000,2000, 100))
	output[i,] = stats.test$stats["mean",]
}
p.2 <- output
for(i in 1:length(b_ltr)){
	data.sub=subset(dist, dist$TEfam == b_ltr[i])
	stats.test=stats.bin(data.sub$distance_2,data.sub$CHH,N=40,breaks=seq(-2000,2000, 100))
	output[i,] = stats.test$stats["mean",]
}
p.3 <- output

b_ltr_cgchg <- cbind(p.1, p.2)

# clustering
b_ltr_cgchg2 <- na.omit(b_ltr_cgchg)
cluster <- (kmeans(b_ltr_cgchg2[,c(1:19,21:59,61:80)], 3))
mydata <- data.frame(b_ltr_cgchg2, cluster$cluster)
write.table(mydata, "b_ltr_meandirection_kmeans3_20Dec2018.txt", sep="\t", quote=F)

cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster$cluster.cluster[cluster$cluster.cluster == 1] <- "B"
cluster$cluster.cluster[cluster$cluster.cluster == 2] <- "C"
cluster$cluster.cluster[cluster$cluster.cluster == 3] <- "A"
mydata <- cluster
write.table(mydata, "b_ltr_meandirection_kmeans3_20Dec2018.txt", sep="\t", quote=F)

mydata <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")

## Heatmap
library(pheatmap)
library(viridis)

newdata <- mydata[order(mydata[,81]),]
annotation <- data.frame(paste("class", newdata[,81]))
colnames(annotation) <- "class"
rownames(annotation) <- rownames(newdata)

pdf("b_ltr_nonNested_class_heatmap_meandirection_kmeans3_cluster_20Dec2018.pdf")
pheatmap(newdata[,1:80],cluster_cols=F,cluster_rows = F, color=inferno(100), breaks=seq(0,1,length.out=100), show_rownames=TRUE, show_colnames = TRUE, border_color = FALSE, fontsize = 4, annotation_row = annotation)
dev.off()
png("b_ltr_nonNested_class_heatmap_meandirection_kmeans3_cluster_20Dec2018.png")
pheatmap(newdata[,1:80],cluster_cols=F,cluster_rows = F, color=inferno(100), breaks=seq(0,1,length.out=100), show_rownames=TRUE, show_colnames = TRUE, border_color = FALSE, fontsize = 4, annotation_row = annotation)
dev.off()
#pdf("b_Root_ltr_nonNested_class_heatmap_meandirection_kmeans3_cluster_20Dec2018.pdf")
#pheatmap(newdata[,1:80],cluster_cols=F,cluster_rows = F, color=inferno(100), breaks=seq(0,1,length.out=100), show_rownames=TRUE, show_colnames = TRUE, border_color = FALSE, fontsize = 4, annotation_row = annotation)
#dev.off()
#pdf("b_Leaf_ltr_nonNested_class_heatmap_meandirection_kmeans3_cluster_20Dec2018.pdf")
#pheatmap(newdata[,1:80],cluster_cols=F,cluster_rows = F, color=inferno(100), breaks=seq(0,1,length.out=100), show_rownames=TRUE, show_colnames = TRUE, border_color = FALSE, fontsize = 4, annotation_row = annotation)
#dev.off()
```

##### internal TE mC boxplot
```{r eval=F}
setwd("/home/springer/nosha003/methylation_spreading/")
df <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
df_cg <- df[,c(20,81)]
rownames(df_cg) <- row.names(df)
colnames(df_cg) <- c("cg", "cluster")

group.colors <- c(A = "cornflowerblue", B = "springgreen3", C = "lightcoral")

library(ggplot2)
pdf("TEfam_internalCG_boxplot_LTR.pdf")
ggplot(df_cg) + geom_boxplot(aes(x=cluster, y=cg, fill=cluster)) + theme_classic() + scale_color_manual(values=group.colors) 
dev.off()

#########

module load bedtools
bedtools intersect -wo -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_cov.tab -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.noctg.gff3 > /home/springer/nosha003/methylation_spreading/b_bins_overlappingTE.txt

# module load R
setwd("/home/springer/nosha003/methylation_spreading/")
te_bins <- read.delim("b_bins_overlappingTE.txt", header=F, sep="\t")
te_bins$fam <- substr(te_bins$V27, 4, 11)
te_bins$TE <- substr(te_bins$V27, 4, 24)
df <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
df$fam <- rownames(df)
df_fam <- df[,c(81:82)]

library(dplyr)
te_bins_df <- subset(te_bins, te_bins$fam %in% df$fam)
te_bins_full <- subset(te_bins_df, te_bins_df$V28 == 99)
te_bins_cluster <- left_join(te_bins_full, df_fam, by="fam")
te_bins_cg <- te_bins_cluster %>% group_by(TE, cluster.cluster) %>% mutate(CG = mean(V4, na.rm = T))
cg_cluster <- te_bins_cg[,c(31:32)]
colnames(cg_cluster) <- c("cluster", "CG")
te_bins_chg <- te_bins_cluster %>% group_by(TE, cluster.cluster) %>% mutate(CG = mean(V9, na.rm = T))
chg_cluster <- te_bins_chg[,c(31:32)]
colnames(chg_cluster) <- c("cluster", "CHG")

cg_cluster2 <- cg_cluster %>% mutate(cluster = ifelse(cluster == "A", "H", ifelse(cluster == "B", "M", ifelse(cluster == "C", "L", "NA"))))
chg_cluster2 <- chg_cluster %>% mutate(cluster = ifelse(cluster == "A", "H", ifelse(cluster == "B", "M", ifelse(cluster == "C", "L", "NA"))))

#group.colors <- c(A = "cornflowerblue", B = "springgreen3", C = "lightcoral")
group.colors <- c("#3972c9ff", "#a43434ff", "#e7a275ff")

library(ggplot2)
pdf("TEfam_internalCG_fullIntersect_boxplot_LTR_20Dec.pdf")
#png("TEfam_internalCG_fullIntersect_boxplot_LTR_20Dec.png")
ggplot(cg_cluster2) + geom_boxplot(aes(x=factor(cluster, levels=c("H", "M", "L")), y=CG, fill=cluster)) + ylim(0,1) + scale_fill_manual(values=group.colors) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank()) 
dev.off()
pdf("TEfam_internalCHG_fullIntersect_boxplot_LTR_20Dec.pdf")
#png("TEfam_internalCHG_fullIntersect_boxplot_LTR_20Dec.png")
ggplot(chg_cluster2) + geom_boxplot(aes(x=factor(cluster, levels=c("H", "M", "L")), y=CHG, fill=cluster)) + ylim(0,1) + scale_fill_manual(values=group.colors) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank()) 
dev.off()
```


### TIR
- create heatmap with corrected orientation 
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=1,mem=40G

library(fields)
library(tidyverse)
library(dplyr)

setwd("/home/springer/nosha003/methylation_spreading/")
b_tir <- read.delim("B73_tir_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
#b_tir <- read.delim("B73_Root_tir_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
#b_tir <- read.delim("B73_Leaf_tir_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
b_tir_fam <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018.txt", sep="\t", header=F)

b_tir$TEfam <- substr(b_tir$TE, 4, 11)
b_tir$TE <- substr(b_tir$TE, 4, 24)
b_tir2 <- subset(b_tir, b_tir$TEfam %in% b_tir_fam$V2)
write.table(b_tir2,"B73_tir_cluster_heatmap_TEs_20Dec2018.txt", row.names=F, quote=F, sep="\t")

dist = subset(b_tir2, b_tir2$distance_2 > -2000 & b_tir2$distance_2 <= 2000)
b_tir_elements <- unique(b_tir2$TE)

dist_na <- na.omit(dist)
for(i in 1:length(b_tir_elements)){
  data.sub=subset(dist_na, dist_na$TE == b_tir_elements[i])
  Lmean <- mean(data.sub$cg[data.sub$V16<0], na.rm=T)
  Rmean <- mean(data.sub$cg[data.sub$V16>0], na.rm=T)
  if ((!is.na(Lmean < Rmean)) & (Lmean < Rmean)) {
    dist_na[(dist_na$TE == b_tir_elements[i]), "meandist"] <- dist_na[(dist_na$TE == b_tir_elements[i]), "distance_2"] * -1
  }
  else if ((!is.na(Lmean > Rmean)) & (Lmean > Rmean)) {
    dist_na[(dist_na$TE == b_tir_elements[i]), "meandist"] <- dist_na[(dist_na$TE == b_tir_elements[i]), "distance_2"]
  }
}
dist <- dist_na

b_tir <- unique(b_tir2$TEfam)
output <- data.frame(matrix(NA, nrow=length(b_tir), ncol=40))
rownames(output) <- b_tir
colnames(output) <- seq(-1950,1950, 100)
for(i in 1:length(b_tir)){
	data.sub=subset(dist, dist$TEfam == b_tir[i])
	stats.test=stats.bin(data.sub$distance_2,data.sub$CG,N=40,breaks=seq(-2000,2000, 100))
	output[i,] = stats.test$stats["mean",]
}
write.table(output, "b_tir_output_cg_meandirection_20Dec.txt", quote=FALSE, sep="\t")
p.1 <- output
for(i in 1:length(b_tir)){
	data.sub=subset(dist, dist$TEfam == b_tir[i])
	stats.test=stats.bin(data.sub$distance_2,data.sub$CHG,N=40,breaks=seq(-2000,2000, 100))
	output[i,] = stats.test$stats["mean",]
}
p.2 <- output
for(i in 1:length(b_tir)){
	data.sub=subset(dist, dist$TEfam == b_tir[i])
	stats.test=stats.bin(data.sub$distance_2,data.sub$CHH,N=40,breaks=seq(-2000,2000, 100))
	output[i,] = stats.test$stats["mean",]
}
p.3 <- output

b_tir_cgchg <- cbind(p.1, p.2)

# clustering
b_tir_cgchg2 <- na.omit(b_tir_cgchg)
cluster <- (kmeans(b_tir_cgchg2[,c(1:19,21:59,61:80)], 3))
#mydata <- data.frame(b_tir_cgchg2, cluster$cluster)
#write.table(mydata, "b_tir_meandirection_kmeans3_20Dec2018.txt", sep="\t", quote=F)

# cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
# cluster$cluster.cluster[cluster$cluster.cluster == 1] <- "A"
# cluster$cluster.cluster[cluster$cluster.cluster == 2] <- "C"
# cluster$cluster.cluster[cluster$cluster.cluster == 3] <- "B"
# mydata <- cluster
# write.table(mydata, "b_tir_meandirection_kmeans3_20Dec2018.txt", sep="\t", quote=F)

mydata <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")

## Heatmap
library(pheatmap)
library(viridis)
library(RColorBrewer)

newdata <- mydata[order(mydata[,81]),]
annotation <- data.frame(paste("class", newdata[,81]))
colnames(annotation) <- "class"
rownames(annotation) <- rownames(newdata)

group.colors <- c(A = "cornflowerblue", B = "springgreen3", C = "lightcoral")

pdf("b_tir_class_heatmap_meandirection_kmeans3_cluster_20Dec2018.pdf")
pheatmap(newdata[,1:80],cluster_cols=F,cluster_rows = F, color=inferno(100), breaks=seq(0,1,length.out=100), show_rownames=TRUE, show_colnames = FALSE, border_color = FALSE, fontsize = 5, annotation_row = annotation)
dev.off()
png("b_tir_class_heatmap_meandirection_kmeans3_cluster_20Dec2018.png")
pheatmap(newdata[,1:80],cluster_cols=F,cluster_rows = F, color=inferno(100), breaks=seq(0,1,length.out=100), show_rownames=TRUE, show_colnames = FALSE, border_color = FALSE, fontsize = 5, annotation_row = annotation)
dev.off()
#pdf("b_Root_tir_class_heatmap_meandirection_kmeans3_cluster_20Dec2018.pdf")
#pheatmap(newdata[,1:80],cluster_cols=F,cluster_rows = F, color=inferno(100), breaks=seq(0,1,length.out=100), show_rownames=TRUE, show_colnames = FALSE, border_color = FALSE, fontsize = 5, annotation_row = annotation)
#dev.off()
#pdf("b_Leaf_tir_class_heatmap_meandirection_kmeans3_cluster_20Dec2018.pdf")
#pheatmap(newdata[,1:80],cluster_cols=F,cluster_rows = F, color=inferno(100), breaks=seq(0,1,length.out=100), show_rownames=TRUE, show_colnames = FALSE, border_color = FALSE, fontsize = 5, annotation_row = annotation)
#dev.off()

```

#### internal TE mC boxplot
```{r eval=F}
setwd("/home/springer/nosha003/methylation_spreading/")
df <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
df_cg <- df[,c(20,81)]
rownames(df_cg) <- row.names(df)
colnames(df_cg) <- c("cg", "cluster")

group.colors <- c(A = "cornflowerblue", B = "springgreen3", C = "lightcoral")

library(ggplot2)
pdf("TEfam_internalCG_boxplot_TIR.pdf")
ggplot(df_cg) + geom_boxplot(aes(x=cluster, y=cg, fill=cluster)) + theme_classic() + scale_color_manual(values=group.colors) 
dev.off()

#########

module load bedtools
bedtools intersect -wo -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_cov.tab -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.noctg.gff3 > /home/springer/nosha003/methylation_spreading/b_bins_overlappingTE.txt

# module load R
setwd("/home/springer/nosha003/methylation_spreading/")
te_bins <- read.delim("b_bins_overlappingTE.txt", header=F, sep="\t")
te_bins$fam <- substr(te_bins$V27, 4, 11)
te_bins$TE <- substr(te_bins$V27, 4, 24)
df <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
df$fam <- rownames(df)
df_fam <- df[,c(81:82)]

library(dplyr)
te_bins_df <- subset(te_bins, te_bins$fam %in% df$fam)
te_bins_full <- subset(te_bins_df, te_bins_df$V28 == 99)
te_bins_cluster <- left_join(te_bins_full, df_fam, by="fam")
te_bins_cg <- te_bins_cluster %>% group_by(TE, cluster.cluster) %>% mutate(CG = mean(V4, na.rm = T))
cg_cluster <- te_bins_cg[,c(31:32)]
colnames(cg_cluster) <- c("cluster", "CG")
te_bins_chg <- te_bins_cluster %>% group_by(TE, cluster.cluster) %>% mutate(CG = mean(V9, na.rm = T))
chg_cluster <- te_bins_chg[,c(31:32)]
colnames(chg_cluster) <- c("cluster", "CHG")

cg_cluster2 <- cg_cluster %>% mutate(cluster = ifelse(cluster == "A", "H", ifelse(cluster == "B", "M", ifelse(cluster == "C", "L", "NA"))))
chg_cluster2 <- chg_cluster %>% mutate(cluster = ifelse(cluster == "A", "H", ifelse(cluster == "B", "M", ifelse(cluster == "C", "L", "NA"))))

#group.colors <- c(A = "cornflowerblue", B = "springgreen3", C = "lightcoral")
group.colors <- c("#3972c9ff", "#a43434ff", "#e7a275ff")

library(ggplot2)
pdf("TEfam_internalCG_fullIntersect_boxplot_TIR_20Dec.pdf")
ggplot(cg_cluster2) + geom_boxplot(aes(x=factor(cluster, levels=c("H", "M", "L")), y=CG, fill=cluster)) + ylim(0,1) + scale_fill_manual(values=group.colors) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank()) 
dev.off()
pdf("TEfam_internalCHG_fullIntersect_boxplot_TIR_20Dec.pdf")
ggplot(chg_cluster2) + geom_boxplot(aes(x=factor(cluster, levels=c("H", "M", "L")), y=CHG, fill=cluster)) + ylim(0,1) + scale_fill_manual(values=group.colors) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank()) 
dev.off()
```

## genome-wide metaplot
### CG/CHG with B73 20+ non-nested
- orientated by methylation levels (like heatmaps)
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

setwd("/home/springer/nosha003/database/B73v4/te")
#b_tir_fam <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018.txt", sep="\t", header=F)
#b_ltr_fam <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", sep="\t", header=F)
b_tir_fam <- read.delim("B73.structuralTEv2.2018.12.20.filteredTE.TIRnonnested.gff3", sep="\t", header=F)
b_ltr_fam <- read.delim("B73.structuralTEv2.2018.12.20.filteredTE.LTRnonnested.gff3", sep="\t", header=T)
b_tir_fam$TE <- substr(b_tir_fam$V9, 4, 24)
b_ltr_fam$TE <- substr(b_ltr_fam$V9, 4, 24)

#b_ltr_fam <- read.delim("B73_ltr_fam_count20_nonnested_16Aug2018.txt", header=F, sep="\t")
#b_tir_fam <- read.delim("B73_tir_fam_count20_nonnested_16Aug2018.txt", header=F, sep="\t")

setwd("/home/springer/nosha003/methylation_spreading")
ltr_orientation <- read.delim("B73_ltr_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
tir_orientation <- read.delim("B73_tir_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
ltr_orientation$TE <- substr(ltr_orientation$TE, 4, 24)
ltr_orientation2 <- unique(ltr_orientation[,c(8,10)])
tir_orientation$TE <- substr(tir_orientation$TE, 4, 24)
tir_orientation2 <- unique(tir_orientation[,c(8,10)])

# CG
library(fields)
library(dplyr)
setwd("/scratch.global/nosha003/methylation_spreading")
data2 <- read.table(file="b73_v2_CG_TIR_metaplot.txt", header=T, sep="\t")
data3 <- read.table(file="b73_v2_CG_LTR_metaplot.txt", header=T, sep="\t")
data2$TE <- substr(data2$gene, 4, 24)
data3$TE <- substr(data3$gene, 4, 24)
data2$fam <- substr(data2$TE, 1, 8)
data3$fam <- substr(data3$TE, 1, 8)

data2_orient <- left_join(data2, tir_orientation2, by="TE")
data3_orient <- left_join(data3, ltr_orientation2, by="TE")

#data2.1 <- subset(data2_orient, data2$fam %in% b_tir_fam$V2)
#data3.1 <- subset(data3_orient, data3$fam %in% b_ltr_fam$V2)
data2.1 <- subset(data2_orient, data2$TE %in% b_tir_fam$TE)
data3.1 <- subset(data3_orient, data3$TE %in% b_ltr_fam$TE)
data2.1 <- data2.1 %>% mutate(dist_orient = ifelse(correction == -1, relative_distance*-1, relative_distance))
data3.1 <- data3.1 %>% mutate(dist_orient = ifelse(correction == -1, relative_distance*-1, relative_distance))

#look just at flanking 1kb of FGS
data.sub2=subset(data2.1,data2.1$dist_orient < 2000 & data2.1$dist_orient > -1000)
data.sub3=subset(data3.1,data3.1$dist_orient < 2000 & data3.1$dist_orient > -1000)

#get 100 bins across the relative gene distance and get stats on your data column
stats.test2=stats.bin(data.sub2$relative_distance,data.sub2$count,N=100)
stats.test3=stats.bin(data.sub3$relative_distance,data.sub3$count,N=100)

p.2=cbind(matrix(stats.test2$centers,ncol=1),stats.test2$stats["mean",])
p.3=cbind(matrix(stats.test3$centers,ncol=1),stats.test3$stats["mean",])

pdf("b73_nonnested_methylation_CG_metaplot_n100_oriented_20Dec.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='CG methylation across TE')
lines(p.2,col="black",lwd=1)
lines(p.3,col="blue",lwd=1)
xline(0,lty=2,col='grey')
xline(1000,lty=2,col='grey')
legend("topright",c('TIR', 'LTR'),col=c("black","blue"),lty=c(1,1,1),lwd=2,cex=0.7)
dev.off()


# CHG
library(dplyr)
library(fields)
setwd("/scratch.global/nosha003/methylation_spreading")
data2 <- read.table(file="b73_v2_CHG_TIR_metaplot.txt", header=T, sep="\t")
data3 <- read.table(file="b73_v2_CHG_LTR_metaplot.txt", header=T, sep="\t")
data2$TE <- substr(data2$gene, 4, 24)
data3$TE <- substr(data3$gene, 4, 24)
data2$fam <- substr(data2$TE, 1, 8)
data3$fam <- substr(data3$TE, 1, 8)

data2_orient <- left_join(data2, tir_orientation2, by="TE")
data3_orient <- left_join(data3, ltr_orientation2, by="TE")

#data2.1 <- subset(data2_orient, data2$fam %in% b_tir_fam$V2)
#data3.1 <- subset(data3_orient, data3$fam %in% b_ltr_fam$V2)
data2.1 <- subset(data2_orient, data2$TE %in% b_tir_fam$TE)
data3.1 <- subset(data3_orient, data3$TE %in% b_ltr_fam$TE)
data2.1$dist_orient <- data2.1$relative_distance
data3.1$dist_orient <- data3.1$relative_distance 

#look just at flanking 1kb of FGS
data.sub2=subset(data2.1,data2.1$dist_orient < 2000 & data2.1$dist_orient > -1000)
data.sub3=subset(data3.1,data3.1$dist_orient < 2000 & data3.1$dist_orient > -1000)

#get 100 bins across the relative gene distance and get stats on your data column
stats.test2=stats.bin(data.sub2$relative_distance,data.sub2$count,N=100)
stats.test3=stats.bin(data.sub3$relative_distance,data.sub3$count,N=100)

p.2=cbind(matrix(stats.test2$centers,ncol=1),stats.test2$stats["mean",])
p.3=cbind(matrix(stats.test3$centers,ncol=1),stats.test3$stats["mean",])

pdf("b73_nonnested_methylation_CHG_metaplot_n100_oriented_20Dec.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='CHG methylation across TE')
lines(p.2,col="black",lwd=1)
lines(p.3,col="blue",lwd=1)
xline(0,lty=2,col='grey')
xline(1000,lty=2,col='grey')
legend("topright",c('TIR', 'LTR'),col=c("black","blue"),lty=c(1,1,1),lwd=2,cex=0.7)
dev.off()



# CHH
library(dplyr)
library(fields)
setwd("/scratch.global/nosha003/methylation_spreading")
data2 <- read.table(file="b73_v2_CHH_TIR_metaplot.txt", header=T, sep="\t")
data3 <- read.table(file="b73_v2_CHH_LTR_metaplot.txt", header=T, sep="\t")
data2$TE <- substr(data2$gene, 4, 24)
data3$TE <- substr(data3$gene, 4, 24)
data2$fam <- substr(data2$TE, 1, 8)
data3$fam <- substr(data3$TE, 1, 8)

data2_orient <- left_join(data2, tir_orientation2, by="TE")
data3_orient <- left_join(data3, ltr_orientation2, by="TE")

data2.1 <- subset(data2_orient, data2$TE %in% b_tir_fam$TE)
data3.1 <- subset(data3_orient, data3$TE %in% b_ltr_fam$TE)
data2.1$dist_orient <- data2.1$relative_distance 
data3.1 <- data3.1 %>% mutate(dist_orient = ifelse(correction == -1, relative_distance*-1, relative_distance))

#look just at flanking 1kb of FGS
data.sub2=subset(data2.1,data2.1$dist_orient < 2000 & data2.1$dist_orient > -1000)
data.sub3=subset(data3.1,data3.1$dist_orient < 2000 & data3.1$dist_orient > -1000)

#get 100 bins across the relative gene distance and get stats on your data column
stats.test2=stats.bin(data.sub2$relative_distance,data.sub2$count,N=100)
stats.test3=stats.bin(data.sub3$relative_distance,data.sub3$count,N=100)

p.2=cbind(matrix(stats.test2$centers,ncol=1),stats.test2$stats["mean",])
p.3=cbind(matrix(stats.test3$centers,ncol=1),stats.test3$stats["mean",])

pdf("b73_nonnested_methylation_CHH_metaplot_n100_oriented_20Dec.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='CHH methylation across TE')
lines(p.2,col="black",lwd=1)
lines(p.3,col="blue",lwd=1)
xline(0,lty=2,col='grey')
xline(1000,lty=2,col='grey')
legend("topright",c('TIR', 'LTR'),col=c("black","blue"),lty=c(1,1,1),lwd=2,cex=0.7)
dev.off()

pdf("b73_nonnested_methylation_CHH_metaplot_n100_oriented_20Dec_scale.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,0.2),xlab="",ylab='read count',main='CHH methylation across TE')
lines(p.2,col="black",lwd=1)
lines(p.3,col="blue",lwd=1)
xline(0,lty=2,col='grey')
xline(1000,lty=2,col='grey')
legend("topright",c('TIR', 'LTR'),col=c("black","blue"),lty=c(1,1,1),lwd=2,cex=0.7)
dev.off()


# /home/springer/nosha003/scripts/bw_mC_metaplot_CGCHG_B73_20Dec.R
```

```{r eval=F}
#3Oct2018
#bw_mC_metaplot
#metaplots for CG, CHG, CHH methylation patterns surrounding genes, TIRs, LTRs for B73 and W22
	
#!/bin/bash
#PBS -l walltime=24:00:00,nodes=1:ppn=6,mem=60gb
#PBS -o /scratch.global/nosha003/e_o/metaplot_o
#PBS -e /scratch.global/nosha003/e_o/metaplot_e
#PBS -N metaplot
#PBS -V
#PBS -r n

cd /home/springer/nosha003/scripts/
module load R
R CMD BATCH bw_mC_metaplot_CGCHG_B73_20Dec.R
wait

# qsub /home/springer/nosha003/scripts/bw_mC_metaplot.sh
```


** just LTRs **
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

setwd("/home/springer/nosha003/database/B73v4/te")
b_ltr_fam <- read.delim("B73.structuralTEv2.2018.12.20.filteredTE.LTRnonnested.gff3", sep="\t", header=T)
b_ltr_fam$TE <- substr(b_ltr_fam$V9, 4, 24)

setwd("/home/springer/nosha003/methylation_spreading")
ltr_orientation <- read.delim("B73_ltr_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
ltr_orientation$TE <- substr(ltr_orientation$TE, 4, 24)
ltr_orientation2 <- unique(ltr_orientation[,c(6,8)])

# CG
library(fields)
library(dplyr)
setwd("/scratch.global/nosha003/methylation_spreading")
data <- read.table(file="b73_v2_CG_LTR_metaplot.txt", header=T, sep="\t")
data$TE <- substr(data$gene, 4, 24)
data$fam <- substr(data$TE, 1, 8)

data_orient <- left_join(data, ltr_orientation2, by="TE")

data2 <- subset(data_orient, data$TE %in% b_ltr_fam$TE)

#look just at flanking 1kb of FGS
data.sub=subset(data2,data2$relative_distance < 2000 & data2$relative_distance > -1000)

#get 100 bins across the relative gene distance and get stats on your data column
stats.test=stats.bin(data.sub$relative_distance,data.sub$count,N=100)

p.1=cbind(matrix(stats.test$centers,ncol=1),stats.test$stats["mean",])

pdf("b73_LTR_nonnested_methylation_CG_metaplot_n100_oriented_20Dec.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='Methylation across LTR')
lines(p.1,col="black",lwd=2)
xline(0,lty=2,col='grey')
xline(1000,lty=2,col='grey')
dev.off()


# CHG
library(fields)
library(dplyr)
setwd("/scratch.global/nosha003/methylation_spreading")
data <- read.table(file="b73_v2_CHG_LTR_metaplot.txt", header=T, sep="\t")
data$TE <- substr(data$gene, 4, 24)
data$fam <- substr(data$TE, 1, 8)

data_orient <- left_join(data, ltr_orientation2, by="TE")

data2 <- subset(data_orient, data$TE %in% b_ltr_fam$TE)

#look just at flanking 1kb of FGS
data.sub=subset(data2,data2$relative_distance < 2000 & data2$relative_distance > -1000)

#get 100 bins across the relative gene distance and get stats on your data column
stats.test=stats.bin(data.sub$relative_distance,data.sub$count,N=100)

p.1=cbind(matrix(stats.test$centers,ncol=1),stats.test$stats["mean",])

pdf("b73_LTR_nonnested_methylation_CHG_metaplot_n100_oriented_20Dec.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='Methylation across LTR')
lines(p.1,col="black",lwd=2)
xline(0,lty=2,col='grey')
xline(1000,lty=2,col='grey')
dev.off()

```


** just TIRs **
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

setwd("/home/springer/nosha003/database/B73v4/te")
b_tir_fam <- read.delim("B73.structuralTEv2.2018.12.20.filteredTE.TIRnonnested.gff3", sep="\t", header=T)
b_tir_fam$TE <- substr(b_tir_fam$V9, 4, 24)

setwd("/home/springer/nosha003/methylation_spreading")
tir_orientation <- read.delim("B73_tir_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
tir_orientation$TE <- substr(tir_orientation$TE, 4, 24)
tir_orientation2 <- unique(tir_orientation[,c(6,8)])

# CG
library(fields)
library(dplyr)
setwd("/scratch.global/nosha003/methylation_spreading")
data <- read.table(file="b73_v2_CG_TIR_metaplot.txt", header=T, sep="\t")
data$TE <- substr(data$gene, 4, 24)
data$fam <- substr(data$TE, 1, 8)

data_orient <- left_join(data, tir_orientation2, by="TE")

data2 <- subset(data_orient, data$TE %in% b_tir_fam$TE)

#look just at flanking 1kb of FGS
data.sub=subset(data2,data2$relative_distance < 2000 & data2$relative_distance > -1000)

#get 100 bins across the relative gene distance and get stats on your data column
stats.test=stats.bin(data.sub$relative_distance,data.sub$count,N=100)

p.1=cbind(matrix(stats.test$centers,ncol=1),stats.test$stats["mean",])

pdf("b73_TIR_nonnested_methylation_CG_metaplot_n100_oriented_20Dec.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='Methylation across TIR')
lines(p.1,col="black",lwd=2)
xline(0,lty=2,col='grey')
xline(1000,lty=2,col='grey')
dev.off()


# CHG
library(fields)
library(dplyr)
setwd("/scratch.global/nosha003/methylation_spreading")
data <- read.table(file="b73_v2_CHG_TIR_metaplot.txt", header=T, sep="\t")
data$TE <- substr(data$gene, 4, 24)
data$fam <- substr(data$TE, 1, 8)

data_orient <- left_join(data, tir_orientation2, by="TE")

data2 <- subset(data_orient, data$TE %in% b_tir_fam$TE)

#look just at flanking 1kb of FGS
data.sub=subset(data2,data2$relative_distance < 2000 & data2$relative_distance > -1000)

#get 100 bins across the relative gene distance and get stats on your data column
stats.test=stats.bin(data.sub$relative_distance,data.sub$count,N=100)

p.1=cbind(matrix(stats.test$centers,ncol=1),stats.test$stats["mean",])

pdf("b73_TIR_nonnested_methylation_CHG_metaplot_n100_oriented_20Dec.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='Methylation across TIR')
lines(p.1,col="black",lwd=2)
xline(0,lty=2,col='grey')
xline(1000,lty=2,col='grey')
dev.off()

```




# Figure S3: Tissue Heatmaps

## Data
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=1,mem=40G

##### B73 tissues
grep -v 'ctg' /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 | sort -k 1,1 -k 4,4n > /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3
grep -v 'ctg' /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 | sort -k 1,1 -k 4,4n > /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3

module load bedtools
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_Root_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_root_tir_closestTEtoBin_20Dec2018.txt
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_Root_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_root_ltr_closestTEtoBin_20Dec2018.txt

bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/wgbs/dec2017/bins/B73_merged_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_leaf_tir_closestTEtoBin_20Dec2018.txt
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/wgbs/dec2017/bins/B73_merged_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_leaf_ltr_closestTEtoBin_20Dec2018.txt

## orientation
library(tidyverse)
setwd("/home/springer/nosha003/methylation_spreading/")

# read in desired columns only
#TE_distance <- read_tsv("B73_root_ltr_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))
#TE_distance <- read_tsv("B73_root_tir_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))
#TE_distance <- read_tsv("B73_leaf_ltr_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))
TE_distance <- read_tsv("B73_leaf_tir_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))

# calculate which TE to flip
correction_values <- TE_distance %>%
  mutate(flank = ifelse(distance < 0, "L", ifelse(distance > 0, "R", ifelse(distance == 0, "TE_body", "what_the")))) %>%
  group_by(TE, flank) %>%
  select(TE, CG, distance, flank) %>%
  summarise(mean_CG = mean(CG, na.rm = T)) %>%
  group_by(TE) %>%
  spread(key = flank, value = mean_CG) %>%
  mutate(correction = ifelse(L > R, 1, ifelse(R > L, -1, 0))) %>%
  select(TE, correction)

# join on flip values
TE_distance <- TE_distance %>%
  left_join(correction_values, by = "TE")  

# flip values
TE_distance <- TE_distance %>%
  mutate(distance_2 = distance * correction)

#write.table(TE_distance, "B73_root_ltr_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)
#write.table(TE_distance, "B73_root_tir_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)
#write.table(TE_distance, "B73_leaf_ltr_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)
write.table(TE_distance, "B73_leaf_tir_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)

```


## Heatmaps

```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=1,mem=40G

library(fields)
library(tidyverse)
library(dplyr)
setwd("/home/springer/nosha003/methylation_spreading/")
b_ltr_fam <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", sep="\t", header=F)
b_ltr <- read.delim("B73_leaf_ltr_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
#b_ltr <- read.delim("B73_root_ltr_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
#b_ltr_fam <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018.txt", sep="\t", header=F)
#b_ltr <- read.delim("B73_leaf_tir_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
#b_ltr <- read.delim("B73_root_tir_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)

b_ltr$TEfam <- substr(b_ltr$TE, 4, 11)
b_ltr2 <- subset(b_ltr, b_ltr$TEfam %in% b_ltr_fam$V2)

dist = subset(b_ltr2, b_ltr2$distance_2 > -2000 & b_ltr2$distance_2 <= 2000)
b_ltr <- unique(b_ltr2$TEfam)

b_ltr <- unique(b_ltr2$TEfam)
output <- data.frame(matrix(NA, nrow=length(b_ltr), ncol=40))
rownames(output) <- b_ltr
colnames(output) <- seq(-1950,1950, 100)
for(i in 1:length(b_ltr)){
	data.sub=subset(dist, dist$TEfam == b_ltr[i])
	stats.test=stats.bin(data.sub$distance_2,data.sub$CG,N=40,breaks=seq(-2000,2000, 100))
	output[i,] = stats.test$stats["mean",]
}
p.1 <- output
for(i in 1:length(b_ltr)){
	data.sub=subset(dist, dist$TEfam == b_ltr[i])
	stats.test=stats.bin(data.sub$distance_2,data.sub$CHG,N=40,breaks=seq(-2000,2000, 100))
	output[i,] = stats.test$stats["mean",]
}
p.2 <- output
for(i in 1:length(b_ltr)){
	data.sub=subset(dist, dist$TEfam == b_ltr[i])
	stats.test=stats.bin(data.sub$distance_2,data.sub$CHH,N=40,breaks=seq(-2000,2000, 100))
	output[i,] = stats.test$stats["mean",]
}
p.3 <- output

b_ltr_cgchg <- cbind(p.1, p.2)
b_ltr_cgchg2 <- na.omit(b_ltr_cgchg)

# clustering based on B73 order
colnames(b_ltr_cgchg2) <- c("X.1950.x", "X.1850.x", "X.1750.x", "X.1650.x", "X.1550.x", "X.1450.x", "X.1350.x", "X.1250.x", "X.1150.x", "X.1050.x", "X.950.x", "X.850.x", "X.750.x", "X.650.x", "X.550.x", "X.450.x", "X.350.x", "X.250.x", "X.150.x", "X.50.x", "X50.x", "X150.x", "X250.x", "X350.x", "X450.x", "X550.x", "X650.x", "X750.x", "X850.x", "X950.x", "X1050.x", "X1150.x", "X1250.x", "X1350.x", "X1450.x", "X1550.x", "X1650.x", "X1750.x", "X1850.x", "X1950.x", "X.1950.y", "X.1850.y", "X.1750.y", "X.1650.y", "X.1550.y", "X.1450.y", "X.1350.y", "X.1250.y", "X.1150.y", "X.1050.y", "X.950.y", "X.850.y", "X.750.y", "X.650.y", "X.550.y", "X.450.y", "X.350.y", "X.250.y", "X.150.y", "X.50.y", "X50.y", "X150.y", "X250.y", "X350.y", "X450.y", "X550.y", "X650.y", "X750.y", "X850.y", "X950.y", "X1050.y", "X1150.y", "X1250.y", "X1350.y", "X1450.y", "X1550.y", "X1650.y", "X1750.y", "X1850.y", "X1950.y")
b_ltr_cgchg2$V1 <- rownames(b_ltr_cgchg2)

cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
#cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster$V1 <- row.names(cluster)
cluster_id <- cluster[,81:82]
mydata <- right_join(b_ltr_cgchg2, cluster_id, by='V1')
rownames(mydata) <- mydata$V1
mydata <- mydata[,c(1:80,82)]

## Heatmap
library(pheatmap)
library(viridis)

newdata <- mydata[order(mydata[,81]),]
annotation <- data.frame(paste("class", newdata[,81]))
colnames(annotation) <- "class"
rownames(annotation) <- rownames(newdata)

#pdf("b_root_ltr_class_heatmap_meandirection_kmeans3_cluster_20Dec2018.pdf")
pdf("b_leaf_ltr_class_heatmap_meandirection_kmeans3_cluster_20Dec2018.pdf")
#pdf("b_root_tir_class_heatmap_meandirection_kmeans3_cluster_20Dec2018.pdf")
#pdf("b_leaf_tir_class_heatmap_meandirection_kmeans3_cluster_20Dec2018.pdf")
pheatmap(newdata[,1:80],cluster_cols=F,cluster_rows = F, color=inferno(100), breaks=seq(0,1,length.out=100), show_rownames=TRUE, show_colnames = TRUE, border_color = FALSE, fontsize = 4, annotation_row = annotation)
dev.off()
```

```{r eval=F}
#9Jan2019
#tissue_heatmaps.sh
#make tir and ltr clustered heatmaps for multiple tissues
	
#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/tissue_heatmap_o
#PBS -e /scratch.global/nosha003/e_o/tissue_heatmap_e
#PBS -N tissue_heatmap
#PBS -V
#PBS -r n

cd /home/springer/nosha003/scripts/
module load R
R CMD BATCH ltr_root_heatmap.R
R CMD BATCH ltr_leaf_heatmap.R
R CMD BATCH tir_root_heatmap.R
R CMD BATCH tir_leaf_heatmap.R
wait

# qsub /home/springer/nosha003/scripts/tissue_heatmaps.sh
```


# Figure S4: Genotype Heatmaps

## Data
- determine non-nested families with >20 members across all genotypes (B73, W22, PH207, and Mo17)
- identify shared >20 non-nested member families between all 4 genotypes 
- make heatmaps for DNA methylation data unique to each genotype based on shared families and B73 clusters

#### B73
```{r, eval=F}
# B73 
module load bedtools

grep -v 'ctg' /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 | sort -k 1,1 -k 4,4n > /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3
grep -v 'ctg' /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 | sort -k 1,1 -k 4,4n > /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3

bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_20Dec2018.txt
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_20Dec2018.txt


## orientation
library(tidyverse)
setwd("/home/springer/nosha003/methylation_spreading/")

# read in desired columns only
TE_distance <- read_tsv("B73_ltr_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))
TE_distance <- read_tsv("B73_tir_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))

# calculate which TE to flip
correction_values <- TE_distance %>%
  mutate(flank = ifelse(distance < 0, "L", ifelse(distance > 0, "R", ifelse(distance == 0, "TE_body", "what_the")))) %>%
  group_by(TE, flank) %>%
  select(TE, CG, distance, flank) %>%
  summarise(mean_CG = mean(CG, na.rm = T)) %>%
  group_by(TE) %>%
  spread(key = flank, value = mean_CG) %>%
  mutate(correction = ifelse(L > R, 1, ifelse(R > L, -1, 0))) %>%
  select(TE, correction)

# join on flip values
TE_distance <- TE_distance %>%
  left_join(correction_values, by = "TE")  

# flip values
TE_distance <- TE_distance %>%
  mutate(distance_2 = distance * correction)

write.table(TE_distance, "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)
write.table(TE_distance, "B73_tir_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)


## families that have at least non-nested 20 members
# R
setwd("/home/springer/sna/B73")
nested <- read.delim("B73_nested_filteredTE_20Dec18.txt", header=T)
setwd("/home/springer/nosha003/database/B73v4/te/")
TIR <- read.delim("B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3", header=F, sep="\t")
TIR$id <- substr(TIR$V9, 4, 24)
LTR <- read.delim("B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3", header=F, sep="\t")
LTR$id <- substr(LTR$V9, 4, 24)

library(dplyr)
TIR_nonnested <- anti_join(TIR, nested, by = c("id" = "nested_te"))
LTR_nonnested <- anti_join(LTR, nested, by = c("id" = "nested_te"))
write.table(TIR_nonnested[,1:9], "B73.structuralTEv2.2018.12.20.filteredTE.TIRnonnested.gff3", quote=F, row.names=F, sep="\t")
write.table(LTR_nonnested[,1:9], "B73.structuralTEv2.2018.12.20.filteredTE.LTRnonnested.gff3", quote=F, row.names=F, sep="\t")


# unix
cut -f 9 /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/B73_tir_fam_count_nonnested20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/B73_tir_fam_count_nonnested20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/B73_tir_fam_count20_nonnested_20Dec2018.txt 
cut -f 9 /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/B73_ltr_fam_count_nonnested_20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/B73_ltr_fam_count_nonnested_20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/B73_ltr_fam_count20_nonnested_20Dec2018.txt

cp /home/springer/nosha003/database/B73v4/te/B73*_fam_count20_nonnested_20Dec2018.txt /home/springer/nosha003/methylation_spreading/.
```

#### W22
```{r eval=F}
# W22 
module load bedtools

grep -v 'ctg' /home/springer/sna/W22/W22.structuralTEv2.10.08.2018.filteredTE.gff3 | grep 'terminal_inverted_repeat' | sort -k 1,1 -k 4,4n > /home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.TIR.noctg.gff3
grep -v 'ctg' /home/springer/sna/W22/W22.structuralTEv2.10.08.2018.filteredTE.gff3 | grep 'LTR' | sort -k 1,1 -k 4,4n > /home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.LTR.noctg.gff3

bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/W22_tir_closestTEtoBin_20Dec2018.txt
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/W22_ltr_closestTEtoBin_20Dec2018.txt


## orientation
library(tidyverse)
setwd("/home/springer/nosha003/methylation_spreading/")

# read in desired columns only
#TE_distance <- read_tsv("W22_ltr_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))
TE_distance <- read_tsv("W22_tir_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))

# calculate which TE to flip
correction_values <- TE_distance %>%
  mutate(flank = ifelse(distance < 0, "L", ifelse(distance > 0, "R", ifelse(distance == 0, "TE_body", "what_the")))) %>%
  group_by(TE, flank) %>%
  select(TE, CG, distance, flank) %>%
  summarise(mean_CG = mean(CG, na.rm = T)) %>%
  group_by(TE) %>%
  spread(key = flank, value = mean_CG) %>%
  mutate(correction = ifelse(L > R, 1, ifelse(R > L, -1, 0))) %>%
  select(TE, correction)

# join on flip values
TE_distance <- TE_distance %>%
  left_join(correction_values, by = "TE")  

# flip values
TE_distance <- TE_distance %>%
  mutate(distance_2 = distance * correction)

#write.table(TE_distance, "W22_ltr_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)
write.table(TE_distance, "W22_tir_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)


## families that have at least non-nested 20 members
# R
setwd("/home/springer/sna/W22")
nested <- read.delim("W22_nested_filteredTE_20Dec18.txt", header=T)
setwd("/home/springer/nosha003/database/B73v4/te/")
TIR <- read.delim("W22.structuralTEv2.10.08.2018.filteredTE.TIR.noctg.gff3", header=F, sep="\t")
TIR$id <- substr(TIR$V9, 4, 24)
LTR <- read.delim("W22.structuralTEv2.10.08.2018.filteredTE.LTR.noctg.gff3", header=F, sep="\t")
LTR$id <- substr(LTR$V9, 4, 24)

library(dplyr)
TIR_nonnested <- anti_join(TIR, nested, by = c("id" = "nested_te"))
LTR_nonnested <- anti_join(LTR, nested, by = c("id" = "nested_te"))
write.table(TIR_nonnested[,1:9], "W22.structuralTEv2.10.08.2018.filteredTE.TIRnonnested.gff3", quote=F, row.names=F, sep="\t")
write.table(LTR_nonnested[,1:9], "W22.structuralTEv2.10.08.2018.filteredTE.LTRnonnested.gff3", quote=F, row.names=F, sep="\t")

# unix
cut -f 9 /home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.TIRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/W22_tir_fam_count_nonnested20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/W22_tir_fam_count_nonnested20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/W22_tir_fam_count20_nonnested_20Dec2018.txt 
cut -f 9 /home/springer/nosha003/database/B73v4/te/W22.structuralTEv2.10.08.2018.filteredTE.LTRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/W22_ltr_fam_count_nonnested_20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/W22_ltr_fam_count_nonnested_20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/W22_ltr_fam_count20_nonnested_20Dec2018.txt

cp /home/springer/nosha003/database/B73v4/te/W22*_fam_count20_nonnested_20Dec2018.txt /home/springer/nosha003/methylation_spreading/.
```


#### PH207
```{r eval=F}
# PH207 
module load bedtools

grep -v 'ctg' /home/springer/sna/PH207/PH207.structuralTEv2.2018.10.30.filteredTE.gff3 | grep -v 'scaffold' | grep 'terminal_inverted_repeat' |sort -k 1,1 -k 4,4n > /home/springer/nosha003/database/B73v4/te/PH207.structuralTEv2.2018.10.30.filteredTE.TIR.noctg.gff3
grep -v 'ctg' /home/springer/sna/PH207/PH207.structuralTEv2.2018.10.30.filteredTE.gff3 | grep -v 'scaffold' | grep 'LTR' |sort -k 1,1 -k 4,4n > /home/springer/nosha003/database/B73v4/te/PH207.structuralTEv2.2018.10.30.filteredTE.LTR.noctg.gff3

awk '{print "chr"$1"\t"$2"\t"$3"\t"$4"\t"$5"\t"$6}' /home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio.txt > /home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio_chr.txt

bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/PH207.structuralTEv2.2018.10.30.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio_chr.txt > /home/springer/nosha003/methylation_spreading/PH207_tir_closestTEtoBin_20Dec2018.txt
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/PH207.structuralTEv2.2018.10.30.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio_chr.txt > /home/springer/nosha003/methylation_spreading/PH207_ltr_closestTEtoBin_20Dec2018.txt

## orientation
library(tidyverse)
setwd("/home/springer/nosha003/methylation_spreading/")

# read in desired columns only
#TE_distance <- read_tsv("PH207_ltr_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))
TE_distance <- read_tsv("PH207_tir_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))

# calculate which TE to flip
correction_values <- TE_distance %>%
  mutate(flank = ifelse(distance < 0, "L", ifelse(distance > 0, "R", ifelse(distance == 0, "TE_body", "what_the")))) %>%
  group_by(TE, flank) %>%
  select(TE, CG, distance, flank) %>%
  summarise(mean_CG = mean(CG, na.rm = T)) %>%
  group_by(TE) %>%
  spread(key = flank, value = mean_CG) %>%
  mutate(correction = ifelse(L > R, 1, ifelse(R > L, -1, 0))) %>%
  select(TE, correction)

# join on flip values
TE_distance <- TE_distance %>%
  left_join(correction_values, by = "TE")  

# flip values
TE_distance <- TE_distance %>%
  mutate(distance_2 = distance * correction)

#write.table(TE_distance, "PH207_ltr_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)
write.table(TE_distance, "PH207_tir_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)


## families that have at least non-nested 20 members
# R
setwd("/home/springer/sna/PH207")
nested <- read.delim("PH207_nested_filteredTE_NOscaffold_20Dec18.txt", header=T)
setwd("/home/springer/nosha003/database/B73v4/te/")
TIR <- read.delim("PH207.structuralTEv2.2018.10.30.filteredTE.TIR.noctg.gff3", header=F, sep="\t")
TIR$id <- substr(TIR$V9, 4, 24)
LTR <- read.delim("PH207.structuralTEv2.2018.10.30.filteredTE.LTR.noctg.gff3", header=F, sep="\t")
LTR$id <- substr(LTR$V9, 4, 24)

library(dplyr)
TIR_nonnested <- anti_join(TIR, nested, by = c("id" = "nested_te"))
LTR_nonnested <- anti_join(LTR, nested, by = c("id" = "nested_te"))
write.table(TIR_nonnested[,1:9], "PH207.structuralTEv2.2018.10.30.filteredTE.TIRnonnested.gff3", quote=F, row.names=F, sep="\t")
write.table(LTR_nonnested[,1:9], "PH207.structuralTEv2.2018.10.30.filteredTE.LTRnonnested.gff3", quote=F, row.names=F, sep="\t")


# unix
cut -f 9 /home/springer/nosha003/database/B73v4/te/PH207.structuralTEv2.2018.10.30.filteredTE.TIRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/PH207_tir_fam_count_nonnested20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/PH207_tir_fam_count_nonnested20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/PH207_tir_fam_count20_nonnested_20Dec2018.txt 
cut -f 9 /home/springer/nosha003/database/B73v4/te/PH207.structuralTEv2.2018.10.30.filteredTE.LTRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/PH207_ltr_fam_count_nonnested_20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/PH207_ltr_fam_count_nonnested_20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/PH207_ltr_fam_count20_nonnested_20Dec2018.txt

cp /home/springer/nosha003/database/B73v4/te/PH207*_fam_count20_nonnested_20Dec2018.txt /home/springer/nosha003/methylation_spreading/.
```


#### Mo17
```{r eval=F}
# Mo17 
module load bedtools

grep -v 'ctg' /home/springer/sna/Mo17/Mo17.structuralTEv2.2018.12.28.filteredTE.gff3 | grep 'terminal_inverted_repeat' | sort -k 1,1 -k 4,4n > /home/springer/nosha003/database/B73v4/te/Mo17.structuralTEv2.2018.12.28.filteredTE.TIR.noctg.gff3
grep -v 'ctg' /home/springer/sna/Mo17/Mo17.structuralTEv2.2018.12.28.filteredTE.gff3 | grep 'LTR' | sort -k 1,1 -k 4,4n > /home/springer/nosha003/database/B73v4/te/Mo17.structuralTEv2.2018.12.28.filteredTE.LTR.noctg.gff3

bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/Mo17.structuralTEv2.2018.12.28.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/Mo17_tir_closestTEtoBin_20Dec2018.txt
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/database/B73v4/te/Mo17.structuralTEv2.2018.12.28.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/Mo17_ltr_closestTEtoBin_20Dec2018.txt

## orientation
library(tidyverse)
setwd("/home/springer/nosha003/methylation_spreading/")

# read in desired columns only
#TE_distance <- read_tsv("Mo17_ltr_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))
TE_distance <- read_tsv("Mo17_tir_closestTEtoBin_20Dec2018.txt", col_names = c("chr", "start", "stop", "CG", "CHG", "CHH", "x7", "x8", "x9", "TEstart", "TEstop", "score", "strand", "x14", "TE", "distance"), cols_only("chr" = col_character(), "start" = col_double(), "CG" = col_double(), "CHG" = col_double(), "CHH" = col_double(), "TE" = col_character(), "TEstart" = col_double(),"TEstop"=col_double(), "distance" = col_double()))

# calculate which TE to flip
correction_values <- TE_distance %>%
  mutate(flank = ifelse(distance < 0, "L", ifelse(distance > 0, "R", ifelse(distance == 0, "TE_body", "what_the")))) %>%
  group_by(TE, flank) %>%
  select(TE, CG, distance, flank) %>%
  summarise(mean_CG = mean(CG, na.rm = T)) %>%
  group_by(TE) %>%
  spread(key = flank, value = mean_CG) %>%
  mutate(correction = ifelse(L > R, 1, ifelse(R > L, -1, 0))) %>%
  select(TE, correction)

# join on flip values
TE_distance <- TE_distance %>%
  left_join(correction_values, by = "TE")  

# flip values
TE_distance <- TE_distance %>%
  mutate(distance_2 = distance * correction)

#write.table(TE_distance, "Mo17_ltr_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)
write.table(TE_distance, "Mo17_tir_closestTEtoBin_orientation_20Dec2018.txt", quote=F, sep="\t", row.names=F)



## families that have at least non-nested 20 members
# R
setwd("/home/springer/sna/Mo17")
nested <- read.delim("Mo17_nested_filteredTE_all_hosts_28Dec18.txt", header=T)
setwd("/home/springer/nosha003/database/B73v4/te/")
TIR <- read.delim("Mo17.structuralTEv2.2018.12.28.filteredTE.TIR.noctg.gff3", header=F, sep="\t")
TIR$id <- substr(TIR$V9, 4, 24)
LTR <- read.delim("Mo17.structuralTEv2.2018.12.28.filteredTE.LTR.noctg.gff3", header=F, sep="\t")
LTR$id <- substr(LTR$V9, 4, 24)

library(dplyr)
TIR_nonnested <- anti_join(TIR, nested, by = c("id" = "nested_te"))
LTR_nonnested <- anti_join(LTR, nested, by = c("id" = "nested_te"))
write.table(TIR_nonnested[,1:9], "Mo17.structuralTEv2.2018.12.28.filteredTE.TIRnonnested.gff3", quote=F, row.names=F, sep="\t")
write.table(LTR_nonnested[,1:9], "Mo17.structuralTEv2.2018.12.28.filteredTE.LTRnonnested.gff3", quote=F, row.names=F, sep="\t")

# unix
cut -f 9 /home/springer/nosha003/database/B73v4/te/Mo17.structuralTEv2.2018.12.28.filteredTE.TIRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/Mo17_tir_fam_count_nonnested20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/Mo17_tir_fam_count_nonnested20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/Mo17_tir_fam_count20_nonnested_20Dec2018.txt 
cut -f 9 /home/springer/nosha003/database/B73v4/te/Mo17.structuralTEv2.2018.12.28.filteredTE.LTRnonnested.gff3 | sed 's/ID=//g' | cut -c1-8 | awk ' { tot[$0]++ } END { for (i in tot) print tot[i],i } ' > /home/springer/nosha003/database/B73v4/te/Mo17_ltr_fam_count_nonnested_20Dec2018.txt
tr ' ' \\t < /home/springer/nosha003/database/B73v4/te/Mo17_ltr_fam_count_nonnested_20Dec2018.txt | awk '{if ($1 >= 20) print $0}' > /home/springer/nosha003/database/B73v4/te/Mo17_ltr_fam_count20_nonnested_20Dec2018.txt

cp /home/springer/nosha003/database/B73v4/te/Mo17*_fam_count20_nonnested_20Dec2018.txt /home/springer/nosha003/methylation_spreading/.
```


### Shared Families
```{r eval=F}
setwd("/home/springer/nosha003/methylation_spreading/")
b73_tir <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
b73_ltr <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
w22_tir <- read.delim("W22_tir_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
w22_ltr <- read.delim("W22_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
ph207_tir <- read.delim("PH207_tir_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
ph207_ltr <- read.delim("PH207_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
mo17_tir <- read.delim("Mo17_tir_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
mo17_ltr <- read.delim("Mo17_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")

library(dplyr)
bw_tir <- inner_join(b73_tir, w22_tir, by="V2")
bwp_tir <- inner_join(bw_tir, ph207_tir, by="V2")
bwpm_tir <- inner_join(bwp_tir, mo17_tir, by="V2")

bw_ltr <- inner_join(b73_ltr, w22_ltr, by="V2")
bwp_ltr <- inner_join(bw_ltr, ph207_ltr, by="V2")
bwpm_ltr <- inner_join(bwp_ltr, mo17_ltr, by="V2")

write.table(bwpm_tir, "BWPM_tir_fam_count20_nonnested_20Dec2018.txt", quote=F, sep="\t", row.names=F)
write.table(bwpm_ltr, "BWPM_ltr_fam_count20_nonnested_20Dec2018.txt", quote=F, sep="\t", row.names=F)
```

## Heatmaps
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=1,mem=40G

library(fields)
library(tidyverse)
library(dplyr)
setwd("/home/springer/nosha003/methylation_spreading/")
#b_ltr_fam <- read.delim("BWPM_ltr_fam_count20_nonnested_20Dec2018.txt", sep="\t", header=F)
#b_ltr <- read.delim("Mo17_ltr_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)
b_ltr_fam <- read.delim("BWPM_tir_fam_count20_nonnested_20Dec2018.txt", sep="\t", header=F)
b_ltr <- read.delim("PH207_tir_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T)

b_ltr$TEfam <- substr(b_ltr$TE, 4, 11)
b_ltr2 <- subset(b_ltr, b_ltr$TEfam %in% b_ltr_fam$V2)

dist = subset(b_ltr2, b_ltr2$distance_2 > -2000 & b_ltr2$distance_2 <= 2000)
b_ltr <- unique(b_ltr2$TEfam)

b_ltr <- unique(b_ltr2$TEfam)
output <- data.frame(matrix(NA, nrow=length(b_ltr), ncol=40))
rownames(output) <- b_ltr
colnames(output) <- seq(-1950,1950, 100)
for(i in 1:length(b_ltr)){
	data.sub=subset(dist, dist$TEfam == b_ltr[i])
	stats.test=stats.bin(data.sub$distance_2,data.sub$CG,N=40,breaks=seq(-2000,2000, 100))
	output[i,] = stats.test$stats["mean",]
}
p.1 <- output
for(i in 1:length(b_ltr)){
	data.sub=subset(dist, dist$TEfam == b_ltr[i])
	stats.test=stats.bin(data.sub$distance_2,data.sub$CHG,N=40,breaks=seq(-2000,2000, 100))
	output[i,] = stats.test$stats["mean",]
}
p.2 <- output
for(i in 1:length(b_ltr)){
	data.sub=subset(dist, dist$TEfam == b_ltr[i])
	stats.test=stats.bin(data.sub$distance_2,data.sub$CHH,N=40,breaks=seq(-2000,2000, 100))
	output[i,] = stats.test$stats["mean",]
}
p.3 <- output

b_ltr_cgchg <- cbind(p.1, p.2)
b_ltr_cgchg2 <- na.omit(b_ltr_cgchg)

# clustering
#cluster <- (kmeans(b_ltr_cgchg2[,c(1:19,21:59,61:80)], 3))
#mydata <- data.frame(b_ltr_cgchg2, cluster$cluster)
#write.table(mydata, "b_ltr_shared_meandirection_kmeans3_20Dec2018.txt", sep="\t", quote=F)
#write.table(mydata, "b_tir_shared_meandirection_kmeans3_20Dec2018.txt", sep="\t", quote=F)
#cluster <- read.delim("b_ltr_shared_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
#cluster$cluster.cluster[cluster$cluster.cluster == 1] <- "A"
#cluster$cluster.cluster[cluster$cluster.cluster == 2] <- "B"
#cluster$cluster.cluster[cluster$cluster.cluster == 3] <- "C"
#mydata <- cluster
#write.table(mydata, "b_ltr_shared_meandirection_kmeans3_20Dec2018.txt", sep="\t", quote=F)
#mydata <- read.delim("b_ltr_shared_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
#mydata <- read.delim("b_tir_shared_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")

# clustering based on B73 order
colnames(b_ltr_cgchg2) <- c("X.1950.x", "X.1850.x", "X.1750.x", "X.1650.x", "X.1550.x", "X.1450.x", "X.1350.x", "X.1250.x", "X.1150.x", "X.1050.x", "X.950.x", "X.850.x", "X.750.x", "X.650.x", "X.550.x", "X.450.x", "X.350.x", "X.250.x", "X.150.x", "X.50.x", "X50.x", "X150.x", "X250.x", "X350.x", "X450.x", "X550.x", "X650.x", "X750.x", "X850.x", "X950.x", "X1050.x", "X1150.x", "X1250.x", "X1350.x", "X1450.x", "X1550.x", "X1650.x", "X1750.x", "X1850.x", "X1950.x", "X.1950.y", "X.1850.y", "X.1750.y", "X.1650.y", "X.1550.y", "X.1450.y", "X.1350.y", "X.1250.y", "X.1150.y", "X.1050.y", "X.950.y", "X.850.y", "X.750.y", "X.650.y", "X.550.y", "X.450.y", "X.350.y", "X.250.y", "X.150.y", "X.50.y", "X50.y", "X150.y", "X250.y", "X350.y", "X450.y", "X550.y", "X650.y", "X750.y", "X850.y", "X950.y", "X1050.y", "X1150.y", "X1250.y", "X1350.y", "X1450.y", "X1550.y", "X1650.y", "X1750.y", "X1850.y", "X1950.y")
b_ltr_cgchg2$V1 <- rownames(b_ltr_cgchg2)

#cluster <- read.delim("b_ltr_shared_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster <- read.delim("b_tir_shared_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster$V1 <- row.names(cluster)
cluster_id <- cluster[,81:82]
mydata <- right_join(b_ltr_cgchg2, cluster_id, by='V1')
rownames(mydata) <- mydata$V1
mydata <- mydata[,c(1:80,82)]

## Heatmap
library(pheatmap)
library(viridis)

newdata <- mydata[order(mydata[,81]),]
annotation <- data.frame(paste("class", newdata[,81]))
colnames(annotation) <- "class"
rownames(annotation) <- rownames(newdata)

#pdf("m_ltr_SharedNonNested_heatmap_20Dec2018.pdf")
pdf("p_tir_SharedNonNested_heatmap_20Dec2018.pdf")
pheatmap(newdata[,1:80],cluster_cols=F,cluster_rows = F, color=inferno(100), breaks=seq(0,1,length.out=100), show_rownames=TRUE, show_colnames = TRUE, border_color = FALSE, fontsize = 4, annotation_row = annotation)
dev.off()
```

```{r eval=F}
#9Jan2019
#genotype_heatmaps.sh
#make tir and ltr clustered heatmaps for multiple genotypes with shared families
	
#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/genotype_heatmap_o
#PBS -e /scratch.global/nosha003/e_o/genotype_heatmap_e
#PBS -N genotype_heatmap
#PBS -V
#PBS -r n

cd /home/springer/nosha003/scripts/
module load R
R CMD BATCH b_ltr_heatmap.R
R CMD BATCH b_tir_heatmap.R
R CMD BATCH w_ltr_heatmap.R
R CMD BATCH w_tir_heatmap.R
R CMD BATCH p_ltr_heatmap.R
R CMD BATCH p_tir_heatmap.R
R CMD BATCH m_ltr_heatmap.R
R CMD BATCH m_tir_heatmap.R
wait

# qsub /home/springer/nosha003/scripts/genotype_heatmaps.sh
```




# Figure 3
- LTR 
    - families
 A  B  C 
71 32 23
    - elements
    A     B     C 
34454 13538  2890

- TIR (>1kb)
    - families
 A  B  C 
61 53 77
    - elements
  A   B   C 
525 911 322 


## Metaplot
- use heatmap output... subset by cluster... average all data per bin... 
- need distance within TE (1kb)

```{r eval=F}
#3Jan2019
#metaplots.sh
#make tir and ltr methylation and chromatin metaplots
	
#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/metaplot_o
#PBS -e /scratch.global/nosha003/e_o/metaplot_e
#PBS -N metaplot
#PBS -V
#PBS -r n

module load R
module load bedtools

## B73 methylation metaplot data table
perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance2.pl -i /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_orientation_20Dec2018.txt -o /scratch.global/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_orientation_metaplot_20Dec2018.txt
perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance2.pl -i /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_orientation_20Dec2018.txt -o /scratch.global/nosha003/methylation_spreading/B73_tir_closestTEtoBin_orientation_metaplot_20Dec2018.txt

## methylation metaplots
cd /home/springer/nosha003/scripts/
R CMD BATCH percent_bins_metaplot.R

## chromatin data correction
cd /home/springer/nosha003/scripts/
R CMD BATCH chromatin_data.R

## chromatin H3 corrected bedtools
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_ATACseq_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_ATACseq.txt
#bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_ATACseq_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_ATACseq.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K4me3_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K4me3.txt
#bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K4me3_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K4me3.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K27me3_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K27me3.txt
#bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K27me3_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K27me3.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K23ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K23ac.txt
#bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K23ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K23ac.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K27ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K27ac.txt
#bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K27ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K27ac.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K36me3_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K36me3.txt
#bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K36me3_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K36me3.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K4me1_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K4me1.txt
#bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K4me1_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K4me1.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K56ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K56ac.txt
#bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K56ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K56ac.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K9ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K9ac.txt
#bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K9ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K9ac.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_Pol_II_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_Pol_II.txt
#bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_Pol_II_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_Pol_II.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K9me2_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K9me2.txt
#bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K9me2_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K9me2.txt

## chromatin orientation
cd /scratch.global/nosha003/
R CMD BATCH chromatin_orientation_ltr.R
R CMD BATCH chromatin_orientation_tir.R

## chromatin metaplot data
perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance4.pl -i /scratch.global/nosha003/methylation_spreading/B73_ltr_${ID}_orientation_20Dec.txt -o /scratch.global/nosha003/methylation_spreading/B73_ltr_${ID}_orientation_metaplot.txt
perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance4.pl -i /scratch.global/nosha003/methylation_spreading/B73_tir_${ID}_orientation_20Dec.txt -o /scratch.global/nosha003/methylation_spreading/B73_tir_${ID}_orientation_metaplot.txt

## chromatin percent metaplots
cd /home/springer/nosha003/scripts/
R CMD BATCH chromatin_percent_metaplot.R
wait

# qsub /home/springer/nosha003/scripts/metaplots.sh
```

### metaplot data table (perl script)
```{r eval=F}
#! /usr/bin/perl -w
# use bedtools closest output, after orientation corrected based on mean values greater on left side, to generate real distance from TE for metaplots
# metaplot_class_distance.pl
# 18_June_2018
# Jaclyn_Noshay

use warnings;
use strict;
use Getopt::Std;

#set soft coded files
my $usage = "\n0 -i in -o out\n";
our ($opt_i, $opt_o, $opt_h);
getopts("i:o:h") || die "$usage";

#check that all files are defined
if ( (!(defined $opt_i)) || (!(defined $opt_o)) || (defined $opt_h) ) {
  print "$usage";
}

#read in bedtools intersect file 
open (my $in_fh, '<', $opt_i) || die;
open (my $out_fh, '>', $opt_o) || die;

print $out_fh "chr\tbinstart\tbinstop\tbinid\tcg\tchg\tchh\tTE\tTEstart\tTEend\tTEsize\tdistance\trelative_distance\treal_distance\n";

my ($relative_dist, $real_dist, $TEsize, diff, $location, $binTEstart, $binTEstop);

my $header = <$in_fh>;
while (my $line = <$in_fh>) {
  chomp $line;

  my ($chr, $start, $cg, $chg, $chh, $TEstart, $TEend, $TEID, undef, undef, $dist) = split("\t", $line);

  my $end = $start + 99;
  my $binid = $chr . "_" . $start . "_" . $end;

  my ($TE_name, undef) = split (";", $TEID);
  $TE_name =~ s/ID=//;
  $TEsize = abs($TEend - $TEstart);

  if ($dist eq "NA") {
    $relative_dist = "NA";
    $real_dist = "NA";
  }
  elsif ($dist == 0) {
    my $decimal = $diff / $TEsize;
    $relative_dist = $decimal * 1000;
    
    $binTEstart = $stop - $TEstart;
    $binTEstop = $TEstop - $start;
    
    if ($binTEstart < $binTEstop) {
      $location = $TEstart;
      $real_dist = $binTEstart;
    }
    elsif ($binTEstop < $binTEstart) {
      $location = $TEstop;
      $real_dist = $binTEstop + 1200;
    }
    
    if ($real_dist < 1200 && $real_dist > 1000) {
      $real_dist = "NA";
    }
    if ($real_dist > 2200) {
      $real_dsit = "NA";
    }
  }
  elsif ($dist > 0) {
    $relative_dist = $dist;
    $real_dist = $dist + 2200;
  }
  elsif ($dist < 0) {
    $relative_dist = $dist;
    $real_dist = $dist
  }

  print $out_fh "$chr\t$start\t$end\t$binid\t$cg\t$chg\t$chh\t$TE_name\t$TEstart\t$TEend\t$TEsize\t$dist\t$relative_dist\t$real_dist\n";
}

close $in_fh;
close $out_fh;
exit;
```

```{r eval=F}
## B73
perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance2.pl -i /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_orientation_20Dec2018.txt -o /scratch.global/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_orientation_metaplot_20Dec2018.txt
perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance2.pl -i /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_orientation_20Dec2018.txt -o /scratch.global/nosha003/methylation_spreading/B73_tir_closestTEtoBin_orientation_metaplot_20Dec2018.txt
```


### % bins Metaplot
- Make metaplots based on % bins with methylation instead of average methylation levels

- try making metaplot as % of bins with methylation instead of average methylation level --> try with CG first
    - make cutoff... >40% = methylated
    - make binary (0=not methylated, 1=methylated)
    - sum / n = % methylated
    - metaplot of % methylated for each class

#### individual TEs to include
- non-nested TEs within families that have >20 non-nested members
```{r eval=F}
setwd("/home/springer/nosha003/database/B73v4/te")
b_te <- read.delim("B73.structuralTEv2.2018.12.20.filteredTE.noctg.gff3", header=F, sep="\t")
colnames(b_te) <- c("chr", "source", "te_type", "start", "end", "dot", "strand", "dot2", "id")
b_te$TE <- substr(b_te$id, 4, 24)

setwd("/home/springer/sna/B73/")
nested <- read.delim("B73_nested_filteredTE_20Dec18.txt", header=T, sep="\t")
nested$TE <- nested$nested_te

library(dplyr)
nonnested <- anti_join(b_te, nested, by="TE")
nonnested$fam <- substr(nonnested$TE, 1, 8)

# TIR
setwd("/home/springer/nosha003/database/B73v4/te/") 
fam <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
colnames(fam) <- c("count", "fam")
nonnested_te <- inner_join(nonnested, fam, by="fam")
setwd("/scratch.global/nosha003/methylation_spreading")
write.table(nonnested_te, "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt", quote=F, row.names=F, sep="\t")
# 49586
# 1758 (>1kb length)

# LTR
setwd("/home/springer/nosha003/database/B73v4/te/") 
fam <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
colnames(fam) <- c("count", "fam")
nonnested_te <- inner_join(nonnested, fam, by="fam")
setwd("/scratch.global/nosha003/methylation_spreading")
write.table(nonnested_te, "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt", quote=F, row.names=F, sep="\t")
# 50882
# 50867 > 1,000bp
# 48586 > 5,000bp
# 29353 > 10,000bp
```



##### Methylation
** FUNCTION = metaplot_mC_PercentBins **
```{r eval=F}
metaplot_mC_PercentBins <- function(input_dir, file, fam_file, class_file, genotype, TEtype, date, cluster) {
library(fields)
library(dplyr)
library(reshape2)
library(ggplot2)
library(Rmisc)

setwd(input_dir)
b_ltr <- read.delim(file, sep="\t", header=T)
#b_ltr_fam <- read.delim(fam_file, sep="\t", header=F)
b_ltr_fam <- read.delim(fam_file, sep="\t", header=T)
class <- read.delim(class_file, sep="\t", header=T)
class$TEfam <- rownames(class)
class_fam <- class[,81:82]

# setwd("/scratch.global/nosha003/methylation_spreading")
# b_ltr <- read.delim("B73_tir_closestTEtoBin_orientation_metaplot_20Dec2018.txt", sep="\t", header=T)
# #b_ltr_fam <- read.delim(fam_file, sep="\t", header=F)
# b_ltr_fam <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt", sep="\t", header=T)
# class <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", sep="\t", header=T)
# class$TEfam <- rownames(class)
# class_fam <- class[,81:82]

b_ltr_fam$length <- abs(b_ltr_fam$end - b_ltr_fam$start)
b_ltr_fam2 <- subset(b_ltr_fam, b_ltr_fam$length > 1000)
  
b_ltr$TEfam <- substr(b_ltr$TE, 1, 8)
#b_ltr2 <- subset(b_ltr, b_ltr$TEfam %in% b_ltr_fam$V2)
b_ltr2 <- subset(b_ltr, b_ltr$TE %in% b_ltr_fam$TE)
#b_ltr2 <- subset(b_ltr, b_ltr$TE %in% b_ltr_fam2$TE)

b_ltr_cg <- as.numeric(paste(b_ltr2$cg))
b_ltr2["cg"] <- b_ltr_cg
b_ltr2$CG <- ifelse(b_ltr2$cg >=0.40, 1, 0)
b_ltr2$CHG <- ifelse(b_ltr2$chg >=0.40, 1, 0)
b_ltr2$CHH <- ifelse(b_ltr2$chh >=0.015, 1, 0)
b_ltr3 <- left_join(b_ltr2, class, by="TEfam")
dist = subset(b_ltr3, b_ltr3$relative_distance > -2000 & b_ltr3$relative_distance <= 3000)

dist_class1 <- subset(dist, dist$cluster.cluster == "A")
dist_class2 <- subset(dist, dist$cluster.cluster == "B")
dist_class3 <- subset(dist, dist$cluster.cluster == "C")

dist_class1_group <-  transform(dist_class1, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class1_sumCG <- do.call(data.frame,aggregate(dist_class1_group$CG~dist_class1_group$group, dist_class1_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class1_sumCG) <- c("group", "count", "sum")
dist_class1_sumCG["percent_methylated"] <- dist_class1_sumCG$sum / dist_class1_sumCG$count
dist_class1_sumCG["cluster.cluster"] <- "H"
dist_class1_sumCHG <- do.call(data.frame,aggregate(dist_class1_group$CHG~dist_class1_group$group, dist_class1_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class1_sumCHG) <- c("group", "count", "sum")
dist_class1_sumCHG["percent_methylated"] <- dist_class1_sumCHG$sum / dist_class1_sumCHG$count
dist_class1_sumCHG["cluster.cluster"] <- "H"
dist_class1_sumCHH <- do.call(data.frame,aggregate(dist_class1_group$CHH~dist_class1_group$group, dist_class1_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class1_sumCHH) <- c("group", "count", "sum")
dist_class1_sumCHH["percent_methylated"] <- dist_class1_sumCHH$sum / dist_class1_sumCHH$count
dist_class1_sumCHH["cluster.cluster"] <- "H"

dist_class2_group <-  transform(dist_class2, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class2_sumCG <- do.call(data.frame,aggregate(dist_class2_group$CG~dist_class2_group$group, dist_class2_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class2_sumCG) <- c("group", "count", "sum")
dist_class2_sumCG["percent_methylated"] <- dist_class2_sumCG$sum / dist_class2_sumCG$count
dist_class2_sumCG["cluster.cluster"] <- "M"
dist_class2_sumCHG <- do.call(data.frame,aggregate(dist_class2_group$CHG~dist_class2_group$group, dist_class2_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class2_sumCHG) <- c("group", "count", "sum")
dist_class2_sumCHG["percent_methylated"] <- dist_class2_sumCHG$sum / dist_class2_sumCHG$count
dist_class2_sumCHG["cluster.cluster"] <- "M"
dist_class2_sumCHH <- do.call(data.frame,aggregate(dist_class2_group$CHH~dist_class2_group$group, dist_class2_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class2_sumCHH) <- c("group", "count", "sum")
dist_class2_sumCHH["percent_methylated"] <- dist_class2_sumCHH$sum / dist_class2_sumCHH$count
dist_class2_sumCHH["cluster.cluster"] <- "M"

dist_class3_group <-  transform(dist_class3, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class3_sumCG <- do.call(data.frame,aggregate(dist_class3_group$CG~dist_class3_group$group, dist_class3_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class3_sumCG) <- c("group", "count", "sum")
dist_class3_sumCG["percent_methylated"] <- dist_class3_sumCG$sum / dist_class3_sumCG$count
dist_class3_sumCG["cluster.cluster"] <- "L"
dist_class3_sumCHG <- do.call(data.frame,aggregate(dist_class3_group$CHG~dist_class3_group$group, dist_class3_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class3_sumCHG) <- c("group", "count", "sum")
dist_class3_sumCHG["percent_methylated"] <- dist_class3_sumCHG$sum / dist_class3_sumCHG$count
dist_class3_sumCHG["cluster.cluster"] <- "L"
dist_class3_sumCHH <- do.call(data.frame,aggregate(dist_class3_group$CHH~dist_class3_group$group, dist_class3_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class3_sumCHH) <- c("group", "count", "sum")
dist_class3_sumCHH["percent_methylated"] <- dist_class3_sumCHH$sum / dist_class3_sumCHH$count
dist_class3_sumCHH["cluster.cluster"] <- "L"

p.1 <- rbind(dist_class1_sumCG, dist_class2_sumCG, dist_class3_sumCG)
p.2 <- rbind(dist_class1_sumCG, dist_class2_sumCHG, dist_class3_sumCHG)
p.3 <- rbind(dist_class1_sumCHH, dist_class2_sumCHH, dist_class3_sumCHH)
write.table(p.1, paste(genotype, TEtype, "class_metaplot_data_cg_cluster_3_PercentMethylated", date, ".txt", sep="_"), sep="\t", quote=F)
write.table(p.2, paste(genotype, TEtype, "class_metaplot_data_chg_cluster_3_PercentMethylated", date, ".txt", sep="_"), sep="\t", quote=F)
write.table(p.3, paste(genotype, TEtype, "class_metaplot_data_chh_cluster_3_PercentMethylated", date, ".txt", sep="_"), sep="\t", quote=F)

## Metaplot of percent bins methylated by class
#group.colors <- c(A = "cornflowerblue", B = "springgreen3", C = "lightcoral")
#group.colors <- c("cornflowerblue", "springgreen3", "lightcoral")
#group.colors <- c("#3972c9ff", "#e7a275ff", "#a43434ff")
group.colors <- c("#3972c9ff", "#a43434ff", "#e7a275ff")

ggplot(p.1, aes(x = group, y = percent_methylated, group = cluster.cluster, colour = cluster.cluster)) + geom_line()

pdf(paste(genotype, TEtype, "class_metaplot_cg_cluster", cluster, date, "PercentMethylated.pdf", sep="_"))
#pdf(paste(genotype, TEtype, "class_metaplot_cg_cluster", cluster, date, "PercentMethylated.length1kb.pdf", sep="_"))
#plot1 <- ggplot(p.1, aes(x = group, y = percent_methylated, group = cluster.cluster, colour = cluster.cluster)) + geom_line() + scale_color_manual(values=group.colors) + theme_classic()
plot1 <- ggplot(p.1, aes(x = group, y = percent_methylated, group = cluster.cluster, colour = cluster.cluster)) + geom_line(size=1) + scale_color_manual(values=group.colors) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot1)
dev.off()
pdf(paste(genotype, TEtype, "class_metaplot_chg_cluster", cluster, date, "PercentMethylated.pdf", sep="_"))
#pdf(paste(genotype, TEtype, "class_metaplot_chg_cluster", cluster, date, "PercentMethylated.length1kb.pdf", sep="_"))
#plot2 <- ggplot(p.2, aes(x = group, y = percent_methylated, group = cluster.cluster, colour = cluster.cluster)) + geom_line() + scale_color_manual(values=group.colors) + theme_classic()
plot2 <- ggplot(p.2, aes(x = group, y = percent_methylated, group = cluster.cluster, colour = cluster.cluster)) + geom_line(size=1) + scale_color_manual(values=group.colors) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot2)
dev.off()
pdf(paste(genotype, TEtype, "class_metaplot_chh_cluster", cluster, date, "PercentMethylated.pdf", sep="_"))
#pdf(paste(genotype, TEtype, "class_metaplot_chh_cluster", cluster, date, "PercentMethylated.length1kb.pdf", sep="_"))
#plot3 <- ggplot(p.3, aes(x = group, y = percent_methylated, group = cluster.cluster, colour = cluster.cluster)) + geom_line() + scale_color_manual(values=group.colors) + theme_classic()
plot3 <- ggplot(p.3, aes(x = group, y = percent_methylated, group = cluster.cluster, colour = cluster.cluster)) + geom_line(size=1) + scale_color_manual(values=group.colors) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot3)
dev.off()
}

```
```{r, eval=F}
scp nosha003@login.msi.umn.edu:/scratch.global/nosha003/methylation_spreading/*class_metaplot*PercentMethylated.pdf /Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/spreading/.
```

```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

## B73
### TIR
metaplot_mC_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_closestTEtoBin_orientation_metaplot_20Dec2018.txt",
  #fam_file= "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  cluster = "3",
  #date = "20Dec2018"
  date = "Jan2019"
)

### LTR
metaplot_mC_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_closestTEtoBin_orientation_metaplot_20Dec2018.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  cluster = "3",
  date = "Jan2019"
)
```


```{r eval=F}
metaplot_mC_PercentBins <- function(input_dir, file, fam_file, class_file, genotype, TEtype, date) {
library(fields)
library(dplyr)
library(reshape2)
library(ggplot2)
library(Rmisc)

setwd(input_dir)
b_ltr <- read.delim(file, sep="\t", header=T)
b_ltr_fam <- read.delim(fam_file, sep="\t", header=T)
class <- read.delim(class_file, sep="\t", header=T)
class$TEfam <- rownames(class)
class_fam <- class[,81:82]

b_ltr_fam$length <- abs(b_ltr_fam$end - b_ltr_fam$start)
b_ltr_fam2 <- subset(b_ltr_fam, b_ltr_fam$length > 1000)
  
b_ltr$TEfam <- substr(b_ltr$TE, 1, 8)
b_ltr2 <- subset(b_ltr, b_ltr$TE %in% b_ltr_fam$TE)

b_ltr_cg <- as.numeric(paste(b_ltr2$cg))
b_ltr2["cg"] <- b_ltr_cg
b_ltr2$CG <- ifelse(b_ltr2$cg >=0.40, 1, 0)
b_ltr2$CHG <- ifelse(b_ltr2$chg >=0.40, 1, 0)
b_ltr2$CHH <- ifelse(b_ltr2$chh >=0.015, 1, 0)
b_ltr3 <- left_join(b_ltr2, class, by="TEfam")
dist = subset(b_ltr3, b_ltr3$relative_distance > -2000 & b_ltr3$relative_distance <= 3000)

dist_group <-  transform(dist, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_sumCG <- do.call(data.frame,aggregate(dist_group$CG~dist_group$group, dist_group, FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_sumCG) <- c("group", "count", "sum")
dist_sumCG["percent_methylated"] <- dist_sumCG$sum / dist_sumCG$count
dist_sumCG["cluster.cluster"] <- "CG"
dist_sumCHG <- do.call(data.frame,aggregate(dist_group$CHG~dist_group$group, dist_group, FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_sumCHG) <- c("group", "count", "sum")
dist_sumCHG["percent_methylated"] <- dist_sumCHG$sum / dist_sumCHG$count
dist_sumCHG["cluster.cluster"] <- "CHG"

p.1 <- rbind(dist_sumCG, dist_sumCHG)

pdf(paste(genotype, TEtype, "class_metaplot_cg_chg", date, "PercentMethylated.pdf", sep="_"))
plot1 <- ggplot(p.1, aes(x = group, y = percent_methylated, group = cluster.cluster, colour = cluster.cluster)) + geom_line(size=1) + scale_color_manual(values=group.colors) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot1)
dev.off()
}
```

```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

## B73
### TIR
metaplot_mC_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_closestTEtoBin_orientation_metaplot_20Dec2018.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019"
)

### LTR
metaplot_mC_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_closestTEtoBin_orientation_metaplot_20Dec2018.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019"
)
```




###### Average Methylation
** LTR and TIR plots based on average methylation within each cluster **
```{r eval=F}
perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance_nonoriented.pl -i /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_20Dec2018.txt -o /scratch.global/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_metaplot_20Dec2018.txt
perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance_nonoriented.pl -i /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_20Dec2018.txt -o /scratch.global/nosha003/methylation_spreading/B73_tir_closestTEtoBin_metaplot_20Dec2018.txt

library(fields)
library(dplyr)
library(reshape2)
library(ggplot2)
library(Rmisc)

# LTR
setwd("/scratch.global/nosha003/methylation_spreading")
b_ltr <- read.delim("B73_ltr_closestTEtoBin_metaplot_20Dec2018.txt", sep="\t", header=T)
b_ltr_fam <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt", sep="\t", header=T)
setwd("/home/springer/nosha003/methylation_spreading")
class <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", sep="\t", header=T)
class$TEfam <- rownames(class)
class_fam <- class[,81:82]

b_ltr$TEfam <- substr(b_ltr$TE, 1, 8)
b_ltr2 <- subset(b_ltr, b_ltr$TE %in% b_ltr_fam$TE)

b_ltr_cg <- as.numeric(paste(b_ltr2$cg))
b_ltr2["cg"] <- b_ltr_cg
b_ltr2$CG <- b_ltr2$cg
b_ltr2$CHG <- b_ltr2$chg
b_ltr2$CHH <- b_ltr2$chh
b_ltr3 <- left_join(b_ltr2, class, by="TEfam")
dist = subset(b_ltr3, b_ltr3$relative_distance > -2000 & b_ltr3$relative_distance <= 3000)

dist_class1 <- subset(dist, dist$cluster.cluster == "A")
dist_class2 <- subset(dist, dist$cluster.cluster == "B")
dist_class3 <- subset(dist, dist$cluster.cluster == "C")

stats.test1=stats.bin(dist_class1$relative_distance,dist_class1$CHH,N=100)
stats.test2=stats.bin(dist_class2$relative_distance,dist_class2$CHH,N=100)
stats.test3=stats.bin(dist_class3$relative_distance,dist_class3$CHH,N=100)

p.1=cbind(matrix(stats.test1$centers,ncol=1),stats.test1$stats["mean",])
p.2=cbind(matrix(stats.test2$centers,ncol=1),stats.test2$stats["mean",])
p.3=cbind(matrix(stats.test3$centers,ncol=1),stats.test3$stats["mean",])

pdf("b73_LTR_cluster_CHH_metaplot_Mar2019.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,0.03),xlab="",ylab='read count',main='Methylation across LTR')
lines(p.1,col="#3972c9ff",lwd=2)
lines(p.2,col="#e7a275ff",lwd=2)
lines(p.3,col="#a43434ff",lwd=2)
xline(0,lty=2,col='grey')
xline(1000,lty=2,col='grey')
legend("topright",c('High', 'Moderate', 'Low'),col=c("#3972c9ff","#e7a275ff", "#a43434ff"),lty=c(1,1,1),lwd=2,cex=0.7)
dev.off()


# TIR
setwd("/scratch.global/nosha003/methylation_spreading")
b_tir <- read.delim("B73_tir_closestTEtoBin_metaplot_20Dec2018.txt", sep="\t", header=T)
b_tir_fam <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt", sep="\t", header=T)
setwd("/home/springer/nosha003/methylation_spreading")
class <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", sep="\t", header=T)
class$TEfam <- rownames(class)
class_fam <- class[,81:82]

b_tir$TEfam <- substr(b_tir$TE, 1, 8)
b_tir2 <- subset(b_tir, b_tir$TE %in% b_tir_fam$TE)

b_tir_cg <- as.numeric(paste(b_tir2$cg))
b_tir2["cg"] <- b_tir_cg
b_tir2$CG <- b_tir2$cg
b_tir2$CHG <- b_tir2$chg
b_tir2$CHH <- b_tir2$chh
b_tir3 <- left_join(b_tir2, class, by="TEfam")
dist = subset(b_tir3, b_tir3$relative_distance > -2000 & b_tir3$relative_distance <= 3000)

dist_class1 <- subset(dist, dist$cluster.cluster == "A")
dist_class2 <- subset(dist, dist$cluster.cluster == "B")
dist_class3 <- subset(dist, dist$cluster.cluster == "C")

stats.test1=stats.bin(dist_class1$relative_distance,dist_class1$CHG,N=100)
stats.test2=stats.bin(dist_class2$relative_distance,dist_class2$CHG,N=100)
stats.test3=stats.bin(dist_class3$relative_distance,dist_class3$CHG,N=100)

p.1=cbind(matrix(stats.test1$centers,ncol=1),stats.test1$stats["mean",])
p.2=cbind(matrix(stats.test2$centers,ncol=1),stats.test2$stats["mean",])
p.3=cbind(matrix(stats.test3$centers,ncol=1),stats.test3$stats["mean",])

pdf("b73_TIR_cluster_CHG_metaplot_Jan2019.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='Methylation across TIR')
lines(p.1,col="#3972c9ff",lwd=2)
lines(p.2,col="#a43434ff",lwd=2)
lines(p.3,col="#e7a275ff",lwd=2)
xline(0,lty=2,col='grey')
xline(1000,lty=2,col='grey')
legend("topright",c('High', 'Moderate', 'Low'),col=c("#3972c9ff","#a43434ff", "#e7a275ff"),lty=c(1,1,1),lwd=2,cex=0.7)
dev.off()

# TIR > 1kb
setwd("/scratch.global/nosha003/methylation_spreading")
b_tir <- read.delim("B73_tir_closestTEtoBin_metaplot_20Dec2018.txt", sep="\t", header=T)
b_tir_fam <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt", sep="\t", header=T)
setwd("/home/springer/nosha003/methylation_spreading")
class <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", sep="\t", header=T)
class$TEfam <- rownames(class)
class_fam <- class[,81:82]

b_tir_fam$length <- abs(b_tir_fam$end - b_tir_fam$start)
b_tir_fam2 <- subset(b_tir_fam, b_tir_fam$length > 1000)

b_tir$TEfam <- substr(b_tir$TE, 1, 8)
#b_tir2 <- subset(b_tir, b_tir$TE %in% b_tir_fam$TE)
b_tir2 <- subset(b_tir, b_tir$TE %in% b_tir_fam2$TE)

b_tir_cg <- as.numeric(paste(b_tir2$cg))
b_tir2["cg"] <- b_tir_cg
b_tir2$CG <- b_tir2$cg
b_tir2$CHG <- b_tir2$chg
b_tir2$CHH <- b_tir2$chh
b_tir3 <- left_join(b_tir2, class, by="TEfam")
dist = subset(b_tir3, b_tir3$relative_distance > -2000 & b_tir3$relative_distance <= 3000)

dist_class1 <- subset(dist, dist$cluster.cluster == "A")
dist_class2 <- subset(dist, dist$cluster.cluster == "B")
dist_class3 <- subset(dist, dist$cluster.cluster == "C")

stats.test1=stats.bin(dist_class1$relative_distance,dist_class1$CHH,N=100)
stats.test2=stats.bin(dist_class2$relative_distance,dist_class2$CHH,N=100)
stats.test3=stats.bin(dist_class3$relative_distance,dist_class3$CHH,N=100)

p.1=cbind(matrix(stats.test1$centers,ncol=1),stats.test1$stats["mean",])
p.2=cbind(matrix(stats.test2$centers,ncol=1),stats.test2$stats["mean",])
p.3=cbind(matrix(stats.test3$centers,ncol=1),stats.test3$stats["mean",])

#pdf("b73_TIR_cluster_CHG_metaplot_Jan2019.pdf")
pdf("b73_TIR_gr1kb_cluster_CHH_metaplot_Mar2019.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,0.03),xlab="",ylab='read count',main='Methylation across TIR')
lines(p.1,col="#3972c9ff",lwd=2)
lines(p.2,col="#e7a275ff",lwd=2)
lines(p.3,col="#a43434ff",lwd=2)
xline(0,lty=2,col='grey')
xline(1000,lty=2,col='grey')
legend("topright",c('High', 'Moderate', 'Low'),col=c("#3972c9ff","#e7a275ff", "#a43434ff"),lty=c(1,1,1),lwd=2,cex=0.7)
dev.off()


```


###### CHH metaplot by family (Revisions)
- create a metaplot for each family of the lowly methylated group for LTRs... look at CHH level at TE boundaries

```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

library(fields)
library(dplyr)
library(reshape2)
library(ggplot2)
library(Rmisc)

# LTR
setwd("/scratch.global/nosha003/methylation_spreading")
b_ltr <- read.delim("B73_ltr_closestTEtoBin_metaplot_20Dec2018.txt", sep="\t", header=T)
b_ltr_fam <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt", sep="\t", header=T)
setwd("/home/springer/nosha003/methylation_spreading")
class <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", sep="\t", header=T)
class$TEfam <- rownames(class)
class_fam <- class[,81:82]

b_ltr$TEfam <- substr(b_ltr$TE, 1, 8)
b_ltr2 <- subset(b_ltr, b_ltr$TE %in% b_ltr_fam$TE)

b_ltr_cg <- as.numeric(paste(b_ltr2$cg))
b_ltr2["cg"] <- b_ltr_cg
b_ltr2$CG <- b_ltr2$cg
b_ltr2$CHG <- b_ltr2$chg
b_ltr2$CHH <- b_ltr2$chh
b_ltr3 <- left_join(b_ltr2, class, by="TEfam")
dist = subset(b_ltr3, b_ltr3$relative_distance > -2000 & b_ltr3$relative_distance <= 3000)

dist_class <- subset(dist, dist$cluster.cluster == "C")
dist_class$family <- as.factor(dist_class$TEfam)

#stats.test <- stats.bin(dist_class$relative_distance,dist_class$CHH,N=100)	
#p.chh <- cbind(matrix(stats.test$centers,ncol=1),stats.test$stats["mean",])
#df <- cbind(matrix(stats.test$centers,ncol=1),stats.test$stats["mean",])

fam <- as.factor(unique(dist_class$TEfam))

df <- data.frame(matrix(NA, nrow=100, ncol=length(fam)))
colnames(df) <- fam
df[1,] <- fam
par(mfrow = c(5,5))
for (i in 1:ncol(df)){
  id <- df[1,i]
  dist_class_fam <- dist_class[dist_class$TEfam == id,]
  stats.test <- stats.bin(dist_class_fam$relative_distance,dist_class_fam$CHH,N=100)
 	output1 <- cbind(matrix(stats.test$centers,ncol=1),stats.test$stats["mean",])
 	df[,i] <- output1[,2]
}

# find where things went wrong with traceback()
# why can't i get families to subset?? 

head(df)
df2 <- data.frame(df)
df2$dist <- output1[,1]


df_melt <- melt(df2, id="dist")

setwd("/scratch.global/nosha003/methylation_spreading")

pdf("chh_low_mC_family.pdf")
ggplot(df_melt, aes(x=dist, y=value, colour=variable, group=variable)) + geom_line() + theme_classic() + theme(legend.position = "none")
dev.off()

pdf("chh_low_mC_family_split.pdf")
ggplot(df_melt, aes(x=dist, y=value)) + geom_line() + facet_grid(variable ~ .) + theme_classic()
dev.off()
pdf("chh_low_mC_family_split2.pdf")
ggplot(df_melt, aes(x=dist, y=value)) + geom_line() + facet_grid(. ~ variable) + theme_classic()
dev.off()




## Calculate average of TE flanks and inner TE in comparison to the average of the potential peak regions for each family... if peak region higher than average of other bins... peak family
head(df_melt)
df_chh <- dcast(df_melt, variable ~ dist)
df_chh$avg <- rowMeans(df_chh[,c(2:40,63:101)])
df_chh$peak <- rowMeans(df_chh[,c(42,61,62)])
peak <- df_chh %>% mutate(peak_call = ifelse(peak > avg, "peak_family", "NA"))
peak_df <- peak[,c(1,102:104)]
# 22/23 low mC LTR families have a CHH peak

#    variable         avg        peak   peak_call
# 1  RLC00032 0.008122063 0.012314807 peak_family
# 2  RLC00030 0.015504083 0.047820148 peak_family
# 3  RLX00824 0.012670004 0.036653358 peak_family
# 4  RLC00063 0.011158606 0.010471649          NA
# 5  RLG00011 0.013336428 0.023179192 peak_family
# 6  RLC00070 0.013258476 0.016672882 peak_family
# 7  RLC00064 0.007261253 0.011956428 peak_family
# 8  RLX00094 0.007623379 0.007863840 peak_family
# 9  RLC00046 0.017284315 0.023871771 peak_family
# 10 RLX00172 0.008868913 0.011286256 peak_family
# 11 RLC00143 0.016709933 0.041088682 peak_family
# 12 RLC00148 0.012575324 0.040907760 peak_family
# 13 RLC00184 0.013797157 0.037762740 peak_family
# 14 RLC00034 0.015124269 0.017957397 peak_family
# 15 RLC00158 0.014268806 0.018853975 peak_family
# 16 RLX00065 0.008102694 0.011648652 peak_family
# 17 RLC00074 0.014068500 0.021013337 peak_family
# 18 RLG00159 0.007321409 0.009715917 peak_family
# 19 RLG00053 0.014459273 0.024073908 peak_family
# 20 RLC00137 0.014875357 0.020520569 peak_family
# 21 RLC00149 0.017232899 0.017545078 peak_family
# 22 RLX00168 0.004775170 0.007240919 peak_family
# 23 RLG00111 0.012121508 0.019864630 peak_family

```



### Chromatin Metaplots
#### data
##### H3 corrected

** Function: h3_corrected **
```{r eval=F}
h3_corrected <- function(input_dir, file, h3_dir, h3_file, mark, genotype, date) {
setwd(h3_dir)
h3 <- read.delim(h3_file, header=F, sep="\t")
colnames(h3) <- c("chr", "start", "end", "h3")
setwd(input_dir)
mark <- read.delim(file, header=F, sep="\t")
mark <- mark[,1:4]
colnames(mark) <- c("chr", "start", "end", "mark")

library(dplyr)
data <- left_join(h3, mark, by=c("chr", "start", "end"))
data2 <- ifelse(data$h3 == 0, 0.001, data$h3)
data3 <- ifelse(data$mark == 0, 0.001, data$mark)
data$h3 <- data2
data$mark <- data3
data$log2 <- log2(data$mark/data$h3)
df <- data[,c(1:3,5)]
df$mark <- data$log2
colnames(df) <- NULL
write.table(df, paste(genotype, mark, date, "H3corrected.txt", sep="_"), quote=F, sep="\t", row.names=F)
#write.table(df, "B73_mark_H3corrected.txt", quote=F, sep="\t", row.names=F)
}
```
** Why am I getting so many Inf values ?? --> USE log2 instead of log(2,) **

```{r eval=F}
h3_corrected(
  input_dir = "/home/springer/nosha003/schmitz/atacseq/counts/",
  file = "atacseq_counts_matrix_fpkm_B73.bed",
  h3_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  h3_file = "Chip_B73_leaf_H3_avg.100bpcounts.rpm.txt",
  mark = "ATACseq",
  genotype = "B73",
  date = "20Dec2018"
)
h3_corrected(
  input_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  file = "Chip_B73_leaf_H3K4me3_avg.100bpcounts.rpm.txt",
  h3_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  h3_file = "Chip_B73_leaf_H3_avg.100bpcounts.rpm.txt",
  mark = "H3K4me3",
  genotype = "B73",
  date = "20Dec2018"
)
h3_corrected(
  input_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  file = "Chip_B73_leaf_H3K27me3_avg.100bpcounts.rpm.txt",
  h3_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  h3_file = "Chip_B73_leaf_H3_avg.100bpcounts.rpm.txt",
  mark = "H3K27me3",
  genotype = "B73",
  date = "20Dec2018"
)
h3_corrected(
  input_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  file = "Chip_B73_leaf_H3K23ac_avg.100bpcounts.rpm.txt",
  h3_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  h3_file = "Chip_B73_leaf_H3_avg.100bpcounts.rpm.txt",
  mark = "H3K23ac",
  genotype = "B73",
  date = "20Dec2018"
)
h3_corrected(
  input_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  file = "Chip_B73_leaf_H3K27ac_avg.100bpcounts.rpm.txt",
  h3_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  h3_file = "Chip_B73_leaf_H3_avg.100bpcounts.rpm.txt",
  mark = "H3K27ac",
  genotype = "B73",
  date = "20Dec2018"
)
h3_corrected(
  input_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  file = "Chip_B73_leaf_H3K36me3_avg.100bpcounts.rpm.txt",
  h3_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  h3_file = "Chip_B73_leaf_H3_avg.100bpcounts.rpm.txt",
  mark = "H3K36me3",
  genotype = "B73",
  date = "20Dec2018"
)
h3_corrected(
  input_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  file = "Chip_B73_leaf_H3K4me1_avg.100bpcounts.rpm.txt",
  h3_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  h3_file = "Chip_B73_leaf_H3_avg.100bpcounts.rpm.txt",
  mark = "H3K4me1",
  genotype = "B73",
  date = "20Dec2018"
)
h3_corrected(
  input_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  file = "Chip_B73_leaf_H3K56ac_avg.100bpcounts.rpm.txt",
  h3_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  h3_file = "Chip_B73_leaf_H3_avg.100bpcounts.rpm.txt",
  mark = "H3K56ac",
  genotype = "B73",
  date = "20Dec2018"
)
h3_corrected(
  input_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  file = "Chip_B73_leaf_H3K9ac_avg.100bpcounts.rpm.txt",
  h3_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  h3_file = "Chip_B73_leaf_H3_avg.100bpcounts.rpm.txt",
  mark = "H3K9ac",
  genotype = "B73",
  date = "20Dec2018"
)
h3_corrected(
  input_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  file = "Chip_B73_leaf_Pol_II_avg.100bpcounts.rpm.txt",
  h3_dir = "/home/springer/nosha003/chipseq_schmitz/bins/",
  h3_file = "Chip_B73_leaf_H3_avg.100bpcounts.rpm.txt",
  mark = "Pol_II",
  genotype = "B73",
  date = "20Dec2018"
)
```

##### bedtools
- ATACseq, H3K4me1, H3K27me3 and H3K9me2
```{r eval=F}
module load bedtools

## remove contigs and sort gff files
#/home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3
#/home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_ATACseq_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_ATACseq.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_ATACseq_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_ATACseq.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K4me3_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K4me3.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K4me3_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K4me3.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K27me3_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K27me3.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K27me3_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K27me3.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K23ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K23ac.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K23ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K23ac.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K27ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K27ac.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K27ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K27ac.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K36me3_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K36me3.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K36me3_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K36me3.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K4me1_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K4me1.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K4me1_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K4me1.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K56ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K56ac.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K56ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K56ac.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K9ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K9ac.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K9ac_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K9ac.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_Pol_II_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_Pol_II.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_Pol_II_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_Pol_II.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K9me2_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestTEtoBin_H3K9me2.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/B73_H3K9me2_H3corrected.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_H3K9me2.txt
```

#### orientation correction based on methylation data
```{r eval=F}
#20Aug2018
#chromatin_orientation
#orient chromatin data based on CG/CHG methylation patterns
	
#!/bin/bash
#PBS -l walltime=96:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/chromatin_orientation_o
#PBS -e /scratch.global/nosha003/e_o/chromatin_orientation_e
#PBS -N chromatin_orientation
#PBS -V
#PBS -r n

#cd /home/springer/nosha003/scripts/
cd /scratch.global/nosha003/
module load R
R CMD BATCH chromatin_orientation_ltr.R
R CMD BATCH chromatin_orientation_tir.R
wait

# qsub /scratch.global/nosha003/chromatin_orientation.sh
# qsub /home/springer/nosha003/scripts/chromatin_orientation.sh
```

** FUNCTION: chromatin_metaplot_orientation **
```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G
# module load R
# R

chromatin_metaplot_orientation <- function(input_dir, input_dir2, orientation_file, chromatin_file, genotype, chromatin, TEtype, date) {
library(dplyr)
  
#setwd(input_dir)
#orientation <- read.delim(orientation_file, sep="\t", header=T, stringsAsFactors = F)
#chromatin <- read.delim(chromatin_file, sep="\t", header=F, stringsAsFactors = F)

setwd("/home/springer/nosha003/methylation_spreading")
orientation <- read.delim("B73_tir_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T, stringsAsFactors = F)

setwd("/home/springer/nosha003/methylation_spreading")
chromatin <- read.delim("B73_tir_closestTEtoBin_ATACseq.txt", sep="\t", header=F, stringsAsFactors = F)
setwd("/scratch.global/nosha003/methylation_spreading")

colnames(chromatin) <- c("chr", "start", "end", "count", "chr_te", "te_type", "te_type2", "te_start", "te_end", "dot", "strand", "dot2", "te_id", "dist")
chromatin$id <- substr(chromatin$te_id, 4, 24)
orientation$id <- substr(orientation$TE, 4, 24)
orientation_id <- unique(orientation[,c(10,12)])

#setwd(input_dir2)
chromatin_orientation <- left_join(chromatin, orientation_id, by="id")
df <- chromatin_orientation %>% mutate(correction = replace(correction, is.na(correction), 1))
df2 <- df %>% mutate(chromatin_distance = ifelse(correction == 1, dist, ifelse(correction == "NA", dist, ifelse(correction == -1, (dist * -1), "NA"))))
df3 <- df2[,c(1,2,4,8,9,15,17)]

write.table(df3, "B73_tir_ATACseq_orientation_20Dec.txt", quote=FALSE, sep="\t", row.names=F)

#write.table(df3, paste(genotype, "_", TEtype, "_", chromatin, "_orientation_", date, ".txt", sep=""), quote=FALSE, sep="\t", row.names=F)
}
```
```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

## ltr
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K4me3.txt",
  chromatin = "H3K4me3",
  genotype = "B73",
  TEtype = "ltr",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K27me3.txt",
  chromatin = "H3K27me3",
  genotype = "B73",
  TEtype = "ltr",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K23ac.txt",
  chromatin = "H3K23ac",
  genotype = "B73",
  TEtype = "ltr",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K27ac.txt",
  chromatin = "H3K27ac",
  genotype = "B73",
  TEtype = "ltr",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K36me3.txt",
  chromatin = "H3K36me3",
  genotype = "B73",
  TEtype = "ltr",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K4me1.txt",
  chromatin = "H3K4me1",
  genotype = "B73",
  TEtype = "ltr",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K56ac.txt",
  chromatin = "H3K56ac",
  genotype = "B73",
  TEtype = "ltr",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K9ac.txt",
  chromatin = "H3K9ac",
  genotype = "B73",
  TEtype = "ltr",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_ATACseq.txt",
  chromatin = "ATACseq",
  genotype = "B73",
  TEtype = "ltr",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_Pol_II.txt",
  chromatin = "Pol_II",
  genotype = "B73",
  TEtype = "ltr",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K9me2.txt",
  chromatin = "H3K9me2",
  genotype = "B73",
  TEtype = "ltr",
  date="20Dec"
)

## tir
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K4me3.txt",
  chromatin = "H3K4me3",
  genotype = "B73",
  TEtype = "tir",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K27me3.txt",
  chromatin = "H3K27me3",
  genotype = "B73",
  TEtype = "tir",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K23ac.txt",
  chromatin = "H3K23ac",
  genotype = "B73",
  TEtype = "tir",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K27ac.txt",
  chromatin = "H3K27ac",
  genotype = "B73",
  TEtype = "tir",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K36me3.txt",
  chromatin = "H3K36me3",
  genotype = "B73",
  TEtype = "tir",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K4me1.txt",
  chromatin = "H3K4me1",
  genotype = "B73",
  TEtype = "tir",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K56ac.txt",
  chromatin = "H3K56ac",
  genotype = "B73",
  TEtype = "tir",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K9ac.txt",
  chromatin = "H3K9ac",
  genotype = "B73",
  TEtype = "tir",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_ATACseq.txt",
  chromatin = "ATACseq",
  genotype = "B73",
  TEtype = "tir",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_Pol_II.txt",
  chromatin = "Pol_II",
  genotype = "B73",
  TEtype = "tir",
  date="20Dec"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/methylation_spreading",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K9me2.txt",
  chromatin = "H3K9me2",
  genotype = "B73",
  TEtype = "tir",
  date="20Dec"
)
```


#### perl script for metaplot data
```{r eval=F}
#20Aug2018
#metaplot perl scripts for corrected chromatin data
	
#!/bin/bash
#PBS -l walltime=24:00:00,nodes=1:ppn=6,mem=20gb
#PBS -o /scratch.global/nosha003/e_o/chromatin_o
#PBS -e /scratch.global/nosha003/e_o/chromatin_e
#PBS -N chromatin
#PBS -V
#PBS -r n

ID="$(/bin/sed -n ${PBS_ARRAYID}p ${LIST} | cut -f 1)"

module load bedtools

# perl script

## corrected TE orientation based on CG methylation
perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance4.pl -i /scratch.global/nosha003/methylation_spreading/B73_ltr_${ID}_orientation_20Dec.txt -o /scratch.global/nosha003/methylation_spreading/B73_ltr_${ID}_orientation_metaplot.txt

perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance4.pl -i /scratch.global/nosha003/methylation_spreading/B73_tir_${ID}_orientation_20Dec.txt -o /scratch.global/nosha003/methylation_spreading/B73_tir_${ID}_orientation_metaplot.txt

#qsub -t 1-11 -v LIST=/home/springer/nosha003/methylation_spreading/chromatin_marks_name.txt /home/springer/nosha003/methylation_spreading/chromatin_meatplot_orientation.sh
```

#### metaplot
```{r eval=F}
#20Aug2018
#chromatin_metaplot
#generate metaplot for chromatin data oriented based on CG/CHG methylation patterns
	
#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/chromatin_metaplot_o
#PBS -e /scratch.global/nosha003/e_o/chromatin_metaplot_e
#PBS -N chromatin_metaplot
#PBS -V
#PBS -r n

#cd /home/springer/nosha003/scripts/
cd /scratch.global/nosha003/
module load R
R CMD BATCH chromatin_metaplot.R
wait

# qsub /scratch.global/nosha003/chromatin_metaplot.sh
# qsub /home/springer/nosha003/scripts/chromatin_metaplot.sh
```

** FUNCTION = metaplot_mC_SE_realTElength **
```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

metaplot_mC_SE_realTElength <- function(input_dir, file, fam_file, genotype, chromatin, TEtype, date, class_file) {
library(fields)
library(dplyr)
setwd(input_dir)
b_ltr <- read.delim(file, sep="\t", header=T)
b_ltr_fam <- read.delim(fam_file, sep="\t", header=F)
class <- read.delim(class_file, sep="\t", header=T)
colnames(class) <- c("family", "class")

b_ltr$TEfam <- substr(b_ltr$TE, 1, 8)
b_ltr2 <- subset(b_ltr, b_ltr$TEfam %in% b_ltr_fam$V2)

b_ltr_count <- as.numeric(paste(b_ltr2$count))

b_ltr2["count"] <- b_ltr_count

b_ltr3 <- na.omit(b_ltr2)

dist = subset(b_ltr3, b_ltr3$real_dist > -1000 & b_ltr3$real_dist <= 3200)
b_ltr <- unique(dist$TEfam)
output <- data.frame(matrix(NA, nrow=length(b_ltr), ncol=21))
rownames(output) <- b_ltr
colnames(output) <- seq(-950,3150, 200)
for(i in 1:length(b_ltr)) {
	data.sub=subset(dist, dist$TEfam == b_ltr[i])
	stats.test=stats.bin(data.sub$real_distance,data.sub$count,N=21,breaks=seq(-1000,3200, 200))
	output[i,] = stats.test$stats["mean",]
}
write.table(output, paste(genotype, "_", TEtype, "_", chromatin, "_output_directionality_realTEdist_", date, ".txt", sep=""), quote=FALSE, sep="\t")
p.1 <- output
p.1["family"] <- rownames(p.1)

library(reshape2)
library(ggplot2)

chromatin <- left_join(p.1, class, by="family")
mean <- aggregate(chromatin[, 1:21], list(chromatin$class), mean, na.rm=TRUE, na.action=NULL)
data <- t(mean[,2:22])
melt_df = melt(data, id = "row.names")

melt_df$Var2[melt_df$Var2 == 1] <- "A"
melt_df$Var2[melt_df$Var2 == 2] <- "B"
melt_df$Var2[melt_df$Var2 == 3] <- "C"

df_na <- subset(melt_df, melt_df$Var1 > 1000 & melt_df$Var1 < 1200)
values <- subset(melt_df, melt_df$Var1 <= 1000 | melt_df$Var1 >= 1200)
values_na <- na.omit(values)
plot <- rbind(df_na, values_na)

write.table(plot, paste(genotype, "_", TEtype, "_", chromatin, "_directionality_realTEdist_metaplot_", date, ".txt", sep=""), quote=FALSE, sep="\t", row.names=F)

group.colors <- c(A = "cornflowerblue", B = "springgreen3", C = "lightcoral")

pdf(paste(genotype, TEtype, chromatin, date, "class_realdist_metaplot.pdf", sep="_"))
plot1 <- ggplot(plot, aes(x = Var1, y = value, group = Var2, colour = Var2)) + geom_line() + scale_color_manual(values=group.colors) + theme_classic() + geom_vline(xintercept = 150, linetype="dotted", color = "grey") + geom_vline(xintercept = 2050, linetype="dotted", color = "grey")
print(plot1)
dev.off()
}
```

```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

# TIR
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K4me3_orientation_metaplot.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  chromatin = "H3K4me3",
  date="20Dec",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K27me3_orientation_metaplot.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  chromatin = "H3K27me3",
  date="20Dec",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K23ac_orientation_metaplot.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  chromatin = "H3K23ac",
  date="20Dec",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K27ac_orientation_metaplot.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  chromatin = "H3K27ac",
  date="20Dec",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K36me3_orientation_metaplot.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  chromatin = "H3K36me3",
  date="20Dec",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K4me1_orientation_metaplot.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  chromatin = "H3K4me1",
  date="20Dec",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K56ac_orientation_metaplot.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  chromatin = "H3K56ac",
  date="20Dec",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K9ac_orientation_metaplot.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  chromatin = "H3K9ac",
  date="20Dec",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_ATACseq_orientation_metaplot.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  chromatin = "ATACseq",
  date="20Dec",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_Pol_II_orientation_metaplot.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  chromatin = "Pol_II",
  date="20Dec",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K9me2_orientation_metaplot.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  chromatin = "H3K9me2",
  date="20Dec",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)


# LTR
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K4me3_orientation_metaplot.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  chromatin = "H3K4me3",
  date="20Dec",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K27me3_orientation_metaplot.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  chromatin = "H3K27me3",
  date="20Dec",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K23ac_orientation_metaplot.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  chromatin = "H3K23ac",
  date="20Dec",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K27ac_orientation_metaplot.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  chromatin = "H3K27ac",
  date="20Dec",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K36me3_orientation_metaplot.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  chromatin = "H3K36me3",
  date="20Dec",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K4me1_orientation_metaplot.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  chromatin = "H3K4me1",
  date="20Dec",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K56ac_orientation_metaplot.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  chromatin = "H3K56ac",
  date="20Dec",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K9ac_orientation_metaplot.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  chromatin = "H3K9ac",
  date="20Dec",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_ATACseq_orientation_metaplot.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  chromatin = "ATACseq",
  date="20Dec",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_Pol_II_orientation_metaplot.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  chromatin = "Pol_II",
  date="20Dec",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
metaplot_mC_SE_realTElength(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K9me2_orientation_metaplot.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  chromatin = "H3K9me2",
  date="20Dec",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)

```


#### Percent Chromatin
```{r eval=F}
#18Oct2018
#chromatin_metaplot_percent
#generate metaplot for chromatin data oriented based on CG/CHG methylation patterns
	
#!/bin/bash
#PBS -l walltime=24:00:00,nodes=1:ppn=8,mem=60gb
#PBS -o /scratch.global/nosha003/e_o/chromatin_metaplot_o
#PBS -e /scratch.global/nosha003/e_o/chromatin_metaplot_e
#PBS -N chromatin_metaplot
#PBS -V
#PBS -r n

#cd /home/springer/nosha003/scripts/
cd /scratch.global/nosha003/
module load R
R CMD BATCH chromatin_metaplot_percent_tir.R
R CMD BATCH chromatin_metaplot_percent_ltr.R
wait

# qsub /scratch.global/nosha003/chromatin_metaplot_percent.sh
```


** FUNCTION = metaplot_chromatin_PercentBins **
```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

metaplot_chromatin_PercentBins <- function(input_dir, file, fam_file, class_file, genotype, TEtype, date, context) {
library(fields)
library(dplyr)
library(reshape2)
library(ggplot2)
library(Rmisc)

setwd(input_dir)
b_tir <- read.delim(file, sep="\t", header=T)
#b_tir_fam <- read.delim(fam_file, sep="\t", header=F)
b_tir_fam <- read.delim(fam_file, sep="\t", header=T)
class <- read.delim(class_file, sep="\t", header=T)
class$TEfam <- row.names(class)
class <- class[,c(81:82)]
colnames(class) <- c("class", "TEfam")

# setwd("/scratch.global/nosha003/methylation_spreading") 
# b_tir <-read.delim("B73_tir_ATACseq_orientation_metaplot.txt", sep="\t", header=T)
# b_tir_fam <-read.delim("B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",sep="\t", header=T) 
# class <-read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", sep="\t", header=T)
# class$TEfam <- row.names(class) 
# class <- class[,c(81:82)] 
# colnames(class) <-c("class", "TEfam")


b_tir$TEfam <- substr(b_tir$TE, 1, 8)
#b_tir2 <- subset(b_tir, b_tir$TEfam %in% b_tir_fam$V2)
b_tir2 <- subset(b_tir, b_tir$TE %in% b_tir_fam$TE)

b_tir_cg <- as.numeric(paste(b_tir2$count))
b_tir2["cg"] <- b_tir_cg
b_tir2$chromatin <- ifelse(b_tir2$cg >=0.05, 1, 0)
b_tir3 <- left_join(b_tir2, class, by="TEfam")
dist = subset(b_tir3, b_tir3$relative_distance > -2000 & b_tir3$relative_distance <= 3000)

dist_class1 <- subset(dist, dist$class == "A")
dist_class2 <- subset(dist, dist$class == "B")
dist_class3 <- subset(dist, dist$class == "C")

dist_class1_group <-  transform(dist_class1, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class1_sum <- do.call(data.frame,aggregate(dist_class1_group$chromatin~dist_class1_group$group, dist_class1_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class1_sum) <- c("group", "count", "sum")
dist_class1_sum["percent_methylated"] <- dist_class1_sum$sum / dist_class1_sum$count
dist_class1_sum["class"] <- "H"

dist_class2_group <-  transform(dist_class2, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class2_sum <- do.call(data.frame,aggregate(dist_class2_group$chromatin~dist_class2_group$group, dist_class2_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class2_sum) <- c("group", "count", "sum")
dist_class2_sum["percent_methylated"] <- dist_class2_sum$sum / dist_class2_sum$count
dist_class2_sum["class"] <- "M"

dist_class3_group <-  transform(dist_class3, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class3_sum <- do.call(data.frame,aggregate(dist_class3_group$chromatin~dist_class3_group$group, dist_class3_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class3_sum) <- c("group", "count", "sum")
dist_class3_sum["percent_methylated"] <- dist_class3_sum$sum / dist_class3_sum$count
dist_class3_sum["class"] <- "L"

p.1 <- rbind(dist_class1_sum, dist_class2_sum, dist_class3_sum)

## Metaplot of percent bins methylated by class
#group.colors <- c("cornflowerblue", "springgreen3", "lightcoral")
#group.colors <- c("#3972c9ff", "#e7a275ff", "#a43434ff")
group.colors <- c("#3972c9ff", "#a43434ff", "#e7a275ff")

ggplot(p.1, aes(x = group, y = percent_methylated, group = class, colour = class)) + geom_line()

pdf(paste(genotype, TEtype, "class_metaplot", context, date, "PercentMethylated.pdf", sep="_"))
#pdf("B73_tir_class_metaplot_ATACseq_20Dec_PercentMethylated.pdf")
plot1 <- ggplot(p.1, aes(x = group, y = percent_methylated, group = class, colour = class)) + geom_line(size=1) + theme_classic() + scale_color_manual(values=group.colors) + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
#plot1 <- ggplot(p.1, aes(x = group, y = percent_methylated, group = class, colour = class)) + geom_line(size=1) + theme_classic() + scale_color_manual(values=group.colors)
print(plot1)
dev.off()
}
```
```{r, eval=F}
scp nosha003@login.msi.umn.edu:/home/springer/nosha003/methylation_spreading/*class_metaplot*PercentMethylated.png /Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/spreading/.
```

```{r eval=F}

# TIR
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K4me3_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K4me3"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K23ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K23ac"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K27ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K27ac"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K36me3_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K36me3"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K4me1_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K4me1"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K56ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K56ac"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K9ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K9ac"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_ATACseq_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "ATACseq"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_Pol_II_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "Pol_II"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K9me2_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K9me2"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K27me3_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K27me3"
)


# LTR
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K27me3_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K27me3"
)

metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K4me3_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K4me3"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K23ac_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K23ac"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K27ac_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K27ac"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K36me3_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K36me3"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K4me1_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K4me1"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K56ac_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K56ac"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K9ac_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K9ac"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_ATACseq_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "ATACseq"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_Pol_II_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "Pol_II"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K9me2_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K9me2"
)
```


###### subset to TIRs >1kb in length
```{r eval=F}
#Jan2019
#chromatin_metaplot_percent_tir_length
#generate metaplot for chromatin data oriented based on CG/CHG methylation patterns --> TIRs >1kb in length
	
#!/bin/bash
#PBS -l walltime=24:00:00,nodes=1:ppn=8,mem=60gb
#PBS -o /scratch.global/nosha003/e_o/chromatin_metaplot2_o
#PBS -e /scratch.global/nosha003/e_o/chromatin_metaplot2_e
#PBS -N chromatin_metaplot
#PBS -V
#PBS -r n

#cd /home/springer/nosha003/scripts/
cd /scratch.global/nosha003/
module load R
R CMD BATCH chromatin_metaplot_percent_tir_1kb.R
wait

# qsub /home/springer/nosha003/scripts/chromatin_metaplot_percent_tir_length.sh
```

** FUNCTION = metaplot_chromatin_PercentBins_1kb **
```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

metaplot_chromatin_PercentBins_1kb <- function(input_dir, file, fam_file, class_file, genotype, TEtype, date, context) {
library(fields)
library(dplyr)
library(reshape2)
library(ggplot2)
library(Rmisc)

setwd(input_dir)
b_tir <- read.delim(file, sep="\t", header=T)
b_tir_fam <- read.delim(fam_file, sep="\t", header=T)
class <- read.delim(class_file, sep="\t", header=T)
class$TEfam <- row.names(class)
class <- class[,c(81:82)]
colnames(class) <- c("class", "TEfam")

# setwd("/scratch.global/nosha003/methylation_spreading")
# b_tir <-read.delim("B73_tir_H3K56ac_orientation_metaplot.txt", sep="\t", header=T)
# b_tir_fam <-read.delim("B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",sep="\t", header=T)
# class <-read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", sep="\t", header=T)
# class$TEfam <- row.names(class)
# class <- class[,c(81:82)]
# colnames(class) <-c("class", "TEfam")

b_tir_fam$length <- abs(b_tir_fam$end - b_tir_fam$start)
b_tir_fam2 <- subset(b_tir_fam, b_tir_fam$length > 1000)
  
b_tir$TEfam <- substr(b_tir$TE, 1, 8)
#b_tir2 <- subset(b_tir, b_tir$TEfam %in% b_tir_fam$V2)
b_tir2 <- subset(b_tir, b_tir$TE %in% b_tir_fam2$TE)

b_tir_cg <- as.numeric(paste(b_tir2$count))
b_tir2["cg"] <- b_tir_cg
b_tir2$chromatin <- ifelse(b_tir2$cg >=0.05, 1, 0)
b_tir3 <- left_join(b_tir2, class, by="TEfam")
dist = subset(b_tir3, b_tir3$relative_distance > -2000 & b_tir3$relative_distance <= 3000)

dist_class1 <- subset(dist, dist$class == "A")
dist_class2 <- subset(dist, dist$class == "B")
dist_class3 <- subset(dist, dist$class == "C")

dist_class1_group <-  transform(dist_class1, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class1_sum <- do.call(data.frame,aggregate(dist_class1_group$chromatin~dist_class1_group$group, dist_class1_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class1_sum) <- c("group", "count", "sum")
dist_class1_sum["percent_methylated"] <- dist_class1_sum$sum / dist_class1_sum$count
dist_class1_sum["class"] <- "H"

dist_class2_group <-  transform(dist_class2, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class2_sum <- do.call(data.frame,aggregate(dist_class2_group$chromatin~dist_class2_group$group, dist_class2_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class2_sum) <- c("group", "count", "sum")
dist_class2_sum["percent_methylated"] <- dist_class2_sum$sum / dist_class2_sum$count
dist_class2_sum["class"] <- "M"

dist_class3_group <-  transform(dist_class3, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class3_sum <- do.call(data.frame,aggregate(dist_class3_group$chromatin~dist_class3_group$group, dist_class3_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_class3_sum) <- c("group", "count", "sum")
dist_class3_sum["percent_methylated"] <- dist_class3_sum$sum / dist_class3_sum$count
dist_class3_sum["class"] <- "L"

p.1 <- rbind(dist_class1_sum, dist_class2_sum, dist_class3_sum)

## Metaplot of percent bins methylated by class
#group.colors <- c("cornflowerblue", "springgreen3", "lightcoral")
#group.colors <- c("#3972c9ff", "#e7a275ff", "#a43434ff")
group.colors <- c("#3972c9ff", "#a43434ff", "#e7a275ff")

ggplot(p.1, aes(x = group, y = percent_methylated, group = class, colour = class)) + geom_line()

pdf(paste(genotype, TEtype, "class_metaplot", context, date, "PercentMethylated_length1kb.pdf", sep="_"))
#pdf("B73_tir_class_metaplot_H3K56ac_20Dec_PercentMethylated_length1kb.pdf")
plot1 <- ggplot(p.1, aes(x = group, y = percent_methylated, group = class, colour = class)) + geom_line(size=1) + theme_classic() + scale_color_manual(values=group.colors) + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot1)
dev.off()
}
```

```{r eval=F}

# TIR
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K4me3_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K4me3"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K23ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K23ac"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K27ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K27ac"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K36me3_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K36me3"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K4me1_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K4me1"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K56ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K56ac"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K9ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K9ac"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_ATACseq_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "ATACseq"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_Pol_II_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "Pol_II"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K9me2_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K9me2"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K27me3_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K27me3"
)
```







### Chromatin Heatmaps 
```{r eval=F}
#14Jan2019
#chromatin_heatmap
#generate closest TE to bin file for oriented chromatin data for heatmap generation
	
#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/chromatin_heatmap_o
#PBS -e /scratch.global/nosha003/e_o/chromatin_heatmap_e
#PBS -N chromatin_heatmap
#PBS -V
#PBS -r n

SAMPLE="$(/bin/sed -n ${PBS_ARRAYID}p ${LIST} | cut -f 1)"

module load bedtools

module load R
R CMD BATCH /scratch.global/nosha003/methylation_spreading/chromatin_heatmap_tir.R
R CMD BATCH /scratch.global/nosha003/methylation_spreading/chromatin_heatmap_ltr.R
wait

#qsub -t 1-11 -v LIST=/home/springer/nosha003/chipseq_schmitz/chipseq_schmitz.txt /scratch.global/nosha003/chromatin_heatmap.sh
#qsub /scratch.global/nosha003/chromatin_heatmap.sh
```


```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=1,mem=40G

heatmap_cluster <- function(input_dir, file, fam_file, genotype, TEtype, mark, date, class_file) {
library(fields)
library(tidyverse)
library(dplyr)
#setwd(input_dir)
#b_ltr <- read.delim(file, sep="\t", header=T)
#b_ltr_fam <- read.delim(fam_file, sep="\t", header=F)

setwd("/home/springer/nosha003/methylation_spreading/")
b_ltr <- read.delim("B73_tir_closestTEtoBin_H3K27me3.txt", sep="\t", header=F)
b_ltr_fam <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018.txt", sep="\t", header=F)

b_ltr$TEfam <- substr(b_ltr$V13, 4, 11)
b_ltr2 <- subset(b_ltr, b_ltr$TEfam %in% b_ltr_fam$V2)

b_ltr2$length <- b_ltr2$V9 - b_ltr2$V8
normalized <- b_ltr2 %>% mutate(dist_norm = ifelse(V14 == 0, ((V3-V8)/length)*1000, ifelse(V14 > 0, V14+1000, V14)))

#dist = subset(b_ltr2, b_ltr2$V14 > -2000 & b_ltr2$V14 <= 2000)
dist = subset(normalized, normalized$dist_norm > -2000 & normalized$dist_norm <= 3000)

b_ltr_names <- unique(b_ltr2$TEfam)

output <- data.frame(matrix(NA, nrow=length(b_ltr_names), ncol=50))
rownames(output) <- b_ltr_names
colnames(output) <- seq(-1950,2950, 100)
for(i in 1:length(b_ltr_names)){
	data.sub=subset(dist, dist$TEfam == b_ltr_names[i])
	stats.test=stats.bin(data.sub$dist_norm,data.sub$V4,N=40,breaks=seq(-2000,3000, 100))
	output[i,] = stats.test$stats["mean",]
}
p.1 <- output

p.2 <- na.omit(p.1)

# clustering based on B73 order
colnames(p.2) <- c("X.1950.x", "X.1850.x", "X.1750.x", "X.1650.x", "X.1550.x", "X.1450.x", "X.1350.x", "X.1250.x", "X.1150.x", "X.1050.x", "X.950.x", "X.850.x", "X.750.x", "X.650.x", "X.550.x", "X.450.x", "X.350.x", "X.250.x", "X.150.x", "X.50.x", "X50.x", "X150.x", "X250.x", "X350.x", "X450.x", "X550.x", "X650.x", "X750.x", "X850.x", "X950.x", "X1050.x", "X1150.x", "X1250.x", "X1350.x", "X1450.x", "X1550.x", "X1650.x", "X1750.x", "X1850.x", "X1950.x", "X2050.x", "X2150.x", "X2250.x", "X2350.x", "X2450.x", "X2550.x", "X2650.x", "X2750.x", "X2850.x", "X2950.x")
p.2$V1 <- rownames(p.2)

#cluster <- read.delim(class_file, header=T, sep="\t")
cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")

cluster$V1 <- row.names(cluster)
#cluster_id <- cluster[,81:82]
#mydata <- right_join(p.2, cluster_id, by='V1')
#rownames(mydata) <- mydata$V1
#mydata <- mydata[,c(1:50,52)]

kmeans <- (kmeans(p.2[,c(1:50)], 3))
mydata <- data.frame(p.2, kmeans$cluster)
mydata2 <- left_join(mydata, cluster[,81:82], by="V1")


## Heatmap
library(pheatmap)
library(viridis)


mydata3 <- mydata2[!is.infinite(rowSums(mydata2[,1:50])),]

newdata <- mydata3[order(mydata3[,52]),]
annotation <- data.frame(paste("class", newdata[,53]))
colnames(annotation) <- "class"
rownames(annotation) <- rownames(newdata)

#pdf(paste(genotype, TEtype, mark, "heatmap.pdf", sep="_"))
pdf("H3K27me3_tir_heatmap.pdf")
#plot1 <- pheatmap(newdata[,1:50],cluster_cols=F,cluster_rows = F, color=inferno(100), breaks=seq(-2,2,length.out=100), show_rownames=TRUE, show_colnames = TRUE, border_color = FALSE, fontsize = 4, annotation_row = annotation)
plot1 <- pheatmap(newdata[,1:50],cluster_cols=F,cluster_rows = F, color=inferno(100), show_rownames=TRUE, show_colnames = TRUE, border_color = FALSE, fontsize = 4, annotation_row = annotation)
print(plot1)
dev.off()
}

```

```{r eval=F}
# TIR
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_closestTEtoBin_H3K4me3.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  mark = "H3K4me3",
  date="20Dec2018",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_closestTEtoBin_H3K27me3.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  mark = "H3K27me3",
  date="20Dec2018",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_closestTEtoBin_H3K23ac.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  mark = "H3K23ac",
  date="20Dec2018",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_closestTEtoBin_H3K27ac.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  mark = "H3K27ac",
  date="20Dec2018",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_closestTEtoBin_H3K36me3.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  mark = "H3K36me3",
  date="20Dec2018",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_closestTEtoBin_H3K4me1.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  mark = "H3K4me1",
  date="20Dec2018",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_closestTEtoBin_H3K56ac.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  mark = "H3K56ac",
  date="20Dec2018",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_closestTEtoBin_H3K9ac.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  mark = "H3K9ac",
  date="20Dec2018",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_closestTEtoBin_ATACseq.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  mark = "ATACseq",
  date="20Dec2018",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_closestTEtoBin_Pol_II.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  mark = "Pol_II",
  date="20Dec2018",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_closestTEtoBin_H3K9me2.txt",
  fam_file = "B73_tir_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  mark = "H3K9me2",
  date="20Dec2018",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt"
)


# ltr
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_closestTEtoBin_H3K4me3.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  mark = "H3K4me3",
  date="20Dec2018",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_closestTEtoBin_H3K27me3.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  mark = "H3K27me3",
  date="20Dec2018",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_closestTEtoBin_H3K23ac.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  mark = "H3K23ac",
  date="20Dec2018",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_closestTEtoBin_H3K27ac.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  mark = "H3K27ac",
  date="20Dec2018",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_closestTEtoBin_H3K36me3.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  mark = "H3K36me3",
  date="20Dec2018",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_closestTEtoBin_H3K4me1.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  mark = "H3K4me1",
  date="20Dec2018",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_closestTEtoBin_H3K56ac.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  mark = "H3K56ac",
  date="20Dec2018",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_closestTEtoBin_H3K9ac.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  mark = "H3K9ac",
  date="20Dec2018",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_closestTEtoBin_ATACseq.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  mark = "ATACseq",
  date="20Dec2018",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_closestTEtoBin_Pol_II.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  mark = "Pol_II",
  date="20Dec2018",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)
heatmap_cluster(
  input_dir = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_closestTEtoBin_H3K9me2.txt",
  fam_file = "B73_ltr_fam_count20_nonnested_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  mark = "H3K9me2",
  date="20Dec2018",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt"
)

```


#### Lexiang Heatmap Method
- Scripts:
/home/springer/nosha003/scripts/Script_01_concatenate_files.sh
/home/springer/nosha003/scripts/Script_02_run_k_means.R
/home/springer/nosha003/scripts/Script_03_heatmap.R
- The first one is used to concatenate different input files, this is a shell script and you can do it by R.
- The second one is used to run K-means, just a few lines, the key is to have a pre-defined number of clusters.
- The last one is used to plot the heatmap.

```{r, eval=F}
heatmap.2(x,breaks=seq(0,lims[count],numbers[count]),key=TRUE,symkey=FALSE,density.info="none",trace="none",col=colorRampPalette(c("white",colors[count]))(n=100),Colv=FALSE,Rowv=FALSE,cexCol=0.8,cexRow=0.01,main=type)


```





# Figure S5: Attributes and chromatin patterns
- Various attributes were tested to identify distinguishing features of each k-means cluster defined by TE family CG and CHG methylation for LTRs as well as genome-wide patterns using all LTR elements in a family containing >= 20 members.  A-D are formatted with cluster A, B, C, and genome from left to right respectively.  A) Stacked bar plot of the relative superfamily (RLG, RLC, and RLX) frequency within each cluster.  B) Distance to closest gene for each element categorized into 4 categories: within gene (yellow), within 1kb of gene (green), 1-5kb from gene (light blue), and >5kb from gene (dark blue).  C) Frequency of family size based on 4 groups of copy number: 20-50 members, 50-100 members, 100-200 members, and >200 members from light to dark color respectively.  D) Violin plot of disjoined TE length for all members of each cluster.  Chromatin patterns are displayed in E-H where 100bp bins were assigned to a single annotated TE in the B73v4 genome. All TEs were oriented as described in Figure 2. All TEs belonging to the same kmeans cluster (as defined by CG and CHG methylation) were averaged for bins within and 1kb flanking each TE.  Metaplots of the levels of CHH methylation for B73 (solid lines) and W22 (dashed lines) within and flanking LTR elements for cluster A (blue), B (green) and C (pink).  In addition, the enrichment for H3K4me1, H3K27me2, and H3K9me2 based on ChIP-seq for LTR (E-H) elements for the same regions.    

## LTR
### Distance to Gene
```{r eval=F}
module load bedtools
bedtools closest -k 1 -D a -a /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -b /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.32.gene.sort2.gff3  > /home/springer/nosha003/methylation_spreading/B73_ltr_closestGene_20Dec2018.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/methylation_spreading/B73_ltr_closestGene_20Dec2018.txt -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_ltr_closestGene_closestTEtoBin_20Dec2018.txt

cd /home/springer/nosha003/methylation_spreading
awk '{if ($82 == "A") print $1}' b_ltr_meandirection_kmeans3_20Dec2018.txt > b_ltr_meandirection_kmeans3_class1_20Dec.txt
awk '{if ($82 == "B") print $1}' b_ltr_meandirection_kmeans3_20Dec2018.txt > b_ltr_meandirection_kmeans3_class2_20Dec.txt
awk '{if ($82 == "C") print $1}' b_ltr_meandirection_kmeans3_20Dec2018.txt > b_ltr_meandirection_kmeans3_class3_20Dec.txt
```

```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=1,mem=40G

library(fields)
library(ggplot2)
setwd("/home/springer/nosha003/methylation_spreading/")
b_ltr <- read.delim("B73_ltr_closestGene_closestTEtoBin_20Dec2018.txt", sep="\t", header=F)
class1 <- read.delim("b_ltr_meandirection_kmeans3_class1_20Dec.txt", header=F, sep="\t")
class2 <- read.delim("b_ltr_meandirection_kmeans3_class2_20Dec.txt", header=F, sep="\t")
class3 <- read.delim("b_ltr_meandirection_kmeans3_class3_20Dec.txt", header=F, sep="\t")

b_ltr$TE <- substr(b_ltr$V15, 4, 11)
b_ltr_class1 <- subset(b_ltr, b_ltr$TE %in% class1$V1)
b_ltr_class2 <- subset(b_ltr, b_ltr$TE %in% class2$V1)
b_ltr_class3 <- subset(b_ltr, b_ltr$TE %in% class3$V1)

b_ltr_class1["class"] <- "A"
b_ltr_class2["class"] <- "B"
b_ltr_class3["class"] <- "C"

b_ltr_class <- rbind(b_ltr_class1, b_ltr_class2, b_ltr_class3)

group.colors <- c("cornflowerblue", "springgreen3", "lightcoral")

library(dplyr)
library(reshape2)
b_ltr_class_bin <- b_ltr_class %>% mutate(category = ifelse(V24 == 0, "A", ifelse(V24 < 1000, "B", ifelse(V24 < 5000, "C", ifelse(V24 > 5000, "D", "NA")))))
b_ltr_class_bin_table <- with(b_ltr_class_bin, table(class, category))
b_ltr_class_bin_table2 <- b_ltr_class_bin_table / rowSums(b_ltr_class_bin_table)
b_ltr_melt <- melt(b_ltr_class_bin_table2)

# add genome column
setwd("/home/springer/nosha003/methylation_spreading/")
copy <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
copy_20 <- subset(copy, copy$V1 >= 20)
setwd("/home/springer/nosha003/methylation_spreading/")
b_ltr_all <- subset(b_ltr, b_ltr$TE %in% copy_20$V2)
b_ltr_all$class <- "LTR"
b_ltr_bin <- b_ltr_all %>% mutate(category = ifelse(V25 == 0, "A", ifelse(V25 < 1000, "B", ifelse(V25 < 5000, "C", ifelse(V25 > 5000, "D", "NA")))))
b_ltr_bin_table <- with(b_ltr_bin, table(class, category))
b_ltr_bin_table2 <- b_ltr_bin_table / rowSums(b_ltr_bin_table)

b_ltr_all <- rbind(b_ltr_class_bin_table2, b_ltr_bin_table2)
b_melt <- melt(b_ltr_all)

pdf("ltr_class_gene_barplot_genome_kmeans3_Jan2019.pdf")
ggplot(b_melt, aes(x=Var1, y=value, fill=Var2)) + geom_bar(colour="black", stat = "identity") + scale_fill_brewer(palette="YlGnBu") + theme_classic()
dev.off()
```

### Superfamily classification
- generate a stacked barplot of superfamily proportion for each cluster for TIRs and LTRs
```{r eval=F}
setwd("/home/springer/nosha003/methylation_spreading/")
class <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
class$fam <- rownames(class)
class$superfam <- substr(class$fam, 1, 3)
df <- class[,81:83]
colnames(df) <- c("cluster.cluster", "fam", "superfam")
write.table(df, "ltr_kmeans3_te_superfam_20Dec.txt", quote=F, sep="\t")

library(ggplot2)
library(reshape2)
library(RColorBrewer)

df_count <- with(df, table(cluster.cluster, superfam))
df_prop <- df_count/rowSums(df_count)
df_melt <- melt(df_prop)

png("ltr_class_superfamily_barplot_kmeans3_20Dec2018.png")
ggplot(df_melt, aes(x=cluster.cluster, y=value, fill=superfam)) + geom_bar(stat = "identity") + scale_fill_brewer(palette="RdYlBu")
dev.off()

## look at genome wide expected
#df_melt["all"] <- "LTR"
#ggplot(df_melt, aes(x=all, y=value, fill=superfam)) + geom_bar(stat = "identity") + scale_fill_brewer(palette="RdYlBu")

setwd("/home/springer/nosha003/methylation_spreading/")
copy <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
copy_20 <- subset(copy, copy$V1 >= 20)
setwd("/home/springer/nosha003/methylation_spreading/")
df_20 <- subset(df, df$fam %in% copy_20$V2)
df_20$cluster.cluster <- "LTR"
df_20_count <- with(df_20, table(cluster.cluster, superfam))
df_20_prop <- df_20_count/rowSums(df_20_count)
df_20_melt <- melt(df_20_prop)
df <- rbind(df_melt, df_20_melt)
  
pdf("ltr_class_superfamily_barplot_genome_kmeans3_20Dec2018.pdf")
ggplot(df, aes(x=cluster.cluster, y=value, fill=superfam)) + geom_bar(stat = "identity") + scale_fill_brewer(palette="RdYlBu") + theme_classic()
dev.off()
```

### Copy Number Classification
- for each cluster... generate a density plot of copy number
```{r eval=F}
setwd("/home/springer/nosha003/methylation_spreading/")
copy <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
cluster <- read.delim("ltr_kmeans3_te_superfam_20Dec.txt", header=T, sep="\t")
colnames(copy) <- c("count", "fam")

cluster_copy <- merge(cluster, copy, by="fam")

library(ggplot2)
library(reshape2)

### Look at superfamily relatedness to copy number?
library(tidyverse)
copy$V2 <- substr(copy$V2, 1, 3)
copy %>% 
  group_by(V2) %>%
  summarise(mean=mean(V1, na.rm=T))
#   V2     mean
# 1 RLC   629  
# 2 RLG   799  
# 3 RLX    91.6
 
# make observed/expected bar plot based on genome-wide values
setwd("/home/springer/nosha003/methylation_spreading/")
copy <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
cluster <- read.delim("ltr_kmeans3_te_superfam.txt", header=T, sep="\t")
colnames(copy) <- c("count", "fam")

cluster_copy <- merge(cluster, copy, by="fam")

library(dplyr)
library(reshape2)
cluster_copy_20 <- subset(cluster_copy, cluster_copy$count >= 20)
cluster_copy_bin <- cluster_copy_20 %>% mutate(category = ifelse(count >= 20 & count < 50, "A", ifelse(count >= 50 & count < 100, "B", ifelse(count >= 100 & count < 200, "C", ifelse(count >= 200, "D", "NA")))))
cluster_copy_bin_table <- with(cluster_copy_bin, table(cluster.cluster, category))
cluster_copy_bin_table2 <- cluster_copy_bin_table / rowSums(cluster_copy_bin_table)

copy_20 <- subset(copy, copy$count >= 20)
copy_bin <- copy_20 %>% mutate(category = ifelse(count >= 20 & count < 50, "A", ifelse(count >= 50 & count < 100, "B", ifelse(count >= 100 & count < 200, "C", ifelse(count >= 200, "D", "NA")))))
copy_bin$cluster.cluster <- "LTR"
copy_bin_table <- with(copy_bin, table(cluster.cluster, category))
copy_bin_table2 <- copy_bin_table / rowSums(copy_bin_table)

df <- rbind(cluster_copy_bin_table2, copy_bin_table2)
df_melt <- melt(df)

pdf("ltr_class_copynumber_barplot_genome_kmeans3_20Dec2018.pdf")
ggplot(df_melt, aes(x=Var1, y=value, fill=Var2)) + geom_bar(colour="black", stat = "identity") + scale_fill_brewer(palette="BuGn") + theme_classic()
dev.off()
```

### Age
```{r eval=F}
library(fields)
library(ggplot2)
setwd("/home/springer/nosha003/database/B73v4/te/")
age <- read.delim("LTR_ages_1June.txt", sep="\t", header=T)
colnames(age) <- c("tename", "tbl")
setwd("/home/springer/nosha003/methylation_spreading/")
class1 <- read.delim("b_ltr_meandirection_kmeans3_class1_20Dec.txt", header=F, sep="\t")
class2 <- read.delim("b_ltr_meandirection_kmeans3_class2_20Dec.txt", header=F, sep="\t")
class3 <- read.delim("b_ltr_meandirection_kmeans3_class3_20Dec.txt", header=F, sep="\t")

age$fam <- substr(age$tename, 1, 8)
age_class1 <- subset(age, age$fam %in% class1$V1)
age_class2 <- subset(age, age$fam %in% class2$V1)
age_class3 <- subset(age, age$fam %in% class3$V1)

age_class1["class"] <- "A"
age_class2["class"] <- "B"
age_class3["class"] <- "C"

age_class <- rbind(age_class1, age_class2, age_class3)
group.colors <- c("cornflowerblue", "springgreen3", "lightcoral")

# violin plot
pdf("ltr_class_age_violinplot_kmeans3_20Dec.pdf")
ggplot(age_class, aes(x=factor(class), y=abs(NA.2), fill=factor(class))) + geom_violin() + scale_y_continuous(name = "TE age") + scale_fill_manual(values=group.colors) + theme_classic() + xlab("")
dev.off()
```

### Length
```{r eval=F}
library(fields)
library(ggplot2)
setwd("/home/springer/nosha003/database/B73v4/te/")
length <- read.delim("TE_length_disjoined_20Dec2018.txt", sep="\t", header=T)
setwd("/home/springer/nosha003/methylation_spreading/")
class1 <- read.delim("b_ltr_meandirection_kmeans3_class1_20Dec.txt", header=F, sep="\t")
class2 <- read.delim("b_ltr_meandirection_kmeans3_class2_20Dec.txt", header=F, sep="\t")
class3 <- read.delim("b_ltr_meandirection_kmeans3_class3_20Dec.txt", header=F, sep="\t")

length$fam <- substr(length$te, 1, 8)
length_class1 <- subset(length, length["fam"] %in% class1$V1)
length_class2 <- subset(length, length$fam %in% class2$V1)
length_class3 <- subset(length, length$fam %in% class3$V1)

length_class1["class"] <- "A"
length_class2["class"] <- "B"
length_class3["class"] <- "C"

length_class <- rbind(length_class1, length_class2, length_class3)
group.colors <- c("cornflowerblue", "springgreen3", "lightcoral")

library(dplyr)
library(reshape2)

# add genome column
setwd("/home/springer/nosha003/methylation_spreading/")
copy <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
copy_20 <- subset(copy, copy$V1 >= 20)
setwd("/home/springer/nosha003/methylation_spreading/")

length_all <- subset(length, length$fam %in% copy_20$V2)
length_all$class <- "LTR"
df <- rbind(length_class, length_all)
group.colors <- c("cornflowerblue", "springgreen3", "lightcoral", "grey")

pdf("ltr_class_length_violinplot_genome_kmeans3_ylim_20Dec2018.pdf")
ggplot(df, aes(x=factor(class), y=abs(V2), fill=factor(class))) + geom_violin() + scale_y_continuous(name = "TE length") + scale_fill_manual(values=group.colors) + theme_classic() + xlab("") + ylim(0,30000)
dev.off()
```


## TIR
### Distance to Gene
```{r eval=F}
module load bedtools
bedtools closest -k 1 -D a -a /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -b /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.32.gene.sort2.gff3  > /home/springer/nosha003/methylation_spreading/B73_tir_closestGene_20Dec2018.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/methylation_spreading/B73_tir_closestGene_20Dec2018.txt -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/methylation_spreading/B73_tir_closestGene_closestTEtoBin_20Dec2018.txt

cd /home/springer/nosha003/methylation_spreading
awk '{if ($82 == "A") print $1}' b_tir_meandirection_kmeans3_20Dec2018.txt > b_tir_meandirection_kmeans3_class1_20Dec.txt
awk '{if ($82 == "B") print $1}' b_tir_meandirection_kmeans3_20Dec2018.txt > b_tir_meandirection_kmeans3_class2_20Dec.txt
awk '{if ($82 == "C") print $1}' b_tir_meandirection_kmeans3_20Dec2018.txt > b_tir_meandirection_kmeans3_class3_20Dec.txt
```

```{r eval=F}
library(fields)
library(ggplot2)
setwd("/home/springer/nosha003/methylation_spreading/")
b_tir <- read.delim("B73_tir_closestGene_closestTEtoBin_20Dec2018.txt", sep="\t", header=F)
class1 <- read.delim("b_tir_meandirection_kmeans3_class1_20Dec.txt", header=F, sep="\t")
class2 <- read.delim("b_tir_meandirection_kmeans3_class2_20Dec.txt", header=F, sep="\t")
class3 <- read.delim("b_tir_meandirection_kmeans3_class3_20Dec.txt", header=F, sep="\t")

b_tir$V15 <- substr(b_tir$V15, 4, 11)
b_tir_class1 <- subset(b_tir, b_tir$V15 %in% class1$V1)
b_tir_class2 <- subset(b_tir, b_tir$V15 %in% class2$V1)
b_tir_class3 <- subset(b_tir, b_tir$V15 %in% class3$V1)

b_tir_class1["class"] <- "A"
b_tir_class2["class"] <- "B"
b_tir_class3["class"] <- "C"

b_tir_class <- rbind(b_tir_class1, b_tir_class2, b_tir_class3)


library(dplyr)
library(reshape2)
b_tir_class_bin <- b_tir_class %>% mutate(category = ifelse(V24 == 0, "A", ifelse(V24 < 1000, "B", ifelse(V24 < 5000, "C", ifelse(V24 > 5000, "D", "NA")))))
b_tir_class_bin_table <- with(b_tir_class_bin, table(class, category))
b_tir_class_bin_table2 <- b_tir_class_bin_table / rowSums(b_tir_class_bin_table)
b_tir_melt <- melt(b_tir_class_bin_table2)

# add genome column
setwd("/home/springer/nosha003/database/B73v4/te/")
copy <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
copy_20 <- subset(copy, copy$V1 >= 20)
setwd("/home/springer/nosha003/methylation_spreading/")
b_tir_all <- subset(b_tir, b_tir$V15 %in% copy_20$V2)
b_tir_all$class <- "TIR"
b_tir_bin <- b_tir_all %>% mutate(category = ifelse(V25 == 0, "A", ifelse(V25 < 1000, "B", ifelse(V25 < 5000, "C", ifelse(V25 > 5000, "D", "NA")))))
b_tir_bin_table <- with(b_tir_bin, table(class, category))
b_tir_bin_table2 <- b_tir_bin_table / rowSums(b_tir_bin_table)

b_tir_all <- rbind(b_tir_class_bin_table2, b_tir_bin_table2)
b_melt <- melt(b_tir_all)

pdf("tir_class_gene_barplot_genome_kmeans3_Jan2019.pdf")
ggplot(b_melt, aes(x=Var1, y=value, fill=Var2)) + geom_bar(colour="black", stat = "identity") + scale_fill_brewer(palette="YlGnBu") + theme_classic()
dev.off()
```

### Superfamily classification
- generate a stacked barplot of superfamily proportion for each cluster for TIRs and LTRs
```{r eval=F}
setwd("/home/springer/nosha003/methylation_spreading/")
class <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
class$fam <- rownames(class)
class$superfam <- substr(class$fam, 1, 3)
df <- class[,81:83]
colnames(df) <- c("cluster.cluster", "fam", "superfam")
write.table(df, "tir_kmeans3_te_superfam_20Dec.txt", quote=F, sep="\t")

library(ggplot2)
library(reshape2)
library(RColorBrewer)

df_count <- with(df, table(cluster.cluster, superfam))
df_prop <- df_count/rowSums(df_count)
df_melt <- melt(df_prop)

png("tir_class_superfamily_barplot_kmeans3_20Dec2018.png")
ggplot(df_melt, aes(x=cluster.cluster, y=value, fill=superfam)) + geom_bar(stat = "identity") + scale_fill_brewer(palette="RdYlBu")
dev.off()

setwd("/home/springer/nosha003/methylation_spreading/")
copy <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
copy_20 <- subset(copy, copy$V1 >= 20)
setwd("/home/springer/nosha003/methylation_spreading/")
df_20 <- subset(df, df$fam %in% copy_20$V2)
df_20$cluster.cluster <- "TIR"
df_20_count <- with(df_20, table(cluster.cluster, superfam))
df_20_prop <- df_20_count/rowSums(df_20_count)
df_20_melt <- melt(df_20_prop)
df <- rbind(df_melt, df_20_melt)
  
pdf("tir_class_superfamily_barplot_genome_kmeans3_20Dec2018.pdf")
ggplot(df, aes(x=cluster.cluster, y=value, fill=superfam)) + geom_bar(stat = "identity") + scale_fill_brewer(palette="RdYlBu") + theme_classic()
dev.off()
```

### Copy Number Classification
- for each cluster... generate a density plot of copy number
```{r eval=F}
setwd("/home/springer/nosha003/methylation_spreading/")
copy <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
cluster <- read.delim("tir_kmeans3_te_superfam_20Dec.txt", header=T, sep="\t")
colnames(copy) <- c("count", "fam")

cluster_copy <- merge(cluster, copy, by="fam")

library(ggplot2)
library(reshape2)

########### 7 Aug 2018 ########### 
# make observed/expected bar plot based on genome-wide values
setwd("/home/springer/nosha003/methylation_spreading/")
copy <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
cluster <- read.delim("tir_kmeans3_te_superfam_20Dec.txt", header=T, sep="\t")
colnames(copy) <- c("count", "fam")

cluster_copy <- merge(cluster, copy, by="fam")

library(dplyr)
library(reshape2)
cluster_copy_20 <- subset(cluster_copy, cluster_copy$count >= 20)
cluster_copy_bin <- cluster_copy_20 %>% mutate(category = ifelse(count >= 20 & count < 50, "A", ifelse(count >= 50 & count < 100, "B", ifelse(count >= 100 & count < 200, "C", ifelse(count >= 200, "D", "NA")))))
cluster_copy_bin_table <- with(cluster_copy_bin, table(cluster.cluster, category))
cluster_copy_bin_table2 <- cluster_copy_bin_table / rowSums(cluster_copy_bin_table)

copy_20 <- subset(copy, copy$count >= 20)
copy_bin <- copy_20 %>% mutate(category = ifelse(count >= 20 & count < 50, "A", ifelse(count >= 50 & count < 100, "B", ifelse(count >= 100 & count < 200, "C", ifelse(count >= 200, "D", "NA")))))
copy_bin$cluster.cluster <- "TIR"
copy_bin_table <- with(copy_bin, table(cluster.cluster, category))
copy_bin_table2 <- copy_bin_table / rowSums(copy_bin_table)

df <- rbind(cluster_copy_bin_table2, copy_bin_table2)
df_melt <- melt(df)

pdf("tir_class_copynumber_barplot_genome_kmeans3_20Dec2018.pdf")
ggplot(df_melt, aes(x=Var1, y=value, fill=Var2)) + geom_bar(colour="black", stat = "identity") + scale_fill_brewer(palette="BuGn") + theme_classic()
dev.off()
```

### Length
```{r eval=F}
library(fields)
library(ggplot2)
setwd("/home/springer/nosha003/database/B73v4/te/")
length <- read.delim("TE_length_disjoined_20Dec2018.txt", sep="\t", header=F)
setwd("/home/springer/nosha003/methylation_spreading/")
class1 <- read.delim("b_tir_meandirection_kmeans3_class1_20Dec.txt", header=F, sep="\t")
class2 <- read.delim("b_tir_meandirection_kmeans3_class2_20Dec.txt", header=F, sep="\t")
class3 <- read.delim("b_tir_meandirection_kmeans3_class3_20Dec.txt", header=F, sep="\t")

length$fam <- substr(length$V1, 1, 8)
length_class1 <- subset(length, length$fam %in% class1$V1)
length_class2 <- subset(length, length$fam %in% class2$V1)
length_class3 <- subset(length, length$fam %in% class3$V1)

length_class1["class"] <- "A"
length_class2["class"] <- "B"
length_class3["class"] <- "C"

length_class <- rbind(length_class1, length_class2, length_class3)
group.colors <- c("cornflowerblue", "springgreen3", "lightcoral")

# boxplot
png("tir_class_length_boxplot_kmeans3_20Dec.png")
ggplot(length_class, aes(x=factor(class), y=abs(V2), fill=factor(class))) + geom_boxplot() + scale_y_continuous(name = "TE length") + scale_fill_manual(values=group.colors) + theme_classic() + xlab("")
dev.off()

# violin plot - zoomed in 
png("tir_class_length_violinplot_kmeans3_ylim_20Dec2018.png")
ggplot(length_class, aes(x=factor(class), y=abs(V2), fill=factor(class))) + geom_violin() + scale_y_continuous(name = "TE length") + scale_fill_manual(values=group.colors) + theme_classic() + xlab("") + ylim(0,2000)
dev.off()

library(dplyr)
library(reshape2)

# add genome column
setwd("/home/springer/nosha003/database/B73v4/te/")
copy <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
copy_20 <- subset(copy, copy$V1 >= 20)
setwd("/home/springer/nosha003/methylation_spreading/")

length_all <- subset(length, length$fam %in% copy_20$V2)
length_all$class <- "TIR"
df <- rbind(length_class, length_all)
group.colors <- c("cornflowerblue", "springgreen3", "lightcoral", "grey")

pdf("tir_class_length_violinplot_genome_kmeans3_ylim_20Dec2018.pdf")
ggplot(df, aes(x=factor(class), y=abs(V2), fill=factor(class))) + geom_violin() + scale_y_continuous(name = "TE length") + scale_fill_manual(values=group.colors) + theme_classic() + xlab("") + ylim(0,2000)
dev.off()
```




# Figure S6: TE expression
```{r eval=F}
### heatmap for each family... mean TE expression of each tissue
library(Rmisc)
library(RColorBrewer)
library(gridExtra)
library(fields)
library(pheatmap)
library(viridis)
library(dplyr)
library(reshape2)
library(ggplot2)

setwd("/home/springer/nosha003/database/B73v4/te")
#te_expression <- read.delim("family_rpm_combined_devatlas_3May17.txt", header=T, sep=" ")
te_expression <- read.delim("Walley_expresslion_18Jan19.txt", header=T, sep="\t")
te_expression$fam <- rownames(te_expression)
setwd("/home/springer/nosha003/methylation_spreading/")
ltr_cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
tir_cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster <- rbind(ltr_cluster, tir_cluster)
cluster$fam <- rownames(cluster)
cluster <- cluster[,81:82]
colnames(cluster) <- c("cluster", "fam")

df <- left_join(te_expression, cluster, by="fam")
df_melt <- melt(df)
df_family <- df_melt %>% group_by(fam, variable) %>% summarise_at(vars(value), funs(mean(., na.rm=TRUE)))
df_family_dcast <- dcast(df_family, fam ~ variable)
df_family_cluster <- left_join(df_family_dcast, cluster, by="fam")
rownames(df_family_cluster) <- df_family_cluster$fam
df_family_cluster_na <- na.omit(df_family_cluster)
df_family_cluster_na$class <- substr(df_family_cluster_na$fam, 1, 2)
df_family_cluster_na_ltr <- subset(df_family_cluster_na, df_family_cluster_na$class == "RL")
df_family_cluster_na_tir <- subset(df_family_cluster_na, df_family_cluster_na$class == "DT")

newdata_ltr <- df_family_cluster_na_ltr[order(df_family_cluster_na_ltr$cluster),] 
annotation_ltr <- as.data.frame(paste("cluster", newdata_ltr$cluster))
colnames(annotation_ltr) <- "cluster"
rownames(annotation_ltr) <- rownames(newdata_ltr)
newdata_tir <- df_family_cluster_na_tir[order(df_family_cluster_na_tir$cluster),] 
annotation_tir <- as.data.frame(paste("cluster", newdata_tir$cluster))
colnames(annotation_tir) <- "cluster"
rownames(annotation_tir) <- rownames(newdata_tir)

setwd("/home/springer/nosha003/methylation_spreading")
pdf("ltr_te_expr_heatmap_20Dec2018.pdf")
pheatmap(newdata_ltr[,2:69],cluster_cols=F,cluster_rows = F,show_rownames=TRUE, show_colnames = TRUE, color=inferno(100), breaks=seq(0,100), border_color = FALSE, fontsize = 5, annotation_row = annotation_ltr)
dev.off()
pdf("ltr_te_expr_heatmap_20_20Dec2018.pdf")
pheatmap(newdata_ltr[,2:69],cluster_cols=F,cluster_rows = F,show_rownames=TRUE, show_colnames = TRUE, color=inferno(20), breaks=seq(0,20), border_color = FALSE, fontsize = 5, annotation_row = annotation_ltr)
dev.off()

pdf("tir_te_expr_heatmap_20Dec2018.pdf")
pheatmap(newdata_tir[,2:69],cluster_cols=F,cluster_rows = F,show_rownames=TRUE, show_colnames = TRUE, color=inferno(100), breaks=seq(0,100), border_color = FALSE, fontsize = 5, annotation_row = annotation_tir)
dev.off()
pdf("tir_te_expr_heatmap_20_20Dec2018.pdf")
pheatmap(newdata_tir[,2:69],cluster_cols=F,cluster_rows = F,show_rownames=TRUE, show_colnames = TRUE, color=inferno(20), breaks=seq(0,20), border_color = FALSE, fontsize = 5, annotation_row = annotation_tir)
dev.off()




# count number of tissues expressed
df_n <- as.data.frame(rowSums(df[,c(1:68)] > 1))
colnames(df_n) <- "n_tissues_expressed"
df_n_data <- cbind(df, df_n)

data_bin <- df_n_data %>% mutate(n_tissue_bin = ifelse(n_tissues_expressed == 0, "A.silent", ifelse(n_tissues_expressed > 0 & n_tissues_expressed < 5, "B.tissue_specific", ifelse(n_tissues_expressed >= 5 & n_tissues_expressed < 50, "C.intermediate_1", ifelse(n_tissues_expressed > 50 & n_tissues_expressed <= 100, "D.intermediate_2", ifelse(n_tissues_expressed > 100 & n_tissues_expressed <= 200, "E.intermediate_3", ifelse(n_tissues_expressed > 200, "F.constitutive", "NA")))))))
data_bin$class <- substr(data_bin$fam, 1, 2)
data_bin_ltr <- subset(data_bin, data_bin$class == "RL")
data_bin_tir <- subset(data_bin, data_bin$class == "DT")

dist <- with(data_bin_ltr, table(cluster, n_tissue_bin))
dist2 <- dist / rowSums(dist)
dist_melt <- melt(dist2)
dist_melt_na <- subset(dist_melt, dist_melt$n_tissue_bin != "NA")
pdf("ltr_expression_tissue_bin_cluster_20Dec2018.pdf")
ggplot(dist_melt_na, aes(x=cluster, y=value, fill=n_tissue_bin)) + geom_bar(colour="black", stat = "identity") + theme_classic()
dev.off()

dist <- with(data_bin_tir, table(cluster, n_tissue_bin))
dist2 <- dist / rowSums(dist)
dist_melt <- melt(dist2)
dist_melt_na <- subset(dist_melt, dist_melt$n_tissue_bin != "NA")
pdf("tir_expression_tissue_bin_cluster_20Dec2018.pdf")
ggplot(dist_melt_na, aes(x=cluster, y=value, fill=n_tissue_bin)) + geom_bar(colour="black", stat = "identity") + theme_classic()
dev.off()

dist <- with(data_bin, table(class, n_tissue_bin))
dist2 <- dist / rowSums(dist)
dist_melt <- melt(dist2)
dist_melt_na <- subset(dist_melt, dist_melt$n_tissue_bin != "NA")
pdf("te_expression_tissue_bin_cluster_all_20Dec2019.pdf")
ggplot(dist_melt_na, aes(x=class, y=value, fill=n_tissue_bin)) + geom_bar(colour="black", stat = "identity") + theme_classic()
dev.off()
```


#### Percent of TE families that are silent, tissue specific, constitutive, intermediate
```{r eval=F}
setwd("/home/springer/nosha003/database/B73v4/te")
te_expression <- read.delim("Walley_expresslion_18Jan19.txt", header=T, sep="\t")
te_expression$fam <- rownames(te_expression)
setwd("/home/springer/nosha003/methylation_spreading/")
ltr_cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
tir_cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster <- rbind(ltr_cluster, tir_cluster)
cluster$fam <- rownames(cluster)
cluster <- cluster[,81:82]
colnames(cluster) <- c("cluster", "fam")

library(dplyr)
library(reshape2)
df <- left_join(te_expression, cluster, by="fam")
df_melt <- melt(df)
df_family <- df_melt %>% group_by(fam, variable) %>% summarise_at(vars(value), funs(mean(., na.rm=TRUE)))
df_family_dcast <- dcast(df_family, fam ~ variable)
df_family_cluster <- left_join(df_family_dcast, cluster, by="fam")
rownames(df_family_cluster) <- df_family_cluster$fam
df_family_cluster_na <- na.omit(df_family_cluster)
df_family_cluster_na$class <- substr(df_family_cluster_na$fam, 1, 2)

# 68 tissues
## silent = 0, tissue specific = 14 (<5%), constitutive = 265 (>90%), intermediate

df_n <- as.data.frame(rowSums(df_family_cluster_na[,c(1:69)] > 1))
colnames(df_n) <- "n_tissues_expressed"
df_n_data <- cbind(df_family_cluster_na[,c(1,70:71)], df_n)
df_n_classify <- df_n_data %>% mutate(classification = ifelse(n_tissues_expressed == 0, "silent", ifelse(n_tissues_expressed < 14, "tissue_specific", ifelse(n_tissues_expressed > 265, "constitutive", "intermediate"))))
df_n_count <- df_n_classify %>% group_by(class, cluster, classification) %>% summarise(count = n())
df_n_total <- df_n_count %>% group_by(class, cluster) %>% mutate(total = sum(count))
df_n_prop <- df_n_total %>% mutate(prop = count / total)

library(ggplot2)
setwd("/scratch.global/nosha003/methylation_spreading")
pdf("te_expressed_prop_20Dec2018.pdf")
ggplot(df_n_prop) + geom_bar(aes(x=cluster, y=prop, fill=classification), stat="identity") + theme_classic() + facet_grid(. ~ class)
dev.off()
```


#### % of families with detectable expression
```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

setwd("/home/springer/nosha003/database/B73v4/te")
te_expression <- read.delim("Walley_expresslion_18Jan19.txt", header=T, sep="\t")
te_expression$fam <- rownames(te_expression)

setwd("/home/springer/nosha003/methylation_spreading/")
ltr_cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
tir_cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster <- rbind(ltr_cluster, tir_cluster)
cluster$fam <- rownames(cluster)
cluster <- cluster[,81:82]
colnames(cluster) <- c("cluster", "fam")

library(dplyr)
library(reshape2)

te_melt <- melt(te_expression)
te_melt_avg <- te_melt %>% group_by(fam, variable) %>% mutate(avg = mean(value)) %>% group_by(fam, variable)
te_avg <- data.frame(unique(te_melt_avg[,c(1,4)]))

te_expressed <- te_avg %>% mutate(type = ifelse(avg < 1, "silent", ifelse(avg >= 1, "expressed", "NA")))
te_expressed$order <- substr(te_expressed$fam, 1, 2)
te_expressed_order <- subset(te_expressed, te_expressed$order == "DT" | te_expressed$order == "RL")
te_expressed_cluster <- left_join(te_expressed_order, cluster, by="fam")

d <- subset(te_expressed_cluster, te_expressed_cluster$type == "expressed")
d_uniq <- unique(d[,c(1,4,5)])
# 3052
# order   A   B   C
#    DT  70  45 124
#    RL  64  29  22

te_cluster_count <- te_expressed_cluster %>% group_by(order, cluster, type) %>% summarise(count = n())
te_cluster_total <- te_cluster_count %>% group_by(order, cluster) %>% mutate(total = sum(count)) 
te_cluster_prop <- te_cluster_total %>% group_by(order, cluster) %>% mutate(prop = count / total)
te_cluster_detectable <- subset(te_cluster_prop, te_cluster_prop$type == "expressed")


library(ggplot2)
setwd("/scratch.global/nosha003/methylation_spreading")
pdf("te_prop_detectable_expression_cluster_20Dec2018.pdf")
ggplot(te_cluster_detectable) + geom_bar(aes(x=cluster, y=prop, fill=cluster), stat="identity") + theme_classic() + facet_grid(order ~ .)
dev.off()


te_count <- te_expressed_order %>% group_by(order, type) %>% summarise(count = n())
te_total <- te_count %>% group_by(order) %>% mutate(total = sum(count))
te_prop <- te_total %>% group_by(order) %>% mutate(prop = count / total)
te_detectable <- subset(te_prop, te_prop$type == "expressed")
te_detectable$cluster <- "all"
te_all <- rbind(te_cluster_detectable, te_detectable)

pdf("te_prop_detectable_expression_20Dec2018.pdf")
ggplot(te_all) + geom_bar(aes(x=factor(cluster, levels=c("all", "H", "M", "L", "NA")), y=prop, fill=cluster), stat="identity") + theme_classic() + facet_grid(order ~ .)
dev.off()
```


#### Heatmap of expressed TEs
** separate out LTRs and TIRs --> also use log2 values **
```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

setwd("/home/springer/nosha003/database/B73v4/te")
te_expression <- read.delim("Walley_expresslion_18Jan19.txt", header=T, sep="\t")
te_expression$fam <- rownames(te_expression)

setwd("/home/springer/nosha003/methylation_spreading/")
ltr_cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
tir_cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster <- rbind(ltr_cluster, tir_cluster)
cluster$fam <- rownames(cluster)
cluster <- cluster[,81:82]
colnames(cluster) <- c("cluster", "fam")

library(dplyr)
library(reshape2)
library(cluster)
library(pheatmap)

te_melt <- melt(te_expression)
te_melt_avg <- te_melt %>% group_by(fam) %>% mutate(avg = mean(value))

te_expressed <- te_melt_avg %>% mutate(type = ifelse(avg < 1, "silent", ifelse(avg >= 1, "expressed", "NA")))
te_expressed2 <- subset(te_expressed, te_expressed$type == "expressed")
te_expressed_order_corrected <- te_expressed2 %>% mutate(corrected_value = value/avg)
te_values <- dcast(te_expressed_order_corrected[,c(1,2,6)], fam ~ variable)
te_values_na <- na.omit(te_values)

te_values_cluster <- left_join(te_values_na, cluster, by="fam")
rownames(te_values_cluster) <- te_values_cluster$fam
te_values_A <- na.omit(subset(te_values_cluster, te_values_cluster$cluster == "A"))
te_values_B <- subset(te_values_cluster, te_values_cluster$cluster == "B")
te_values_C <- subset(te_values_cluster, te_values_cluster$cluster == "C")

te_values_A_clust <- hclust(dist(te_values_A[,2:69]))
A <- data.frame(te_values_A_clust$labels[c(te_values_A_clust$order)])
colnames(A) <- "fam"
A$cluster <- "A"
te_values_B_clust <- hclust(dist(te_values_B[,2:69]))
B <- data.frame(te_values_B_clust$labels[c(te_values_B_clust$order)])
colnames(B) <- "fam"
B$cluster <- "B"
te_values_C_clust <- hclust(dist(te_values_C[,2:69]))
C <- data.frame(te_values_C_clust$labels[c(te_values_C_clust$order)])
colnames(C) <- "fam"
C$cluster <- "C"
abc <- rbind(A, B, C)

df <- left_join(abc, te_values_cluster, by="fam")
df_na <- na.omit(df)
df_heatmap <- df[,3:70]
df_heatmap[df_heatmap > 10] <- 10
annotate <- data.frame(df_na[,2])
rownames(annotate) <- df_na[,1]
rownames(df_heatmap) <- df_na[,1]

setwd("/scratch.global/nosha003/methylation_spreading/")
pdf("te_expression_normalized_heatmap_20Dec2018.pdf")
pheatmap(df_heatmap, cluster_cols=T, cluster_rows=F, show_rownames=F, show_colnames=T, border_color=F, fontsize=5, na.rm=T, annotation_row=annotate)
dev.off()




abc$order <- substr(abc$fam, 1, 2)
ltr <- subset(abc, abc$order == "RL")
ltr_abc <- ltr[,1:2]
tir <- subset(abc, abc$order == "DT")
tir_abc <- tir[,1:2]

df_ltr <- left_join(ltr_abc, te_values_cluster, by="fam")
df_ltr_na <- na.omit(df_ltr)
df_ltr_heatmap <- df_ltr_na[,3:70]
df_ltr_heatmap[df_ltr_heatmap > 10] <- 10
ltr_annotate <- data.frame(df_ltr_na[,2])
rownames(ltr_annotate) <- df_ltr_na[,1]
rownames(df_ltr_heatmap) <- df_ltr_na[,1]
df_ltr_heatmap[df_ltr_heatmap == 0] <- 0.0001
df_ltr_log2 <- log2(df_ltr_heatmap)

df_tir <- left_join(tir_abc, te_values_cluster, by="fam")
df_tir_na <- na.omit(df_tir)
df_tir_heatmap <- df_tir_na[,3:70]
df_tir_heatmap[df_tir_heatmap > 10] <- 10
tir_annotate <- data.frame(df_tir_na[,2])
rownames(tir_annotate) <- df_tir_na[,1]
rownames(df_tir_heatmap) <- df_tir_na[,1]
df_tir_heatmap[df_tir_heatmap == 0] <- 0.0001
df_tir_log2 <- log2(df_tir_heatmap)

setwd("/scratch.global/nosha003/methylation_spreading/")
pdf("ltr_expression_normalized_heatmap_20Dec2018.pdf")
pheatmap(df_ltr_log2, cluster_cols=T, cluster_rows=F, show_rownames=F, show_colnames=T, border_color=F, fontsize=2, na.rm=T, annotation_row=ltr_annotate)
dev.off()
pdf("tir_expression_normalized_heatmap_20Dec2018.pdf")
pheatmap(df_tir_log2, cluster_cols=T, cluster_rows=F, show_rownames=F, show_colnames=T, border_color=F, fontsize=2, na.rm=T, annotation_row=tir_annotate)
dev.off()
```


** 29 January 2019 **
```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

setwd("/home/springer/nosha003/database/B73v4/te")
te_expression <- read.delim("Walley_expresslion_18Jan19.txt", header=T, sep="\t")
te_expression$fam <- rownames(te_expression)

setwd("/home/springer/nosha003/methylation_spreading/")
ltr_cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
tir_cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster <- rbind(ltr_cluster, tir_cluster)
cluster$fam <- rownames(cluster)
cluster <- cluster[,81:82]
colnames(cluster) <- c("cluster", "fam")

library(dplyr)
library(reshape2)
library(cluster)
library(pheatmap)

te_melt <- melt(te_expression)
te_melt_avg <- te_melt %>% group_by(fam, variable) %>% mutate(fam_avg = mean(value))

te_expressed <- te_melt_avg %>% mutate(type = ifelse(fam_avg < 1, "silent", ifelse(fam_avg >= 1, "expressed", "NA")))

te_expressed2 <- te_expressed %>% group_by(fam) %>% filter(any(type == "expressed"))
#te_expressed2 <- subset(te_expressed, te_expressed$type == "expressed")
te_values <- dcast(te_expressed2[,c(1,2,4)], fam ~ variable)

te_values_cluster <- left_join(te_values, cluster, by="fam")
rownames(te_values_cluster) <- te_values_cluster$fam
te_values_A <- na.omit(subset(te_values_cluster, te_values_cluster$cluster == "A"))
te_values_B <- subset(te_values_cluster, te_values_cluster$cluster == "B")
te_values_C <- subset(te_values_cluster, te_values_cluster$cluster == "C")

te_values_A_clust <- hclust(dist(te_values_A[,2:69]))
A <- data.frame(te_values_A_clust$labels[c(te_values_A_clust$order)])
colnames(A) <- "fam"
A$cluster <- "A"
te_values_B_clust <- hclust(dist(te_values_B[,2:69]))
B <- data.frame(te_values_B_clust$labels[c(te_values_B_clust$order)])
colnames(B) <- "fam"
B$cluster <- "B"
te_values_C_clust <- hclust(dist(te_values_C[,2:69]))
C <- data.frame(te_values_C_clust$labels[c(te_values_C_clust$order)])
colnames(C) <- "fam"
C$cluster <- "C"
abc <- rbind(A, B, C)

abc$order <- substr(abc$fam, 1, 2)
ltr <- subset(abc, abc$order == "RL")
ltr_abc <- ltr[,1:2]
tir <- subset(abc, abc$order == "DT")
tir_abc <- tir[,1:2]

df_ltr <- left_join(ltr_abc, te_values_cluster, by="fam")
df_ltr_na <- na.omit(df_ltr)
df_ltr_heatmap <- df_ltr_na[,3:70]
ltr_annotate <- data.frame(df_ltr_na[,2])
rownames(ltr_annotate) <- df_ltr_na[,1]
rownames(df_ltr_heatmap) <- df_ltr_na[,1]
df_ltr_log2 <- log2(1 + df_ltr_heatmap)

df_tir <- left_join(tir_abc, te_values_cluster, by="fam")
df_tir_na <- na.omit(df_tir)
df_tir_heatmap <- df_tir_na[,3:70]
tir_annotate <- data.frame(df_tir_na[,2])
rownames(tir_annotate) <- df_tir_na[,1]
rownames(df_tir_heatmap) <- df_tir_na[,1]
df_tir_log2 <- log2(1 + df_tir_heatmap)

setwd("/scratch.global/nosha003/methylation_spreading/")
pdf("ltr_expression_log2_heatmap_jan2019.pdf")
pheatmap(df_ltr_log2, cluster_cols=T, cluster_rows=F, show_rownames=F, show_colnames=T, border_color=F, fontsize=3, na.rm=T, annotation_row=ltr_annotate)
dev.off()
pdf("tir_expression_log2_heatmap_jan2019.pdf")
pheatmap(df_tir_log2, cluster_cols=T, cluster_rows=F, show_rownames=F, show_colnames=T, border_color=F, fontsize=3, na.rm=T, annotation_row=tir_annotate)
dev.off()
```





# Figure 4: Insertion Site Preference
** 13 Nov 2018 **
- realized that the insertion sites I am using are based on the cluster families (not specific TEs) and the all sites category is not filtered at all
    - separate into insertion sites that are nested vs non-nested?


** 3 October 2018 **
- empty site (all data combined) barplot of mC, intermediate, C
    - what % of each family is mC?
        - limit to families that have X number of empty sites
    - when sites are found in multiple genotypes...
        - x% consistently placed in same bin (mC, intermediate, C)
        
- take unmethylated bins...
    - how many have methylation when TE is present? --> spreading
    
    
** Sarah Files updated 10 Sept 2018 **
- You can find the insertion sites in the following directory:
/home/springer/sna/polyTE/sna/
  - The "output1" files will have elements you want encoded as "non.shared" in the "class_group" column and the coordinates in the target genome are in the "W22.coordinates" (or equivalent) column. 
  - The "output2" files will have elements you want encoded as "poly.te.site" with the coordinates in bed format relative to the target genome (W22 or PH207). 
  
  
** January 2019 files **
/home/springer/sna/polyTE/sna/
  - dated 1Jan19 or 3Jan19
  
  
### Bins to TEs
```{r eval=F}
awk  '{gsub(":","\t",$0); print;}' /home/springer/sna/polyTE/sna/B73.to.W22.output2_3Jan19.bed | awk  '{gsub("-","\t",$0); print;}' | awk '{print $7"\t"$8"\t"$9"\t"$10"\t"$4}' | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/B73.to.W22.output2_3Jan19.B73.sort.bed
cut -f 1-4,8 /home/springer/nosha003/sarah/TEpolymorphism/B73.to.W22.output2_3Jan19.B73.sort.bed | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/B73.to.W22.output2_3Jan19.W22.sort.bed

module load bedtools
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/B73.to.W22.output2_3Jan19.B73.sort.bed -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /scratch.global/nosha003/B73.to.W22.output2.B73bins.txt
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/B73.to.W22.output2_3Jan19.W22.sort.bed -a /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /scratch.global/nosha003/B73.to.W22.output2.W22bins.txt


## All genotypes

awk  '{gsub(":","\t",$0); print;}' /home/springer/sna/polyTE/sna/B73.to.PH207.output2_3Jan19.bed | awk  '{gsub("-","\t",$0); print;}' | awk '{print $7"\t"$8"\t"$9"\t"$10"\t"$4}' | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/B73.to.PH207.output2_3Jan19.B73.sort.bed
awk  '{gsub(":","\t",$0); print;}' /home/springer/sna/polyTE/sna/B73.to.Mo17.output2_1Jan19.bed | awk  '{gsub("-","\t",$0); print;}' | awk '{print $7"\t"$8"\t"$9"\t"$10"\t"$4}' | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/B73.to.Mo17.output2_1Jan19.B73.sort.bed
awk  '{gsub(":","\t",$0); print;}' /home/springer/sna/polyTE/sna/W22.to.B73.output2_20Sept18.bed | awk  '{gsub("-","\t",$0); print;}' | awk '{print $7"\t"$8"\t"$9"\t"$10"\t"$4}' | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/W22.to.B73.output2_10Sept18.W22.sort.bed
awk  '{gsub(":","\t",$0); print;}' /home/springer/sna/polyTE/sna/W22.to.PH207.output2_25Sept18.bed | awk  '{gsub("-","\t",$0); print;}' | awk '{print $7"\t"$8"\t"$9"\t"$10"\t"$4}' | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/W22.to.PH207.output2_10Sept18.W22.sort.bed
awk  '{gsub(":","\t",$0); print;}' /home/springer/sna/polyTE/sna/W22.to.Mo17.output2_26Sept18.bed | awk  '{gsub("-","\t",$0); print;}' | awk '{print $7"\t"$8"\t"$9"\t"$10"\t"$4}' | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/W22.to.Mo17.output2_26Sept18.W22.sort.bed
awk  '{gsub(":","\t",$0); print;}' /home/springer/sna/polyTE/sna/PH207.to.B73.output2_18Sept18.bed | awk  '{gsub("-","\t",$0); print;}' | awk '{print $7"\t"$8"\t"$9"\t"$10"\t"$4}' | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.B73.output2_10Sept18.PH207.sort.bed
awk  '{gsub(":","\t",$0); print;}' /home/springer/sna/polyTE/sna/PH207.to.W22.output2_18Sept18.bed | awk  '{gsub("-","\t",$0); print;}' | awk '{print $7"\t"$8"\t"$9"\t"$10"\t"$4}' | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.W22.output2_10Sept18.PH207.sort.bed
awk  '{gsub(":","\t",$0); print;}' /home/springer/sna/polyTE/sna/PH207.to.Mo17.output2_26Sept18.bed | awk  '{gsub("-","\t",$0); print;}' | awk '{print $7"\t"$8"\t"$9"\t"$10"\t"$4}' | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.M017.output2_26Sept18.PH207.sort.bed
awk  '{gsub(":","\t",$0); print;}' /home/springer/sna/polyTE/sna/Mo17.to.W22.output2_3Jan19.bed | awk  '{gsub("-","\t",$0); print;}' | awk '{print $7"\t"$8"\t"$9"\t"$10"\t"$4}' | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.W22.output2_3Jan19.Mo17.sort.bed
awk  '{gsub(":","\t",$0); print;}' /home/springer/sna/polyTE/sna/Mo17.to.PH207.output2_3Jan19.bed | awk  '{gsub("-","\t",$0); print;}' | awk '{print $7"\t"$8"\t"$9"\t"$10"\t"$4}' | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.PH207.output2_3Jan19.Mo17.sort.bed
awk  '{gsub(":","\t",$0); print;}' /home/springer/sna/polyTE/sna/Mo17.to.B73.output2_3Jan19.bed | awk  '{gsub("-","\t",$0); print;}' | awk '{print $7"\t"$8"\t"$9"\t"$10"\t"$4}' | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.B73.output2_3Jan19.Mo17.sort.bed

module load bedtools
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/B73.to.PH207.output2_3Jan19.B73.sort.bed -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /scratch.global/nosha003/B73.to.PH207.output2.B73bins.txt
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/B73.to.Mo17.output2_1Jan19.B73.sort.bed -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /scratch.global/nosha003/B73.to.PH207.output2.B73bins.txt
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/W22.to.B73.output2_10Sept18.W22.sort.bed -a /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /scratch.global/nosha003/W22.to.B73.output2.W22bins.txt
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/W22.to.PH207.output2_10Sept18.W22.sort.bed -a /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /scratch.global/nosha003/W22.to.PH207.output2.W22bins.txt
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/W22.to.Mo17.output2_26Sept18.bed -a /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /scratch.global/nosha003/W22.to.Mo17.output2.W22bins.txt
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.B73.output2_10Sept18.PH207.sort.bed -a /home/springer/nosha003/wgbs_schmitz/tiles_output/PH207_V2_100bp_ratio.txt > /scratch.global/nosha003/PH207.to.B73.output2.PH207bins.txt
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.W22.output2_10Sept18.PH207.sort.bed -a /home/springer/nosha003/wgbs_schmitz/tiles_output/PH207_V2_100bp_ratio.txt > /scratch.global/nosha003/PH207.to.W22.output2.PH207bins.txt
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.Mo17.output2_26Sept18.PH207.sort.bed -a /home/springer/nosha003/wgbs_schmitz/tiles_output/PH207_V2_100bp_ratio.txt > /scratch.global/nosha003/PH207.to.Mo17.output2.PH207bins.txt
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.W22.output2_3Jan19.Mo17.sort.bed -a /home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt > /scratch.global/nosha003/Mo17.to.W22.output2.Mo17bins.txt
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.PH207.output2_3Jan19.Mo17.sort.bed -a /home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt > /scratch.global/nosha003/Mo17.to.PH207.output2.Mo17bins.txt
bedtools closest -k 1 -t first -D b -b /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.B73.output2_3Jan19.Mo17.sort.bed -a /home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt > /scratch.global/nosha003/Mo17.to.B73.output2.Mo17bins.txt
```

### Bins to Sites
```{r, eval=F}
#/home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt
#/home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio.txt
#/home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt
#/home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt

# empty sites in W22, PH207, and Mo17 based on B73 TE annotation
sort -k 1,1 -k 2,2n /home/springer/sna/polyTE/sna/B73.to.W22.output2_3Jan19.bed | grep 'poly.te.site' > /home/springer/nosha003/sarah/TEpolymorphism/B73.to.W22.output2_3Jan19_site_sort.bed
sort -k 1,1 -k 2,2n /home/springer/sna/polyTE/sna/B73.to.PH207.output2_3Jan19.bed | grep 'poly.te.site' > /home/springer/nosha003/sarah/TEpolymorphism/B73.to.PH207.output2_3Jan19_site_sort.bed
sort -k 1,1 -k 2,2n /home/springer/sna/polyTE/sna/B73.to.Mo17.output2_1Jan19.bed | grep 'poly.te.site' > /home/springer/nosha003/sarah/TEpolymorphism/B73.to.Mo17.output2_1Jan19_site_sort.bed

module load bedtools
bedtools closest -k 1 -t first -D a -a /home/springer/nosha003/sarah/TEpolymorphism/B73.to.W22.output2_3Jan19_site_sort.bed -b /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/W22emptysite_methylation.txt
bedtools closest -D a -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/B73.to.W22.output2_3Jan19_site_sort.bed -a /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/W22emptysite_methylation_bins.txt
bedtools closest -k 1 -t first -D a -a /home/springer/nosha003/sarah/TEpolymorphism/B73.to.PH207.output2_3Jan19_site_sort.bed -b /home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/PH207emptysite_methylation.txt
bedtools closest -D a -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/B73.to.PH207.output2_3Jan19_site_sort.bed -a /home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/PH207emptysite_methylation_bins.txt
bedtools closest -k 1 -t first -D a -a /home/springer/nosha003/sarah/TEpolymorphism/B73.to.Mo17.output2_1Jan19_site_sort.bed -b /home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/Mo17emptysite_methylation.txt
bedtools closest -D a -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/B73.to.Mo17.output2_1Jan19_site_sort.bed -a /home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/Mo17emptysite_methylation_bins.txt


# empty sites in B73 based on W22, PH207, and Mo17 TE annotations
sort -k 1,1 -k 2,2n /home/springer/sna/polyTE/sna/W22.to.B73.output2_20Sept18.bed | grep 'poly.te.site' > /home/springer/nosha003/sarah/TEpolymorphism/W22.to.B73.output2_20Sept18_site_sort.bed
sort -k 1,1 -k 2,2n /home/springer/sna/polyTE/sna/PH207.to.B73.output2_18Sept18.bed | grep 'poly.te.site' > /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.B73.output2_18Sept18_site_sort.bed
sort -k 1,1 -k 2,2n /home/springer/sna/polyTE/sna/Mo17.to.B73.output2_3Jan19.bed | grep 'poly.te.site' > /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.B73.output2_3Jan19_site_sort.bed

module load bedtools
bedtools closest -k 1 -t first -D a -a /home/springer/nosha003/sarah/TEpolymorphism/W22.to.B73.output2_20Sept18_site_sort.bed -b /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/W22_B73emptysite_methylation.txt
bedtools closest -D a -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/W22.to.B73.output2_20Sept18_site_sort.bed -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/W22_B73emptysite_methylation_bins.txt
bedtools closest -k 1 -t first -D a -a /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.B73.output2_18Sept18_site_sort.bed -b /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/PH207_B73emptysite_methylation.txt
bedtools closest -D a -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.B73.output2_18Sept18_site_sort.bed -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/PH207_B73emptysite_methylation_bins.txt
bedtools closest -k 1 -t first -D a -a /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.B73.output2_3Jan19_site_sort.bed -b /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/Mo17_B73emptysite_methylation.txt
bedtools closest -D a -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.B73.output2_3Jan19_site_sort.bed -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/Mo17_B73emptysite_methylation_bins.txt


# all other contrasts
## W22
sort -k 1,1 -k 2,2n /home/springer/sna/polyTE/sna/W22.to.PH207.output2_25Sept18.bed | grep 'poly.te.site' > /home/springer/nosha003/sarah/TEpolymorphism/W22.to.PH207.output2_25Sept18_site_sort.bed
sort -k 1,1 -k 2,2n /home/springer/sna/polyTE/sna/W22.to.Mo17.output2_26Sept18.bed | grep 'poly.te.site' > /home/springer/nosha003/sarah/TEpolymorphism/W22.to.Mo17.output2_26Sept18_site_sort.bed
## PH207
sort -k 1,1 -k 2,2n /home/springer/sna/polyTE/sna/PH207.to.W22.output2_18Sept18.bed | grep 'poly.te.site' > /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.W22.output2_18Sept18_site_sort.bed
sort -k 1,1 -k 2,2n /home/springer/sna/polyTE/sna/PH207.to.Mo17.output2_26Sept18.bed | grep 'poly.te.site' > /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.Mo17.output2_26Sept18_site_sort.bed
##Mo17
sort -k 1,1 -k 2,2n /home/springer/sna/polyTE/sna/Mo17.to.W22.output2_3Jan19.bed | grep 'poly.te.site' > /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.W22.output2_3Jan19_site_sort.bed
sort -k 1,1 -k 2,2n /home/springer/sna/polyTE/sna/Mo17.to.PH207.output2_3Jan19.bed | grep 'poly.te.site' > /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.PH207.output2_3Jan19_site_sort.bed

module load bedtools
bedtools closest -k 1 -t first -D a -a /home/springer/nosha003/sarah/TEpolymorphism/W22.to.PH207.output2_25Sept18_site_sort.bed -b /home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/W22_PH207emptysite_methylation.txt
bedtools closest -D a -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/W22.to.PH207.output2_25Sept18_site_sort.bed -a /home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/W22_PH207emptysite_methylation_bins.txt
bedtools closest -k 1 -t first -D a -a /home/springer/nosha003/sarah/TEpolymorphism/W22.to.Mo17.output2_26Sept18_site_sort.bed -b /home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/W22_Mo17emptysite_methylation.txt
bedtools closest -D a -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/W22.to.Mo17.output2_26Sept18_site_sort.bed -a /home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/W22_Mo17emptysite_methylation_bins.txt

bedtools closest -k 1 -t first -D a -a /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.W22.output2_18Sept18_site_sort.bed -b /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/PH207_W22emptysite_methylation.txt
bedtools closest -D a -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.W22.output2_18Sept18_site_sort.bed -a /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/PH207_W22emptysite_methylation_bins.txt
bedtools closest -k 1 -t first -D a -a /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.Mo17.output2_26Sept18_site_sort.bed -b /home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/PH207_Mo17emptysite_methylation.txt
bedtools closest -D a -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/PH207.to.Mo17.output2_26Sept18_site_sort.bed -a /home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/PH207_Mo17emptysite_methylation_bins.txt

bedtools closest -k 1 -t first -D a -a /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.W22.output2_3Jan19_site_sort.bed -b /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/Mo17_W22emptysite_methylation.txt
bedtools closest -D a -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.W22.output2_3Jan19_site_sort.bed -a /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/Mo17_W22emptysite_methylation_bins.txt
bedtools closest -k 1 -t first -D a -a /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.PH207.output2_3Jan19_site_sort.bed -b /home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/Mo17_PH207emptysite_methylation.txt
bedtools closest -D a -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/Mo17.to.PH207.output2_3Jan19_site_sort.bed -a /home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/Mo17_PH207emptysite_methylation_bins.txt

```

### number of site-defined polymorphisms among genotypes
```{r}
B.W <- 7373
B.P <- 6588
B.M <- 7748
W.B <- 6207
W.P <- 5910
W.M <- 6489
P.B <- 2439
P.W <- 2538
P.M <- 2445
M.B <- 7450
M.W <- 7441
M.P <- 6664

df <- data.frame(base_genotype = c("B73", "B73", "B73", "W22", "W22", "W22", "PH207", "PH207", "PH207", "Mo17", "Mo17", "Mo17"), site_genotype = c("W22", "PH207", "Mo17", "B73", "PH207", "Mo17", "B73", "W22", "Mo17", "B73", "W22", "PH207"), genotype_comparison = c("B73.to.W22", "B73.to.PH207", "B73.to.Mo17", "W22.to.B73", "W22.to.PH207", "W22.to.Mo17", "PH207.to.B73", "PH207.to.W22", "PH207.to.Mo17", "Mo17.to.B73", "Mo17.to.W22", "Mo17.to.PH207"), site_defined_polymorphisms = c(B.W, B.P, B.M, W.B, W.P, W.M, P.B, P.W, P.M, M.B, M.W, M.P))

# +B.P+B.M+W.B+W.P+W.M+P.B+P.W+P.M+M.B+M.W+M.P = 69292

library(ggplot2)
setwd("/Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/te/")
pdf("site_defined_polymorphisms.pdf")
ggplot(df) + geom_bar(aes(x=genotype_comparison, y=site_defined_polymorphisms, fill=genotype_comparison), stat="identity") + labs(x="genotype_comparison", y="site_defined_polymorphisms") + scale_fill_brewer(palette="RdYlBu") + theme_classic() 
dev.off()
pdf("site_defined_polymorphisms_stacked.pdf")
ggplot(df) + geom_bar(aes(x=base_genotype, y=site_defined_polymorphisms, fill=site_genotype), stat="identity") + labs(x="base_genotype", y="site_defined_polymorphisms") + scale_fill_brewer(palette="RdYlBu") + theme_classic() 
dev.off()
```


#### how many site-defined polymorphisms are in TEs (nested)?
```{r eval=F}
# B73 TEs with empty site in W22
module load bedtools
bedtools intersect -wo -a /home/springer/nosha003/sarah/TEpolymorphism/B73.to.W22.output2_3Jan19_site_sort.bed -b /home/maize/shared/databases/genomes/Zea_mays/W22/W22.structuralTEv2.2018-09-12.allTE_chr.gff3 > /home/springer/nosha003/sarah/TEpolymorphism/B73.to.W22.output2_10Sept18_site_nested.bed
# 3798 / 7373 = 0.5151227 of W22 empty sites intersect (are nested within) a W22 TE 
```



### combine data
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

setwd("/home/springer/nosha003/sarah/TEpolymorphism/")
w22 <- read.delim("W22emptysite_methylation_bins.txt", header=F, sep="\t")
ph207 <- read.delim("PH207emptysite_methylation_bins.txt", header=F, sep="\t")
mo17 <- read.delim("Mo17emptysite_methylation_bins.txt", header=F, sep="\t")
B73_w <- read.delim("W22_B73emptysite_methylation_bins.txt", header=F, sep="\t")
B73_p <- read.delim("PH207_B73emptysite_methylation_bins.txt", header=F, sep="\t")
B73_m <- read.delim("Mo17_B73emptysite_methylation_bins.txt", header=F, sep="\t")
W22_p <- read.delim("PH207_W22emptysite_methylation_bins.txt", header=F, sep="\t")
W22_m <- read.delim("Mo17_W22emptysite_methylation_bins.txt", header=F, sep="\t")
PH207_w <- read.delim("W22_PH207emptysite_methylation_bins.txt", header=F, sep="\t")
PH207_m <- read.delim("Mo17_PH207emptysite_methylation_bins.txt", header=F, sep="\t")
Mo17_w <- read.delim("W22_Mo17emptysite_methylation_bins.txt", header=F, sep="\t")
Mo17_p <- read.delim("PH207_Mo17emptysite_methylation_bins.txt", header=F, sep="\t")

w22$fam <- substr(w22$V10, 1, 8)
w22$element <- substr(w22$V10, 17, 21)
ph207$fam <- substr(ph207$V10, 1, 8)
ph207$element <- substr(ph207$V10, 17, 21)
mo17$fam <- substr(mo17$V10, 1, 8)
mo17$element <- substr(mo17$V10, 17, 21)
B73_w$fam <- substr(B73_w$V10, 1, 8)
B73_w$element <- substr(B73_w$V10, 17, 21)
B73_p$fam <- substr(B73_p$V10, 1, 8)
B73_p$element <- substr(B73_p$V10, 17, 21)
B73_m$fam <- substr(B73_m$V10, 1, 8)
B73_m$element <- substr(B73_m$V10, 17, 21)
W22_p$fam <- substr(W22_p$V10, 1, 8)
W22_p$element <- substr(W22_p$V10, 17, 21)
W22_m$fam <- substr(W22_m$V10, 1, 8)
W22_m$element <- substr(W22_m$V10, 17, 21)
PH207_w$fam <- substr(PH207_w$V10, 1, 8)
PH207_w$element <- substr(PH207_w$V10, 17, 21)
PH207_m$fam <- substr(PH207_m$V10, 1, 8)
PH207_m$element <- substr(PH207_m$V10, 17, 21)
Mo17_w$fam <- substr(Mo17_w$V10, 1, 8)
Mo17_w$element <- substr(Mo17_w$V10, 17, 21)
Mo17_p$fam <- substr(Mo17_p$V10, 1, 8)
Mo17_p$element <- substr(Mo17_p$V10, 17, 21)

library(dplyr)
# B73_wp <- full_join(B73_w, B73_p, by=c("fam", "element"))
# B73_all <- full_join(B73_wp, B73_m, by=c("fam", "element"))
# W22_mp <- full_join(W22_m, W22_p, by=c("fam", "element"))
# W22_all <- full_join(W22_mp, W22_b, by=c("fam", "element"))
# PH207_mw <- full_join(PH207_m, PH207_w, by=c("fam", "element"))
# PH207_all <- full_join(PH207_mw, PH207_b, by=c("fam", "element"))
# Mo17_wp <- full_join(Mo17_w, Mo17_p, by=c("fam", "element"))
# Mo17_all <- full_join(Mo17_wp, Mo17_b, by=c("fam", "element"))
# 
# BW <- full_join(B73_all, W22_all, by=c("fam", "element"))
# BWP <- full_join(BW, PH207_all, by=c("fam", "element"))
# BWPM <- full_join(BWP, Mo17_all, by=c("fam", "element"))

#B73 <- rbind(B73_w, B73_p, B73_m)
#B73 <- rbind(B73_w, B73_p)
#B73_avg <- B73 %>% group_by(fam, element) %>% mutate(B73_CG_avg = mean(V4), B73_CHG_avg = mean(V5), B73_CHH_avg = mean(V6))
#write.table(B73_avg, "B73_empty_average.txt", quote=F, row.names=F, sep="\t")

BWPM <- rbind(B73_w, B73_p, B73_m, w22, ph207, mo17, W22_p, W22_m, PH207_w, PH207_m, Mo17_w, Mo17_p)
BWPM_avg <- BWPM %>% group_by(fam, element) %>% mutate(CG_avg = mean(V4), CHG_avg = mean(V5), CHH_avg = mean(V6))
write.table(BWPM_avg, "all_empty_average.txt", quote=F, row.names=F, sep="\t")

BWPM$id <- paste(BWPM$fam,"_",BWPM$element, sep="")

BWPM_duplicates <- data.frame(BWPM[duplicated(BWPM[,18])])
write.table(BWPM_duplicates, "all_empty_duplicates.txt", quote=F, row.names=F, sep="\t")
```

```{r eval=F}
#4Oct2018
#merge_empty_sites.sh
#merge empty site files
	
#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/merge_o
#PBS -e /scratch.global/nosha003/e_o/merge_e
#PBS -N merge
#PBS -V
#PBS -r n

cd /home/springer/nosha003/scripts/
module load R
#R CMD BATCH merge_empty_sites.R
#R CMD BATCH sites_spreading.R

R CMD BATCH combine_data_sites.R
wait

# qsub /home/springer/nosha003/scripts/merge_empty_sites.sh
```

### categorize empty site methylation as mC, intermediate, or C
```{r eval=F}
#20Jan2019
#insertion_sites.sh

#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/insertion_sites_o
#PBS -e /scratch.global/nosha003/e_o/insertion_sites_e
#PBS -N insertion_sites
#PBS -V
#PBS -r n

cd /home/springer/nosha003/scripts/
module load R
R CMD BATCH /home/springer/nosha003/scripts/insertion_sites.R
wait

# qsub /home/springer/nosha003/scripts/insertion_sites.sh
```

```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G
# /home/springer/nosha003/scripts/insertion_sites.R

library(reshape2)
library(dplyr)
library(ggplot2)
library(RColorBrewer)

setwd("/home/springer/nosha003/methylation_spreading/")
ltr_cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
tir_cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster <- rbind(ltr_cluster, tir_cluster)
cluster$fam <- row.names(cluster)
cluster <- cluster[,c(81:82)]
colnames(cluster) <- c("cluster", "fam")

setwd("/home/springer/nosha003/sarah/TEpolymorphism")
#w22 <- read.delim("W22emptysite_methylation_bins.txt", header=F, sep="\t")
w22 <- read.delim("all_empty_average.txt", header=T, sep="\t")
#w22 <- df[,c(1:3,18:20,7:15)]
#w22_empty <- subset(w22, w22$V15 == 0)
w22_empty <- subset(w22, w22$V15 != 0 & abs(w22$V15) < 100)
w22_empty_melt <- melt(w22_empty[,c(4:6,10)], id="V10")
colnames(w22_empty_melt) <- c("TE", "context", "mC_ratio")
w22_empty_melt_na <- na.omit(w22_empty_melt)
w22_empty_melt_na$fam <- substr(w22_empty_melt_na$TE, 1, 8)
w22_empty_melt_na$class <- substr(w22_empty_melt_na$TE, 1, 2)
w22_empty_melt_cluster <- left_join(w22_empty_melt_na, cluster, by="fam")
setwd("/scratch.global/nosha003/methylation_spreading")
#write.table(w22_empty_melt_cluster, "w22_empty_melt_cluster.txt", quote=F, row.names=F, sep="\t")
write.table(w22_empty_melt_cluster, "all_empty_melt_cluster_20Dec.txt", quote=F, row.names=F, sep="\t")

# by class and cluster
w22_empty_region_cluster <- w22_empty_melt_cluster %>% group_by(context, class, cluster) %>% summarise(region_count = n()) 
w22_empty_region_cluster_na <- na.omit(w22_empty_region_cluster)
w22_empty_category_cluster <- w22_empty_melt_cluster %>% group_by(context, class, cluster) %>% mutate(mC_state = ifelse(context == "V4", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "V5", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "V6", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(context, class, cluster, mC_state) 
w22_empty_category_cluster_na <- na.omit(w22_empty_category_cluster)
write.table(w22_empty_category_cluster, "all_empty_melt_cluster2_20Dec.txt", quote=F, row.names=F, sep="\t")
w22_empty_category_cluster_count <- w22_empty_category_cluster_na %>% summarise(count = n())

w22_empty_category_cluster_state <- inner_join(w22_empty_region_cluster_na, w22_empty_category_cluster_count, by=c("context", "class", "cluster"))
w22_empty_category_cluster_prop <- w22_empty_category_cluster_state %>% mutate(prop = count/region_count)
w22_empty_category_cluster_prop %>% print(n = Inf)

setwd("/scratch.global/nosha003/methylation_spreading")
#write.table(w22_empty_category_cluster_prop, "w22_empty_category_cluster_prop.txt", row.names=F, quote=F, sep="\t")
write.table(w22_empty_category_cluster_prop, "all_empty_category_cluster_prop_20Dec.txt", row.names=F, quote=F, sep="\t")
#pdf("w22_empty_mC_barplot.pdf")
pdf("bwmp_empty_mC_barplot_20Dec.pdf")
ggplot(data=w22_empty_category_cluster_prop) + geom_bar(aes(x=cluster, y=prop, fill=mC_state), stat="identity") + facet_grid(class ~ context) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_brewer(palette="Set2")
dev.off()

# TIR: A(5), B(128), C(318)
# LTR: A(2141), B(781), C(171) 

## including all, A, B, C, NA
library(reshape2)
library(dplyr)
library(ggplot2)
library(RColorBrewer)

setwd("/home/springer/nosha003/methylation_spreading/")
ltr_cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
tir_cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster <- rbind(ltr_cluster, tir_cluster)
cluster$fam <- row.names(cluster)
cluster <- cluster[,c(81:82)]
colnames(cluster) <- c("cluster", "fam")

# setwd("/home/springer/nosha003/sarah/TEpolymorphism")
#w22 <- read.delim("W22emptysite_methylation_bins.txt", header=F, sep="\t")
# w22 <- read.delim("all_empty_average.txt", header=T, sep="\t", stringsAsFactors = FALSE)
# w22_empty <- subset(w22, w22$V15 != 0 & abs(w22$V15) < 100)
# w22_empty_melt <- melt(w22_empty[,c(4:6,10)], id="V10")
# colnames(w22_empty_melt) <- c("TE", "context", "mC_ratio")
# w22_empty_melt_cg <- subset(w22_empty_melt, w22_empty_melt$context == "V4")
# w22_empty_melt_na <- na.omit(w22_empty_melt_cg)
# w22_empty_melt_na$fam <- substr(w22_empty_melt_na$TE, 1, 8)
# w22_empty_melt_na$class <- substr(w22_empty_melt_na$TE, 1, 2)
# w22_empty_melt_cluster <- left_join(w22_empty_melt_na, cluster, by="fam")

setwd("/scratch.global/nosha003/methylation_spreading")
w22_empty_melt_cluster <- read.delim("all_empty_melt_cluster_20Dec.txt", header=T, sep="\t")
w22_empty_melt_na <- subset(w22_empty_melt_cluster, w22_empty_melt_cluster$mC_ratio != "NA")

w22_empty_region_cluster <- w22_empty_melt_na %>% group_by(context, class, cluster) %>% summarise(region_count = n()) 
w22_empty_category_cluster <- w22_empty_melt_na %>% group_by(context, class, cluster) %>% mutate(mC_state = ifelse(context == "V4", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "V5", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "V6", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(context, class, cluster, mC_state) 
w22_empty_category_cluster_count <- w22_empty_category_cluster %>% summarise(count = n())
w22_empty_category_cluster_state <- inner_join(w22_empty_region_cluster, w22_empty_category_cluster_count, by=c("context", "class", "cluster"))
w22_empty_category_cluster_prop <- w22_empty_category_cluster_state %>% mutate(prop = count/region_count)
w22_empty_category_cluster_prop_RL <- subset(w22_empty_category_cluster_prop, w22_empty_category_cluster_prop$class == "RL")
w22_empty_category_cluster_prop_DT <- subset(w22_empty_category_cluster_prop, w22_empty_category_cluster_prop$class == "DT")

w22_empty_melt <- w22_empty_melt_na
w22_empty_melt$cluster <- "all"
w22_empty_region <- w22_empty_melt %>% group_by(context, class, cluster) %>% summarise(region_count = n()) 
w22_empty_category <- w22_empty_melt %>% group_by(context, class, cluster) %>% mutate(mC_state = ifelse(context == "V4", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "V5", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "V6", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(context, class, cluster, mC_state) 
w22_empty_category_count <- w22_empty_category %>% summarise(count = n())
w22_empty_category_state <- inner_join(w22_empty_region, w22_empty_category_count, by=c("context", "class", "cluster"))
w22_empty_category_prop <- w22_empty_category_state %>% mutate(prop = count/region_count)
w22_empty_category_prop_RL <- subset(w22_empty_category_prop, w22_empty_category_prop$class == "RL")
w22_empty_category_prop_DT <- subset(w22_empty_category_prop, w22_empty_category_prop$class == "DT")

prop <- rbind(w22_empty_category_cluster_prop, w22_empty_category_prop)
#write.table(prop, "w22_empty_category_cluster_prop_clusters.txt", quote=F, row.names=F, sep="\t")
write.table(prop, "all_empty_category_cluster_prop_clusters_20Dec.txt", quote=F, row.names=F, sep="\t")
prop_RL <- rbind(w22_empty_category_cluster_prop_RL, w22_empty_category_prop_RL)
prop_DT <- rbind(w22_empty_category_cluster_prop_DT, w22_empty_category_prop_DT)

setwd("/scratch.global/nosha003/methylation_spreading")
#pdf("w22_empty_mC_barplot_CG_all.pdf")
pdf("all_empty_mC_barplot_CG_all_20Dec.pdf")
ggplot(data=prop) + geom_bar(aes(x=factor(cluster, levels=c("all", "A", "B", "C", "NA")), y=prop, fill=factor(mC_state, levels=c("mC", "intermediate", "C"))), stat="identity") + facet_grid(class ~ context) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()
#pdf("w22_empty_mC_barplot_CG_ltr.pdf")
pdf("all_empty_mC_barplot_CG_ltr_20Dec.pdf")
ggplot(data=prop_RL) + geom_bar(aes(x=factor(cluster, levels=c("all", "A", "B", "C", "NA")), y=prop, fill=factor(mC_state, levels=c("mC", "intermediate", "C"))), stat="identity") + facet_grid(class ~ context) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()
#pdf("w22_empty_mC_barplot_CG_all_TIR.pdf")
pdf("all_empty_mC_barplot_CG_all_TIR_20Dec.pdf")
ggplot(data=prop_DT) + geom_bar(aes(x=factor(cluster, levels=c("all", "A", "B", "C", "NA")), y=prop, fill=factor(mC_state, levels=c("mC", "intermediate", "C"))), stat="identity") + facet_grid(class ~ context) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()


#########################################################
library(reshape2)
library(dplyr)
library(ggplot2)
library(RColorBrewer)

setwd("/home/springer/nosha003/methylation_spreading/")
ltr_cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
tir_cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster <- rbind(ltr_cluster, tir_cluster)
cluster$fam <- row.names(cluster)
cluster <- cluster[,c(81:82)]
colnames(cluster) <- c("cluster", "fam")

setwd("/scratch.global/nosha003/methylation_spreading")
#w22_empty_melt_na <- read.delim("w22_empty_melt_cluster.txt", header=T, sep="\t")
w22_empty_melt_na <- read.delim("all_empty_melt_cluster_20Dec.txt", header=T, sep="\t")

# by family
w22_empty_region_fam <- w22_empty_melt_na %>% group_by(context, fam) %>% summarise(region_count = n()) 
w22_empty_region_fam_na <- na.omit(w22_empty_region_fam)
w22_empty_category_fam <- w22_empty_melt_na %>% group_by(context, fam) %>% mutate(mC_state = ifelse(context == "V4", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "V5", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "V6", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(context, fam, mC_state) 
w22_empty_category_fam_na <- na.omit(w22_empty_category_fam)
w22_empty_category_fam_count <- w22_empty_category_fam_na %>% summarise(count = n())

w22_empty_category_fam_state <- inner_join(w22_empty_region_fam_na, w22_empty_category_fam_count, by=c("context", "fam"))
w22_empty_category_fam_prop <- w22_empty_category_fam_state %>% mutate(prop = count/region_count)
w22_empty_category_fam_prop %>% print(n = Inf)

w22_empty_category_fam_prop$class <- substr(w22_empty_category_fam_prop$fam, 1, 2)
w22_empty_category_fam_prop_cluster <- left_join(w22_empty_category_fam_prop, cluster, by="fam")
w22_empty_category_fam_prop_cluster_C <- subset(w22_empty_category_fam_prop_cluster, w22_empty_category_fam_prop_cluster$mC_state == "C")
w22_empty_category_fam_prop_cluster_C_order <- w22_empty_category_fam_prop_cluster_C[order(w22_empty_category_fam_prop_cluster_C$prop),]
#write.table(w22_empty_category_fam_prop_cluster_C_order, "w22_empty_category_fam_prop_cluster_C_order.txt", row.names=F, quote=F, sep="\t")
write.table(w22_empty_category_fam_prop_cluster_C_order, "all_empty_category_fam_prop_cluster_C_order_20Dec.txt", row.names=F, quote=F, sep="\t")

w22_empty_category_fam_prop_cluster_C_order_label <- w22_empty_category_fam_prop_cluster_C_order %>% group_by(class, cluster) %>% mutate(id = row_number())
#write.table(w22_empty_category_fam_prop_cluster_C_order_label, "w22_empty_category_fam_prop_cluster_C_order_label.txt", quote=F, row.names=F, sep="\t")
write.table(w22_empty_category_fam_prop_cluster_C_order_label, "all_empty_category_fam_prop_cluster_C_order_label_20Dec.txt", quote=F, row.names=F, sep="\t")

#ggplot(w22_empty_category_fam_prop_cluster_C_order) + geom_point(aes(x=id, y=prop, color=cluster)) + facet_grid(. ~ class)

w22_empty_category_fam_prop_cluster_C_order_label_cg <- subset(w22_empty_category_fam_prop_cluster_C_order_label, w22_empty_category_fam_prop_cluster_C_order_label$context == "V4")

tir_a <- subset(w22_empty_category_fam_prop_cluster_C_order_label_cg, w22_empty_category_fam_prop_cluster_C_order_label_cg$class == "DT" & w22_empty_category_fam_prop_cluster_C_order_label_cg$cluster == "A")
tir_b <- subset(w22_empty_category_fam_prop_cluster_C_order_label_cg, w22_empty_category_fam_prop_cluster_C_order_label_cg$class == "DT" & w22_empty_category_fam_prop_cluster_C_order_label_cg$cluster == "B")
tir_c <- subset(w22_empty_category_fam_prop_cluster_C_order_label_cg, w22_empty_category_fam_prop_cluster_C_order_label_cg$class == "DT" & w22_empty_category_fam_prop_cluster_C_order_label_cg$cluster == "C")
#pdf("tir_abc_fam.pdf")
pdf("bwpm_tir_abc_fam_cg_20Dec.pdf")
ggplot() + geom_point(data=tir_a, aes(x=id, y=prop), color="cornflowerblue", size=3) + geom_point(data=tir_b, aes(x=id, y=prop), color="springgreen3", size=3) + geom_point(data=tir_c, aes(x=id, y=prop), color="lightcoral", size=3) + theme_classic() 
dev.off()
#pdf("bwp_tir_abc_fam_connected.pdf")
#ggplot() + geom_point(data=tir_a, aes(x=id, y=prop), color="cornflowerblue", size=3) + geom_point(data=tir_b, aes(x=id, y=prop), color="springgreen3", size=3) + geom_line() + geom_point(data=tir_c, aes(x=id, y=prop), color="lightcoral", size=3) + theme_classic() + facet_grid(context ~ .)
#dev.off()

ltr_a <- subset(w22_empty_category_fam_prop_cluster_C_order_label_cg, w22_empty_category_fam_prop_cluster_C_order_label_cg$class == "RL" & w22_empty_category_fam_prop_cluster_C_order_label_cg$cluster == "A")
ltr_b <- subset(w22_empty_category_fam_prop_cluster_C_order_label_cg, w22_empty_category_fam_prop_cluster_C_order_label_cg$class == "RL" & w22_empty_category_fam_prop_cluster_C_order_label_cg$cluster == "B")
ltr_c <- subset(w22_empty_category_fam_prop_cluster_C_order_label_cg, w22_empty_category_fam_prop_cluster_C_order_label_cg$class == "RL" & w22_empty_category_fam_prop_cluster_C_order_label_cg$cluster == "C")
#pdf("ltr_abc_fam.pdf")
pdf("bwpm_ltr_abc_fam_cg_20Dec.pdf")
ggplot() + geom_point(data=ltr_a, aes(x=id, y=prop), color="cornflowerblue", size=3) + geom_point(data=ltr_b, aes(x=id, y=prop), color="springgreen3", size=3) + geom_point(data=ltr_c, aes(x=id, y=prop), color="lightcoral", size=3) + theme_classic() 
dev.off()
#pdf("bwp_ltr_abc_fam_connected.pdf")
#ggplot() + geom_point(data=ltr_a, aes(x=id, y=prop), color="cornflowerblue", size=3) + geom_point(data=ltr_b, aes(x=id, y=prop), color="springgreen3", size=3) + geom_line() + geom_point(data=ltr_c, aes(x=id, y=prop), color="lightcoral", size=3) + theme_classic()
#dev.off()


--------------------------------------

library(ggplot2)
library(dplyr)

setwd("/scratch.global/nosha003/methylation_spreading")
w22_empty_category_fam_prop_cluster_C_order <- read.delim("all_empty_category_fam_prop_cluster_C_order_20Dec.txt", header=T, sep="\t")
w22_empty_category_fam_prop_cluster_C_order_label <- w22_empty_category_fam_prop_cluster_C_order %>% group_by(class, cluster) %>% mutate(id = row_number())

group.colors <- c(A = "cornflowerblue", B = "springgreen3", C = "lightcoral")

df <- subset(w22_empty_category_fam_prop_cluster_C_order_label, w22_empty_category_fam_prop_cluster_C_order_label$cluster != "NA" & w22_empty_category_fam_prop_cluster_C_order_label$context != "V6")
df_tir <- subset(df, df$class == "DT")
df_ltr <- subset(df, df$class == "RL")

#pdf("w_tir_consistent_fam.pdf")
pdf("bwpm_tir_consistent_fam_20Dec.pdf")
ggplot() + geom_point(data=df_tir, aes(x=id, y=prop, color=cluster), size=2) + theme_classic() + facet_grid(. ~ context) + scale_color_manual(values=group.colors)
dev.off()
#pdf("w_ltr_consistent_fam.pdf")
pdf("bwpm_ltr_consistent_fam_20Dec.pdf")
ggplot() + geom_point(data=df_ltr, aes(x=id, y=prop, color=cluster), size=2) + theme_classic() + facet_grid(. ~ context) + scale_color_manual(values=group.colors)
dev.off()
```


#### metaplot of unmethylated insertion sites
```{r eval=F}
#16Feb2019
#site_C_metaplot.sh

#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=6,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/metaplot_o
#PBS -e /scratch.global/nosha003/e_o/metaplot_e
#PBS -N metaplot
#PBS -V
#PBS -r n

cd /scratch.global/nosha003/
module load R
R CMD BATCH site_C_metaplot.R
wait

# qsub /scratch.global/nosha003/site_C_metaplot.sh
```

```{r eval=F}
# qsub -I -l walltime=3:00:00,nodes=1:ppn=6,mem=90G

library(fields)
library(dplyr)
library(reshape2)
library(ggplot2)
library(Rmisc)

setwd("/home/springer/nosha003/sarah/TEpolymorphism")
df <- read.delim("all_empty_average.txt", header=T, sep="\t")
setwd("/scratch.global/nosha003/methylation_spreading")

df_C <- subset(df, df$V15 == 0 & df$V4 < 0.20)
df_C_TE <- data.frame(TE = df_C$V10)

df_C_TE_bins <- subset(df, df$V10 %in% df_C_TE$TE)

df_C_TE_bins$CG <- ifelse(df_C_TE_bins$V4 >=0.40, 1, 0)
df_C_TE_bins$CHG <- ifelse(df_C_TE_bins$V5 >=0.40, 1, 0)
df_C_TE_bins$CHH <- ifelse(df_C_TE_bins$V6 >=0.015, 1, 0)
dist = subset(df_C_TE_bins, df_C_TE_bins$V15 > -1000 & df_C_TE_bins$V15 <= 1000)
dist$order <- substr(dist$fam, 1, 2)
  
tir <- subset(dist, dist$order == "DT")
ltr <- subset(dist, dist$order == "RL")

tir_group <-  transform(tir, group=cut(as.numeric(sub('[%]', '', V15)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000)))
tir_sumCG <- do.call(data.frame,aggregate(tir_group$CG~tir_group$group, tir_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(tir_sumCG) <- c("group", "count", "sum")
tir_sumCG["percent_methylated"] <- tir_sumCG$sum / tir_sumCG$count
tir_sumCG["order"] <- "TIR"
tir_sumCHG <- do.call(data.frame,aggregate(tir_group$CHG~tir_group$group, tir_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(tir_sumCHG) <- c("group", "count", "sum")
tir_sumCHG["percent_methylated"] <- tir_sumCHG$sum / tir_sumCHG$count
tir_sumCHG["order"] <- "TIR"
tir_sumCHH <- do.call(data.frame,aggregate(tir_group$CHH~tir_group$group, tir_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(tir_sumCHH) <- c("group", "count", "sum")
tir_sumCHH["percent_methylated"] <- tir_sumCHH$sum / tir_sumCHH$count
tir_sumCHH["order"] <- "TIR"

ltr_group <-  transform(ltr, group=cut(as.numeric(sub('[%]', '', V15)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000)))
ltr_sumCG <- do.call(data.frame,aggregate(ltr_group$CG~ltr_group$group, ltr_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(ltr_sumCG) <- c("group", "count", "sum")
ltr_sumCG["percent_methylated"] <- ltr_sumCG$sum / ltr_sumCG$count
ltr_sumCG["order"] <- "LTR"
ltr_sumCHG <- do.call(data.frame,aggregate(ltr_group$CHG~ltr_group$group, ltr_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(ltr_sumCHG) <- c("group", "count", "sum")
ltr_sumCHG["percent_methylated"] <- ltr_sumCHG$sum / ltr_sumCHG$count
ltr_sumCHG["order"] <- "LTR"
ltr_sumCHH <- do.call(data.frame,aggregate(ltr_group$CHH~ltr_group$group, ltr_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(ltr_sumCHH) <- c("group", "count", "sum")
ltr_sumCHH["percent_methylated"] <- ltr_sumCHH$sum / ltr_sumCHH$count
ltr_sumCHH["order"] <- "LTR"

p.1 <- rbind(tir_sumCG, ltr_sumCG)
p.2 <- rbind(tir_sumCHG, ltr_sumCHG)
p.3 <- rbind(tir_sumCHH, ltr_sumCHH)

## Metaplot of percent bins methylated by class
pdf("CG_unmethylated_sites_metaplot.pdf")
plot1 <- ggplot(p.1, aes(x = group, y = percent_methylated, group = order, colour = order)) + geom_line(size=1) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot1)
dev.off()
pdf("CHG_unmethylated_sites_metaplot.pdf")
plot2 <- ggplot(p.2, aes(x = group, y = percent_methylated, group = order, colour = order)) + geom_line(size=1) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot2)
dev.off()
pdf("CHH_unmethylated_sites_metaplot.pdf")
plot3 <- ggplot(p.3, aes(x = group, y = percent_methylated, group = order, colour = order)) + geom_line(size=1) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot3)
dev.off()
}

```

```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

## B73
metaplot_mC_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "all_empty_category_fam_prop_cluster_C_order_label_20Dec.txt",
  genotype = "B73",
  date = "Feb2019"
)

```



## How close are other TE's to polymorphic sites?
```{r eval=F}
cd /home/springer/nosha003/sarah/TEpolymorphism
bedtools closest -d -io -a B73.to.PH207.output2_10Sept18.sort.bed -b B73.to.PH207.output2_10Sept18.sort.bed > /scratch.global/nosha003/B73.to.PH207.output2.closest.txt
# 97972 --> 6065 poly.te.site

awk '{if ($4 != $12 && $17 <= 500 && ($8 == "poly.te.site")) print $0}' /scratch.global/nosha003/B73.to.PH207.output2.closest.txt > /scratch.global/nosha003/B73.to.PH207.output2.closest.poly.txt
# 415 / 6065 = 0.06842539 of polymorphic sites have another TE within 500bp (but not overlapping)
    228 conserved
     73 conserved.left.flank
     55 conserved.right.flank
     59 poly.te.site
# 191 / 6065 = 0.03149217 of polymorphic sites have another TE within 300bp (but not overlapping)
    100 conserved
     36 conserved.left.flank
     23 conserved.right.flank
     32 poly.te.site
# 62 / 6065 =  0.01022259 of polymorphic sites have another TE within 200bp (but not overlapping)
```



## Empty site consistency
- if we have a TE in one genotype but an empty site in the other 3... for those with data look to see what the methylation state is at the empty site... for those with at least one unmethylated site, is it unmethylated in 1,2,or 3 genotypes 
- asking if the mC state at the empty site is likely to truly be the ancestral state (consistent across multiple genotypes lacking the TE)

- find all B73 TE insertions... TE absent in 3 other genotypes (N = x)
- overlap with empty site methylation for each of the 3 genotypes
- subset to TEs with at least one unmethylated empty site (N = x)
- determine proportion of unmethylated empty sites for each TE
```{r eval=F}
# qsub -I -l walltime=3:00:00,nodes=1:ppn=6,mem=90G

library(dplyr)
library(ggplot2)
library(reshape2)
library(tidyr)

# setwd("/home/springer/nosha003/sarah/TEpolymorphism/")
# te <- read.delim("B73_allContrasts_filteredTE_13Jan19.bed", header=F, sep="\t", stringsAsFactors = F)
# te$te <- substr(te$V4, 1, 21)
# te_class <- separate(te, V4, into=c("te", "W22", "Mo17", "PH207"), sep=";")
# 
# setwd("/home/springer/nosha003/methylation_spreading/")
# mC <- read.delim("all_empty_melt_cluster2_20Dec.txt", header=T, sep="\t")
# mC_site_count <- mC %>% group_by(TE) %>% mutate(site_count = n())
# mC_state_binary
# mC_state_binary <- mC %>% group_by(TE) %>% mutate(state_binary = ifelse(mC_state == "mC", 1, ifelse(mC_state == "C", 0, "NA"))) 
# mC_state_binary_na <- na.omit(mC_state_binary[,c()])
# mC_state_count <- mC_state_binary %>% group_by(TE) %>% mutate(state_count = sum(state_binary))


setwd("/home/springer/nosha003/sarah/TEpolymorphism/")
ph207 <- read.delim("PH207emptysite_insertion_methylation.txt", header=F, sep="\t")
mo17 <- read.delim("Mo17emptysite_insertion_methylation.txt", header=F, sep="\t")
w22 <- read.delim("W22emptysite_insertion_methylation.txt", header=F, sep="\t")

ph207_site <- subset(ph207, ph207$V15 != 0 & ph207$V15 < 100)
ph207_site_mC <- ph207_site %>% group_by(V10) %>% mutate(mean_mC = mean(V4, na.rm=T)) %>% mutate(mC_state = ifelse(mean_mC < 0.40, "C", ifelse(mean_mC > 0.60, "mC", ifelse(mean_mC <= 0.60 & mean_mC >= 0.40, "intermediate", "na"))))
ph207_site_mC$site_genotype <- "ph207"
ph207_site_mC_uniq <- unique(ph207_site_mC[,c(10,16:18)])

mo17_site <- subset(mo17, mo17$V15 != 0 & mo17$V15 < 100)
mo17_site_mC <- mo17_site %>% group_by(V10) %>% mutate(mean_mC = mean(V4, na.rm=T)) %>% mutate(mC_state = ifelse(mean_mC < 0.40, "C", ifelse(mean_mC > 0.60, "mC", ifelse(mean_mC <= 0.60 & mean_mC >= 0.40, "intermediate", "na"))))
mo17_site_mC$site_genotype <- "mo17"
mo17_site_mC_uniq <- unique(mo17_site_mC[,c(10,16:18)])

w22_site <- subset(w22, w22$V15 != 0 & w22$V15 < 100)
w22_site_mC <- w22_site %>% group_by(V10) %>% mutate(mean_mC = mean(V4, na.rm=T)) %>% mutate(mC_state = ifelse(mean_mC < 0.40, "C", ifelse(mean_mC > 0.60, "mC", ifelse(mean_mC <= 0.60 & mean_mC >= 0.40, "intermediate", "na"))))
w22_site_mC$site_genotype <- "w22"
w22_site_mC_uniq <- unique(w22_site_mC[,c(10,16:18)])

pm_site_mC <- rbind(ph207_site_mC_uniq, mo17_site_mC_uniq)
pmw_site_mC <- rbind(pm_site_mC, w22_site_mC_uniq)
pmw_site_mC_na <- na.omit(pmw_site_mC)
pmw_site_mC_count <- pmw_site_mC_na %>% group_by(V10) %>% mutate(site_count = n())
# 19353
pmw_site_mC_count_3 <- subset(pmw_site_mC_count, pmw_site_mC_count$site_count == 3)
# 5535
pmw_site_mC_count_3_unmethylated <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "C")
# 116
pmw_site_mC_count_3_unmethylated_count <- pmw_site_mC_count_3_unmethylated %>% group_by(V10) %>% mutate(unmethylated_site_count = n())
pmw_site_mC_count_3_unmethylated_count2 <- unique(pmw_site_mC_count_3_unmethylated_count[,c(1,5,6)])
# 202
pmw_site_mC_count_3_unmethylated_count_consistency <- pmw_site_mC_count_3_unmethylated_count2 %>% group_by(unmethylated_site_count) %>% mutate(consistency_count = n())
pmw_site_mC_count_3_unmethylated_count_consistency2 <- unique(pmw_site_mC_count_3_unmethylated_count_consistency[,c(3,4)])
#   unmethylated_site_count consistency_count
#                    <int>             <int>
#1                       2                17
#2                       1                52
#3                       3                10
## 27/79=34% of unmethylated insertion sites are unmethylated in at least 2 of the 3 genotypes with insertion sites

### Not good... means the unmethylated insertion sites are not consistent across genotypes and therefore may not be representative of the ancestral state... try including intermediate mC state
pmw_site_mC_count_3_unmethylated <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "C" | pmw_site_mC_count_3$mC_state == "intermediate")
# 320
pmw_site_mC_count_3_unmethylated_count <- pmw_site_mC_count_3_unmethylated %>% group_by(V10) %>% mutate(unmethylated_site_count = n())
pmw_site_mC_count_3_unmethylated_count2 <- unique(pmw_site_mC_count_3_unmethylated_count[,c(1,5,6)])
# 202
pmw_site_mC_count_3_unmethylated_count_consistency <- pmw_site_mC_count_3_unmethylated_count2 %>% group_by(unmethylated_site_count) %>% mutate(consistency_count = n())
pmw_site_mC_count_3_unmethylated_count_consistency2 <- unique(pmw_site_mC_count_3_unmethylated_count_consistency[,c(3,4)])
#  unmethylated_site_count consistency_count
#                    <int>             <int>
#1                       1               112
#2                       2                62
#3                       3                28

#### still not good :/



## How many are methylated
pmw_site_mC_count_3_methylated <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "mC")
# 5215
pmw_site_mC_count_3_methylated_count <- pmw_site_mC_count_3_methylated %>% group_by(V10) %>% mutate(methylated_site_count = n())
pmw_site_mC_count_3_methylated_count2 <- unique(pmw_site_mC_count_3_methylated_count[,c(1,5,6)])
# 1817
pmw_site_mC_count_3_methylated_count_consistency <- pmw_site_mC_count_3_methylated_count2 %>% group_by(methylated_site_count) %>% mutate(consistency_count = n())
pmw_site_mC_count_3_methylated_count_consistency2 <- unique(pmw_site_mC_count_3_methylated_count_consistency[,c(3,4)])

pmw_site_mC_count_3_mCstate <- pmw_site_mC_count_3 %>% group_by(V10, mC_state) %>% mutate(state_count = n())




# make a matrix
pmw_site_mC_count_3_methylated <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "mC")
pmw_site_mC_count_3_unmethylated <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "C")
pmw_site_mC_count_3_intermediate <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "intermediate")
pmw_site_mC_count_3_methylated2 <- pmw_site_mC_count_3_methylated %>% group_by(V10) %>% mutate(mC_count = n())
pmw_site_mC_count_3_unmethylated2 <- pmw_site_mC_count_3_unmethylated %>% group_by(V10) %>% mutate(C_count = n())
pmw_site_mC_count_3_intermediate2 <- pmw_site_mC_count_3_intermediate %>% group_by(V10) %>% mutate(intermediate_count = n())

pmw_site_mC_count_3_mC_C <- full_join(unique(pmw_site_mC_count_3_methylated2[,c(1,6)]), unique(pmw_site_mC_count_3_unmethylated2[,c(1,6)]), by="V10")
pmw_site_mC_count_3_mC_C_intermediate <- full_join(pmw_site_mC_count_3_mC_C, unique(pmw_site_mC_count_3_intermediate2[,c(1,6)]), by="V10")

pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated <- subset(pmw_site_mC_count_3_mC_C_intermediate, pmw_site_mC_count_3_mC_C_intermediate$C_count > 0)
pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated[is.na(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated)] <- 0
table(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated[,2:4])

dcast(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated, mC_count + intermediate_count ~ C_count)
#   mC_count intermediate_count  1  2  3
# 1        0                  0  0  0 10
# 2        0                  1  0  2  0
# 3        0                  2  8  0  0
# 4        1                  0  0 15  0
# 5        1                  1 19  0  0
# 6        2                  0 25  0  0

# 20/79= 25% have all 3 either unmethylated or intermediate (13% all unmethylated)
# 34/79= 43% have at least 2 as unmethylated/intermediate with 1 methylated
# 25/79= 32% have only 1 unmethylated and 2 methylated


# so if we take away the one unmethylated empty site that caused us to include the TE event in the subset... how many of the other 158 (79 events * 2 alleles) sites are methylated
table(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated$mC_count)
#  0  1  2 
# 20 34 25 

# 25*2 + 34 = 84/(79*2) = 53%








########################### should be looking at the site tile, not the flanking tiles ###################################

ph207_site <- subset(ph207, ph207$V15 == 0)
ph207_site_mC <- ph207_site %>% group_by(V10) %>% mutate(mean_mC = mean(V4, na.rm=T)) %>% mutate(mC_state = ifelse(mean_mC < 0.40, "C", ifelse(mean_mC > 0.60, "mC", ifelse(mean_mC <= 0.60 & mean_mC >= 0.40, "intermediate", "na"))))
ph207_site_mC$site_genotype <- "ph207"
ph207_site_mC_uniq <- unique(ph207_site_mC[,c(10,16:18)])

mo17_site <- subset(mo17, mo17$V15 == 0)
mo17_site_mC <- mo17_site %>% group_by(V10) %>% mutate(mean_mC = mean(V4, na.rm=T)) %>% mutate(mC_state = ifelse(mean_mC < 0.40, "C", ifelse(mean_mC > 0.60, "mC", ifelse(mean_mC <= 0.60 & mean_mC >= 0.40, "intermediate", "na"))))
mo17_site_mC$site_genotype <- "mo17"
mo17_site_mC_uniq <- unique(mo17_site_mC[,c(10,16:18)])

w22_site <- subset(w22, w22$V15 == 0)
w22_site_mC <- w22_site %>% group_by(V10) %>% mutate(mean_mC = mean(V4, na.rm=T)) %>% mutate(mC_state = ifelse(mean_mC < 0.40, "C", ifelse(mean_mC > 0.60, "mC", ifelse(mean_mC <= 0.60 & mean_mC >= 0.40, "intermediate", "na"))))
w22_site_mC$site_genotype <- "w22"
w22_site_mC_uniq <- unique(w22_site_mC[,c(10,16:18)])

pm_site_mC <- rbind(ph207_site_mC_uniq, mo17_site_mC_uniq)
pmw_site_mC <- rbind(pm_site_mC, w22_site_mC_uniq)
pmw_site_mC_na <- na.omit(pmw_site_mC)
pmw_site_mC_count <- pmw_site_mC_na %>% group_by(V10) %>% mutate(site_count = n())
# 15617
pmw_site_mC_count_3 <- subset(pmw_site_mC_count, pmw_site_mC_count$site_count == 3)
# 3744

# make a matrix
pmw_site_mC_count_3_methylated <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "mC")
pmw_site_mC_count_3_unmethylated <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "C")
pmw_site_mC_count_3_intermediate <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "intermediate")
pmw_site_mC_count_3_methylated2 <- pmw_site_mC_count_3_methylated %>% group_by(V10) %>% mutate(mC_count = n())
pmw_site_mC_count_3_unmethylated2 <- pmw_site_mC_count_3_unmethylated %>% group_by(V10) %>% mutate(C_count = n())
pmw_site_mC_count_3_intermediate2 <- pmw_site_mC_count_3_intermediate %>% group_by(V10) %>% mutate(intermediate_count = n())

pmw_site_mC_count_3_mC_C <- full_join(unique(pmw_site_mC_count_3_methylated2[,c(1,6)]), unique(pmw_site_mC_count_3_unmethylated2[,c(1,6)]), by="V10")
pmw_site_mC_count_3_mC_C_intermediate <- full_join(pmw_site_mC_count_3_mC_C, unique(pmw_site_mC_count_3_intermediate2[,c(1,6)]), by="V10")

pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated <- subset(pmw_site_mC_count_3_mC_C_intermediate, pmw_site_mC_count_3_mC_C_intermediate$C_count > 0)
pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated[is.na(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated)] <- 0
table(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated[,2:4])

dcast(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated, mC_count + intermediate_count ~ C_count)
#   mC_count intermediate_count  1  2   3
# 1        0                  0  0  0 225
# 2        0                  1  0  9   0
# 3        1                  0  0 31   0
# 4        1                  1 10  0   0
# 5        2                  0 47  0   0

# 238/322= 74% have all 3 either unmethylated or intermediate (70% all unmethylated)
# 41/322= 12% have at least 2 as unmethylated/intermediate with 1 methylated
# 47/322= 14% have only 1 unmethylated and 2 methylated


# so if we take away the one unmethylated empty site that caused us to include the TE event in the subset... how many of the other 158 (79 events * 2 alleles) sites are methylated
table(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated$mC_count)
#   0   1   2 
# 234  41  47 

# 47*2 + 41 = 135/(322*2) = 21%


# split by LTR and TIR
pmw_site_mC_count_3$order <- substr(pmw_site_mC_count_3$V10, 1, 2)
pmw_site_mC_count_3_methylated <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "mC")
pmw_site_mC_count_3_unmethylated <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "C")
pmw_site_mC_count_3_intermediate <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "intermediate")
pmw_site_mC_count_3_methylated2 <- pmw_site_mC_count_3_methylated %>% group_by(V10, order) %>% mutate(mC_count = n())
pmw_site_mC_count_3_unmethylated2 <- pmw_site_mC_count_3_unmethylated %>% group_by(V10, order) %>% mutate(C_count = n())
pmw_site_mC_count_3_intermediate2 <- pmw_site_mC_count_3_intermediate %>% group_by(V10, order) %>% mutate(intermediate_count = n())

pmw_site_mC_count_3_mC_C <- full_join(unique(pmw_site_mC_count_3_methylated2[,c(1,6,7)]), unique(pmw_site_mC_count_3_unmethylated2[,c(1,6,7)]), by=c("V10", "order"))
pmw_site_mC_count_3_mC_C_intermediate <- full_join(pmw_site_mC_count_3_mC_C, unique(pmw_site_mC_count_3_intermediate2[,c(1,6,7)]), by=c("V10", "order"))

pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated <- subset(pmw_site_mC_count_3_mC_C_intermediate, pmw_site_mC_count_3_mC_C_intermediate$C_count > 0)
pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated[is.na(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated)] <- 0
table(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated[,2:5])

pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated_tir <- subset(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated, pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated$order == "DT")
pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated_ltr <- subset(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated, pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated$order == "RL")
dcast(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated_tir, mC_count + intermediate_count ~ C_count)
#   mC_count intermediate_count  1  2   3
# 1        0                  0  0  0 143
# 2        0                  1  0  2   0
# 3        1                  0  0 12   0
# 4        1                  1  7  0   0
# 5        2                  0 14  0   0

# 145/178=81% all 3 sites unmethylated/intermediate
# 19/178=11% at least 2 sites unmethylated/intermediate with 1 methylated
# 14/178=8% with 1 site unmethylated and 2 sites methylated

dcast(pmw_site_mC_count_3_mC_C_intermediate_atleast1unmethylated_ltr, mC_count + intermediate_count ~ C_count)
#   mC_count intermediate_count  1  2  3
# 1        0                  0  0  0 71
# 2        0                  1  0  5  0
# 3        1                  0  0 17  0
# 4        1                  1  1  0  0
# 5        2                  0 28  0  0

# 76/122=62% all 3 sites unmethylated/intermediate
# 18/122=15% at least 2 sites unmethylated/intermediate with 1 methylated
# 28/122=23% 1 site unmethylated and 2 sites methylated

# make table for figure
prop <- matrix(c(81,11,8,62,15,23), ncol=3, byrow=TRUE)
colnames(prop) <- c("3 genotypes", "2 genotypes", "1 genotype")
rownames(prop) <- c("TIR", "LTR")
prop <- as.table(prop)
prop_melt <- melt(prop)

setwd("/scratch.global/nosha003/methylation_spreading")
pdf("site_unmethylated_consistency.pdf")
ggplot(prop_melt) + geom_bar(aes(x=Var1, y=value, fill=factor(Var2, levels=c("1 genotype", "2 genotypes", "3 genotypes"))), stat="identity") + theme_classic() + xlab("") + ylab("Proportion of genotypes (PH207, W22, and Mo17) with unmethylated insertion sites")
dev.off()
```


```{r eval=F}
# qsub -I -l walltime=3:00:00,nodes=1:ppn=6,mem=90G

library(dplyr)
library(ggplot2)
library(reshape2)
library(tidyr)

setwd("/home/springer/nosha003/sarah/TEpolymorphism/")
ph207 <- read.delim("PH207emptysite_insertion_methylation.txt", header=F, sep="\t")
mo17 <- read.delim("Mo17emptysite_insertion_methylation.txt", header=F, sep="\t")
w22 <- read.delim("W22emptysite_insertion_methylation.txt", header=F, sep="\t")

## consistency of methylated or unmethylated across the 3 empty sites
ph207_site <- subset(ph207, ph207$V15 == 0)
ph207_site_mC <- ph207_site %>% group_by(V10) %>% mutate(mean_mC = mean(V4, na.rm=T)) %>% mutate(mC_state = ifelse(mean_mC < 0.40, "C", ifelse(mean_mC > 0.60, "mC", ifelse(mean_mC <= 0.60 & mean_mC >= 0.40, "intermediate", "na"))))
ph207_site_mC$site_genotype <- "ph207"
ph207_site_mC_uniq <- unique(ph207_site_mC[,c(10,16:18)])

mo17_site <- subset(mo17, mo17$V15 == 0)
mo17_site_mC <- mo17_site %>% group_by(V10) %>% mutate(mean_mC = mean(V4, na.rm=T)) %>% mutate(mC_state = ifelse(mean_mC < 0.40, "C", ifelse(mean_mC > 0.60, "mC", ifelse(mean_mC <= 0.60 & mean_mC >= 0.40, "intermediate", "na"))))
mo17_site_mC$site_genotype <- "mo17"
mo17_site_mC_uniq <- unique(mo17_site_mC[,c(10,16:18)])

w22_site <- subset(w22, w22$V15 == 0)
w22_site_mC <- w22_site %>% group_by(V10) %>% mutate(mean_mC = mean(V4, na.rm=T)) %>% mutate(mC_state = ifelse(mean_mC < 0.40, "C", ifelse(mean_mC > 0.60, "mC", ifelse(mean_mC <= 0.60 & mean_mC >= 0.40, "intermediate", "na"))))
w22_site_mC$site_genotype <- "w22"
w22_site_mC_uniq <- unique(w22_site_mC[,c(10,16:18)])

pm_site_mC <- rbind(ph207_site_mC_uniq, mo17_site_mC_uniq)
pmw_site_mC <- rbind(pm_site_mC, w22_site_mC_uniq)
pmw_site_mC_na <- na.omit(pmw_site_mC)
pmw_site_mC_count <- pmw_site_mC_na %>% group_by(V10) %>% mutate(site_count = n())
# 15617
pmw_site_mC_count_3 <- subset(pmw_site_mC_count, pmw_site_mC_count$site_count == 3)
# 3744

# make a matrix
pmw_site_mC_count_3_methylated <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "mC")
pmw_site_mC_count_3_unmethylated <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "C")
pmw_site_mC_count_3_intermediate <- subset(pmw_site_mC_count_3, pmw_site_mC_count_3$mC_state == "intermediate")
pmw_site_mC_count_3_methylated2 <- pmw_site_mC_count_3_methylated %>% group_by(V10) %>% mutate(mC_count = n())
pmw_site_mC_count_3_unmethylated2 <- pmw_site_mC_count_3_unmethylated %>% group_by(V10) %>% mutate(C_count = n())
pmw_site_mC_count_3_intermediate2 <- pmw_site_mC_count_3_intermediate %>% group_by(V10) %>% mutate(intermediate_count = n())

pmw_site_mC_count_3_mC_C <- full_join(unique(pmw_site_mC_count_3_methylated2[,c(1,6)]), unique(pmw_site_mC_count_3_unmethylated2[,c(1,6)]), by="V10")
pmw_site_mC_count_3_mC_C_intermediate <- full_join(pmw_site_mC_count_3_mC_C, unique(pmw_site_mC_count_3_intermediate2[,c(1,6)]), by="V10")

pmw_site_mC_count_3_mC_C_intermediate[is.na(pmw_site_mC_count_3_mC_C_intermediate)] <- 0
table(pmw_site_mC_count_3_mC_C_intermediate[,2:4])

#         C_count
# mC_count   0   1   2   3
#        0   0   0   0 225
#        1   0   0  31   0
#        2   0  47   0   0
#        3 890   0   0   0
# 
# , , intermediate_count = 1
# 
#         C_count
# mC_count   0   1   2   3
#        0   0   0   9   0
#        1   0  10   0   0
#        2  36   0   0   0
#        3   0   0   0   0
       
## (225+890) / (225+890+47+31) = 0.9346186 = 93.5%
## including intermediate --> (225+890+36+9) / (225+890+47+31+36+9+10) = 93%

```




# Figure 5: Spreading
- low empty site methylation --> TE present methylation

```{r eval=F}
#20Jan2019
#site_spreading.sh

#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=8,mem=60gb
#PBS -o /scratch.global/nosha003/e_o/site_spreading_o
#PBS -e /scratch.global/nosha003/e_o/site_spreading_e
#PBS -N site_spreading
#PBS -V
#PBS -r n

cd /home/springer/nosha003/scripts/
module load R
R CMD BATCH site_spreading.R
R CMD BATCH prop_spreading.R
wait

# qsub /home/springer/nosha003/scripts/site_spreading.sh
```

```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

# setwd("/scratch.global/nosha003")
# b <- read.delim("B73.to.W22.output2.B73bins.txt", header=F, sep="\t")
# b$TE <- substr(b$V11, 1, 21)
# df1 <- read.delim("B73.to.W22.output2.B73bins.txt", header=F, sep="\t")
# df2 <- read.delim("B73.to.PH207.output2.B73bins.txt", header=F, sep="\t")
# df3 <- read.delim("W22.to.B73.output2.W22bins.txt", header=F, sep="\t")
# df4 <- read.delim("W22.to.PH207.output2.W22bins.txt", header=F, sep="\t")
# df5 <- read.delim("PH207.to.B73.output2.PH207bins.txt", header=F, sep="\t")
# df6 <- read.delim("PH207.to.W22.output2.PH207bins.txt", header=F, sep="\t")
# df7 <- read.delim("PH207.to.B73.output2.PH207bins.txt", header=F, sep="\t")
# df8 <- read.delim("PH207.to.W22.output2.PH207bins.txt", header=F, sep="\t")
# b <- rbind(df1, df2, df3, df4, df5, df6, df7, df8)
# b$TE <- substr(b$V11, 1, 21)
# write.table(b, "all.to.all.output2.bins.txt", quote=F, row.names=F, sep="\t")

setwd("/home/springer/nosha003/methylation_spreading")
b <- read.delim("all.to.all.output2.bins.txt", header=T, sep="\t")

setwd("/scratch.global/nosha003/methylation_spreading")
#w <- read.delim("w22_empty_melt_cluster.txt", header=T, sep="\t")
w <- read.delim("all_empty_melt_cluster_20Dec.txt", header=T, sep="\t")
#w_low <- subset(w, w$mC_ratio <= 0.20)
##### Need to subset to just CG so that I am not including TEs that only have low empty insertion sites in CHG and CHH #######
w_cg <- subset(w, w$context == "V4")
w_low <- subset(w_cg, w_cg$mC_ratio <= 0.20)
w_low$TE <- substr(w_low$TE, 1, 21)

library(dplyr)
library(reshape2)
w_low_b <- inner_join(w_low, b, by="TE")
w_low_b_flank <- subset(w_low_b, w_low_b$V12 != 0 & abs(w_low_b$V12) < 200)
#w_low_b_flank_na <- na.omit(w_low_b_flank)
cluster <- data.frame(unique(w_low_b_flank[,c(1, 6)]))

w_cg$TE <- substr(w_cg$TE, 1, 21)
w_b <- inner_join(w_cg, b, by="TE")
w_b_flank <- subset(w_b, w_b$V12 != 0 & abs(w_b$V12) < 200)
w_b_flank_category <- w_b_flank %>% group_by(TE) %>% summarise(cg = mean(V4), chg = mean(V5), chh = mean(V6))
write.table(w_b_flank_category, "w_b_flank_mC_20Dec.txt", quote=F, row.names=F, sep="\t")

w_low_b_flank_category <- w_low_b_flank %>% group_by(TE) %>% summarise(cg = mean(V4), chg = mean(V5), chh = mean(V6))
#write.table(w_low_b_flank_category, "w_low_b_flank_mC.txt", quote=F, row.names=F, sep="\t")
write.table(w_low_b_flank_category, "all_low_b_flank_mC_20Dec.txt", quote=F, row.names=F, sep="\t")
w_low_b_flank_category_cluster <- left_join(w_low_b_flank_category, cluster, by="TE")
w_low_b_flank_category_cluster$order <- substr(w_low_b_flank_category$TE, 1, 2)

w_low_b_flank_melt <- melt(w_low_b_flank_category_cluster)
w_low_b_flank_region <- w_low_b_flank_melt %>% group_by(variable, order, cluster) %>% summarise(region_count = n()) 
w_low_b_flank_category_cluster <- w_low_b_flank_melt %>% group_by(variable, order, cluster) %>% mutate(mC_state = ifelse(variable == "cg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chh", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(variable, order, cluster, mC_state) 
setwd("/scratch.global/nosha003/methylation_spreading")
#write.table(w_low_b_flank_category_cluster, "w_low_b_flank_category_cluster.txt", quote=F, row.names=F, sep="\t")
write.table(w_low_b_flank_category_cluster, "all_low_b_flank_category_cluster_20Dec.txt", quote=F, row.names=F, sep="\t")

w_low_b_flank_count <- w_low_b_flank_category_cluster %>% summarise(count = n())
w_low_b_flank_state <- inner_join(w_low_b_flank_region, w_low_b_flank_count, by=c("variable", "order", "cluster"))
w_low_b_flank_prop <- w_low_b_flank_state %>% mutate(prop = count/region_count)
w_low_b_flank_prop %>% print(n = Inf)

setwd("/scratch.global/nosha003/methylation_spreading")
#write.table(w_low_b_flank_prop, "w_low_b_flank_prop_24Oct.txt", row.names=F, quote=F, sep="\t")
write.table(w_low_b_flank_prop, "all_low_b_flank_prop_20Dec.txt", row.names=F, quote=F, sep="\t")
#pdf("w_low_b_flank_prop_24Oct.pdf")
pdf("all_low_b_flank_prop_24Oct.pdf")
#ggplot(data=w_low_b_flank_prop) + geom_bar(aes(x=cluster, y=prop, fill=mC_state), stat="identity") + facet_grid(order ~ variable) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_brewer(palette="Set2")
ggplot(data=w_low_b_flank_prop) + geom_bar(aes(x=cluster, y=prop, fill=factor(mC_state, levels=c("mC", "intermediate", "C"))), stat="identity") + facet_grid(order ~ variable) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()
# scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))


# TIR: A(N=1), B(N=41), C(176)
# LTR: A(62), B(34), C(70)




w_low_b_flank_melt <- melt(w_low_b_flank_category_cluster)
w_low_b_flank_melt_na <- subset(w_low_b_flank_melt, w_low_b_flank_melt$value != "NA")

w_low_b_flank_region_cluster <- w_low_b_flank_melt_na %>% group_by(variable, order, cluster) %>% summarise(region_count = n()) 
w_low_b_flank_category_cluster <- w_low_b_flank_melt_na %>% group_by(variable, order, cluster) %>% mutate(mC_state = ifelse(variable == "cg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chh", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(variable, order, cluster, mC_state) 
w_low_b_flank_count_cluster <- w_low_b_flank_category_cluster %>% summarise(count = n())
w_low_b_flank_state_cluster <- inner_join(w_low_b_flank_region_cluster, w_low_b_flank_count_cluster, by=c("variable", "order", "cluster"))
w_low_b_flank_prop_cluster <- w_low_b_flank_state_cluster %>% mutate(prop = count/region_count)
w_low_b_flank_prop_cluster_RL <- swell thubset(w_low_b_flank_prop_cluster, w_low_b_flank_prop_cluster$order == "RL")
w_low_b_flank_prop_cluster_DT <- subset(w_low_b_flank_prop_cluster, w_low_b_flank_prop_cluster$order == "DT")

w_low_b_flank_melt2 <- w_low_b_flank_melt_na
w_low_b_flank_melt2$cluster <- "all"
w_low_b_flank_region <- w_low_b_flank_melt2 %>% group_by(variable, order, cluster) %>% summarise(region_count = n()) 
w_low_b_flank_category <- w_low_b_flank_melt2 %>% group_by(variable, order, cluster) %>% mutate(mC_state = ifelse(variable == "cg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chh", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(variable, order, cluster, mC_state) 
w_low_b_flank_count <- w_low_b_flank_category %>% summarise(count = n())
w_low_b_flank_state <- inner_join(w_low_b_flank_region, w_low_b_flank_count, by=c("variable", "order", "cluster"))
w_low_b_flank_prop <- w_low_b_flank_state %>% mutate(prop = count/region_count)
write.table(w_low_b_flank_prop, "all_low_b_flank_prop_all_20Dec.txt", row.names=F, quote=F, sep="\t")
w_low_b_flank_prop_RL <- subset(w_low_b_flank_prop, w_low_b_flank_prop$order == "RL")
w_low_b_flank_prop_DT <- subset(w_low_b_flank_prop, w_low_b_flank_prop$order == "DT")

prop_ltr <- rbind(w_low_b_flank_prop_cluster_RL, w_low_b_flank_prop_RL) 
prop_ltr_cg <- subset(prop_ltr, prop_ltr$variable == "cg")
prop_tir <- rbind(w_low_b_flank_prop_cluster_DT, w_low_b_flank_prop_DT) 
prop_tir_cg <- subset(prop_tir, prop_tir$variable == "cg")

library(ggplot2)
setwd("/scratch.global/nosha003/methylation_spreading")
#pdf("w_low_b_flank_prop_all_CG_LTR.pdf")
pdf("all_low_b_flank_prop_all_CG_LTR_20Dec.pdf")
ggplot(data=prop_ltr_cg, aes(x=factor(cluster, levels=c("NA", "C", "B", "A", "all")), y=prop)) + geom_bar(aes(fill=factor(mC_state, levels=c("mC", "intermediate", "C"))), stat="identity") + facet_grid(. ~ variable) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()
#pdf("w_low_b_flank_prop_all_CG_TIR.pdf")
pdf("all_low_b_flank_prop_all_CG_TIR_20Dec.pdf")
ggplot(data=prop_tir_cg, aes(x=factor(cluster, levels=c("all", "A", "B", "C", "NA")), y=prop)) + geom_bar(aes(fill=factor(mC_state, levels=c("mC", "intermediate", "C"))), stat="identity") + facet_grid(. ~ variable) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()





####################################################################################

## All clusters together
setwd("/scratch.global/nosha003")
#b <- read.delim("B73.to.W22.output2.B73bins.txt", header=F, sep="\t")
#b$TE <- substr(b$V11, 1, 21)
b <- read.delim("all.to.all.output2.bins.txt", header=T, sep="\t")

setwd("/scratch.global/nosha003/methylation_spreading")
#w <- read.delim("w22_empty_melt_cluster.txt", header=T, sep="\t")
w <- read.delim("all_empty_melt_cluster_20Dec.txt", header=T, sep="\t")
w_cg <- subset(w, w$context == "V4")
w_low <- subset(w_cg, w_cg$mC_ratio <= 0.20)
w_low$TE <- substr(w_low$TE, 1, 21)

library(dplyr)
library(reshape2)
#w_low_b <- left_join(b, w_low, by="TE")
w_low_b <- inner_join(w_low, b, by="TE")
w_low_b_flank <- subset(w_low_b, w_low_b$V12 != 0 & abs(w_low_b$V12) < 200)
w_low_b_flank_na <- na.omit(w_low_b_flank)
#cluster <- data.frame(unique(w_low_b_flank_na[,c(13,18)]))
cluster <- data.frame(unique(w_low_b_flank_na[,c(1,5)]))

w_low_b_flank_category <- w_low_b_flank_na %>% group_by(TE) %>% summarise(cg = mean(V4), chg = mean(V5), chh = mean(V6))
w_low_b_flank_category$order <- substr(w_low_b_flank_category$TE, 1, 2)

w_low_b_flank_melt <- melt(w_low_b_flank_category, id=c("TE", "order"))
#w_low_b_flank_melt2 <- w_low_b_flank_melt[,c(1:4,6)]
w_low_b_flank_region <- w_low_b_flank_melt %>% group_by(variable, order) %>% summarise(region_count = n()) 

w_low_b_flank_category <- w_low_b_flank_melt %>% group_by(variable, order) %>% mutate(mC_state = ifelse(variable == "cg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chh", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(variable, order, mC_state) 
w_low_b_flank_count <- w_low_b_flank_category %>% summarise(count = n())
w_low_b_flank_state <- inner_join(w_low_b_flank_region, w_low_b_flank_count, by=c("variable", "order"))
w_low_b_flank_prop <- w_low_b_flank_state %>% mutate(prop = count/region_count)
w_low_b_flank_prop %>% print(n = Inf)
w_low_b_flank_prop_ltr <- subset(w_low_b_flank_prop, w_low_b_flank_prop$order == "RL")
w_low_b_flank_prop_tir <- subset(w_low_b_flank_prop, w_low_b_flank_prop$order == "DT")

setwd("/scratch.global/nosha003/methylation_spreading")
#write.table(w_low_b_flank_prop, "w_low_b_flank_prop_noCluster_24Oct.txt", row.names=F, quote=F, sep="\t")
write.table(w_low_b_flank_prop, "all_low_b_flank_prop_noCluster_20Dec.txt", row.names=F, quote=F, sep="\t")

#w_low_b_flank_prop <- read.delim("all_low_b_flank_prop_noCluster_20Dec.txt", header=T, sep="\t")

library(ggplot2)
#pdf("w_low_b_flank_prop_LTR_noCluster_24Oct.pdf")
pdf("all_low_b_flank_prop_LTR_noCluster_20Dec.pdf")
#ggplot(data=w_low_b_flank_prop_ltr) + geom_bar(aes(x=variable, y=prop, fill=mC_state), stat="identity") + theme_classic() + theme(text = element_text(size=7)) + scale_fill_brewer(palette="Set2")
ggplot(data=w_low_b_flank_prop_ltr) + geom_bar(aes(x=variable, y=prop, fill=factor(mC_state, levels=c("mC", "intermediate", "C"))), stat="identity") + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()

#pdf("w_low_b_flank_prop_TIR_noCluster_24Oct.pdf")
pdf("all_low_b_flank_prop_TIR_noCluster_20Dec.pdf")
#ggplot(data=w_low_b_flank_prop_tir) + geom_bar(aes(x=variable, y=prop, fill=mC_state), stat="identity") + theme_classic() + theme(text = element_text(size=7)) + scale_fill_brewer(palette="Set2")
ggplot(data=w_low_b_flank_prop_tir) + geom_bar(aes(x=variable, y=prop, fill=factor(mC_state, levels=c("mC", "intermediate", "C"))), stat="identity") + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()



# awk '{if ($5 == "DT" && $2 == "V4" && $3 <= 0.20) print $0}' w22_empty_melt_cluster.txt | wc -l
617 --> only 216 actually with B73 data (in plot)
# awk '{if ($5 == "DT" && $2 == "V4") print $0}' w22_empty_melt_cluster.txt | wc -l
1241
# awk '{if ($5 == "RL" && $2 == "V4" && $3 <= 0.20) print $0}' w22_empty_melt_cluster.txt | wc -l
323 --> only 162 actually with B73 data (in plot)
# awk '{if ($5 == "RL" && $2 == "V4") print $0}' w22_empty_melt_cluster.txt | wc -l
3656
```


##### library(ggalluvial)
```{r eval=F}

tir <- data.frame(site=c("intermediate", "methylated", "unmethylated", "unmethylated", "unmethylated"), flank=c("NA", "NA", "unmethylated", "intermediate", "methylated"), prop=c(2.2, 46.3, (33.2*.515), (14.1*.515), (52.7*.515)))
ltr <- data.frame(site=c("intermediate", "methylated", "unmethylated", "unmethylated", "unmethylated"), flank=c("NA", "NA", "unmethylated", "intermediate", "methylated"), prop=c(0.6, 59.7, (21.4*.097), (12.8*.097), (65.7*.097)))

library(dplyr)
library(ggalluvial)
is_alluvia_form(as.data.frame(tir), axes = 1:2, silent = TRUE)

setwd("/Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/spreading")
pdf("tir_alluvial.pdf")
ggplot(as.data.frame(tir), aes(y = prop, axis1 = site, axis2 = flank)) +
  geom_alluvium(aes(fill = flank), width = 1/12) +
  geom_stratum(width = 1/12, fill = "black", color = "grey") +
  geom_label(stat = "stratum", label.strata = TRUE) +
  scale_x_discrete(limits = c("site", "flank"), expand = c(.05, .05)) +
  scale_fill_brewer(type = "qual", palette = "Set1") + theme_classic()
dev.off()
  
setwd("/Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/spreading")
pdf("ltr_alluvial.pdf")
ggplot(as.data.frame(ltr), aes(y = prop, axis1 = site, axis2 = flank)) +
  geom_alluvium(aes(fill = flank), width = 1/12) +
  geom_stratum(width = 1/12, fill = "black", color = "grey") +
  geom_label(stat = "stratum", label.strata = TRUE) +
  scale_x_discrete(limits = c("site", "flank"), expand = c(.05, .05)) +
  scale_fill_brewer(type = "qual", palette = "Set1") + theme_classic()
dev.off()






ltr <- data.frame(site=c("unmethylated", "unmethylated", "unmethylated"), flank=c("unmethylated", "intermediate", "methylated"), prop=c((21.4*.097), (12.8*.097), (65.7*.097)))
ltr$site_factor <- factor(ltr$site, levels=c("methylated", "intermediate", "unmethylated"))
ltr$flank_factor <- factor(ltr$flank, levels=c("unmethylated", "intermediate", "methylated", "NA"))

setwd("/Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/spreading")
pdf("ltr_alluvial2.pdf")
ggplot(as.data.frame(ltr), aes(y = prop, axis1 = site_factor, axis2 = flank_factor)) +
  geom_alluvium(aes(fill = flank)) +
  geom_stratum(fill = "black", color = "grey") +
  geom_label(stat = "stratum", label.strata = TRUE) +
  scale_x_discrete(limits = c("site", "flank"), expand = c(.05, .05)) +
  scale_fill_brewer(type = "qual", palette = "Set1") + theme_classic()
dev.off()


```




###### Proportion of each family that gained methylation
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=90G

library(dplyr)
library(reshape2)
library(ggplot2)

setwd("/scratch.global/nosha003/methylation_spreading")
#w <- read.delim("w22_empty_melt_cluster.txt", header=T, sep="\t")
w <- read.delim("all_empty_melt_cluster_20Dec.txt", header=T, sep="\t")
cluster <- data.frame(unique(w[,c(1, 6)]))
cluster$fam <- substr(cluster$TE, 1, 8)

setwd("/scratch.global/nosha003/methylation_spreading")
#df <- read.delim("w_low_b_flank_mC.txt", header=T, sep="\t")
df <- read.delim("all_low_b_flank_mC_20Dec.txt", header=T, sep="\t")
df$order <- substr(df$TE, 1, 2)
df$fam <- substr(df$TE, 1, 8)

df_cluster <- left_join(df, cluster[,2:3], by="fam")
df_melt <- melt(df_cluster)
df_category <- df_melt %>% group_by(variable, order, cluster) %>% mutate(mC_state = ifelse(variable == "cg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chh", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(variable, order, cluster, mC_state) 
df_category_na <- na.omit(df_category)
df_category_cg <- subset(df_category, df_category$variable == "cg")
df_region <- df_category_cg %>% group_by(order, fam, cluster) %>% summarise(fam_count = n())
df_count <- df_category_cg %>% group_by(order, fam, cluster, mC_state) %>% summarise(mC_count = n())
df_join <- inner_join(df_region, df_count, by=c("order", "fam", "cluster"))
df_prop <- df_join %>% mutate(prop_mC = mC_count / fam_count)
df_prop_mC <- subset(df_prop, df_prop$mC_state == "mC")

df_prop2 <- subset(df_prop_mC, df_prop_mC$order == "RL" | df_prop_mC$order == "DT")
df_prop_subset <- subset(df_prop2, df_prop2$fam_count > 3)

df_prop_order <- df_prop_subset[order(df_prop_subset$prop_mC),]
df_prop_label <- df_prop_order %>% group_by(order, cluster) %>% mutate(id = row_number())

setwd("/scratch.global/nosha003/methylation_spreading")
#pdf("spreading_fam_prop_mC.pdf")
pdf("all_spreading_fam_prop_mC_20Dec.pdf")
ggplot() + geom_point(data=df_prop_label, aes(x=id, y=prop_mC, color=cluster), size=2) + theme_classic() + facet_grid(. ~ order, scales="free")
dev.off()

df_prop_label2 <- df_prop_order %>% group_by(order) %>% mutate(id = row_number())
setwd("/scratch.global/nosha003/methylation_spreading")
#write.table(df_prop_label2, "low_empty_TE_fam_proportion.txt", quote=F, row.names=F, sep="\t")
write.table(df_prop_label2, "all_low_empty_TE_fam_proportion_20Dec.txt", quote=F, row.names=F, sep="\t")
pdf("spreading_fam_prop_mC2.pdf")
ggplot() + geom_point(data=df_prop_label2, aes(x=id, y=prop_mC, color=cluster), size=2) + theme_classic() + facet_grid(. ~ order, scales="free")
dev.off()

ltr <- subset(df_prop_order, df_prop_order$order == "RL")
tir <- subset(df_prop_order, df_prop_order$order == "DT")
ltr_order <- ltr %>% group_by(order) %>% mutate(id = row_number())
tir_order <- tir %>% group_by(order) %>% mutate(id = row_number())
setwd("/scratch.global/nosha003/methylation_spreading")
#pdf("spreading_fam_prop_mC_LTR.pdf")
pdf("all_spreading_fam_prop_mC_LTR_20Dec.pdf")
ggplot() + geom_point(data=ltr_order, aes(x=id, y=prop_mC, color=cluster), size=2) + theme_classic() 
dev.off()
#pdf("spreading_fam_prop_mC_TIR.pdf")
pdf("all_spreading_fam_prop_mC_TIR_20Dec.pdf")
ggplot() + geom_point(data=tir_order, aes(x=id, y=prop_mC, color=cluster), size=2) + theme_classic() 
dev.off()

fam_count <- df_prop_label[order(df_prop_label$fam_count),]
# 22 families with 2-3 sites, 82 families with > 3 sites (most with < 25 sites) --> almost all have no cluster
## ALL --> 150 FAMILIES WITH > 4 SITES



## LTRs and TIRs in one plot
order <- rbind(ltr_order, tir_order)
pdf("all_spreading_fam_prop_mC_LTR_TIR_20Dec.pdf")
ggplot() + geom_jitter(data=order, aes(x=id, y=prop_mC, color=cluster, shape=order), size=2) + theme_classic()
dev.off()



## Families with 100% spreading
# order	fam	cluster	fam_count	mC_state	mC_count	prop_mC	id
# DT	DTA00019	NA	7	mC	7	1	166 --> spread on one side into methylated region but doesn't spread on other side into genic region (spreads until it runs into another genomic feature)
# DT	DTA00037	NA	6	mC	6	1	167
# DT	DTA00078	NA	16	mC	16	1	168
# DT	DTA00118	B	4	mC	4	1	169
# DT	DTA00137	NA	4	mC	4	1	170
# DT	DTA00148	NA	10	mC	10	1	171
# DT	DTA00159	NA	10	mC	10	1	172
# DT	DTA00191	NA	8	mC	8	1	173
# DT	DTA00311	NA	10	mC	10	1	174
# DT	DTA00312	NA	11	mC	11	1	175
# DT	DTA00319	NA	4	mC	4	1	176
# DT	DTA00343	NA	8	mC	8	1	177
# DT	DTA00344	NA	4	mC	4	1	178
# DT	DTH00024	NA	5	mC	5	1	179
# DT	DTH00486	NA	8	mC	8	1	180
# DT	DTT00042	NA	4	mC	4	1	181
# RL	RLC00099	B	13	mC	13	1	115 --> spreads into methylated region or genomic feature boundary
# RL	RLC00231	NA	5	mC	5	1	116
# RL	RLC00254	NA	12	mC	12	1	117
# RL	RLC00409	NA	10	mC	10	1	118
# RL	RLC00676	NA	5	mC	5	1	119
# RL	RLC00882	NA	18	mC	18	1	120
# RL	RLC02067	NA	5	mC	5	1	121
# RL	RLC02959	NA	4	mC	4	1	122
# RL	RLC14069	NA	4	mC	4	1	123
# RL	RLC21532	NA	4	mC	4	1	124
# RL	RLG00006	A	1110	mC	1110	1	125
# RL	RLG00024	A	256	mC	256	1	126
# RL	RLG00121	NA	6	mC	6	1	127
# RL	RLG00339	NA	30	mC	30	1	128
# RL	RLG00911	NA	6	mC	6	1	129
# RL	RLG01911	NA	12	mC	12	1	130
# RL	RLG02097	NA	5	mC	5	1	131
# RL	RLG03493	NA	4	mC	4	1	132
# RL	RLG04160	NA	6	mC	6	1	133
# RL	RLG04235	NA	4	mC	4	1	134
# RL	RLG05961	NA	6	mC	6	1	135
# RL	RLG28582	NA	6	mC	6	1	136
# RL	RLX03748	NA	10	mC	10	1	137
# RL	RLX10409	NA	6	mC	6	1	138

```



##### Spreading lists for Qing
- make a list of B73 spreading TEs and coordinates and spreading TEs absent in B73 with insertion coordinates

```{r eval=F}
setwd("/home/springer/nosha003/database/B73v4/te")
b_te <- read.delim("B73.structuralTEv2.2018.12.20.filteredTE.noctg.gff3", header=F, sep="\t")
colnames(b_te) <- c("chr", "source", "te_type", "start", "end", "dot", "strand", "dot2", "id")
b_te$TE <- substr(b_te$id, 4, 24)

setwd("/home/springer/nosha003/sarah/TEpolymorphism")
w_site <- read.delim("W22.to.B73.output2_20Sept18_site_sort.bed", header=F, sep="\t")
p_site <- read.delim("PH207.to.B73.output2_18Sept18_site_sort.bed", header=F, sep="\t")
m_site <- read.delim("Mo17.to.B73.output2_3Jan19_site_sort.bed", header=F, sep="\t")
colnames(w_site) <- c("chr", "start", "end", "TE.site", "length", "gap", "te_coord", "poly.te.site")
colnames(p_site) <- c("chr", "start", "end", "TE.site", "length", "gap", "te_coord", "poly.te.site")
colnames(m_site) <- c("chr", "start", "end", "TE.site", "length", "gap", "te_coord", "poly.te.site")
w_site$TE <- substr(w_site$TE.site, 1, 21)
p_site$TE <- substr(p_site$TE.site, 1, 21)
m_site$TE <- substr(m_site$TE.site, 1, 21)
b_site <- rbind(w_site, p_site, m_site)
b_site$genome <- substr(b_site$TE, 9, 16)

setwd("/scratch.global/nosha003/methylation_spreading")
df <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", header=T, sep="\t")
spreading <- subset(df, df$mC_state == "mC")

spreading_dt_rl <- subset(spreading, spreading$order == "DT" | spreading$order == "RL")
spreading_dt_rl$genome <- substr(spreading_dt_rl$TE, 9, 16)

spreading_b <- subset(spreading_dt_rl, spreading_dt_rl$genome == "Zm00001d")
spreading_other <- subset(spreading_dt_rl, spreading_dt_rl$genome != "Zm00001d")

library(dplyr)
spreading_b_te <- left_join(spreading_b, b_te, by="TE")
B73_TE <- spreading_b_te[,c(1, 8,11,12)]
B73_TE$spreading_te <- "B73_present"
B73_TE_na <- na.omit(B73_TE)
# 1289
setwd("/scratch.global/nosha003/methylation_spreading")
write.table(B73_TE_na, "spreading_te_B73present_coord.txt", quote=F, row.names=F, sep="\t")


spreading_other_coord <- left_join(spreading_other, b_site, by="TE")
B73_site <- spreading_other_coord[,c(1,8:10)]
B73_site$spreading_te <- "B73_absent"
B73_site_na <- na.omit(B73_site)
# 1052
setwd("/scratch.global/nosha003/methylation_spreading")
write.table(B73_site_na, "spreading_te_B73absent_coord.txt", quote=F, row.names=F, sep="\t")
```





##### Look at Gene Expression for spreading and non-spreading families
** NOT INCLUDING IN THIS PAPER **

```{r eval=F}
library(dplyr)
library(reshape2)
library(ggplot2)

setwd("/home/springer/zhoux379/projects/briggs/data/71_output")
fpkm <- read.delim("FPKM_B73.tsv", header=T, sep="\t")
setwd("/home/springer/nosha003/methylation_spreading/")
geneTIR <- read.delim("closestGenetoTIR.txt", header=F, sep="\t")
geneLTR <- read.delim("closestGenetoLTR.txt", header=F, sep="\t")

gene <- rbind(geneTIR, geneLTR)
gene$gid <- substr(gene$V14, 6, 19)
gene$TE <- substr(gene$V9, 4, 21)
gene$fam <- substr(gene$V9, 4, 11)
gene$order <- substr(gene$V9, 4, 5)

setwd("/home/springer/nosha003/database/B73v4")
synteny <- read.delim("maize_syntenic_genes_Alex_paper.txt", header=T, sep="\t")

setwd("/scratch.global/nosha003/methylation_spreading")
df <- read.delim("all_low_empty_TE_fam_proportion_20Dec.txt", header=T, sep="\t")

df_spreading <- subset(df, df$prop_mC > 0.80)
# 40
df_insertion <- subset(df, df$prop_mC < 0.40)
# 149

gene_near <- subset(gene, gene$V16 != 0 & gene$V16 < 500)
#gene_near <- subset(gene, gene$V16 == 0)
df_spreading_gene <- left_join(df_spreading, gene_near[,c(16,17,19)], by="fam")
df_spreading_gene_fpkm <- left_join(df_spreading_gene, fpkm, by="gid")
df_insertion_gene <- left_join(df_insertion, gene_near[,c(16,17,19)], by="fam")
df_insertion_gene_fpkm <- left_join(df_insertion_gene, fpkm, by="gid")

df_spreading_synteny <- df_spreading_gene_fpkm %>% mutate(synteny = ifelse(df_spreading_gene_fpkm$gid %in% synteny$B73maize1_Gene, "B1_syntenic", ifelse(df_spreading_gene_fpkm$gid %in% synteny$B73maize2_Gene, "B2_syntenic", "non_syntenic")))
df_insertion_synteny <- df_insertion_gene_fpkm %>% mutate(synteny = ifelse(df_insertion_gene_fpkm$gid %in% synteny$B73maize1_Gene, "B1_syntenic", ifelse(df_insertion_gene_fpkm$gid %in% synteny$B73maize2_Gene, "B2_syntenic", "non_syntenic")))

df_spreading_melt <- melt(df_spreading_synteny[,c(1,3,10:34)], id=c("gid", "order", "cluster", "synteny"))
df_spreading_expr <- df_spreading_melt %>% mutate(expr = ifelse(value > 1, 1, 0))
df_spreading_prop <- df_spreading_expr %>% group_by(order, cluster, variable) %>% summarise(count = n(), expressed = sum(expr)) %>% mutate(prop = expressed / count)
df_spreading_prop_na <- na.omit(df_spreading_prop)
df_spreading_syntenic <- subset(df_spreading_expr, df_spreading_expr$synteny != "non_syntenic")
df_spreading_prop_syntenic <- df_spreading_syntenic %>% group_by(order, cluster, variable) %>% summarise(count = n(), expressed = sum(expr)) %>% mutate(prop = expressed / count)
df_spreading_prop_syntenic_na <- na.omit(df_spreading_prop_syntenic)
df_spreading_nonsyntenic <- subset(df_spreading_expr, df_spreading_expr$synteny == "non_syntenic")
df_spreading_prop_nonsyntenic <- df_spreading_nonsyntenic %>% group_by(order, cluster, variable) %>% summarise(count = n(), expressed = sum(expr)) %>% mutate(prop = expressed / count)
df_spreading_prop_nonsyntenic_na <- na.omit(df_spreading_prop_nonsyntenic)

df_insertion_melt <- melt(df_insertion_synteny[,c(1,3,10:34)], id=c("gid", "order", "cluster", "synteny"))
df_insertion_expr <- df_insertion_melt %>% mutate(expr = ifelse(value > 1, 1, 0))
df_insertion_prop <- df_insertion_expr %>% group_by(order, cluster, variable) %>% summarise(count = n(), expressed = sum(expr)) %>% mutate(prop = expressed / count)
df_insertion_prop_na <- na.omit(df_insertion_prop)
df_insertion_syntenic <- subset(df_insertion_expr, df_insertion_expr$synteny != "non_syntenic")
df_insertion_prop_syntenic <- df_insertion_syntenic %>% group_by(order, cluster, variable) %>% summarise(count = n(), expressed = sum(expr)) %>% mutate(prop = expressed / count)
df_insertion_prop_syntenic_na <- na.omit(df_insertion_prop_syntenic)
df_insertion_nonsyntenic <- subset(df_insertion_expr, df_insertion_expr$synteny == "non_syntenic")
df_insertion_prop_nonsyntenic <- df_insertion_nonsyntenic %>% group_by(order, cluster, variable) %>% summarise(count = n(), expressed = sum(expr)) %>% mutate(prop = expressed / count)
df_insertion_prop_nonsyntenic_na <- na.omit(df_insertion_prop_nonsyntenic)

setwd("/scratch.global/nosha003/methylation_spreading")
pdf("low_empty_spreading_gene_syntenic.pdf")
#pdf("low_empty_spreading_gene_overlap_syntenic.pdf")
ggplot(df_spreading_prop_syntenic_na, aes(x=cluster, y=prop)) + geom_boxplot(aes(color=cluster)) + geom_jitter(shape=16, position=position_jitter(0.2)) + theme_classic() + facet_grid(order ~ .) + ylim(0,1)
dev.off()
pdf("low_empty_spreading_gene_nonsyntenic.pdf")
#pdf("low_empty_spreading_gene_overlap_nonsyntenic.pdf")
ggplot(df_spreading_prop_nonsyntenic_na, aes(x=cluster, y=prop)) + geom_boxplot(aes(color=cluster)) + geom_jitter(shape=16, position=position_jitter(0.2)) + theme_classic() + facet_grid(order ~ .) + ylim(0,1)
dev.off()

pdf("low_empty_insertion_gene_syntenic.pdf")
#pdf("low_empty_insertion_gene_overlap_syntenic.pdf")
ggplot(df_insertion_prop_syntenic_na, aes(x=cluster, y=prop)) + geom_boxplot(aes(color=cluster)) + geom_jitter(shape=16, position=position_jitter(0.2)) + theme_classic() + facet_grid(order ~ .) + ylim(0,1)
dev.off()
pdf("low_empty_insertion_gene_nonsyntenic.pdf")
#pdf("low_empty_insertion_gene_overlap_nonsyntenic.pdf")
ggplot(df_insertion_prop_nonsyntenic_na, aes(x=cluster, y=prop)) + geom_boxplot(aes(color=cluster)) + geom_jitter(shape=16, position=position_jitter(0.2)) + theme_classic() + facet_grid(order ~ .) + ylim(0,1)
dev.off()


df_insertion_prop_nonsyntenic_na$synteny <- "nonsyntenic"
df_insertion_prop_syntenic_na$synteny <- "syntenic"
df_spreading_prop_nonsyntenic_na$synteny <- "nonsyntenic"
df_spreading_prop_syntenic_na$synteny <- "syntenic"
insertion <- rbind(df_insertion_prop_nonsyntenic_na, df_insertion_prop_syntenic_na)
spreading <- rbind(df_spreading_prop_nonsyntenic_na, df_spreading_prop_syntenic_na)

setwd("/scratch.global/nosha003/methylation_spreading")
pdf("low_empty_insertion_gene.pdf")
#pdf("low_empty_insertion_gene_overlap.pdf")
ggplot(insertion, aes(x=synteny, y=prop)) + geom_boxplot(aes(color=synteny)) + geom_jitter(shape=16, position=position_jitter(0.2)) + theme_classic() + facet_grid(order ~ .) + ylim(0,1)
dev.off()
pdf("low_empty_spreading_gene.pdf")
#pdf("low_empty_spreading_gene_overlap.pdf")
ggplot(spreading, aes(x=synteny, y=prop)) + geom_boxplot(aes(color=synteny)) + geom_jitter(shape=16, position=position_jitter(0.2)) + theme_classic() + facet_grid(order ~ .) + ylim(0,1)
dev.off()

```



** Example of a polymorphic site near genes (500bp)… is the gene differentially expressed with and without TE?? **
```{r eval=F}
library(dplyr)
library(reshape2)
library(ggplot2)

setwd("/home/springer/zhoux379/projects/briggs/data/71_output")
#fpkm <- read.delim("FPKM_B73.tsv", header=T, sep="\t")
fpkm <- read.delim("fpkm.tsv", header=T, sep="\t")
fpkm_bm <- fpkm[,c(1:6,8)]

setwd("/home/springer/nosha003/methylation_spreading/")
geneTIR <- read.delim("closestGenetoTIR.txt", header=F, sep="\t")
geneLTR <- read.delim("closestGenetoLTR.txt", header=F, sep="\t")

gene <- rbind(geneTIR, geneLTR)
gene$gid <- substr(gene$V14, 6, 19)
gene$TE <- substr(gene$V9, 4, 21)
gene$fam <- substr(gene$V9, 4, 11)
gene$order <- substr(gene$V9, 4, 5)

setwd("/home/springer/nosha003/database/B73v4")
synteny <- read.delim("maize_syntenic_genes_Alex_paper.txt", header=T, sep="\t")

setwd("/home/springer/nosha003/sarah/TEpolymorphism/")
insertion <- read.delim("Mo17emptysite_methylation.txt", header=F, sep="\t")
df_insertion <- insertion[,1:8]
colnames(df_insertion) <- c("chr", "start", "end", "TE.site", "length", "gap", "coord", "poly.site")
df_insertion$fam <- substr(df_insertion$TE.site, 1, 8)

gene_near <- subset(gene, gene$V16 < 0 & gene$V16 > -500)
insertion_gene <- left_join(df_insertion, gene_near[,c(16,17,19)], by="fam")
insertion_gene_fpkm <- left_join(insertion_gene, fpkm_bm, by="gid")

insertion_synteny <- insertion_gene_fpkm %>% mutate(synteny = ifelse(insertion_gene_fpkm$gid %in% synteny$B73maize1_Gene, "B1_syntenic", ifelse(insertion_gene_fpkm$gid %in% synteny$B73maize2_Gene, "B2_syntenic", "non_syntenic")))
insertion_synteny$B73 <- (insertion_synteny$BR001 + insertion_synteny$BR002 + insertion_synteny$BR004) / 3 
insertion_synteny$Mo17 <- (insertion_synteny$BR003 + insertion_synteny$BR005 + insertion_synteny$BR007) / 3 
insertion_synteny$BM_diff <- log2(insertion_synteny$Mo17) - log2(insertion_synteny$B73)
insertion_synteny_Mhigh_Blow <- subset(insertion_synteny, insertion_synteny$BM_diff > 2 & insertion_synteny$BM_diff != Inf)

# Gene: Zm00001d007245 
# TE 92bp Upstream: RLG00001Zm00001d00962
# B73 TE present 0.06314675
# Mo17 TE absent 41.5773141
# Log2FC = 9.362872


insertion_synteny_RLC00014 <- subset(insertion_synteny, insertion_synteny$fam == "RLC00014")
insertion_synteny_RLC00014_Mhigh <- subset(insertion_synteny_RLC00014, insertion_synteny_RLC00014$B73 < insertion_synteny_RLC00014$Mo17)

# Gene: Zm00001d019622 
# TE 279bp Upstream: RLC00014Zm00001d00804
# B73 TE present 0.000000
# Mo17 TE absent 3.24238229


insertion_synteny_RLG00001 <- subset(insertion_synteny, insertion_synteny$fam == "RLG00001")
insertion_synteny_RLG00001_Mhigh <- subset(insertion_synteny_RLG00001, insertion_synteny_RLG00001$B73 < 0.5 & insertion_synteny_RLG00001$Mo17 > 1)
# Gene: Zm00001d007632
# TE 181bp upstream: RLG00001Zm00001d12603
# B73 TE present 0.17865434
# Mo17 TE absent 2.504393

```




## Spreading Reproducibility
- spreading distance and consistency across genotypes
  → For all insertion sites in B73 with a TE in all W22,PH207,and Mo17…   
  → how often do we see spreading? (plot of proportion X/3 genotypes with a change in methylation)
  → how far does methylation spread? (violin plot of distance)

```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

# find insertion site in B73 with TE in W22, PH207, and Mo17
# setwd("/home/springer/nosha003/sarah/TEpolymorphism/")
# B73_w <- read.delim("W22_B73emptysite_methylation_bins.txt", header=F, sep="\t")
# B73_p <- read.delim("PH207_B73emptysite_methylation_bins.txt", header=F, sep="\t")
# B73_m <- read.delim("Mo17_B73emptysite_methylation_bins.txt", header=F, sep="\t")
# 
# w <- data.frame(unique(B73_w$V10))
# w$Wcount <- 1
# p <- data.frame(unique(B73_p$V10))
# p$Pcount <- 1
# m <- data.frame(unique(B73_m$V10))
# m$Mcount <- 1
# 
# w$id <- substr(w[,1], 17, 21)
# w$fam <- substr(w[,1], 1, 8)
# p$id <- substr(p[,1], 17, 21)
# p$fam <- substr(p[,1], 1, 8)
# m$id <- substr(m[,1], 17, 21)
# m$fam <- substr(m[,1], 1, 8)
# 
# 
# library(dplyr)
# wp <- full_join(w, p, by=c("id", "fam"))
# wpm <- full_join(wp, m, by=c("id", "fam"))
# colnames(wpm) <- c("W22", "Wcount", "id", "fam", "PH207", "Pcount", "Mo17", "Mcount")
# wpm[is.na(wpm)] <- 0
# wpm_count <- wpm %>% mutate(TE_count = (Wcount + Pcount + Mcount))


# determine if the TE has spreaidng in 1, 2, or 3 of the genotypes with the TE
# setwd("/scratch.global/nosha003/methylation_spreading")
# spreading <- read.delim("all_low_b_flank_mC_20Dec.txt", header=T, sep="\t")
# spreading$id <- substr(spreading$TE, 17, 21)
# spreading$fam <- substr(spreading$TE, 1, 8)
# 
# wpm_count2 <- wpm_count[,c(3:4,9)]
# 
# B73_wpm_spreading <- inner_join(wpm_count2, spreading, by=c("id", "fam"))
# B73_wpm_spreading_tally <- B73_wpm_spreading %>% group_by(id, fam) %>% tally
# B73_wpm_spreading_count <- left_join(B73_wpm_spreading_tally, B73_wpm_spreading, by=c("id", "fam"))
# B73_wpm_spreading_count$spreading_count <- B73_wpm_spreading_count$n / B73_wpm_spreading_count$TE_count
# 
# wpm_count <- subset(B73_wpm_spreading_count, B73_wpm_spreading_count$TE_count == 3)
# # 52 TEs 
# wpm_count <- B73_wpm_spreading_count %>% group_by(TE_count, n) %>% summarise(count=n())
# wpm_total <- B73_wpm_spreading_count %>% group_by(TE_count) %>% summarise(total=n())
# wpm_count_total <- left_join(wpm_count, wpm_total, by="TE_count")
# wpm_count_total$prop <- wpm_count_total$count / wpm_count_total$total


# plot 
# library(ggplot2)
# pdf("spreading_reproducibility.pdf")
# ggplot(wpm_count_total) + geom_bar(aes(x=TE_count, y=prop, fill=n), stat="identity") + theme_classic() 
# dev.off()
# 
# pdf("wpm_spreading_reproducibility.pdf")
# wpm_count_total_3 <- subset(wpm_count_total, wpm_count_total$TE_count == 3)
# ggplot(wpm_count_total_3) + geom_bar(aes(x=TE_count, y=prop, fill=n), stat="identity") + theme_classic() 
# dev.off()




#### sarah list of shared TEs in W, P, M and site.defined in B
/home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.txt

sed 's/:/\t/g' /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.txt | sed 's/-/\t/g' | sed '1d' > /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.txt
awk '{print $5"\t"$6"\t"$7"\t"$1"\t""B73.site"}' /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.txt | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.B73coord.txt
awk '{print $2"\t"$3"\t"$4"\t"$1"\t""W22.TE"}' /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.txt | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.W22coord.txt
awk '{print $9"\t"$10"\t"$11"\t"$1"\t""PH207.TE"}' /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.txt | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.PH207coord.txt
awk '{print $13"\t"$14"\t"$15"\t"$1"\t""Mo17.TE"}' /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.txt | sort -k 1,1 -k 2,2n > /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.Mo17coord.txt

module load bedtools
bedtools closest -k 1 -t first -D a -b /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.B73coord.txt -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /scratch.global/nosha003/methylation_spreading/reproducibility.B73.site.txt
bedtools closest -k 1 -t first -D a -b /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.W22coord.txt -a /home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt > /scratch.global/nosha003/methylation_spreading/reproducibility.W22.TE.txt
bedtools closest -k 1 -t first -D a -b /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.PH207coord.txt -a /home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio.txt > /scratch.global/nosha003/methylation_spreading/reproducibility.PH207.TE.txt
bedtools closest -k 1 -t first -D a -b /home/springer/nosha003/sarah/TEpolymorphism/nonshared.site.defined.absent.B73.sep.Mo17coord.txt -a /home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt > /scratch.global/nosha003/methylation_spreading/reproducibility.Mo17.TE.txt

cd /scratch.global/nosha003/methylation_spreading/
awk '{if ($12 == 0 && $4 < 0.20) print $0}' reproducibility.B73.site.txt > reproducibility.B73.site.unmethylated.txt
awk '{if ($12 != 0 && $12 < 100 && $12 > -100 && $4 > 0.40) print $0}' reproducibility.W22.TE.txt > reproducibility.W22.TE.methylated.txt
awk '{if ($12 != 0 && $12 < 100 && $12 > -100 && $4 > 0.40) print $0}' reproducibility.PH207.TE.txt > reproducibility.PH207.TE.methylated.txt
awk '{if ($12 != 0 && $12 < 100 && $12 > -100 && $4 > 0.40) print $0}' reproducibility.Mo17.TE.txt > reproducibility.Mo17.TE.methylated.txt


# R
setwd("/scratch.global/nosha003/methylation_spreading/")
b.site <- read.delim("reproducibility.B73.site.unmethylated.txt", header=F, sep="\t")
w.TE <- read.delim("reproducibility.W22.TE.methylated.txt", header=F, sep="\t")
p.TE <- read.delim("reproducibility.PH207.TE.methylated.txt", header=F, sep="\t")
m.TE <- read.delim("reproducibility.Mo17.TE.methylated.txt", header=F, sep="\t")

b <- data.frame(unique(b.site$V10))
w <- data.frame(unique(w.TE$V10))
p <- data.frame(unique(p.TE$V10))
m <- data.frame(unique(m.TE$V10))
colnames(b) <- "TE"
colnames(w) <- "TE"
colnames(p) <- "TE"
colnames(m) <- "TE"
b$b.site <- 1
w$w.TE <- 1
p$p.TE <- 1
m$m.TE <- 1

library(dplyr)
# b = 88
b.w <- inner_join(b, w, by="TE")
# 66
b.m <- inner_join(b, m, by="TE")
# 60
b.p <- inner_join(b, p, by="TE")
# 62

b.p.m <- inner_join(b.p, b.m, by="TE")
# 54
b.p.w <- inner_join(b.p, b.w, by="TE")
# 55
b.m.w <- inner_join(b.m, b.w, by="TE")
# 55

b.p.m.w <- inner_join(b.p.w, b.m, by="TE")
# 51

bp <- full_join(b, p, by="TE")
bpm <- full_join(bp, m, by="TE")
bpmw <- full_join(bpm, w, by="TE")
# 613
bpmw.site <- subset(bpmw, bpmw$b.site == 1)
# 88

bpmw.site[is.na(bpmw.site)] <- 0
bpmw.site.count <- bpmw.site %>% group_by(TE) %>% mutate(TE_count = p.TE + m.TE + w.TE) 
bpmw.site.count$order <- substr(bpmw.site.count$TE, 1, 2)
bpmw.site.tecount <- bpmw.site.count %>% group_by(order, TE_count) %>% tally
bpmw.site.total <- bpmw.site.count %>% group_by(order, b.site) %>% tally
bpmw.site.prop <- bpmw.site.tecount %>% mutate(prop = ifelse(order == "DH", n/5, ifelse(order == "DT", n/51, ifelse(order == "RI", n/1, ifelse(order == "RL", n/31, NA)))))
bpmw.site.prop.dt.rl <- subset(bpmw.site.prop, bpmw.site.prop$order == "DT" | bpmw.site.prop$order == "RL")

library(ggplot2)
pdf("spreading_reproducibility_plot.pdf")
ggplot(bpmw.site.prop.dt.rl) + geom_bar(aes(x=order, y=prop, fill=TE_count), stat="identity") + theme_classic() 
dev.off()


# only those that have spreading in at least 1 genotype
bpmw.site[is.na(bpmw.site)] <- 0
bpmw.site.count <- bpmw.site %>% group_by(TE) %>% mutate(TE_count = p.TE + m.TE + w.TE) 
bpmw.site.count.spreading <- subset(bpmw.site.count, bpmw.site.count$TE_count > 0)
bpmw.site.count.spreading$order <- substr(bpmw.site.count.spreading$TE, 1, 2)
bpmw.site.tecount <- bpmw.site.count.spreading %>% group_by(order, TE_count) %>% tally
bpmw.site.total <- bpmw.site.count.spreading %>% group_by(order, b.site) %>% tally
bpmw.site.prop <- bpmw.site.tecount %>% mutate(prop = ifelse(order == "DH", n/5, ifelse(order == "DT", n/42, ifelse(order == "RL", n/28, NA))))
bpmw.site.prop.dt.rl <- subset(bpmw.site.prop, bpmw.site.prop$order == "DT" | bpmw.site.prop$order == "RL")

setwd("/scratch.global/nosha003/methylation_spreading")
write.table(bpmw.site.count, "spreading_reproducibility_site.txt", quote=F, row.names=F, sep="\t")
write.table(bpmw.site.prop.dt.rl, "spreading_reproducibility.txt", quote=F, row.names=F, sep="\t")


library(ggplot2)
pdf("spreading_reproducibility_plot2.pdf")
ggplot(bpmw.site.prop.dt.rl) + geom_bar(aes(x=order, y=prop, fill=TE_count), stat="identity") + theme_classic() 
dev.off()


#### examples

# /home/springer/sna/polyTE/sna/W22.to.B73.output2_20Sept18.bed
# /home/springer/sna/polyTE/sna/W22.to.PH207.output2_25Sept18.bed
# /home/springer/sna/polyTE/sna/W22.to.Mo17.output2_26Sept18.bed

# RLC00034Zm00004b00002	1	1	1	1	3 --> W22 (1:10915658-10922421), PH207 (01	10734098	10734098), Mo17 (1	11544018	11550781), B73 (1	10349826	10349831)
# DHH00002Zm00004b04438	1	1	1	1	3 --> W22 (), PH207 (), Mo17 (), B73 ()
# DTH00102Zm00004b00939	1	1	1	1	3 --> W22 (1:65287057-65287193), PH207 (01	62950628	62950764), Mo17 (1	66703048	66703189), B73 (1	64950960	64950963)
# DTH11636Zm00004b00024	1	1	1	1	3 --> W22 (1:90576346-90576575), PH207 (01	87307979	87307979), Mo17 (1	90947407	90947636), B73 (1	90963296	90963299)
# DTA00313Zm00004b00024	1	1	1	1	3 --> W22 (1:109887619-109888168), PH207 (01	106491676	106492225), Mo17 (1	109706111	109706660), B73 (1	109844506	109844514)
#### RLG03666Zm00004b00001	1	1	1	1	3 --> W22 (1:159195690-159201210), PH207 (01	157466790	157477894), Mo17 (1	157915093	157920613), B73 (1	157832417	157832420)
# DTH00098Zm00004b00051	1	1	1	1	3 --> W22 (), PH207 (), Mo17 (), B73 ()
# RLC00064Zm00004b00014	1	1	1	1	3 --> W22 (), PH207 (), Mo17 (), B73 ()
# DTA00117Zm00004b00163	1	1	1	1	3 --> W22 (), PH207 (), Mo17 (), B73 ()
# RLX00172Zm00004b00029	1	1	1	1	3 --> W22 (), PH207 (), Mo17 (), B73 ()
15746679
15747789
```



## Attributes of spreading families / spreading TEs
- differences between spreading and non-spreading members of a family
	→ age
	→ distance to genes
	→ length of TE

```{r eval=F}

```

## Spreading Regions
```{r eval=F}
# find low mC insertion sites in W22 with spreading TEs in B73
library(reshape2)
library(dplyr)
library(ggplot2)
library(RColorBrewer)

setwd("/home/springer/nosha003/methylation_spreading/")
ltr_cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
tir_cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster <- rbind(ltr_cluster, tir_cluster)
cluster$fam <- row.names(cluster)
cluster <- cluster[,c(81:82)]
colnames(cluster) <- c("cluster", "fam")

setwd("/home/springer/nosha003/sarah/TEpolymorphism/")
w22 <- read.delim("W22emptysite_methylation_bins.txt", header=F, sep="\t")
#w22 <- read.delim("all_empty_average.txt", header=F, sep="\t")
w22_empty <- subset(w22, w22$V15 == 0)
w22_empty_melt <- melt(w22_empty[,c(4:6,10)], id="V10")
colnames(w22_empty_melt) <- c("TE", "context", "mC_ratio")
w22_empty_melt_na <- na.omit(w22_empty_melt)
w22_empty_melt_na$fam <- substr(w22_empty_melt_na$TE, 1, 8)
w22_empty_melt_na$class <- substr(w22_empty_melt_na$TE, 1, 2)
w22_empty_melt_cluster <- left_join(w22_empty_melt_na, cluster, by="fam")
w22_empty_region_cluster <- w22_empty_melt_cluster %>% group_by(context, class, cluster) %>% summarise(region_count = n()) 
w22_empty_region_cluster_na <- na.omit(w22_empty_region_cluster)
w22_empty_category_cluster <- w22_empty_melt_cluster %>% group_by(context, class, cluster) %>% mutate(mC_state = ifelse(context == "V4", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "V5", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "V6", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(context, class, cluster, mC_state) 
w22_empty_category_cluster_na <- na.omit(w22_empty_category_cluster)
w_cg <- subset(w22_empty_category_cluster_na, w22_empty_category_cluster_na$context == "V4")
w_low <- subset(w_cg, w_cg$mC_state == "C")
w_low$TE <- substr(w_low$TE, 1, 21)

setwd("/scratch.global/nosha003")
b <- read.delim("B73.to.W22.output2.B73bins.txt", header=F, sep="\t")
#b <- read.delim("all.to.all.output2.bins.txt", header=F, sep="\t")
b$TE <- substr(b$V11, 1, 21)

w_low_b <- inner_join(w_low, b, by="TE")
w_low_b_flank <- subset(w_low_b, w_low_b$V12 != 0 & abs(w_low_b$V12) < 200)
w_low_b_flank_na <- na.omit(w_low_b_flank)
cluster <- data.frame(unique(w_low_b_flank_na[,c(1, 6)]))
w_low_b_flank_category <- w_low_b_flank_na %>% group_by(TE) %>% mutate(cg = mean(V4), chg = mean(V5), chh = mean(V6))
w_low_b_flank_category_cluster <- left_join(w_low_b_flank_category, cluster, by="TE")
w_low_b_flank_category_cluster$order <- substr(w_low_b_flank_category$TE, 1, 2)
w_low_b_flank_melt <- melt(w_low_b_flank_category_cluster)
w_low_b_flank_region <- w_low_b_flank_melt %>% group_by(variable, order, cluster.x) %>% summarise(region_count = n()) 
w_low_b_flank_category_cluster <- w_low_b_flank_melt %>% group_by(variable, order, cluster.x) %>% mutate(mC_state = ifelse(variable == "cg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chh", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(variable, order, cluster.x, mC_state) 
w_low_b_mC <- subset(w_low_b_flank_category_cluster, w_low_b_flank_category_cluster$context == "V4" & w_low_b_flank_category_cluster$mC_state == "mC")
length(unique(w_low_b_mC$TE))
## 312 TEs

# subset to bins surrounding spreading TEs (remove overlapping bins and unmethylated bins)
library(dplyr)
w_low_b_mC_bins <- inner_join(w_low_b_mC, b, by = "TE")
bins.spreading <- subset(w_low_b_mC_bins, w_low_b_mC_bins$V12 != 0 & w_low_b_mC_bins$V4 > 0.40)


# merge spreading bins
library(tidygenomics)
bins.spreading.merge <- genome_cluster(bins.spreading, by=c("V1", "V2", "V3"), max_distance=250)
length(unique(bins.spreading.merge$cluster_id))
## 551 regions

bins.spreading.merge.coord <- bins.spreading.merge %>% group_by(cluster_id) %>% mutate(chr = V1, start = min(V2), end = max(V3))

# take 2 closest regions (upstream and downstream?)
bins.spreading.merge.coord$dist <- abs(bins.spreading.merge.coord$V12)
#bins.spreading.merge.min <- bins.spreading.merge.coord %>% group_by(TE) %>% filter(V12 == min(V12))
bins.spreading.merge.min <- bins.spreading.merge.coord %>% group_by(TE) %>% top_n(-2, dist)
# 1525 --> why is it returning more than 2 lines for each TE?
length(unique(bins.spreading.merge.min$cluster_id))
# 317 --> dataframe lists multiple lines for each cluster since they are unique
length(unique(bins.spreading.merge.min$TE))
# 312

bins.spreading.merge.min.uniq <- data.frame(unique(bins.spreading.merge.min[,c(1,3:5,25:29)]))
# 320
colnames(bins.spreading.merge.min.uniq) <- c("TE", "fam", "order", "cluster", "merge", "chr", "start", "end", "dist")
bins.spreading.merge.min.uniq$length <- bins.spreading.merge.min.uniq$end - bins.spreading.merge.min.uniq$start
bins.spreading.merge.min.uniq$dist2 <- bins.spreading.merge.min.uniq$dist + bins.spreading.merge.min.uniq$length

setwd("/scratch.global/nosha003/methylation_spreading")
write.table(bins.spreading.merge.min.uniq, "spreading_bins.txt", quote=F, sep="\t", row.names=F)
```


## DMRs 
** still not sure about these methods?? **
- what proportion of DMRs (in a single contrast) overlap with spreading regions?

```{r eval=F}
# B73 to W22 DMR calls (high methylation in B (TE), low methylation in W (empty site))
# qsub /home/springer/nosha003/scripts/dss_swift_BW_call.sh
/scratch.global/nosha003/wgbs_schmitz/BW_swift_DSS_dml_nosmoothing_allchr_CG_4sites_2cov.txt


library(dplyr)
library(tidygenomics)
setwd("/scratch.global/nosha003/methylation_spreading")
spreading <- read.delim("spreading_bins.txt", header=T, sep="\t")

# intersect DMRs with spreading regions
setwd("/scratch.global/nosha003/wgbs_schmitz/")
dmr <- read.delim("BW_swift_DSS_dml_nosmoothing_allchr_CG_4sites_2cov.txt", header=T, sep="\t")
# 242877
dmr.coord <- dmr[,c(1:3,15)]
colnames(dmr.coord) <- c("V1", "V2", "V3", "diff")
dmr.merge <- genome_cluster(dmr.coord, by=c("V1", "V2", "V3"), max_distance=200)
dmr.merge.coord <- dmr.merge %>% group_by(cluster_id) %>% mutate(chr = V1, start = min(V2), end = max(V3))
dmr.merge.coord.uniq <- data.frame(unique(dmr.merge.coord[,c(5:8)]))
# 152847

dmr.spreading <- genome_intersect(dmr.merge.coord.uniq, spreading, by=c("chr", "start", "end"), mode="both")
length(unique(dmr.spreading$cluster_id))
# 754 / 152847 = 0.5%

dmr.spreading.max <- dmr.spreading %>% group_by(order) %>% filter(dist == max(dist))
# within 25kb of a TE???
```

- only B > W DMRs (focusing on TEs present in B73 and absent in W22)
- only DMRs not overlapping TEs 
- calculate proportion of DMRs within 500bp of a spreading TE

```{r eval=F}
library(dplyr)
library(tidygenomics)

setwd("/home/springer/nosha003/database/B73v4/te")
b_te <- read.delim("B73.structuralTEv2.2018.12.20.filteredTE.noctg.gff3", header=F, sep="\t")
colnames(b_te) <- c("chr", "source", "te_type", "start", "end", "dot", "strand", "dot2", "id")
b_te$TE <- substr(b_te$id, 4, 24)

setwd("/scratch.global/nosha003/wgbs_schmitz/")
dmr <- read.delim("BW_swift_DSS_dml_nosmoothing_allchr_CG_4sites_2cov.txt", header=T, sep="\t")
# 242877
dmr.coord <- dmr[,c(1:3,15)]
colnames(dmr.coord) <- c("chr", "start", "end", "diff")

dmr.nonTE <- genome_intersect(dmr.coord, b_te, by=c("chr", "start", "end"), mode="anti")
# 171292

dmr.merge <- genome_cluster(dmr.nonTE, by=c("chr", "start", "end"), max_distance=200)
dmr.merge.coord <- dmr.merge %>% group_by(cluster_id) %>% mutate(chr = chr, start = min(start), end = max(end))
dmr.merge.coord.uniq <- data.frame(unique(dmr.merge.coord[,c(1:3,5)]))
# 152847 (when TE overlap is kept)
# 104630 (when TEs are removed)

setwd("/scratch.global/nosha003/methylation_spreading")
spreading <- read.delim("spreading_te_B73present_coord.txt", header=T, sep="\t")
spreading_upstream <- spreading[,c(1:3)]
spreading_upstream$end <- spreading_upstream$start
spreading_upstream$start <- spreading_upstream$start - 500
spreading_downstream <- spreading[,c(1:2,4)]
spreading_downstream$start <- spreading_downstream$end
spreading_downstream$end <- spreading_downstream$end + 500

dmr.spreading.upstream <- genome_intersect(dmr.merge.coord.uniq, spreading_upstream, by=c("chr", "start", "end"), mode="both")
length(unique(dmr.spreading.upstream$cluster_id))
# 343 DMRs are within 500bp UPSTREAM of a spreading TE

dmr.spreading.downstream <- genome_intersect(dmr.merge.coord.uniq, spreading_downstream, by=c("chr", "start", "end"), mode="both")
length(unique(dmr.spreading.downstream$cluster_id))
# 349 DMRs are within 500bp DOWNSTREAM of a spreading TE

dmr.spreading <- rbind(dmr.spreading.upstream, dmr.spreading.downstream)
length(unique(dmr.spreading$cluster_id))
# 682 / 104630 = 0.65% of merged DMRs are within 500bp of a spreading TE


# non-merged
dmr.spreading.upstream <- genome_intersect(dmr.nonTE, spreading_upstream, by=c("chr", "start", "end"), mode="both")
nrow(dmr.spreading.upstream)
# 916 DMRs are within 500bp UPSTREAM of a spreading TE

dmr.spreading.downstream <- genome_intersect(dmr.nonTE, spreading_downstream, by=c("chr", "start", "end"), mode="both")
nrow(dmr.spreading.downstream)
# 869 DMRs are within 500bp DOWNSTREAM of a spreading TE

dmr.spreading <- rbind(dmr.spreading.upstream, dmr.spreading.downstream)
nrow(dmr.spreading)
# 1785 / 171292 = 1.04% of DMRs are within 500bp of a spreading TE

```


##### Spreading distance
** CANNOT ACCURATELY ASSESS AS MANY REGIONS WILL RUN INTO OTHER METHYLATED REGIONS OF THE GENOME **

```{r eval=F}
# how far does spreading go?
setwd("/scratch.global/nosha003/methylation_spreading")
spreading <- read.delim("spreading_bins.txt", header=T, sep="\t")

library(dplyr)
spreading.category <- spreading %>% mutate(category = ifelse(dist2 < 100, "<100", ifelse(dist2 < 500, "<500", ifelse(dist2 < 1000, "<1kb", ">1kb")))) %>% group_by(order, category) %>% mutate(count = n())
spreading.total <- spreading %>% group_by(order) %>% mutate(total = n())
spreading.count <- left_join(spreading.category, spreading.total, by="order")
spreading.prop <- spreading.count[,c(3,12:13,24)] %>% group_by(order, category) %>% mutate(prop = count/total)
df <- unique(spreading.prop)

library(ggplot2)
pdf("spreading_dist_bar.pdf")
ggplot(df) + geom_bar(aes(x=order, y=prop, fill=category), stat="identity") + theme_classic()
dev.off()

```


###### Direction of spreading
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=90G

library(reshape2)
library(tidyverse)
library(ggplot2)
library(RColorBrewer)

setwd("/home/springer/nosha003/methylation_spreading/")
ltr_cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
tir_cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster <- rbind(ltr_cluster, tir_cluster)
cluster$fam <- row.names(cluster)
cluster <- cluster[,c(81:82)]
colnames(cluster) <- c("cluster", "fam")

setwd("/home/springer/nosha003/sarah/TEpolymorphism")
w22 <- read.delim("all_empty_average.txt", header=T, sep="\t", stringsAsFactors = F)
w22_empty_left <- subset(w22, w22$V15 < 0 & w22$V15 > -100 & w22$V4 < 0.20)
w22_empty_right <- subset(w22, w22$V15 > 0 & w22$V15 < 100 & w22$V4 < 0.20)
w22_empty_left$flank <- "upstream"
w22_empty_right$flank <- "downstream"
w22_empty <- rbind(w22_empty_left, w22_empty_right)
w22_empty$chr <- as.integer(w22_empty$V1)
w22_empty$start <- as.integer(w22_empty$V2)
w22_empty$end <- as.integer(w22_empty$V3)

setwd("/home/springer/nosha003/methylation_spreading")
b <- read.delim("all.to.all.output2.bins.txt", header=T, sep="\t", stringsAsFactors = F)
b$chr <- as.integer(b$V1)
b$start <- as.integer(b$V2)
b$end <- as.integer(b$V3)

w22_empty_b <- left_join(w22_empty, b, by=c("chr", "start", "end"))

w22_empty_melt <- melt(w22_empty_b[,c(37,21,28:30)], id=c("TE", "flank"))
colnames(w22_empty_melt) <- c("TE", "flank", "context", "mC_ratio")
w22_empty_melt_na <- na.omit(w22_empty_melt)
w22_empty_melt_na$fam <- substr(w22_empty_melt_na$TE, 1, 8)
w22_empty_melt_na$class <- substr(w22_empty_melt_na$TE, 1, 2)
w22_empty_melt_cluster <- left_join(w22_empty_melt_na, cluster, by="fam")
write.table(w22_empty_melt_cluster, "all_empty_melt_cluster_100bp_Jan2019.txt", quote=F, row.names=F, sep="\t")

# by class and cluster
w22_empty_category_cluster <- w22_empty_melt_cluster %>% group_by(context, flank, class, cluster) %>% mutate(mC_state = ifelse(context == "V4.y", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "V5.y", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), ifelse(context == "V6.y", ifelse(mC_ratio >= 0.40, "mC", ifelse(mC_ratio <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(context, flank, class, cluster, mC_state) 
w22_empty_category_cluster_na <- na.omit(w22_empty_category_cluster)

w22_empty_category_cluster_na_up <- subset(w22_empty_category_cluster_na, w22_empty_category_cluster_na$flank == "upstream")
w22_empty_category_cluster_na_down <- subset(w22_empty_category_cluster_na, w22_empty_category_cluster_na$flank == "downstream")
w22_empty_category_cluster_na_flank <- full_join(w22_empty_category_cluster_na_up, w22_empty_category_cluster_na_down, by=c("TE", "context"))
w22_empty_category_cluster_na_flank_mC <- w22_empty_category_cluster_na_flank %>% unite(flank_mC, c(flank.x, mC_state.x, flank.y, mC_state.y), sep = "_", remove = FALSE)
df <- na.omit(w22_empty_category_cluster_na_flank_mC)

setwd("/scratch.global/nosha003/methylation_spreading")
write.table(df, "all_empty_category_cluster_na_flank_mC_100bp_Jan2019.txt", quote=F, row.names=F, sep="\t")

mC_region <- df %>% group_by(context, class.x, cluster.x) %>% summarise(region_count = n())
mC_count <- df %>% group_by(context, class.x, cluster.x, flank_mC) %>% summarise(count = n())
state <- inner_join(mC_region, mC_count, by=c("context", "class.x", "cluster.x"))
proportion <- state %>% group_by(context, class.x, cluster.x, flank_mC) %>% mutate(prop = count/region_count)

df <- proportion %>% mutate(type = ifelse(flank_mC == "upstream_C_downstream_C", "unmethylated", ifelse(flank_mC == "upstream_C_downstream_intermediate", "unmethylated/intermediate", ifelse(flank_mC == "upstream_C_downstream_mC", "unmethylated/methylated", ifelse(flank_mC == "upstream_intermediate_downstream_C", "unmethylated/intermediate", ifelse(flank_mC == "upstream_intermediate_downstream_intermediate", "intermediate", ifelse(flank_mC == "upstream_intermediate_downstream_mC", "methylated/intermediate", ifelse(flank_mC == "upstream_mC_downstream_C", "unmethylated/methylated", ifelse(flank_mC == "upstream_mC_downstream_intermediate", "methylated/intermediate", ifelse(flank_mC == "upstream_mC_downstream_mC", "methylated", "NA"))))))))))

setwd("/scratch.global/nosha003/methylation_spreading")
pdf("all_empty_100bp_flank_mC_barplot_Jan2019.pdf")
ggplot(data=proportion) + geom_bar(aes(x=cluster.x, y=prop, fill=flank_mC), stat="identity") + facet_grid(class.x ~ context) + theme_classic() + theme(text = element_text(size=7)) 
dev.off()

pdf("all_empty_100bp_flank_mC_barplot2_Jan2019.pdf")
ggplot(data=df) + geom_bar(aes(x=cluster.x, y=prop, fill=type), stat="identity") + facet_grid(class.x ~ context) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_brewer(palette="Set2")
dev.off()
```
qsub /scratch.global/nosha003/direction_spreading.sh


###### subset to just spreading elements... get numbers'

```{r eval=F}
setwd("/home/springer/nosha003/methylation_spreading/")
class <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", sep="\t", header=T)
spreading <- subset(class, class$mC_state == "mC")
non.spreading <- subset(class, class$mC_state == "C")

setwd("/scratch.global/nosha003/methylation_spreading/")
df <- read.delim("all_empty_category_cluster_na_flank_mC_100bp_Jan2019.txt", header=T, sep="\t")
df_cg <- subset(df, df$context == "V4.y")
df_cg_uniq <- (unique(df_cg))
# 10865

library(dplyr)
df_spread <- inner_join(spreading, df_cg_uniq, by="TE")
# 1682
df_spread_uniq <- unique(df_spread[,c(1,7,14,20)])
table(df_spread_uniq[,3:4])
#               mC_state.y
# mC_state.x      C intermediate mC
#   C            76            0 10
#   intermediate  1            0  2
#   mC           13            1 79

# both mC (79/106=75%)
# single mC (27/106=25%)

```

** Look at those elements that were unmethylated on both sides of the empty site... what happens to the TE flanks... **


###### spreading element flanks
- find the spreading elements that have unmethylated empty sites on both sides
- then see how many of them gain methylation on both sides and how many gain methylation on one side

```{r eval=F}
library(dplyr)
library(reshape2)

setwd("/home/springer/nosha003/methylation_spreading")
df <- read.delim("chromatin_ms_supplemental_mC_site_flank.txt", header=T, sep="\t")
spreading <- subset(df, df$site_mC_state == "Unmethylated" & df$flank_mC_state == "Methylated")

empty <- read.delim("all_empty_melt_cluster_100bp_Jan2019.txt", header=T, sep="\t")
spreading_empty <- left_join(spreading, empty, by="TE")
spreading_empty_cg <- subset(spreading_empty, spreading_empty$context == "V4.y")
spreading_empty_uniq <- unique(spreading_empty_cg[,c(1,6,8)])
spreading_empty_group <- spreading_empty_uniq %>% group_by(TE, flank) %>% mutate(mC_ratio_high = max(mC_ratio))
spreading_empty_group_uniq <- unique(spreading_empty_group[,c(1,2,4)])
spreading_empty_dcast <- spreading_empty_group_uniq %>% dcast(TE ~ flank, value.var = "mC_ratio_high")
spreading_empty_dcast_call <- spreading_empty_dcast %>% mutate(flank_call = ifelse(downstream > 0.20 & upstream > 0.20, "both_spread", ifelse(downstream > 0.20 | upstream > 0.20, "single_spread", ifelse(downstream <= 0.20 & upstream <= 0.20, "no_spread", "no_call"))))
# 297

spreading_empty_dcast_call[is.na(spreading_empty_dcast_call)] <- "spread"
table(spreading_empty_dcast_call$flank_call)
#  both_spread     no_spread single_spread        spread 
#          100            78             8           111

# 219 spreading... 119/219=54% spreading on one side, 46% spreading on both sides

```







## Metaprofiles for chromatin of spreading v non-spreading (Figure 6)

```{r eval=F}
#29Jan2019
#chromatin_metaplot_spreading

#!/bin/bash
#PBS -l walltime=24:00:00,nodes=1:ppn=8,mem=60gb
#PBS -o /scratch.global/nosha003/e_o/chromatin_spreading_o
#PBS -e /scratch.global/nosha003/e_o/chromatin_spreading_e
#PBS -N chromatin_spreading
#PBS -V
#PBS -r n

cd /scratch.global/nosha003/
module load R
R CMD BATCH chromatin_metaplot_spreading_tir.R
R CMD BATCH chromatin_metaplot_spreading_ltr.R
R CMD BATCH chromatin_metaplot_spreading_tir1kb.R
wait

# qsub /scratch.global/nosha003/chromatin_metaplot_spreading.sh
```

** FUNCTION = metaplot_chromatin_spreading **
```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

metaplot_chromatin_spreading <- function(input_dir, file, fam_file, spreading, genotype, TEtype, date, context) {
library(fields)
library(dplyr)
library(reshape2)
library(ggplot2)
library(Rmisc)

setwd(input_dir)
b_tir <- read.delim(file, sep="\t", header=T)
b_tir_fam <- read.delim(fam_file, sep="\t", header=T)
class <- read.delim(spreading, sep="\t", header=T)

setwd("/scratch.global/nosha003/methylation_spreading")
b_tir <- read.delim("B73_tir_H3K4me3_orientation_metaplot.txt", sep="\t", header=T)
b_tir_fam <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt", sep="\t", header=T)
class <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", sep="\t", header=T)

b_tir_fam$length <- abs(b_tir_fam$end - b_tir_fam$start)
b_tir_fam2 <- subset(b_tir_fam, b_tir_fam$length > 1000)
  
b_tir$TEfam <- substr(b_tir$TE, 1, 8)
b_tir <- subset(b_tir, b_tir$TE %in% b_tir_fam2$TE)


b_tir_cg <- as.numeric(paste(b_tir$count))
b_tir["cg"] <- b_tir_cg
b_tir$chromatin <- ifelse(b_tir$cg >=0.05, 1, 0)
b_tir2 <- left_join(b_tir, class, by="TE")
dist = subset(b_tir2, b_tir2$relative_distance > -2000 & b_tir2$relative_distance <= 3000)

dist_spreading <- subset(dist, dist$mC_state == "mC")
dist_non.spreading <- subset(dist, dist$mC_state == "C")

dist_spreading_group <-  transform(dist_spreading, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_spreading_sum <- do.call(data.frame,aggregate(dist_spreading_group$chromatin~dist_spreading_group$group, dist_spreading_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_spreading_sum) <- c("group", "count", "sum")
dist_spreading_sum["percent_methylated"] <- dist_spreading_sum$sum / dist_spreading_sum$count
dist_spreading_sum["class"] <- "spreading"

dist_non.spreading_group <-  transform(dist_non.spreading, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_non.spreading_sum <- do.call(data.frame,aggregate(dist_non.spreading_group$chromatin~dist_non.spreading_group$group, dist_non.spreading_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_non.spreading_sum) <- c("group", "count", "sum")
dist_non.spreading_sum["percent_methylated"] <- dist_non.spreading_sum$sum / dist_non.spreading_sum$count
dist_non.spreading_sum["class"] <- "non.spreading"

p.1 <- rbind(dist_spreading_sum, dist_non.spreading_sum)

## Metaplot of percent bins methylated by spreading v non-spreading
#pdf(paste(genotype, TEtype, context, "spreading_metaplot.pdf", sep="_"))
pdf(paste(genotype, TEtype, context, "spreading_metaplot_1kb.pdf", sep="_"))
#pdf("H3K4m3_tir_spreading_test.pdf")
plot1 <- ggplot(p.1, aes(x = group, y = percent_methylated, group = class, colour = class)) + geom_line(size=1) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot1)
dev.off()
}
```

```{r eval=F}

# TIR
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K27me3_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K27me3"
)

metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K4me3_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K4me3"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K23ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K23ac"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K27ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K27ac"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K36me3_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K36me3"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K4me1_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K4me1"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K56ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K56ac"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K9ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K9ac"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_ATACseq_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "ATACseq"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_Pol_II_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "Pol_II"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K9me2_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "Jan2019",
  context = "H3K9me2"
)


# LTR
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K27me3_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K27me3"
)

metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K4me3_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K4me3"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K23ac_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K23ac"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K27ac_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K27ac"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K36me3_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K36me3"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K4me1_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K4me1"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K56ac_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K56ac"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K9ac_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K9ac"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_ATACseq_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "ATACseq"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_Pol_II_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "Pol_II"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K9me2_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "Jan2019",
  context = "H3K9me2"
)
```



##### Metaprofiles for CG, CHG, CHH
```{r eval=F}
#4Feb2019
#methylation_metaplot_spreading

#!/bin/bash
#PBS -l walltime=24:00:00,nodes=1:ppn=8,mem=60gb
#PBS -o /scratch.global/nosha003/e_o/methylation_spreading_o
#PBS -e /scratch.global/nosha003/e_o/methylation_spreading_e
#PBS -N chromatin_spreading
#PBS -V
#PBS -r n

cd /scratch.global/nosha003/
module load R
R CMD BATCH methylation_metaplot_spreading.R
R CMD BATCH methylation_metaplot_spreading_tir1kb.R
R CMD BATCH methylation_metaplot_spreading_ltr_nonoriented.R
wait

# qsub /scratch.global/nosha003/methylation_metaplot_spreading.sh
```

** FUNCTION = metaplot_mC_spreading **
```{r eval=F}
metaplot_mC_spreading <- function(input_dir, file, fam_file, spreading, genotype, TEtype, date, cluster) {
library(fields)
library(dplyr)
library(reshape2)
library(ggplot2)
library(Rmisc)

setwd(input_dir)
b_ltr <- read.delim(file, sep="\t", header=T)
b_ltr_fam <- read.delim(fam_file, sep="\t", header=T)
class <- read.delim(spreading, sep="\t", header=T)

#b_ltr_fam$length <- abs(b_ltr_fam$end - b_ltr_fam$start)
#b_ltr_fam2 <- subset(b_ltr_fam, b_ltr_fam$length > 1000)
  
b_ltr$TEfam <- substr(b_ltr$TE, 1, 8)
b_ltr2 <- subset(b_ltr, b_ltr$TE %in% b_ltr_fam$TE)
#b_ltr2 <- subset(b_ltr, b_ltr$TE %in% b_ltr_fam2$TE)

b_ltr_cg <- as.numeric(paste(b_ltr2$cg))
b_ltr2["cg"] <- b_ltr_cg
b_ltr2$CG <- ifelse(b_ltr2$cg >=0.40, 1, 0)
b_ltr2$CHG <- ifelse(b_ltr2$chg >=0.40, 1, 0)
b_ltr2$CHH <- ifelse(b_ltr2$chh >=0.015, 1, 0)
b_ltr3 <- left_join(b_ltr2, class, by="TE")
dist = subset(b_ltr3, b_ltr3$relative_distance > -2000 & b_ltr3$relative_distance <= 3000)

dist_spreading <- subset(dist, dist$mC_state == "mC")
dist_non.spreading <- subset(dist, dist$mC_state == "C")

dist_spreading_group <-  transform(dist_spreading, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_spreading_sumCG <- do.call(data.frame,aggregate(dist_spreading_group$CG~dist_spreading_group$group, dist_spreading_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_spreading_sumCG) <- c("group", "count", "sum")
dist_spreading_sumCG["percent_methylated"] <- dist_spreading_sumCG$sum / dist_spreading_sumCG$count
dist_spreading_sumCG["cluster.cluster"] <- "spreading"
dist_spreading_sumCHG <- do.call(data.frame,aggregate(dist_spreading_group$CHG~dist_spreading_group$group, dist_spreading_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_spreading_sumCHG) <- c("group", "count", "sum")
dist_spreading_sumCHG["percent_methylated"] <- dist_spreading_sumCHG$sum / dist_spreading_sumCHG$count
dist_spreading_sumCHG["cluster.cluster"] <- "spreading"
dist_spreading_sumCHH <- do.call(data.frame,aggregate(dist_spreading_group$CHH~dist_spreading_group$group, dist_spreading_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_spreading_sumCHH) <- c("group", "count", "sum")
dist_spreading_sumCHH["percent_methylated"] <- dist_spreading_sumCHH$sum / dist_spreading_sumCHH$count
dist_spreading_sumCHH["cluster.cluster"] <- "spreading"

dist_non.spreading_group <-  transform(dist_non.spreading, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_non.spreading_sumCG <- do.call(data.frame,aggregate(dist_non.spreading_group$CG~dist_non.spreading_group$group, dist_non.spreading_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_non.spreading_sumCG) <- c("group", "count", "sum")
dist_non.spreading_sumCG["percent_methylated"] <- dist_non.spreading_sumCG$sum / dist_non.spreading_sumCG$count
dist_non.spreading_sumCG["cluster.cluster"] <- "non.spreading"
dist_non.spreading_sumCHG <- do.call(data.frame,aggregate(dist_non.spreading_group$CHG~dist_non.spreading_group$group, dist_non.spreading_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_non.spreading_sumCHG) <- c("group", "count", "sum")
dist_non.spreading_sumCHG["percent_methylated"] <- dist_non.spreading_sumCHG$sum / dist_non.spreading_sumCHG$count
dist_non.spreading_sumCHG["cluster.cluster"] <- "non.spreading"
dist_non.spreading_sumCHH <- do.call(data.frame,aggregate(dist_non.spreading_group$CHH~dist_non.spreading_group$group, dist_non.spreading_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x))))
colnames(dist_non.spreading_sumCHH) <- c("group", "count", "sum")
dist_non.spreading_sumCHH["percent_methylated"] <- dist_non.spreading_sumCHH$sum / dist_non.spreading_sumCHH$count
dist_non.spreading_sumCHH["cluster.cluster"] <- "non.spreading"

p.1 <- rbind(dist_spreading_sumCG, dist_non.spreading_sumCG)
p.2 <- rbind(dist_spreading_sumCHG, dist_non.spreading_sumCHG)
p.3 <- rbind(dist_spreading_sumCHH, dist_non.spreading_sumCHH)
write.table(p.1, paste(genotype, TEtype, "class_metaplot_data_cg_cluster_3_spreading_metaplot", date, ".txt", sep="_"), sep="\t", quote=F)
write.table(p.2, paste(genotype, TEtype, "class_metaplot_data_chg_cluster_3_spreading_metaplot", date, ".txt", sep="_"), sep="\t", quote=F)
write.table(p.3, paste(genotype, TEtype, "class_metaplot_data_chh_cluster_3_spreading_metaplot", date, ".txt", sep="_"), sep="\t", quote=F)

## Metaplot of percent bins methylated by class
pdf(paste(genotype, TEtype, "class_metaplot_cg_cluster", cluster, date, "spreading_metaplot.pdf", sep="_"))
#pdf(paste(genotype, TEtype, "class_metaplot_cg_cluster", cluster, date, "spreading_metaplot_1kb.pdf", sep="_"))
plot1 <- ggplot(p.1, aes(x = group, y = percent_methylated, group = cluster.cluster, colour = cluster.cluster)) + geom_line(size=1) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot1)
dev.off()
pdf(paste(genotype, TEtype, "class_metaplot_chg_cluster", cluster, date, "spreading_metaplot.pdf", sep="_"))
#pdf(paste(genotype, TEtype, "class_metaplot_chg_cluster", cluster, date, "spreading_metaplot_1kb.pdf", sep="_"))
plot2 <- ggplot(p.2, aes(x = group, y = percent_methylated, group = cluster.cluster, colour = cluster.cluster)) + geom_line(size=1) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot2)
dev.off()
pdf(paste(genotype, TEtype, "class_metaplot_chh_cluster", cluster, date, "spreading_metaplot.pdf", sep="_"))
#pdf(paste(genotype, TEtype, "class_metaplot_chh_cluster", cluster, date, "spreading_metaplot_1kb.pdf", sep="_"))
plot3 <- ggplot(p.3, aes(x = group, y = percent_methylated, group = cluster.cluster, colour = cluster.cluster)) + geom_line(size=1) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot3)
dev.off()
}
```

```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

## B73
### TIR
metaplot_mC_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_closestTEtoBin_orientation_metaplot_20Dec2018.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  cluster = "3",
  #date = "20Dec2018"
  date = "Jan2019"
)

### LTR
metaplot_mC_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_closestTEtoBin_orientation_metaplot_20Dec2018.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  cluster = "3",
  date = "Jan2019"
)


### LTR not oriented (Revisions)
perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance2.pl -i /home/springer/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_20Dec2018.txt -o /scratch.global/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_metaplot_20May2019.txt
cp /scratch.global/nosha003/methylation_spreading/B73_ltr_closestTEtoBin_metaplot_20May2019.txt /home/springer/nosha003/methylation_spreading/.

metaplot_mC_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_closestTEtoBin_metaplot_20May2019.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  cluster = "3",
  date = "May2019"
)

```

#### Asymmetry of methylation calculation
- revisions: calculate the difference in flanking methylation levels for each individual TE (not just the metaplot of the average)
```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

library(dplyr)
library(ggplot2)

# LTR
setwd("/home/springer/nosha003/methylation_spreading")
b_ltr <- read.delim("B73_ltr_closestTEtoBin_metaplot_20May2019.txt", sep="\t", header=T)
spreading <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", header=T, sep="\t")
spreading_cg <- subset(spreading, spreading$variable == "cg")

b_ltr_spreading <- inner_join(b_ltr, spreading_cg, by="TE")

left_flank = subset(b_ltr_spreading, b_ltr_spreading$relative_distance < 0 & b_ltr_spreading$relative_distance > -1000)
left_flank_avg <- left_flank %>% group_by(TE) %>% mutate(left_flank = mean(cg, na.rm = T))
right_flank = subset(b_ltr_spreading, b_ltr_spreading$relative_distance > 0 & b_ltr_spreading$relative_distance < 1000)
right_flank_avg <- right_flank %>% group_by(TE) %>% mutate(right_flank = mean(cg, na.rm = T))

flanks <- inner_join(unique(left_flank_avg[,c(8,15,19,20)]), unique(right_flank_avg[,c(8,15,19,20)]), by=c("TE", "cluster", "mC_state"))
flanks$diff <- abs(flanks$left_flank - flanks$right_flank)
flanks$log2 <- log2(flanks$left_flank / flanks$right_flank)
flanks$fc <- abs(flanks$left_flank / flanks$right_flank)

flanks_spreading <- flanks %>% mutate(spreading = ifelse(mC_state == "mC", "spreading", ifelse(mC_state == "C", "non_spreading", "NA")))
flanks_na <- subset(flanks_spreading, flanks_spreading$spreading == "spreading" | flanks_spreading$spreading == "non_spreading")

pdf("spreading_CG_symmetry.pdf")
ggplot(flanks_na) + geom_violin(aes(x=spreading, y=diff, fill=spreading)) + ylab("Difference in flanking CG") + theme_classic() + xlab("")
dev.off()
pdf("spreading_CG_symmetry_box.pdf")
ggplot(flanks_na) + geom_boxplot(aes(x=spreading, y=diff, fill=spreading)) + ylab("Difference in flanking CG") + theme_classic() + xlab("")
dev.off()
pdf("spreading_CG_symmetry_log2.pdf")
ggplot(flanks_na) + geom_violin(aes(x=spreading, y=log2, fill=spreading)) + ylab("Difference in flanking CG") + theme_classic() + xlab("")
dev.off()
pdf("spreading_CG_symmetry_fc.pdf")
ggplot(flanks_na) + geom_violin(aes(x=spreading, y=fc, fill=spreading)) + ylab("Difference in flanking CG") + theme_classic() + xlab("")
dev.off()

```


# Figure 5: Spreading RE-DONE (Feb 2019)
```{r eval=F}
#26Feb2019
#site_spreading_flank.sh

#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/site_spreading_o
#PBS -e /scratch.global/nosha003/e_o/site_spreading_e
#PBS -N site_spreading
#PBS -V
#PBS -r n

cd /home/springer/nosha003/scripts/
module load R
R CMD BATCH site_spreading_flank.R
wait

# qsub /home/springer/nosha003/scripts/site_spreading_flank.sh
```

```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

setwd("/scratch.global/nosha003")
b <- read.delim("B73.to.W22.output2.B73bins.txt", header=F, sep="\t")
b$TE <- substr(b$V11, 1, 21)
df1 <- read.delim("B73.to.W22.output2.B73bins.txt", header=F, sep="\t")
df2 <- read.delim("B73.to.PH207.output2.B73bins.txt", header=F, sep="\t")
df3 <- read.delim("W22.to.B73.output2.W22bins.txt", header=F, sep="\t")
df4 <- read.delim("W22.to.PH207.output2.W22bins.txt", header=F, sep="\t")
df5 <- read.delim("PH207.to.B73.output2.PH207bins.txt", header=F, sep="\t")
df6 <- read.delim("PH207.to.W22.output2.PH207bins.txt", header=F, sep="\t")
df7 <- read.delim("PH207.to.B73.output2.PH207bins.txt", header=F, sep="\t")
df8 <- read.delim("PH207.to.W22.output2.PH207bins.txt", header=F, sep="\t")
b <- rbind(df1, df2, df3, df4, df5, df6, df7, df8)
b$TE <- substr(b$V11, 1, 21)
write.table(b, "all.to.all.output2.bins.txt", quote=F, row.names=F, sep="\t")
```

```{r eval=F}
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

library(reshape2)
library(dplyr)
library(ggplot2)
library(RColorBrewer)

setwd("/home/springer/nosha003/methylation_spreading/")
ltr_cluster <- read.delim("b_ltr_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
tir_cluster <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", header=T, sep="\t")
cluster <- rbind(ltr_cluster, tir_cluster)
cluster$fam <- row.names(cluster)
cluster <- cluster[,c(81:82)]
colnames(cluster) <- c("cluster", "fam")

setwd("/home/springer/nosha003/sarah/TEpolymorphism")
w22 <- read.delim("all_empty_average.txt", header=T, sep="\t")
w22$fam <- substr(w22$V10, 1, 8)
w22$class <- substr(w22$V10, 1, 2)
w22$TE <- substr(w22$V10, 1, 21)

w22_empty_upstream <- subset(w22, w22$V15 < 0 & w22$V15 > -200)
w22_empty_upstream$flank <- "upstream"
w22_empty_downstream <- subset(w22, w22$V15 > 0 & w22$V15 < 200)
w22_empty_downstream$flank <- "downstream"
w22_empty_flank <- rbind(w22_empty_upstream, w22_empty_downstream)

w22_empty_melt <- melt(w22_empty_flank[,c(4:6,10)], id="V10")
colnames(w22_empty_melt) <- c("TE", "context", "mC_ratio")
w22_empty_melt$fam <- substr(w22_empty_melt$TE, 1, 8)
w22_flank <- left_join(w22_empty_melt, cluster, by="fam")

w22_empty <- subset(w22, w22$V15 != 0 & abs(w22$V15) < 100)
w22_empty_melt <- melt(w22_empty[,c(4:6,10)], id="V10")
colnames(w22_empty_melt) <- c("TE", "context", "mC_ratio")
w22_empty_melt_na <- na.omit(w22_empty_melt)
w22_empty_melt_na$fam <- substr(w22_empty_melt_na$TE, 1, 8)
w22_empty_melt_na$class <- substr(w22_empty_melt_na$TE, 1, 2)
w22_empty_melt_cluster <- left_join(w22_empty_melt_na, cluster, by="fam")
w22_empty_melt_cluster_cg <- subset(w22_empty_melt_cluster, w22_empty_melt_cluster$context == "V4")
w_low <- subset(w22_empty_melt_cluster_cg, w22_empty_melt_cluster_cg$mC_ratio <= 0.20)
w_low$TE <- substr(w_low$TE, 1, 21)

w22_low_flank <- subset(w22_flank, w22_flank$TE %in% w_low$TE)
write.table(w22_low_flank, "w22_low_flank.txt", quote=F, row.names=F, sep="\t")

setwd("/scratch.global/nosha003")
b <- read.delim("all.to.all.output2.bins.txt", header=T, sep="\t")
b_flank_up <- subset(b, b$V12 < 0 & b$V12 > -200)
b_flank_down <- subset(b, b$V12 > 0 & b$V12 < 200)
b_flank_up$flank <- "upstream"
b_flank_down$flank <- "downstream"
b_flank <- rbind(b_flank_up, b_flank_down)

w_low_b_flank <- inner_join(w22_low_flank, b_flank, by=c("TE", "flank"))

cluster <- data.frame(unique(w_low_b_flank[,c(1, 6)]))

w_low_b_flank_cluster <- left_join(w_low_b_flank, cluster, by="TE")
w_low_b_flank_cluster$order <- substr(w_low_b_flank_cluster$TE, 1, 2)

w_low_b_flank_melt <- melt(w_low_b_flank_category_cluster)
w_low_b_flank_region <- unique(w_low_b_flank_melt[,c()]) %>% group_by(variable, order, cluster) %>% summarise(region_count = n()) 
w_low_b_flank_category_cluster <- w_low_b_flank_melt %>% group_by(variable, order, cluster, flank) %>% mutate(mC_state = ifelse(variable == "cg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chh", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(variable, order, cluster, mC_state) 
setwd("/scratch.global/nosha003/methylation_spreading")

w_low_b_flank_call <- w_low_b_flank_category_cluster %>% group_by(TE) %>% mutate(mC_state_call <- ifelse(flank == "upstream" & mC_state == "mC", "mC", ifelse(flank == "downstream" & mC_state == "mC", "mC", ifelse(flank == "upstream" & mC_state == "intermediate", "intermediate", ifelse(flank == "downstream" & mC_state == "intermediate", "intermediate"), "C"))))

w_low_b_flank_count <- w_low_b_flank_call %>% summarise(count = n())
w_low_b_flank_state <- inner_join(w_low_b_flank_region, w_low_b_flank_count, by=c("variable", "order", "cluster"))
w_low_b_flank_prop <- w_low_b_flank_state %>% mutate(prop = count/region_count)
w_low_b_flank_prop %>% print(n = Inf)

setwd("/scratch.global/nosha003/methylation_spreading")
pdf("all_low_b_flank_prop_Feb2019.pdf")
ggplot(data=w_low_b_flank_prop) + geom_bar(aes(x=cluster, y=prop, fill=factor(mC_state, levels=c("mC", "intermediate", "C"))), stat="identity") + facet_grid(order ~ variable) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()


w_low_b_flank_prop_cluster_RL <- subset(w_low_b_flank_prop, w_low_b_flank_prop$order == "RL")
w_low_b_flank_prop_cluster_DT <- subset(w_low_b_flank_prop, w_low_b_flank_prop$order == "DT")

w_low_b_flank_melt2 <- w_low_b_flank_melt
w_low_b_flank_melt2$cluster <- "all"
w_low_b_flank_region <- unique(w_low_b_flank_melt[,c()]) %>% group_by(variable, order, cluster) %>% summarise(region_count = n()) 
w_low_b_flank_category_cluster <- w_low_b_flank_melt %>% group_by(variable, order, cluster, flank) %>% mutate(mC_state = ifelse(variable == "cg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chg", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), ifelse(variable == "chh", ifelse(value >= 0.40, "mC", ifelse(value <= 0.20, "C", "intermediate")), "NA")))) %>% group_by(variable, order, cluster, mC_state) 
setwd("/scratch.global/nosha003/methylation_spreading")
w_low_b_flank_call <- w_low_b_flank_category_cluster %>% group_by(TE) %>% mutate(mC_state_call <- ifelse(flank == "upstream" & mC_state == "mC", "mC", ifelse(flank == "downstream" & mC_state == "mC", "mC", ifelse(flank == "upstream" & mC_state == "intermediate", "intermediate", ifelse(flank == "downstream" & mC_state == "intermediate", "intermediate"), "C"))))
w_low_b_flank_count <- w_low_b_flank_call %>% summarise(count = n())
w_low_b_flank_state <- inner_join(w_low_b_flank_region, w_low_b_flank_count, by=c("variable", "order", "cluster"))
w_low_b_flank_prop <- w_low_b_flank_state %>% mutate(prop = count/region_count)
w_low_b_flank_prop_RL <- subset(w_low_b_flank_prop, w_low_b_flank_prop$order == "RL")
w_low_b_flank_prop_DT <- subset(w_low_b_flank_prop, w_low_b_flank_prop$order == "DT")

prop_ltr <- rbind(w_low_b_flank_prop_cluster_RL, w_low_b_flank_prop_RL) 
prop_ltr_cg <- subset(prop_ltr, prop_ltr$variable == "cg")
prop_tir <- rbind(w_low_b_flank_prop_cluster_DT, w_low_b_flank_prop_DT) 
prop_tir_cg <- subset(prop_tir, prop_tir$variable == "cg")

library(ggplot2)
setwd("/scratch.global/nosha003/methylation_spreading")
pdf("all_low_b_flank_prop_all_CG_LTR_Feb2019.pdf")
ggplot(data=prop_ltr_cg, aes(x=factor(cluster, levels=c("A", "B", "C", "all")), y=prop)) + geom_bar(aes(fill=factor(mC_state, levels=c("mC", "intermediate", "C"))), stat="identity") + facet_grid(. ~ variable) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()
pdf("w_low_b_flank_prop_all_CG_TIR_Feb2019.pdf")
ggplot(data=prop_tir_cg, aes(x=factor(cluster, levels=c("all", "A", "B", "C")), y=prop)) + geom_bar(aes(fill=factor(mC_state, levels=c("mC", "intermediate", "C"))), stat="identity") + facet_grid(. ~ variable) + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()

```







# TSD matching (TE - insertion site)
- write perl script to see if TSD matches between TE annotation and site definted insertions
    - use Sarah's insertion sites file to identify negative gaps... the characters (from fasta file) in those coordinates should match the flanks of the TE.
    
```{r eval=F}
#! /usr/bin/perl -w
# perl script to see if TSD matches between TE annotation and site definted insertions 
# /scratch.global/nosha003/tsd_matching.pl
# 19_Nov_2018
# Jaclyn_Noshay

# Input fasta file (for both genotypes) and input Sarah's insertion site calls and find negative gap coordinates to determine if characters match between ISD empty site and TE flanks

use warnings;
use strict;
#use Getopt::Std;
use Bio::DB::Fasta;
use Getopt::Long();

my $usage = "\nUsage: $0 --fasta_in <Input file> --fasta2_in <Input file> --sites_in <sites file> --fasta_out <Output file> --fasta2_out <output file>\n\n";

my ($fasta_in, $fasta2_in, $sites_in, $fasta_out, $fasta2_out);

Getopt::Long::GetOptions('fasta_in=s' => \$fasta_in,
                         'fasta2_in=s' => \$fasta2_in,
                         'sites_in=s' => \$sites_in,
                         'fasta_out=s' => \$fasta_out,
                         'fasta2_out=s' => \$fasta2_out,
                        );

my $db1 = Bio::DB::Fasta->new($fasta_in);
open (my $fasta_out_fh, '>', $fasta_out) or die "Can't open $fasta_out\n\n";

my @ids1 = $db1->get_all_ids();
foreach my $fasta_name1 (@ids1) {
  #print "$fasta_name1\n";
  my $seq = $db1->get_Seq_by_id($fasta_name1);

  open (my $sites_in_fh, '<', $sites_in) or die "Can't open $sites_in\n\n";
  while (my $line = <$sites_in_fh>) {
  chomp $line;
  my ($chr, $start, $end, $TE, $te_length, $gap_length, $te_coord, $category) = split ("\t", $line);

    if ($gap_length ne "UNK") {
    my $subseq;
      if ($chr eq $fasta_name1) {
        $subseq = $seq->subseq($start => $end);
        print $fasta_out_fh ">$TE\t$gap_length\t$subseq\t$subseq\n";
      }
    }
  }
}

my $db2 = Bio::DB::Fasta->new($fasta2_in);
open (my $fasta2_out_fh, '>', $fasta2_out) or die "Can't open $fasta2_out\n\n";

my ($te_start_true, $te_end_true);
my @ids2 = $db2->get_all_ids();
foreach my $fasta_name2 (@ids2) {
  #print "$fasta_name2\n";
  my $seq = $db2->get_Seq_by_id($fasta_name2);

  open (my $sites_in_fh, '<', $sites_in) or die "Can't open $sites_in\n\n";
  while (my $line = <$sites_in_fh>) {
    chomp $line;
    my ($chr, $start, $end, $TE, $te_length, $gap_length, $te_coord, $category) = split ("\t", $line);
    my ($te_chr, $te_coord2) = split (":", $te_coord);
    my ($te_start, $te_end) = split("-", $te_coord2);

    if ($te_start < $te_end) {
      $te_start_true = $te_start;
      $te_end_true = $te_end;
    }
    elsif ($te_start > $te_end) {
        $te_start_true = $te_end;
        $te_end_true = $te_start;
    }

    if ($gap_length ne "UNK") {
      my $te_start_gap = ($te_start_true-1) - abs($gap_length);
      my $te_end_gap = ($te_end_true) + abs($gap_length);

      my ($subseq, $subseq2);
        if ($chr eq $fasta_name2) {
          $subseq = $seq->subseq($te_start_gap => ($te_start_true-1));
          $subseq2 = $seq->subseq(($te_end_true) => $te_end_gap);
          print $fasta2_out_fh ">$TE\t$gap_length\t$subseq\t$subseq2\n";
        }
    }
  }
close $sites_in_fh;
}

close $fasta2_in;
close $fasta2_out_fh;
close $fasta_in;
close $fasta_out_fh;
exit;
```


```{r eval=F}
module load bioperl/5.16.1 

grep 'ctg' /home/maize/shared/databases/genomes/Zea_mays/B73/Zea_mays.AGPv4.dna.toplevel.fa | head
sed '/B73V4_ctg150 dna:contig contig:AGPv4:B73V4_ctg150:1:1738914:1 REF/,$d' /home/maize/shared/databases/genomes/Zea_mays/B73/Zea_mays.AGPv4.dna.toplevel.fa > /scratch.global/nosha003/Zea_mays.AGPv4.dna.toplevel.noctg.fa
sed 's/chr//g' /home/maize/shared/databases/genomes/Zea_mays/W22/W22__Ver12.genome.normalized.fasta > /scratch.global/nosha003/W22__Ver12.genome.normalized.fasta

### output files include site TSD (TE, gap_length, TSD, TSD) + TE TSD (TE, gap_length, TSD_left, TSD_right)
perl /scratch.global/nosha003/tsd_matching.pl --fasta2_in /scratch.global/nosha003/Zea_mays.AGPv4.dna.toplevel.noctg.fa --fasta_in /scratch.global/nosha003/W22__Ver12.genome.normalized.fasta --sites_in /home/springer/sna/polyTE/sna/B73.to.W22.output2_3Jan19.bed --fasta_out /scratch.global/nosha003/methylation_spreading/B73.to.W22.sitesTSD.Jan19.txt --fasta2_out /scratch.global/nosha003/methylation_spreading/B73.to.W22.teTSD.Jan19.txt
# 172139

# only look at those with negative gap length
awk '{if ($2 < 0) print $0}' /scratch.global/nosha003/methylation_spreading/B73.to.W22.sitesTSD.Jan19.txt | awk -F"\t" '{print $1"\t"$2"\t"substr($3,2)"\t"substr($4,2)}' | sort -k 1,1 -k 2,2n > /scratch.global/nosha003/methylation_spreading/B73.to.W22.sitesTSD.neg2.Jan19.txt 
awk '{if ($2 < 0) print $0}' /scratch.global/nosha003/methylation_spreading/B73.to.W22.teTSD.Jan19.txt  | awk -F"\t" '{print $1"\t"$2"\t"substr($3,2)"\t"substr($4,2)}' | sort -k 1,1 -k 2,2n > /scratch.global/nosha003/methylation_spreading/B73.to.W22.teTSD.neg2.Jan19.txt
# 8214

# module load R
# R
setwd("/scratch.global/nosha003/methylation_spreading/")
sites <- read.delim("B73.to.W22.sitesTSD.neg2.Jan19.txt", header=F, sep="\t")
# 8214
te <- read.delim("B73.to.W22.teTSD.neg2.Jan19.txt", header=F, sep="\t")
# 8214

sites$te <- substr(sites$V1, 2, 22)
sites$site <- substr(sites$V1, 23, 28)
sites_defined <- subset(sites, sites$site == ".site")
te$te <- substr(te$V1, 2, 22)
te$site <- substr(te$V1, 23, 28)
te_defined <- subset(te, te$site == ".site")

library(dplyr)
sites_te_left <- semi_join(sites, te, by=c("V1", "V2", "V3"))
sites_te_right <- semi_join(sites, te, by=c("V1", "V2", "V4"))
sites_te_both <- semi_join(sites, te, by=c("V1", "V2", "V3", "V4"))
sites_te_both$type <- "both_match"
# 5019
sites_te_left_only <- anti_join(sites_te_left, sites_te_both, by=c("V1", "V2"))
sites_te_left_only$type <- "left_match"
# 58
sites_te_right_only <- anti_join(sites_te_right, sites_te_both, by=c("V1", "V2"))
sites_te_right_only$type <- "right_match"
# 1182

sites_te_match <- rbind(sites_te_both, sites_te_left_only, sites_te_right_only)
colnames(sites_te_match) <- c("id", "neg_gap", "tsd1_site", "tsd2_site", "te", "site", "match_type")
sites_te_match_df <- left_join(sites_te_match, te, by="te")
sites_te_match_df2 <- sites_te_match_df[,c(1:3,10:11,5:7)]
colnames(sites_te_match_df2) <- c("id", "neg_gap", "site_tsd", "left_tsd", "right_tsd", "te", "site", "match_type")
write.table(sites_te_match_df2, "tsd_match_Jan19.txt", quote=F, row.names=F, sep="\t")
# 6259

sites_te_nomatch <- anti_join(sites, sites_te_match_df2, by="te")
sites_te_nomatch$type <- "no_match"
sites_te_nomatch_df <- left_join(sites_te_nomatch, te, by="te")
sites_te_nomatch_df2 <- sites_te_nomatch_df[,c(1:3,10:11,5:7)]
colnames(sites_te_nomatch_df2) <- c("id", "neg_gap", "site_tsd", "left_tsd", "right_tsd", "te", "site", "match_type")
# 1955
sites_te_match_nomatch <- rbind(sites_te_match_df2, sites_te_nomatch_df2)
# 8214

nrow(subset(sites_te_nomatch, sites_te_nomatch$site == ".site"))
# 619 (others are conserved TEs)


sites_te_match_nomatch_sitedefined <- subset(sites_te_match_nomatch, sites_te_match_nomatch$site == ".site")
# 6878
sites_te_match_nomatch_sitedefined$number.of.Ns.leftTSD <- lengths(gregexpr('N', sites_te_match_nomatch_sitedefined$left_tsd)) - 1
sites_te_match_nomatch_sitedefined$number.of.Ns.rightTSD <- lengths(gregexpr('N', sites_te_match_nomatch_sitedefined$right_tsd)) - 1
write.table(sites_te_match_nomatch_sitedefined, "tsd_all_Jan19.txt", quote=F, row.names=F, sep="\t")

library(ggplot2)
pdf("tsd_gap_length.pdf")
ggplot(sites_te_match_nomatch_sitedefined) + geom_violin(aes(x=match_type, y=neg_gap, fill=match_type)) + theme_classic() + ylim(-500, 0)
dev.off()
pdf("tsd_N_count.pdf")
ggplot(sites_te_match_nomatch_sitedefined) + geom_violin(aes(x=match_type, y=number.of.Ns.leftTSD, fill=match_type)) + theme_classic() + ylim(0, 10)
dev.off()

## N's play a big role in non-matching TSDs (was due to the conserved TEs that were still in the files due to a negative gap length)




## Look at methylation state of excision events vs insertion events
library(dplyr)
library(ggplot2)

setwd("/scratch.global/nosha003/methylation_spreading/")
tsd <- read.delim("tsd_all_Jan19.txt", header=T, sep="\t")

setwd("/home/springer/nosha003/sarah/TEpolymorphism")
w22 <- read.delim("W22emptysite_methylation_bins.txt", header=F, sep="\t")
w22$te <- substr(w22$V10, 1, 21)
w22_site <- subset(w22, abs(w22$V15) < 200 & w22$V15 != 0)
w22_site_na <- na.omit(w22_site)
w22_site_avg <- w22_site_na %>% group_by(te) %>% mutate(avg_cg = mean(V4), avg_chg = mean(V5), avg_chh = mean(V6)) 
w22_site_avg2 <- unique(w22_site_avg[,c(16,17)])
w22_site_state <- w22_site_avg2 %>% mutate(mC_state = ifelse(avg_cg <= 0.2, "unmethylated", ifelse(avg_cg >= 0.40, "methylated", ifelse(avg_cg > 0.2 & avg_cg < 0.4, "intermediate", "NA"))))

tsd_mC <- left_join(tsd, unique(w22_site_state[,c(1,3)]), by="te")

tsd_mC_count <- tsd_mC %>% group_by(match_type, mC_state) %>% mutate(count = n())
tsd_mC_count2 <- tsd_mC %>% group_by(match_type) %>% mutate(count2 = n())
tsd_mC_count_prop <- left_join(unique(tsd_mC_count[,c(8,11,12)]), unique(tsd_mC_count2[,c(8,12)]), by="match_type")
tsd_mC_count_prop$prop <- tsd_mC_count_prop$count / tsd_mC_count_prop$count2

#    match_type  mC_state     count count2   prop
#    <fct>       <chr>        <int>  <int>  <dbl>
#  1 both_match  unmethylated   622   5019 0.124 
#  2 both_match  methylated    3946   5019 0.786 
#  3 both_match  intermediate   139   5019 0.0277
#  4 both_match  NA             312   5019 0.0622
#  5 left_match  methylated      36     58 0.621 
#  6 left_match  unmethylated    17     58 0.293 
#  7 left_match  intermediate     3     58 0.0517
#  8 left_match  NA               2     58 0.0345
#  9 right_match methylated     750   1182 0.635 
# 10 right_match NA              80   1182 0.0677
# 11 right_match unmethylated   282   1182 0.239 
# 12 right_match intermediate    70   1182 0.0592
# 13 no_match    unmethylated   102    619 0.165 
# 14 no_match    NA              40    619 0.0646
# 15 no_match    intermediate    24    619 0.0388
# 16 no_match    methylated     453    619 0.732 

pdf("tsd_mC.pdf")
ggplot(tsd_mC_count_prop) + geom_bar(aes(x=match_type, y=prop, fill=mC_state), stat="identity") + theme_classic()
dev.off()


tsd_mC_count_label <- tsd_mC %>% mutate(label = ifelse(match_type == "both_match", "Insertion", ifelse(match_type == "right_match" | match_type == "left_match", "Unknown", ifelse(match_type == "no_match", "Excision", "NA"))))
tsd_mC_count <- tsd_mC_count_label %>% group_by(label, mC_state) %>% mutate(count = n())
tsd_mC_count2 <- tsd_mC_count_label %>% group_by(label) %>% mutate(count2 = n())
tsd_mC_count_prop <- left_join(unique(tsd_mC_count[,c(11,12,13)]), unique(tsd_mC_count2[,c(12,13)]), by="label")
tsd_mC_count_prop$prop <- tsd_mC_count_prop$count / tsd_mC_count_prop$count2

pdf("tsd_mC_label.pdf")
ggplot(tsd_mC_count_prop) + geom_bar(aes(x=label, y=prop, fill=mC_state), stat="identity") + theme_classic()
dev.off()

## remove NAs
tsd_mC_count_label <- tsd_mC %>% mutate(label = ifelse(match_type == "both_match", "Insertion", ifelse(match_type == "right_match" | match_type == "left_match", "Unknown", ifelse(match_type == "no_match", "Excision", "NA"))))
tsd_mC_count_na <- na.omit(tsd_mC_count_label)
tsd_mC_count <- tsd_mC_count_na %>% group_by(label, mC_state) %>% mutate(count = n())
tsd_mC_count2 <- tsd_mC_count_na %>% group_by(label) %>% mutate(count2 = n())
tsd_mC_count_prop <- left_join(unique(tsd_mC_count[,c(11,12,13)]), unique(tsd_mC_count2[,c(12,13)]), by="label")
tsd_mC_count_prop$prop <- tsd_mC_count_prop$count / tsd_mC_count_prop$count2
tsd_mC_count_prop

pdf("tsd_mC_label_na.pdf")
ggplot(tsd_mC_count_prop) + geom_bar(aes(x=label, y=prop, fill=factor(mC_state, levels=c("methylated", "intermediate", "unmethylated"))), stat="identity") + theme_classic() + theme(text = element_text(size=7)) + scale_fill_manual(values=c("#8da0cb", "#fc8d62", "#66c2a5"))
dev.off()

#   mC_state     label     count count2   prop
#   <chr>        <chr>     <int>  <int>  <dbl>
# 1 unmethylated Insertion   622   4707 0.132 
# 2 methylated   Insertion  3946   4707 0.838 
# 3 intermediate Insertion   139   4707 0.0295
# 4 methylated   Unknown     786   1158 0.679 
# 5 unmethylated Unknown     299   1158 0.258 
# 6 intermediate Unknown      73   1158 0.0630
# 7 unmethylated Excision    102    579 0.176 
# 8 intermediate Excision     24    579 0.0415
# 9 methylated   Excision    453    579 0.782

## are some of those that have no_match in W22 (potential TE excision event) true insertion sites in PH207 or Mo17? --> could compare methylation state of the different events (if it is a "spreading" TE)... was methylation "remembered" and maintained after TE excised?

```


# Methylation genome-wide (1Mb windows)
- chromosome level plots for B73

```{r eval=F}

setwd("/home/springer/nosha003/wgbs_schmitz/tiles_output/")
b <- read.delim("B73_V2_100bp_ratio.txt", header=F, sep="\t")
colnames(b) <- c("chr", "start", "end", "CG", "CHG", "CHH")

library(dplyr)
library(reshape2)
b$bin <- paste(b$chr,floor(b$start/1e6)*1e6,sep="-")
b_melt <- melt(b[,4:7], id="bin")
b_melt_na <- na.omit(b_melt)
summary.B73.mC <- data.frame(b_melt_na %>% group_by(bin, variable) %>% summarize(count = sum(value)/1e4))
summary.B73.mC.done <- dcast(summary.B73.mC, bin ~ variable)

write.table(summary.B73.mC.done, "summary.B73.mC.txt", quote=F, row.names=F, sep="\t")
```



# Sarah Gene Metaplots
- 4 categories of genes (syntenic, gene in TE (non-syntenic, variable TE, shared TE))
- closest gene to each 100bp bin (overlap Sarah's file with Gff file of gene coordinates)
- metaplot for each category (CG, CHG, CHH different colors) of gene +/- 1kb

```{r eval=F}
# get gene coordinates
setwd("/home/springer/nosha003/database/B73v4")
gene <- read.delim("Zea_mays.AGPv4.32.gene.sort2.gff3", header=F, sep="\t", stringsAsFactors = F)
gene$geneID <- substr(gene$V9, 9, 22)

setwd("/home/springer/nosha003/sarah/TEpolymorphism")
category <- read.delim("genes_lists_for_metaplots.txt", header=T, sep="\t", stringsAsFactors = F)

library(dplyr)
library(tidygenomics)
gene_category <- left_join(category, gene, by="geneID")
gene_gff <- gene_category[,c(3:11, 1:2)]
gene_gff_na <- na.omit(gene_gff)
setwd("/home/springer/nosha003/sarah/TEpolymorphism")
write.table(gene_gff_na, "genes_lists_for_metaplots_coordinates.gff3", quote=F, row.names=F, sep="\t")


# unix --> closest gene to each 100bp bin
sed '1d' /home/springer/nosha003/sarah/TEpolymorphism/genes_lists_for_metaplots_coordinates.gff3 | sort -k 1,1 -k 4,4n > /home/springer/nosha003/sarah/TEpolymorphism/genes_lists_for_metaplots_coordinates_sort.gff3

module load bedtools
bedtools closest -D b -t first -k 1 -b /home/springer/nosha003/sarah/TEpolymorphism/genes_lists_for_metaplots_coordinates_sort.gff3 -a /home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt > /home/springer/nosha003/sarah/TEpolymorphism/genes_lists_for_metaplots_bin.txt

# dataframe
perl /home/springer/nosha003/methylation_spreading/metaplot_gene.pl -i /home/springer/nosha003/sarah/TEpolymorphism/genes_lists_for_metaplots_bin.txt -o /scratch.global/nosha003/genes_lists_for_metaplots_bin_df.txt


# metaplots
awk '{if ($15 == "syntenic") print $0}' /scratch.global/nosha003/genes_lists_for_metaplots_bin_df.txt > /scratch.global/nosha003/genes_lists_for_metaplots_bin_df_syntenic.txt
awk '{if ($15 == "non.syntenic") print $0}' /scratch.global/nosha003/genes_lists_for_metaplots_bin_df.txt > /scratch.global/nosha003/genes_lists_for_metaplots_bin_df_nonsyntenic.txt
awk '{if ($15 == "in.shared.TE") print $0}' /scratch.global/nosha003/genes_lists_for_metaplots_bin_df.txt > /scratch.global/nosha003/genes_lists_for_metaplots_bin_df_sharedTE.txt
awk '{if ($15 == "in.variable.TE") print $0}' /scratch.global/nosha003/genes_lists_for_metaplots_bin_df.txt > /scratch.global/nosha003/genes_lists_for_metaplots_bin_df_varialeTE.txt

# generate metaplot of data 
# qsub -I -l walltime=5:00:00,nodes=1:ppn=6,mem=60G

library(fields)
setwd("/scratch.global/nosha003/")
#setwd("/Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/spreading/")
q1 <- read.table(file="genes_lists_for_metaplots_bin_df_syntenic.txt", header=F, sep="\t")
q2 <- read.table(file="genes_lists_for_metaplots_bin_df_nonsyntenic.txt", header=F, sep="\t")
q3 <- read.table(file="genes_lists_for_metaplots_bin_df_sharedTE.txt", header=F, sep="\t")
q4 <- read.table(file="genes_lists_for_metaplots_bin_df_varialeTE.txt", header=F, sep="\t")
colnames(q1) <- c("chr", "binstart", "binstop", "binid", "cg", "chg", "chh", "TE", "TEstart", "TEend", "TEsize", "distance", "relative_distance", "real_distance", "category") 
colnames(q2) <- c("chr", "binstart", "binstop", "binid", "cg", "chg", "chh", "TE", "TEstart", "TEend", "TEsize", "distance", "relative_distance", "real_distance", "category") 
colnames(q3) <- c("chr", "binstart", "binstop", "binid", "cg", "chg", "chh", "TE", "TEstart", "TEend", "TEsize", "distance", "relative_distance", "real_distance", "category") 
colnames(q4) <- c("chr", "binstart", "binstop", "binid", "cg", "chg", "chh", "TE", "TEstart", "TEend", "TEsize", "distance", "relative_distance", "real_distance", "category") 

look just at flanking 1kb of FGS
data.sub1=subset(q1,q1$relative_distance < 2000 & q1$relative_distance > -1000)
data.sub2=subset(q2,q2$relative_distance < 2000 & q2$relative_distance > -1000)
data.sub3=subset(q3,q3$relative_distance < 2000 & q3$relative_distance > -1000)
data.sub4=subset(q4,q4$relative_distance < 2000 & q4$relative_distance > -1000)

get 100 bins across the relative gene distance and get stats on your data column
stats.test1.cg=stats.bin(data.sub1$relative_distance,data.sub1$cg,N=600)
stats.test1.chg=stats.bin(data.sub1$relative_distance,data.sub1$chg,N=600)
stats.test1.chh=stats.bin(data.sub1$relative_distance,data.sub1$chh,N=600)

stats.test2.cg=stats.bin(data.sub2$relative_distance,data.sub2$cg,N=600)
stats.test2.chg=stats.bin(data.sub2$relative_distance,data.sub2$chg,N=600)
stats.test2.chh=stats.bin(data.sub2$relative_distance,data.sub2$chh,N=600)

stats.test3.cg=stats.bin(data.sub3$relative_distance,data.sub3$cg,N=600)
stats.test3.chg=stats.bin(data.sub3$relative_distance,data.sub3$chg,N=600)
stats.test3.chh=stats.bin(data.sub3$relative_distance,data.sub3$chh,N=600)

stats.test4.cg=stats.bin(data.sub4$relative_distance,data.sub4$cg,N=600)
stats.test4.chg=stats.bin(data.sub4$relative_distance,data.sub4$chg,N=600)
stats.test4.chh=stats.bin(data.sub4$relative_distance,data.sub4$chh,N=600)

p.1.cg=cbind(matrix(stats.test1.cg$centers,ncol=1),stats.test1.cg$stats["mean",])
p.1.chg=cbind(matrix(stats.test1.chg$centers,ncol=1),stats.test1.chg$stats["mean",])
p.1.chh=cbind(matrix(stats.test1.chh$centers,ncol=1),stats.test1.chh$stats["mean",])

p.2.cg=cbind(matrix(stats.test2.cg$centers,ncol=1),stats.test2.cg$stats["mean",])
p.2.chg=cbind(matrix(stats.test2.chg$centers,ncol=1),stats.test2.chg$stats["mean",])
p.2.chh=cbind(matrix(stats.test2.chh$centers,ncol=1),stats.test2.chh$stats["mean",])

p.3.cg=cbind(matrix(stats.test3.cg$centers,ncol=1),stats.test3.cg$stats["mean",])
p.3.chg=cbind(matrix(stats.test3.chg$centers,ncol=1),stats.test3.chg$stats["mean",])
p.3.chh=cbind(matrix(stats.test3.chh$centers,ncol=1),stats.test3.chh$stats["mean",])

p.4.cg=cbind(matrix(stats.test4.cg$centers,ncol=1),stats.test4.cg$stats["mean",])
p.4.chg=cbind(matrix(stats.test4.chg$centers,ncol=1),stats.test4.chg$stats["mean",])
p.4.chh=cbind(matrix(stats.test4.chh$centers,ncol=1),stats.test4.chh$stats["mean",])

setwd("/scratch.global/nosha003/")

# make plots for each gene set with CG, CHG, and CHH on the same plot

pdf("sarah_gene_metaplot_syntenic.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='syntenic')
lines(p.1.cg,col=1,lwd=1)
lines(p.1.chg,col=2,lwd=1)
lines(p.1.chh,col=3,lwd=1)
xline(0,lty=2,col='black')
xline(1000,lty=2,col='black')
legend("topright",c('CG', 'CHG', 'CHH'),col=c(1,2,3),lty=1,lwd=2,cex=0.7)
dev.off()

pdf("sarah_gene_metaplot_nonsyntenic.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='non.syntenic')
lines(p.2.cg,col=1,lwd=1)
lines(p.2.chg,col=2,lwd=1)
lines(p.2.chh,col=3,lwd=1)
xline(0,lty=2,col='black')
xline(1000,lty=2,col='black')
legend("topright",c('CG', 'CHG', 'CHH'),col=c(1,2,3),lty=1,lwd=2,cex=0.7)
dev.off()

pdf("sarah_gene_metaplot_shared.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='in.shared.TE')
lines(p.3.cg,col=1,lwd=1)
lines(p.3.chg,col=2,lwd=1)
lines(p.3.chh,col=3,lwd=1)
xline(0,lty=2,col='black')
xline(1000,lty=2,col='black')
legend("topright",c('CG', 'CHG', 'CHH'),col=c(1,2,3),lty=1,lwd=2,cex=0.7)
dev.off()

pdf("sarah_gene_metaplot_variable.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='in.variable.TE')
lines(p.4.cg,col=1,lwd=1)
lines(p.4.chg,col=2,lwd=1)
lines(p.4.chh,col=3,lwd=1)
xline(0,lty=2,col='black')
xline(1000,lty=2,col='black')
legend("topright",c('CG', 'CHG', 'CHH'),col=c(1,2,3),lty=1,lwd=2,cex=0.7)
dev.off()


# make plots with all gene sets on the same plot (separate plots for CG, CHG, and CHH)
pdf("sarah_gene_metaplot_cg.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='CG')
lines(p.1.cg,col=1,lwd=1)
lines(p.2.cg,col=2,lwd=1)
lines(p.3.cg,col=3,lwd=1)
lines(p.4.cg,col=4,lwd=1)
xline(0,lty=2,col='black')
xline(1000,lty=2,col='black')
legend("topright",c('syntenic', 'non.syntenic', 'in.shared.TE', 'in.variable.TE'),col=c(1,2,3,4),lty=1,lwd=2,cex=0.7)
dev.off()

pdf("sarah_gene_metaplot_chg.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,1),xlab="",ylab='read count',main='CHG')
lines(p.1.chg,col=1,lwd=1)
lines(p.2.chg,col=2,lwd=1)
lines(p.3.chg,col=3,lwd=1)
lines(p.4.chg,col=4,lwd=1)
xline(0,lty=2,col='black')
xline(1000,lty=2,col='black')
legend("topright",c('syntenic', 'non.syntenic', 'in.shared.TE', 'in.variable.TE'),col=c(1,2,3,4),lty=1,lwd=2,cex=0.7)
dev.off()

pdf("sarah_gene_metaplot_chh.pdf")
par(mfrow=c(2,1))
par(mar = rep(2,4))
plot(x=NULL, y=NULL,xlim=c(-1000,2000),ylim=c(0,0.04),xlab="",ylab='read count',main='CHH')
lines(p.1.chh,col=1,lwd=1)
lines(p.2.chh,col=2,lwd=1)
lines(p.3.chh,col=3,lwd=1)
lines(p.4.chh,col=4,lwd=1)
xline(0,lty=2,col='black')
xline(1000,lty=2,col='black')
legend("topright",c('syntenic', 'non.syntenic', 'in.shared.TE', 'in.variable.TE'),col=c(1,2,3,4),lty=1,lwd=2,cex=0.7)
dev.off()


```





# Files for SRA / DRUM

## Fastq files
- B73 shoot and PH207 shoot: /home/springer/data_release/umgc/hiseq/170524_D00635_0241_BCB0RCANXX/Springer_Project_053/*.fastq
- B73 root: /home/springer/data_release/umgc/hiseq/170602_D00635_0242_ACB1FBANXX/Springer_Project_053/JN26_BS_*.fastq
- W22 leaf: /home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057/PAC0020*.fastq
            /home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057/PAC0019*.fastq
- B73 leaf: /home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057/PAC0013*.fastq
            /home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057/PAC0014*.fastq
            /home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057/PAC0015*.fastq
            /home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057/PAC0016*.fastq
            /home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057/PAC0017*.fastq
            /home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057/PAC0018*.fastq
            

/home/springer/data_release/umgc/hiseq/170524_D00635_0241_BCB0RCANXX/Springer_Project_053       JN25  JN25_BS_S8_R1_001.fastq
/home/springer/data_release/umgc/hiseq/170524_D00635_0241_BCB0RCANXX/Springer_Project_053       JN25  JN25_BS_S8_R2_001.fastq
/home/springer/data_release/umgc/hiseq/170524_D00635_0241_BCB0RCANXX/Springer_Project_053       JN27  JN27_BS_S9_R1_001.fastq
/home/springer/data_release/umgc/hiseq/170524_D00635_0241_BCB0RCANXX/Springer_Project_053       JN27  JN27_BS_S9_R2_001.fastq
/home/springer/data_release/umgc/hiseq/170602_D00635_0242_ACB1FBANXX/Springer_Project_053       JN26  JN26_BS_S1_R1_001.fastq
/home/springer/data_release/umgc/hiseq/170602_D00635_0242_ACB1FBANXX/Springer_Project_053       JN26  JN26_BS_S1_R2_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0013 PAC0013_S1_R1_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0013 PAC0013_S1_R2_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0014 PAC0014_S2_R1_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0014 PAC0014_S2_R2_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0015 PAC0015_S3_R1_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0015 PAC0015_S3_R2_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0016 PAC0016_S4_R1_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0016 PAC0016_S4_R2_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0017 PAC0017_S5_R1_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0017 PAC0017_S5_R2_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0018 PAC0018_S6_R1_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0018 PAC0018_S6_R2_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0019 PAC0019_S7_R1_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0019 PAC0019_S7_R2_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0020 PAC0020_S8_R1_001.fastq
/home/springer/data_release/umgc/hiseq/171218_D00635_0314_ACC5BFANXX/Springer_Project_057       PAC0020 PAC0020_S8_R2_001.fastq

```{r eval=F}
#20Feb2019
#fastq_transfer.sh
#pull fastq files from umgc database to msi scratch space and zip before transfering to desktop to input into sra
	
#!/bin/bash
#PBS -l walltime=96:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/fastq_o
#PBS -e /scratch.global/nosha003/e_o/fastq_e
#PBS -N fastq
#PBS -V
#PBS -r n

DIR="$(/bin/sed -n ${PBS_ARRAYID}p ${LIST} | cut -f 1)"
SAMPLE="$(/bin/sed -n ${PBS_ARRAYID}p ${LIST} | cut -f 2)"
ID="$(/bin/sed -n ${PBS_ARRAYID}p ${LIST} | cut -f 3)"

cd /scratch.global/nosha003/fastq

cp ${DIR}/${ID} .

#zip chromatin_ms_fastq.zip /scratch.global/nosha003/fastq/*.fastq

zip B73_shoot_R1.fastq.zip JN25_BS_S8_R1_001.fastq
zip B73_shoot_R2.fastq.zip JN25_BS_S8_R2_001.fastq
zip B73_root_R1.fastq.zip JN26_BS_S1_R1_001.fastq 
zip B73_root_R2.fastq.zip JN26_BS_S1_R2_001.fastq 
zip PH207_shoot_R1.fastq.zip JN27_BS_S9_R1_001.fastq 
zip PH207_shoot_R2.fastq.zip JN27_BS_S9_R2_001.fastq 

zip B73_leaf_sample1_R1.1.fastq.zip PAC0013_S1_R1_001.fastq 
zip B73_leaf_sample1_R2.1.fastq.zip PAC0013_S1_R2_001.fastq 
zip B73_leaf_sample1_R1.2.fastq.zip PAC0014_S2_R1_001.fastq 
zip B73_leaf_sample1_R2.2.fastq.zip PAC0014_S2_R2_001.fastq 
zip B73_leaf_sample2_R1.1.fastq.zip PAC0015_S3_R1_001.fastq 
zip B73_leaf_sample2_R2.1.fastq.zip PAC0015_S3_R2_001.fastq 
zip B73_leaf_sample2_R1.2.fastq.zip PAC0016_S4_R1_001.fastq 
zip B73_leaf_sample2_R2.2.fastq.zip PAC0016_S4_R2_001.fastq 
zip B73_leaf_sample3_R1.1.fastq.zip PAC0017_S5_R1_001.fastq 
zip B73_leaf_sample3_R3.1.fastq.zip PAC0017_S5_R2_001.fastq 
zip B73_leaf_sample3_R1.2.fastq.zip PAC0018_S6_R1_001.fastq 
zip B73_leaf_sample3_R2.2.fastq.zip PAC0018_S6_R2_001.fastq 
zip W22_leaf_sample1_R1.fastq.zip PAC0019_S7_R1_001.fastq 
zip W22_leaf_sample1_R2.fastq.zip PAC0019_S7_R2_001.fastq 
zip W22_leaf_sample2_R1.fastq.zip PAC0020_S8_R1_001.fastq 
zip W22_leaf_sample2_R1.fastq.zip PAC0020_S8_R2_001.fastq 

#qsub -t 1-11 -v LIST=/home/springer/nosha003/methylation_spreading/fastq_transfer_names.txt /scratch.global/nosha003/fastq/fastq_transfer.sh
#qsub /scratch.global/nosha003/fastq/fastq_transfer_individual.sh
```
scp nosha003@login.msi.umn.edu:/scratch.global/nosha003/fastq/*.fastq.zip /Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/fastq/.

```{r eval=F}
#25Feb2019
#fastq_transfer_zip.sh
#zip fastq files before transfer
	
#!/bin/bash
#PBS -l walltime=96:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/fastq_o
#PBS -e /scratch.global/nosha003/e_o/fastq_e
#PBS -N fastq
#PBS -V
#PBS -r n

cd /scratch.global/nosha003/fastq

tar -zcvf JN25_BS_S8_R1_001.fastq B73_shoot_R1.fastq
tar -zcvf JN25_BS_S8_R2_001.fastq B73_shoot_R2.fastq
tar -zcvf JN26_BS_S1_R1_001.fastq B73_root_R1.fastq
tar -zcvf JN26_BS_S1_R2_001.fastq B73_root_R2.fastq
tar -zcvf JN27_BS_S9_R1_001.fastq PH207_shoot_R1.fastq
tar -zcvf JN27_BS_S9_R2_001.fastq PH207_shoot_R2.fastq

tar -zcvf PAC0013_S1_R1_001.fastq B73_leaf_sample1_R1.1.fastq
tar -zcvf PAC0013_S1_R2_001.fastq B73_leaf_sample1_R2.1.fastq
tar -zcvf PAC0014_S2_R1_001.fastq B73_leaf_sample1_R1.2.fastq
tar -zcvf PAC0014_S2_R2_001.fastq B73_leaf_sample1_R2.2.fastq
tar -zcvf PAC0015_S3_R1_001.fastq B73_leaf_sample2_R1.1.fastq
tar -zcvf PAC0015_S3_R2_001.fastq B73_leaf_sample2_R2.1.fastq
tar -zcvf PAC0016_S4_R1_001.fastq B73_leaf_sample2_R1.2.fastq
tar -zcvf PAC0016_S4_R2_001.fastq B73_leaf_sample2_R2.2.fastq
tar -zcvf PAC0017_S5_R1_001.fastq B73_leaf_sample3_R1.1.fastq
tar -zcvf PAC0017_S5_R2_001.fastq B73_leaf_sample3_R3.1.fastq
tar -zcvf PAC0018_S6_R1_001.fastq B73_leaf_sample3_R1.2.fastq
tar -zcvf PAC0018_S6_R2_001.fastq B73_leaf_sample3_R2.2.fastq
tar -zcvf PAC0019_S7_R1_001.fastq W22_leaf_sample1_R1.fastq
tar -zcvf PAC0019_S7_R2_001.fastq W22_leaf_sample1_R2.fastq
tar -zcvf PAC0020_S8_R1_001.fastq W22_leaf_sample2_R1.fastq
tar -zcvf PAC0020_S8_R2_001.fastq W22_leaf_sample2_R2.fastq

#qsub /scratch.global/nosha003/fastq/fastq_transfer_zip.sh
```
scp nosha003@login.msi.umn.edu:/scratch.global/nosha003/fastq/chromatin_ms_fastq.gz /Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/fastq/.



#### transfer files to NCBI 
```{r eval=F}
#8March2019
#fastq_transfer_ncbi.sh
#pull fastq files from msi and upload to ncbi for sra
	
#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=8,mem=60gb
#PBS -o /scratch.global/nosha003/e_o/ncbi_o
#PBS -e /scratch.global/nosha003/e_o/ncbi_e
#PBS -N ncbi
#PBS -V
#PBS -r n

module load aspera

cd /scratch.global/nosha003/fastq/bpw_submit/

mkdir subasp@upload.ncbi.nlm.nih.gov:uploads/nosha003_umn.edu_0MdZWHfk/bpw_wgbs
ascp -i /scratch.global/nosha003/fastq/bpw_submit/aspera.openssh -QT -l100m -k1 -d /scratch.global/nosha003/fastq/bpw_submit subasp@upload.ncbi.nlm.nih.gov:uploads/nosha003_umn.edu_0MdZWHfk/bpw_wgbs

cd /scratch.global/nosha003/fastq/b73_submit/

mkdir subasp@upload.ncbi.nlm.nih.gov:uploads/nosha003_umn.edu_0MdZWHfk/b73_wgbs
ascp -i /scratch.global/nosha003/fastq/geo_submit/aspera.openssh -QT -l100m -k1 -d /scratch.global/nosha003/fastq/b73_submit subasp@upload.ncbi.nlm.nih.gov:uploads/nosha003_umn.edu_0MdZWHfk/b73_wgbs

#qsub /scratch.global/nosha003/fastq/fastq_transfer_ncbi.sh
```


- WGBS data sheet: https://docs.google.com/spreadsheets/d/16aj1gv4h7djd331PIMtvkESB5AV_5WMh9TqwXENSlM4/edit#gid=619222645

## SRA metadata
Biosample attributes:

B73 shoot			Zea mays	not applicable	not applicable	not applicable	6 day old seedling	seedling	USA: Georgia	above-ground tissue									B73
B73 root			Zea mays	not applicable	not applicable	not applicable	6 day old seedling	seedling	USA: Georgia	below-ground tissue									B73
B73 leaf			Zea mays	not applicable	not applicable	not applicable	10 day old seedling	seedling	USA: Minneosta	3rd seedling leaf tissue									B73
PH207 shoot			Zea mays	not applicable	not applicable	not applicable	6 day old seedling	seedling	USA: Georgia	above-ground tissue									PH207
W22 leaf			Zea mays	not applicable	not applicable	not applicable	10 day old seedling	seedling	USA: Minneosta	3rd seedling leaf tissue									W22

Metadata:

B73 shoot	B73_Shoot	WGBS of Zea mays: B73 shoot	Bisulfite-Seq	GENOMIC	unspecified	paired	ILLUMINA	Illumina HiSeq 2000	KAPA library preparation kit, bisulfite sodium treatment, library enrichment through PCR, paired-end sequencing	fastq		JN25_BS_S8_R1_001.fastq	JN25_BS_S8_R2_001.fastq						
B73 root	B73_Root	WGBS of Zea mays: B73 root	Bisulfite-Seq	GENOMIC	unspecified	paired	ILLUMINA	Illumina HiSeq 2000	KAPA library preparation kit, bisulfite sodium treatment, library enrichment through PCR, paired-end sequencing	fastq		JN26_BS_S1_R1_001.fastq	JN26_BS_S1_R2_001.fastq						
B73 leaf	B73_Leaf	WGBS of Zea mays: B73 leaf	Bisulfite-Seq	GENOMIC	unspecified	paired	ILLUMINA	Illumina HiSeq 2000	KAPA library preparation kit, bisulfite sodium treatment, library enrichment through PCR, paired-end sequencing	fastq		PAC0013_S1_R1_001.fastq	PAC0013_S1_R2_001.fastq	PAC0014_S2_R1_001.fastq	PAC0014_S2_R2_001.fastq	PAC0015_S3_R1_001.fastq	PAC0015_S3_R2_001.fastq	PAC0016_S4_R1_001.fastq	PAC0016_S4_R2_001.fastq
PH207 shoot	PH207_Shoot	WGBS of Zea mays: PH207 shoot	Bisulfite-Seq	GENOMIC	unspecified	paired	ILLUMINA	Illumina HiSeq 2000	KAPA library preparation kit, bisulfite sodium treatment, library enrichment through PCR, paired-end sequencing	fastq		JN27_BS_S9_R1_001.fastq	JN27_BS_S9_R2_001.fastq						
W22 leaf	W22_Leaf	WGBS of Zea mays: W22 leaf	Bisulfite-Seq	GENOMIC	unspecified	paired	ILLUMINA	Illumina HiSeq 2000	KAPA library preparation kit, bisulfite sodium treatment, library enrichment through PCR, paired-end sequencing	fastq		PAC0019_S7_R1_001.fastq	PAC0019_S7_R2_001.fastq	PAC0020_S8_R1_001.fastq	PAC0020_S8_R2_001.fastq 				

## 100bp tile files
scp nosha003@login.msi.umn.edu:/home/springer/nosha003/wgbs_schmitz/tiles_output/B73_V2_100bp_ratio.txt /Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/100bp_files/B73toB73_shoot_100bp_wgbs.txt
scp nosha003@login.msi.umn.edu:/home/springer/nosha003/wgbs_schmitz/tiles_output/B73_Root_100bp_ratio.txt /Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/100bp_files/B73toB73_root_100bp_wgbs.txt
scp nosha003@login.msi.umn.edu:/home/springer/nosha003/wgbs/dec2017/w22/W22_100bp_ratio.txt /Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/100bp_files/W22toW22_leaf_100bp_wgbs.txt
scp nosha003@login.msi.umn.edu:/home/springer/nosha003/wgbs_schmitz/PH207/PH207_100bp_ratio_chr.txt /Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/100bp_files/PH207toPH207_shoot_100bp_wgbs.txt
scp nosha003@login.msi.umn.edu:/home/springer/nosha003/wgbs_schmitz/Mo17/Mo17_100bp_ratio.txt /Volumes/CBS/Groups/LAB-springer/Jaclyn/Graduate_projects/methylation/100bp_files/Mo17toMo17_leaf_100bp_wgbs.txt




# Supplemental Tables
- list of non-nested TE families 
    - column stating cluster
    - column stating if shared with other 3 genotypes

```{r eval=F}
setwd("/home/springer/nosha003/methylation_spreading/")
fam_ltr <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
fam_tir <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
fam <- rbind(fam_ltr, fam_tir)
colnames(fam) <- c("B73_count", "fam")
tir_shared <- read.delim("BWPM_tir_fam_count20_nonnested_20Dec2018.txt", header=T, sep="\t")
colnames(tir_shared) <- c("B73_count", "fam", "W22_count", "PH207_count", "Mo17_count")
tir_shared$across_genomes <- "shared"
ltr_shared <- read.delim("BWPM_ltr_fam_count20_nonnested_20Dec2018.txt", header=T, sep="\t")
colnames(ltr_shared) <- c("B73_count", "fam", "W22_count", "PH207_count", "Mo17_count")
ltr_shared$across_genomes <- "shared"

library(dplyr)
fam_tir_shared <- left_join(fam, tir_shared, by="fam")
fam_ltr_shared <- left_join(fam, ltr_shared, by="fam")

all <- rbind(fam_tir_shared[,c(2,1,7,4:6)], fam_ltr_shared[,c(2,1,7,4:6)])
write.table(all, "chromatin_ms_supplemental_fam_list.txt", quote=F, row.names=F, sep="\t")
# columns = family, b73 count, shared or NA, w22 count, ph207 count, mo17 count
```

- list of elements in Figure 4
    - state which haplotypes
    - state identification as methylated, intermediate, unmethylated
    - for unmethylated (continue to figure 5) --> state flanking methylation levels

```{r eval=F}
library(dplyr)

setwd("/scratch.global/nosha003/methylation_spreading")
df <- read.delim("all_empty_melt_cluster2_20Dec.txt", header=T, sep="\t")

df$genome <- substr(df$TE, 15, 16)
df$TE <- substr(df$TE, 1, 21)
df_cg <- subset(df, df$context == "V4")
df2 <- df_cg %>% mutate(TE_genotype = ifelse(genome == "1d", "B73", ifelse(genome == "4b", "W22", ifelse(genome == "8a", "PH207", ifelse(genome == "4a", "Mo17", "NA"))))) %>% mutate(site_mC_state = ifelse(mC_state == "mC", "Methylated", ifelse(mC_state == "C", "Unmethylated", ifelse(mC_state == "intermediate", "Intermediate", "NA")))) %>% mutate(category = ifelse(cluster == "A", "H", ifelse(cluster == "B", "M", ifelse(cluster == "C", "L", "NA"))))
df2_genotype <- subset(df2, df2$TE_genotype != "NA")

flank <- read.delim("w_b_flank_mC_20Dec.txt", header=T, sep="\t")
flank2 <- flank %>% mutate(flank_mC_state = ifelse(cg < 0.20, "Unmethylated", ifelse(cg > 0.40, "Methylated", ifelse(cg >= 0.20 & cg <= 0.40, "Intermediate", "NA"))))
df_flank <- left_join(df2_genotype[,c(1,9,11,10)], flank2[,c(1,5)], by="TE")

setwd("/home/springer/nosha003/methylation_spreading/")
write.table(df_flank, "chromatin_ms_supplemental_mC_site_flank.txt", quote=F, row.names=F, sep="\t")
```




# New chromatin data (SRA)
--> DNase-seq 
doi: 10.1186/s13059-017-1273-4
geo: GSE94291 (DNase-seq), GSE94251 (H3K9ac)
sra: SRP098579
     SRR5218002, SRR5218003	(DNase-seq)
     SRX2527280, SRX2527279 (H3K9ac)

--> H3K27me3
doi: 10.1104/pp.17.01467
sra: PRJNA382414 (DNase-seq, H3K4me1, H3K4me2, H3K4e3, H3K4ac, H3K27ac, H3K27me3)
     SRR5436222	(H3K27me3)
     SRX2726177 (DNase-seq)
     
```{r eval=F}
module load sratoolkit

cd /home/springer/nosha003/sra

fastq-dump SRR5436222
fastq-dump SRR5436230	
mv SRR5436222.fastq H3K27me3_sra.fastq
mv SRR5436230.fastq DNase_zhao_sra.fastq

fastq-dump SRR5218002
fastq-dump SRR5218003
fastq-dump SRR5217070	
fastq-dump SRR5217069
mv SRR5218002.fastq DNase_rep1_oka_sra.fastq
mv SRR5218003.fastq DNase_rep2_oka_sra.fastq
mv SRR5217070.fastq H3K9ac_rep1_sra.fastq
mv SRR5217069.fastq H3K9ac_rep2_sra.fastq
```

## Alignment
### Names File
/home/springer/nosha003/sra_chromatin_names.txt
```{r eval=F}
H3K27me3_sra
DNase_zhao_sra
DNase_rep1_oka_sra
DNase_rep2_oka_sra
H3K9ac_rep1_sra
H3K9ac_rep2_sra
```

### fastqc
```{r eval=F}
#8March2019

#!/bin/bash
#PBS -l walltime=08:00:00,nodes=1:ppn=1,mem=5gb
#PBS -o /scratch.global/nosha003/e_o/sra_fastqc_o
#PBS -e /scratch.global/nosha003/e_o/sra_fastqc_e
#PBS -N sra_fastqc
#PBS -V
#PBS -r n

SAMPLE="$(/bin/sed -n ${PBS_ARRAYID}p ${LIST} | cut -f 1)"

module load fastqc

mkdir /scratch.global/nosha003/chromatin_sra
mkdir /scratch.global/nosha003/chromatin_sra/fastqc

fastqc -o /scratch.global/nosha003/chromatin_sra/fastqc --noextract -f fastq /home/springer/nosha003/sra/${SAMPLE}.fastq

# qsub -t 1-6 -v LIST=/home/springer/nosha003/sra_chromatin_names.txt /home/springer/nosha003/scripts/atacseq_fastqc.sh
```

### clean reads
```{r eval=F}
#8March2019
#sra_clean2.sh
#!/bin/bash

#PBS -l walltime=02:00:00,nodes=1:ppn=1,mem=2gb
#PBS -o /scratch.global/nosha003/e_o/sra_clean_o
#PBS -e /scratch.global/nosha003/e_o/sra_clean_e
#PBS -N sra_clean
#PBS -V
#PBS -r n

SAMPLE="$(/bin/sed -n ${PBS_ARRAYID}p ${LIST} | cut -f 1)"

module load cutadapt
module load fastqc
module load fastx_toolkit
module load python
module load trimmomatic

mkdir /scratch.global/nosha003/chromatin_sra/trimmomatic
mkdir /scratch.global/nosha003/chromatin_sra/cutadapt

gunzip /home/springer/nosha003/sra/${SAMPLE}.fastq

# trimmomatix 
java -Xmx7000M -jar $TRIMMOMATIC/trimmomatic.jar SE -threads 4 -phred64 /home/springer/nosha003/sra/${SAMPLE}.fastq /scratch.global/nosha003/chromatin_sra/trimmomatic/${SAMPLE}.fastq ILLUMINACLIP:$TRIMMOMATIC/adapters/all_illumina_adapters.fa:2:30:10:2:true LEADING:20 TRAILING:20 SLIDINGWINDOW:4:20 MINLEN:35

# cutadapt
cutadapt -b AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT -f fastq -m 30 -q 10 --quality-base=33 -o /scratch.global/nosha003/chromatin_sra/cutadapt/${SAMPLE}_cutadapt.fastq /home/springer/nosha003/sra/${SAMPLE}.fastq

gzip /home/springer/nosha003/sra/${SAMPLE}.fastq

# qsub -t 1-6 -v LIST=/home/springer/nosha003/sra_chromatin_names.txt /home/springer/nosha003/scripts/sra_clean2.sh
```

### index
- make bowtie1 index with B73v4
```{r eval=F}
#8March2019
#sra_index.sh
#!/bin/bash

#Bowtie_Alignment
#PBS -l walltime=10:00:00,nodes=1:ppn=6,mem=10gb	
#PBS -o /scratch.global/nosha003/e_o/sra_index_o
#PBS -e /scratch.global/nosha003/e_o/sra_index_e
#PBS -N sra_index
#PBS -V
#PBS -r n

#module load bowtie/2.3.2
#bowtie-build /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.dna.toplevel.fa /home/springer/nosha003/database/bowtie1db/Zea_mays.AGPv4.dna.toplevel

mkdir /home/springer/nosha003/database/bowtie2db

module load bowtie2/2.2.4
bowtie2-build /home/springer/nosha003/database/B73v4/Zea_mays.AGPv4.dna.toplevel.fa /home/springer/nosha003/database/bowtie2db/Zea_mays.AGPv4.dna.toplevel

# qsub /home/springer/nosha003/scripts/sra_index.sh
```


### align to reference
```{r eval=F}
#8March2019
#sra_align2.sh
#!/bin/bash

#Bowtie_Alignment
#PBS -l walltime=10:00:00,nodes=1:ppn=6,mem=10gb	
#PBS -o /scratch.global/nosha003/e_o/sra_align2_o
#PBS -e /scratch.global/nosha003/e_o/sra_align2_e
#PBS -N sra_align2
#PBS -V
#PBS -r n

ID="$(/bin/sed -n ${PBS_ARRAYID}p ${LIST} | cut -f 1)"

module load bowtie/2.2.4
module load samtools/1.3
module load bowtie2/2.2.4

cd /scratch.global/nosha003/chromatin_sra
mkdir /scratch.global/nosha003/chromatin_sra/align

bowtie /home/springer/nosha003/database/bowtie1db/Zea_mays.AGPv4.dna.genome -X 1000 -0 /scratch.global/nosha003/chromatin_sra/${ID}.fastq -S /scratch.global/nosha003/chromatin_sra/align/${ID}_bowtie.sam -t -p 4 -v 0 --best --strata -m 3

bowtie2 -p 6 --phred33 -x /home/maize/shared/databases/bt2/Zea_mays/B73/Zea_mays.AGPv4.dna.toplevel -U /scratch.global/nosha003/chromatin_sra/cutadapt/${ID}_cutadapt.fastq -S /scratch.global/nosha003/chromatin_sra/align/${ID}_bowtie2.sam

samtools view -S -b -o /scratch.global/nosha003/chromatin_sra/align/${ID}_bowtie2.bam /scratch.global/nosha003/chromatin_sra/align/${ID}_bowtie2.sam

# qsub -t 1-6 -v LIST=/home/springer/nosha003/sra_chromatin_names.txt /home/springer/nosha003/scripts/sra_align2.sh
```

### Uniquely mapped reads
```{r eval=F}
#8March2019
#sra_align_unique.sh
#!/bin/bash

#Uniquely mapping reads
#PBS -l walltime=10:00:00,nodes=1:ppn=6,mem=10gb	
#PBS -o /scratch.global/nosha003/e_o/sra_uniq_o
#PBS -e /scratch.global/nosha003/e_o/sra_uniq_e
#PBS -N sra_uniq
#PBS -V
#PBS -r n

ID="$(/bin/sed -n ${PBS_ARRAYID}p ${LIST} | cut -f 1)"

module load samtools

cd /scratch.global/nosha003/chromatin_sra

samtools view -H align/${ID}_bowtie2.bam > align/${ID}_header.sam
samtools view -F 4 align/${ID}_bowtie2.bam | grep -v "XS:" | cat align/${ID}_header.sam - | \
samtools view -b - > align/${ID}_uniq.bam
rm align/${ID}_header.sam

samtools sort -O bam -T temp_${ID} align/${ID}_uniq.bam > align/${ID}_uniq_sort.bam 
samtools index align/${ID}_uniq_sort.bam
samtools view -h -o align/${ID}_uniq_sort.sam align/${ID}_uniq_sort.bam

samtools sort -O bam -T temp_${ID} align/${ID}_bowtie2.bam > align/${ID}_sort.bam 
samtools index align/${ID}_sort.bam
samtools view -h -o align/${ID}_sort.sam align/${ID}_sort.bam

# qsub -t 1-6 -v LIST=/home/springer/nosha003/sra_chromatin_names.txt /home/springer/nosha003/scripts/sra_align_unique.sh
```


### 100bp bins
```{r eval=F}
#8March2019
#sra_count
#count reads for 100 bp tiles with bedtools intersect
	
#!/bin/bash
#PBS -l walltime=08:00:00,nodes=1:ppn=1,mem=50gb
#PBS -o /scratch.global/nosha003/e_o/sra_count_o
#PBS -e /scratch.global/nosha003/e_o/sra_count_e
#PBS -N sra_count
#PBS -V
#PBS -r n

ID="$(/bin/sed -n ${PBS_ARRAYID}p ${LIST} | cut -f 1)"

#load modules
module load bedtools
module load samtools
	
cd /scratch.global/nosha003/chromatin_sra
mkdir /scratch.global/nosha003/chromatin_sra/counts

#count number of reads aligning to each 100 bp window
#grep -v 'ctg' align/${ID}_sort.sam | grep -v 'Mt' | grep -v 'Pt' > align/${ID}_sort_noctg.sam
#grep -v 'ctg' align/${ID}_uniq_sort.sam | grep -v 'Mt' | grep -v 'Pt' > align/${ID}_uniq_sort_noctg.sam

bedtools intersect -a align/${ID}_sort.bam -b /home/springer/nosha003/database/B73v4/B73v4_100bp_bins.bed -bed -wa -wb > counts/${ID}.100bpbins.bed
bedtools intersect -a align/${ID}_uniq_sort.bam -b /home/springer/nosha003/database/B73v4/B73v4_100bp_bins.bed -bed -wa -wb > counts/${ID}.uniq.100bpbins.bed

## all bins
perl /home/springer/nosha003/doebley/scripts/bincounts.pl -i /home/springer/nosha003/database/B73v4/B73v4_100bp_bins.bed -I counts/${ID}.100bpbins.bed -o counts/${ID}.counts.txt
perl /home/springer/nosha003/doebley/scripts/bincounts.pl -i /home/springer/nosha003/database/B73v4/B73v4_100bp_bins.bed -I counts/${ID}.uniq.100bpbins.bed -o counts/${ID}.uniq.counts.txt

## all bins bed count file
#cut -f 13,16-21 counts/${ID}.100bpbins.bed | sort | uniq > counts/${ID}.100bpbins_cut.bed
perl /home/springer/nosha003/wenli_data/scripts/bincounts_bed.pl -i counts/${ID}.counts.txt -I /home/springer/nosha003/database/B73v4/B73v4_100bp_bins.bed -o counts/${ID}.100bpcounts.bed
perl /home/springer/nosha003/wenli_data/scripts/bincounts_bed.pl -i counts/${ID}.uniq.counts.txt -I /home/springer/nosha003/database/B73v4/B73v4_100bp_bins.bed -o counts/${ID}.uniq.100bpcounts.bed

#qsub -t 1-6 -v LIST=/home/springer/nosha003/sra_chromatin_names.txt /home/springer/nosha003/scripts/sra_count.sh
```

#### rpm
```{r eval=F}
chromatin_counts_rpm <- function(input_dir, file, chromatin) {
library(dplyr)
  
setwd(input_dir)
df <- read.delim(file, header=T, sep="\t")

total <- sum(df$count)
pm <- total / 1000000
df$rpm <- df$count / pm

write.table(df[,c(1:3,5)], paste(chromatin, "_rpm.txt", sep=""), quote=FALSE, sep="\t", row.names=F)
}
```

```{r eval=F}
chromatin_counts_rpm(
  input_dir = "/scratch.global/nosha003/chromatin_sra/counts/",
  file = "DNase_rep1_oka_sra.uniq.100bpcounts.bed",
  chromatin = "DNase_rep1_oka"
)
chromatin_counts_rpm(
  input_dir = "/scratch.global/nosha003/chromatin_sra/counts/",
  file = "DNase_rep2_oka_sra.uniq.100bpcounts.bed",
  chromatin = "DNase_rep2_oka"
)
chromatin_counts_rpm(
  input_dir = "/scratch.global/nosha003/chromatin_sra/counts/",
  file = "DNase_zhao_sra.uniq.100bpcounts.bed",
  chromatin = "DNase_zhao"
)
chromatin_counts_rpm(
  input_dir = "/scratch.global/nosha003/chromatin_sra/counts/",
  file = "H3K27me3_sra.uniq.100bpcounts.bed",
  chromatin = "H3K27me3"
)
chromatin_counts_rpm(
  input_dir = "/scratch.global/nosha003/chromatin_sra/counts/",
  file = "H3K9ac_rep1_sra.uniq.100bpcounts.bed",
  chromatin = "H3K9ac_rep1"
)
chromatin_counts_rpm(
  input_dir = "/scratch.global/nosha003/chromatin_sra/counts/",
  file = "H3K9ac_rep2_sra.uniq.100bpcounts.bed",
  chromatin = "H3K9ac_rep2"
)
chromatin_counts_rpm(
  input_dir = "/home/springer/nosha003/chipseq/",
  file = "SRR1482362_100bpcounts_sort.bed",
  chromatin = "H3K9me2"
)
```



## Figure 3 metaplots
### bedtools
```{r eval=F}
#14March2019
#sra_bedtools.sh
	
#!/bin/bash
#PBS -l walltime=08:00:00,nodes=1:ppn=1,mem=50gb
#PBS -o /scratch.global/nosha003/e_o/sra_bedtools_o
#PBS -e /scratch.global/nosha003/e_o/sra_bedtools_e
#PBS -N sra_bedtools
#PBS -V
#PBS -r n

module load bedtools

## remove contigs and sort gff files
#/home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3
#/home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3

cd /scratch.global/nosha003/chromatin_sra/counts/
sort -k 1,1 -k 2,2n DNase_rep1_oka_rpm.txt > DNase_rep1_oka_sra.uniq.100bpcounts.sort.bed
sort -k 1,1 -k 2,2n DNase_rep2_oka_rpm.txt > DNase_rep2_oka_sra.uniq.100bpcounts.sort.bed
sort -k 1,1 -k 2,2n DNase_zhao_rpm.txt > DNase_zhao_sra.uniq.100bpcounts.sort.bed
sort -k 1,1 -k 2,2n H3K27me3_rpm.txt > H3K27me3_sra.uniq.100bpcounts.sort.bed
sort -k 1,1 -k 2,2n H3K9ac_rep1_rpm.txt > H3K9ac_rep1_sra.uniq.100bpcounts.sort.bed
sort -k 1,1 -k 2,2n H3K9ac_rep2_rpm.txt > H3K9ac_rep2_sra.uniq.100bpcounts.sort.bed
sort -k 1,1 -k 2,2n /home/springer/nosha003/chipseq/H3K9me2_rpm.txt > /home/springer/nosha003/chipseq/H3K9me2_sra.uniq.100bpcounts.sort.bed

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /scratch.global/nosha003/chromatin_sra/counts/DNase_rep1_oka_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_tir_closestTEtoBin_DNase.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /scratch.global/nosha003/chromatin_sra/counts/DNase_rep1_oka_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_ltr_closestTEtoBin_DNase.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /scratch.global/nosha003/chromatin_sra/counts/DNase_rep2_oka_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_tir_closestTEtoBin_DNase2.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /scratch.global/nosha003/chromatin_sra/counts/DNase_rep2_oka_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_ltr_closestTEtoBin_DNase2.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /scratch.global/nosha003/chromatin_sra/counts/DNase_zhao_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_tir_closestTEtoBin_DNase3.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /scratch.global/nosha003/chromatin_sra/counts/DNase_zhao_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_ltr_closestTEtoBin_DNase3.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /scratch.global/nosha003/chromatin_sra/counts/H3K27me3_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_tir_closestTEtoBin_H3K27me3.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /scratch.global/nosha003/chromatin_sra/counts/H3K27me3_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_ltr_closestTEtoBin_H3K27me3.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /scratch.global/nosha003/chromatin_sra/counts/H3K9ac_rep1_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_tir_closestTEtoBin_H3K9ac.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /scratch.global/nosha003/chromatin_sra/counts/H3K9ac_rep1_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_ltr_closestTEtoBin_H3K9ac.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /scratch.global/nosha003/chromatin_sra/counts/H3K9ac_rep2_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_tir_closestTEtoBin_H3K9ac2.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /scratch.global/nosha003/chromatin_sra/counts/H3K9ac_rep2_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_ltr_closestTEtoBin_H3K9ac2.txt


# with old data
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/Chip_B73_leaf_H3K56ac_avg.100bpcounts.rpm.txt > /scratch.global/nosha003/chromatin_sra/counts/B73_tir_closestTEtoBin_H3K56ac.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq_schmitz/bins/Chip_B73_leaf_H3K56ac_avg.100bpcounts.rpm.txt > /scratch.global/nosha003/chromatin_sra/counts/B73_ltr_closestTEtoBin_H3K56ac.txt

bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.TIR.noctg.gff3 -a /home/springer/nosha003/chipseq/H3K9me2_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_tir_closestTEtoBin_H3K9me2.txt
bedtools closest -D b -k 1 -b /home/springer/nosha003/database/B73v4/te/B73.structuralTEv2.2018.12.20.filteredTE.LTR.noctg.gff3 -a /home/springer/nosha003/chipseq/H3K9me2_sra.uniq.100bpcounts.sort.bed > /scratch.global/nosha003/chromatin_sra/counts/B73_ltr_closestTEtoBin_H3K9me2.txt

#qsub /home/springer/nosha003/scripts/sra_bedtools.sh
```

### orientation
```{r eval=F}
#11March2019
#chromatin_sra_orientation
#orient chromatin data based on CG/CHG methylation patterns
	
#!/bin/bash
#PBS -l walltime=36:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/chromatin_sra_orientation_o
#PBS -e /scratch.global/nosha003/e_o/chromatin_sra_orientation_e
#PBS -N chromatin_sra_orientation
#PBS -V
#PBS -r n

cd /scratch.global/nosha003/
module load R
#R CMD BATCH chromatin_sra_orientation.R
R CMD BATCH chromatin_sra_orientation_hardcoded.R
wait

# qsub /scratch.global/nosha003/chromatin_sra_orientation.sh
# qsub /scratch.global/nosha003/chromatin_sra_orientation_hardcoded.sh
```

```{r eval=F}
# qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G
# module load R
# R

chromatin_metaplot_orientation <- function(input_dir, input_dir2, orientation_file, chromatin_file, genotype, chromatin, TEtype, date) {
library(dplyr)

#setwd(input_dir)
#orientation <- read.delim(orientation_file, sep="\t", header=T, stringsAsFactors = F)
#setwd(input_dir2)
#chromatin <- read.delim(chromatin_file, sep="\t", header=F, stringsAsFactors = F)

setwd("/home/springer/nosha003/methylation_spreading")
orientation <- read.delim("B73_tir_closestTEtoBin_orientation_20Dec2018.txt", sep="\t", header=T, stringsAsFactors = F)
setwd("/scratch.global/nosha003/chromatin_sra/counts")
chromatin <- read.delim("B73_tir_closestTEtoBin_DNase.txt", sep="\t", header=F, stringsAsFactors = F)

colnames(chromatin) <- c("chr", "start", "end", "count", "chr_te", "te_type", "te_type2", "te_start", "te_end", "dot", "strand", "dot2", "te_id", "dist")
chromatin$id <- substr(chromatin$te_id, 4, 24)
orientation$id <- substr(orientation$TE, 4, 24)
orientation_id <- unique(orientation[,c(10,12)])

chromatin_orientation <- left_join(chromatin, orientation_id, by="id")
df <- chromatin_orientation %>% mutate(correction = replace(correction, is.na(correction), 1))
df2 <- df %>% mutate(chromatin_distance = ifelse(correction == 1, dist, ifelse(correction == "NA", dist, ifelse(correction == -1, (dist * -1), "NA"))))
df3 <- df2[,c(1,2,4,8,9,15,17)]

#write.table(df3, paste(genotype, "_", TEtype, "_", chromatin, "_orientation_", date, ".txt", sep=""), quote=FALSE, sep="\t", row.names=F)
write.table(df3, "B73_tir_DNase_orientation_11March.txt", quote=FALSE, sep="\t", row.names=F)
}
```

```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

## ltr
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K9me2.txt",
  chromatin = "H3K9me2",
  genotype = "B73",
  TEtype = "ltr",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K56ac.txt",
  chromatin = "H3K56ac",
  genotype = "B73",
  TEtype = "ltr",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_DNase.txt",
  chromatin = "DNase",
  genotype = "B73",
  TEtype = "ltr",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_DNase2.txt",
  chromatin = "DNase2",
  genotype = "B73",
  TEtype = "ltr",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_DNase3.txt",
  chromatin = "DNase3",
  genotype = "B73",
  TEtype = "ltr",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K27me3.txt",
  chromatin = "H3K27me3",
  genotype = "B73",
  TEtype = "ltr",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K9ac.txt",
  chromatin = "H3K9ac",
  genotype = "B73",
  TEtype = "ltr",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_ltr_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_ltr_closestTEtoBin_H3K9ac2.txt",
  chromatin = "H3K9ac2",
  genotype = "B73",
  TEtype = "ltr",
  date="11March"
)


## tir
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K9me2.txt",
  chromatin = "H3K9me2",
  genotype = "B73",
  TEtype = "tir",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K56ac.txt",
  chromatin = "H3K56ac",
  genotype = "B73",
  TEtype = "tir",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_DNase.txt",
  chromatin = "DNase",
  genotype = "B73",
  TEtype = "tir",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_DNase2.txt",
  chromatin = "DNase2",
  genotype = "B73",
  TEtype = "tir",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_DNase3.txt",
  chromatin = "DNase3",
  genotype = "B73",
  TEtype = "tir",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K27me3.txt",
  chromatin = "H3K27me3",
  genotype = "B73",
  TEtype = "tir",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K9ac.txt",
  chromatin = "H3K9ac",
  genotype = "B73",
  TEtype = "tir",
  date="11March"
)
chromatin_metaplot_orientation(
  input_dir = "/home/springer/nosha003/methylation_spreading",
  input_dir2 = "/scratch.global/nosha003/chromatin_sra/counts",
  orientation_file = "B73_tir_closestTEtoBin_orientation_20Dec2018.txt",
  chromatin_file = "B73_tir_closestTEtoBin_H3K9ac2.txt",
  chromatin = "H3K9ac2",
  genotype = "B73",
  TEtype = "tir",
  date="11March"
)

```

#### perl script for metaplot data
```{r eval=F}
#11March2019
#metaplot perl scripts for corrected chromatin data
	
#!/bin/bash
#PBS -l walltime=24:00:00,nodes=1:ppn=6,mem=20gb
#PBS -o /scratch.global/nosha003/e_o/chromatin_o
#PBS -e /scratch.global/nosha003/e_o/chromatin_e
#PBS -N chromatin
#PBS -V
#PBS -r n

SAMPLE="$(/bin/sed -n ${PBS_ARRAYID}p ${LIST} | cut -f 1)"
ID="$(/bin/sed -n ${PBS_ARRAYID}p ${LIST} | cut -f 2)"

module load bedtools

# perl script

## corrected TE orientation based on CG methylation
perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance4.pl -i /scratch.global/nosha003/chromatin_sra/counts/B73_ltr_${ID}_orientation_11March.txt -o /scratch.global/nosha003/methylation_spreading/B73_ltr_${ID}_orientation_metaplot.txt

perl /home/springer/nosha003/methylation_spreading/metaplot_class_distance4.pl -i /scratch.global/nosha003/chromatin_sra/counts/B73_tir_${ID}_orientation_11March.txt -o /scratch.global/nosha003/methylation_spreading/B73_tir_${ID}_orientation_metaplot.txt

#qsub -t 1-8 -v LIST=/home/springer/nosha003/sra_chromatin_names.txt /home/springer/nosha003/methylation_spreading/chromatin_sra_meatplot_orientation.sh
```


#### metaplot
```{r eval=F}
#17March2019
#chromatin_sra_metaplot
#generate metaplot for chromatin data oriented based on CG/CHG methylation patterns
	
#!/bin/bash
#PBS -l walltime=48:00:00,nodes=1:ppn=8,mem=90gb
#PBS -o /scratch.global/nosha003/e_o/chromatin_metaplot_o
#PBS -e /scratch.global/nosha003/e_o/chromatin_metaplot_e
#PBS -N chromatin_metaplot
#PBS -V
#PBS -r n

#cd /home/springer/nosha003/scripts/
cd /scratch.global/nosha003/
module load R
R CMD BATCH chromatin_sra_metaplot2.R
wait

# qsub /scratch.global/nosha003/chromatin_sra_metaplot2.sh
```

```{r eval=F}
metaplot_chromatin_PercentBins <- function(input_dir, input_dir2, file, fam_file, class_file, genotype, TEtype, date, context) {
library(fields)
library(dplyr)
library(reshape2)
library(ggplot2)
library(Rmisc)

# setwd("/scratch.global/nosha003/methylation_spreading")
# b_tir <- read.delim("B73_tir_DNase_orientation_metaplot.txt", sep="\t", header=T)
# b_tir_fam <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt", sep="\t", header=T)
# setwd("/home/springer/nosha003/methylation_spreading/")
# class <- read.delim("b_tir_meandirection_kmeans3_20Dec2018.txt", sep="\t", header=T)
# class$TEfam <- row.names(class)
# class <- class[,c(81:82)]
# colnames(class) <- c("class", "TEfam")

setwd(input_dir)
b_tir <- read.delim(file, sep="\t", header=T)
b_tir_fam <- read.delim(fam_file, sep="\t", header=T)
setwd(input_dir2)
class <- read.delim(class_file, sep="\t", header=T)
class$TEfam <- row.names(class)
class <- class[,c(81:82)]
colnames(class) <- c("class", "TEfam")

b_tir$TEfam <- substr(b_tir$TE, 1, 8)
b_tir2 <- subset(b_tir, b_tir$TE %in% b_tir_fam$TE)

b_tir_cg <- as.numeric(paste(b_tir2$count))
b_tir2["rpm"] <- b_tir_cg
b_tir2$log2 <- log2(1+b_tir2$rpm)
b_tir2$chromatin <- ifelse(b_tir2$log2 >= 1, 1, 0)
b_tir3 <- left_join(b_tir2, class, by="TEfam")
dist = subset(b_tir3, b_tir3$relative_distance > -2000 & b_tir3$relative_distance <= 3000)

dist_class1 <- subset(dist, dist$class == "A")
dist_class2 <- subset(dist, dist$class == "B")
dist_class3 <- subset(dist, dist$class == "C")

dist_class1_group <-  transform(dist_class1, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class1_sum <- do.call(data.frame,aggregate(dist_class1_group$rpm~dist_class1_group$group, dist_class1_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x), Avg=mean(x))))
colnames(dist_class1_sum) <- c("group", "count", "sum", "mean")
dist_class1_sum["percent_methylated"] <- dist_class1_sum$sum / dist_class1_sum$count
dist_class1_sum["class"] <- "H"

dist_class2_group <-  transform(dist_class2, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class2_sum <- do.call(data.frame,aggregate(dist_class2_group$rpm~dist_class2_group$group, dist_class2_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x), Avg=mean(x))))
colnames(dist_class2_sum) <- c("group", "count", "sum", "mean")
dist_class2_sum["percent_methylated"] <- dist_class2_sum$sum / dist_class2_sum$count
dist_class2_sum["class"] <- "M"

dist_class3_group <-  transform(dist_class3, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class3_sum <- do.call(data.frame,aggregate(dist_class3_group$rpm~dist_class3_group$group, dist_class3_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x), Avg=mean(x))))
colnames(dist_class3_sum) <- c("group", "count", "sum", "mean")
dist_class3_sum["percent_methylated"] <- dist_class3_sum$sum / dist_class3_sum$count
dist_class3_sum["class"] <- "L"

p.1 <- rbind(dist_class1_sum, dist_class2_sum, dist_class3_sum)

## Metaplot of percent bins methylated by class
group.colors <- c("#3972c9ff", "#a43434ff", "#e7a275ff")

ggplot(p.1, aes(x = group, y = mean, group = class, colour = class)) + geom_line()

pdf(paste(genotype, TEtype, "class_metaplot", context, date, "PercentMethylated_Mean.pdf", sep="_"))
#pdf("DNase_metaplot_percent_mean.pdf")
plot1 <- ggplot(p.1, aes(x = group, y = mean, group = class, colour = class)) + geom_line(size=1) + theme_classic() + scale_color_manual(values=group.colors) + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot1)
dev.off()

# pdf("DNase_metaplot_count.pdf")
# plot1 <- ggplot(p.1, aes(x = group, y = count, group = class, colour = class)) + geom_line(size=1) + theme_classic() + scale_color_manual(values=group.colors) + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
# print(plot1)
# dev.off()

}
```

```{r eval=F}
metaplot_chromatin_PercentBins_1kb <- function(input_dir, input_dir2, file, fam_file, class_file, genotype, TEtype, date, context) {
library(fields)
library(dplyr)
library(reshape2)
library(ggplot2)
library(Rmisc)

setwd(input_dir)
b_tir <- read.delim(file, sep="\t", header=T)
b_tir_fam <- read.delim(fam_file, sep="\t", header=T)
setwd(input_dir2)
class <- read.delim(class_file, sep="\t", header=T)
class$TEfam <- row.names(class)
class <- class[,c(81:82)]
colnames(class) <- c("class", "TEfam")

b_tir_fam$length <- abs(b_tir_fam$end - b_tir_fam$start)
b_tir_fam2 <- subset(b_tir_fam, b_tir_fam$length > 1000)
  
b_tir$TEfam <- substr(b_tir$TE, 1, 8)
b_tir2 <- subset(b_tir, b_tir$TE %in% b_tir_fam2$TE)

b_tir_cg <- as.numeric(paste(b_tir2$count))
b_tir2["rpm"] <- b_tir_cg
b_tir2$log2 <- log2(1+b_tir2$rpm)
b_tir2$chromatin <- ifelse(b_tir2$log2 >= 1, 1, 0)
b_tir3 <- left_join(b_tir2, class, by="TEfam")
dist = subset(b_tir3, b_tir3$relative_distance > -2000 & b_tir3$relative_distance <= 3000)

dist_class1 <- subset(dist, dist$class == "A")
dist_class2 <- subset(dist, dist$class == "B")
dist_class3 <- subset(dist, dist$class == "C")

dist_class1_group <-  transform(dist_class1, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class1_sum <- do.call(data.frame,aggregate(dist_class1_group$rpm~dist_class1_group$group, dist_class1_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x), Avg=mean(x))))
colnames(dist_class1_sum) <- c("group", "count", "sum", "mean")
dist_class1_sum["percent_methylated"] <- dist_class1_sum$sum / dist_class1_sum$count
dist_class1_sum["class"] <- "H"

dist_class2_group <-  transform(dist_class2, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class2_sum <- do.call(data.frame,aggregate(dist_class2_group$rpm~dist_class2_group$group, dist_class2_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x), Avg=mean(x))))
colnames(dist_class2_sum) <- c("group", "count", "sum", "mean")
dist_class2_sum["percent_methylated"] <- dist_class2_sum$sum / dist_class2_sum$count
dist_class2_sum["class"] <- "M"

dist_class3_group <-  transform(dist_class3, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_class3_sum <- do.call(data.frame,aggregate(dist_class3_group$rpm~dist_class3_group$group, dist_class3_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x), Avg=mean(x))))
colnames(dist_class3_sum) <- c("group", "count", "sum", "mean")
dist_class3_sum["percent_methylated"] <- dist_class3_sum$sum / dist_class3_sum$count
dist_class3_sum["class"] <- "L"

p.1 <- rbind(dist_class1_sum, dist_class2_sum, dist_class3_sum)

## Metaplot of percent bins methylated by class
group.colors <- c("#3972c9ff", "#a43434ff", "#e7a275ff")

pdf(paste(genotype, TEtype, "class_metaplot", context, date, "PercentMethylated_Mean.pdf", sep="_"))
plot1 <- ggplot(p.1, aes(x = group, y = mean, group = class, colour = class)) + geom_line(size=1) + theme_classic() + scale_color_manual(values=group.colors) + theme(axis.title.x = element_blank(), axis.title.y = element_blank())
print(plot1)
dev.off()
}
```


```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

# LTR
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_H3K9me2_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "19March",
  context = "H3K9me2"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_H3K56ac_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "19March",
  context = "H3K56ac"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_DNase_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "19March",
  context = "DNase"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_DNase2_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "19March",
  context = "DNase2"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_DNase3_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "19March",
  context = "DNase3"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_H3K27me3_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "19March",
  context = "H3K27me3"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_H3K9ac_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "19March",
  context = "H3K9ac"
)
metaplot_chromatin_PercentBins(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_ltr_H3K9ac2_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_ltr_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "19March",
  context = "H3K9ac2"
)


# TIR
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_H3K9me2_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "19March",
  context = "H3K9me2"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_H3K56ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "19March",
  context = "H3K56ac"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_DNase_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "19March",
  context = "DNase"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_DNase2_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "19March",
  context = "DNase2"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_DNase3_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "19March",
  context = "DNase3"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_H3K27me3_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "19March",
  context = "H3K27me3"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_H3K9ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "19March",
  context = "H3K9ac"
)
metaplot_chromatin_PercentBins_1kb(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  input_dir2 = "/home/springer/nosha003/methylation_spreading/",
  file = "B73_tir_H3K9ac2_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  class_file = "b_tir_meandirection_kmeans3_20Dec2018.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "19March",
  context = "H3K9ac2"
)
```




## Figure 6 metaplots
```{r eval=F}
#19March2019
#chromatin_metaplot_spreading_sra

#!/bin/bash
#PBS -l walltime=24:00:00,nodes=1:ppn=8,mem=60gb
#PBS -o /scratch.global/nosha003/e_o/chromatin_spreading_o
#PBS -e /scratch.global/nosha003/e_o/chromatin_spreading_e
#PBS -N chromatin_spreading
#PBS -V
#PBS -r n

cd /scratch.global/nosha003/
module load R
R CMD BATCH chromatin_metaplot_spreading_ltr_sra.R
R CMD BATCH chromatin_metaplot_spreading_tir1kb_sra.R
wait

# qsub /scratch.global/nosha003/chromatin_metaplot_spreading_sra.sh
```

```{r eval=F}
#qsub -I -l walltime=3:00:00,nodes=1:ppn=1,mem=60G

metaplot_chromatin_spreading <- function(input_dir, file, fam_file, spreading, genotype, TEtype, date, context) {
library(fields)
library(dplyr)
library(reshape2)
library(ggplot2)
library(Rmisc)

setwd(input_dir)
b_tir <- read.delim(file, sep="\t", header=T)
b_tir_fam <- read.delim(fam_file, sep="\t", header=T)
class <- read.delim(spreading, sep="\t", header=T)

#setwd("/scratch.global/nosha003/methylation_spreading")
#b_tir <- read.delim("B73_tir_H3K4me3_orientation_metaplot.txt", sep="\t", header=T)
#b_tir_fam <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt", sep="\t", header=T)
#class <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", sep="\t", header=T)

b_tir_fam$length <- abs(b_tir_fam$end - b_tir_fam$start)
b_tir_fam2 <- subset(b_tir_fam, b_tir_fam$length > 1000)
  
b_tir$TEfam <- substr(b_tir$TE, 1, 8)
b_tir <- subset(b_tir, b_tir$TE %in% b_tir_fam2$TE)

b_tir_cg <- as.numeric(paste(b_tir$count))
b_tir["cg"] <- b_tir_cg
b_tir$chromatin <- ifelse(b_tir$cg >=1, 1, 0)
b_tir2 <- left_join(b_tir, class, by="TE")
dist = subset(b_tir2, b_tir2$relative_distance > -2000 & b_tir2$relative_distance <= 3000)

dist_spreading <- subset(dist, dist$mC_state == "mC")
dist_non.spreading <- subset(dist, dist$mC_state == "C")

dist_spreading_group <-  transform(dist_spreading, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_spreading_sum <- do.call(data.frame,aggregate(dist_spreading_group$cg~dist_spreading_group$group, dist_spreading_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x), Avg=mean(x))))
colnames(dist_spreading_sum) <- c("group", "count", "sum", "mean")
dist_spreading_sum["percent_methylated"] <- dist_spreading_sum$sum / dist_spreading_sum$count
dist_spreading_sum["class"] <- "spreading"

dist_non.spreading_group <-  transform(dist_non.spreading, group=cut(as.numeric(sub('[%]', '', relative_distance)), 
    breaks=c(-1050,-950,-850,-750,-650,-550,-450,-350,-250,-150,-50,50,150,250,350,450,550,650,750,850,950,1050,1150,1250,1350,1450,1550,1650,1750,1850,1950,2050), labels=c(-1000,-900,-800,-700,-600,-500,-400,-300,-200,-100,0,100,200,300,400,500,600,700,800,900,1000,1100,1200,1300,1400,1500,1600,1700,1800,1900,2000)))
dist_non.spreading_sum <- do.call(data.frame,aggregate(dist_non.spreading_group$cg~dist_non.spreading_group$group, dist_non.spreading_group, 
        FUN=function(x) c(Count=length(x), Sum=sum(x), Avg=mean(x))))
colnames(dist_non.spreading_sum) <- c("group", "count", "sum", "mean")
dist_non.spreading_sum["percent_methylated"] <- dist_non.spreading_sum$sum / dist_non.spreading_sum$count
dist_non.spreading_sum["class"] <- "non.spreading"

p.1 <- rbind(dist_spreading_sum, dist_non.spreading_sum)

group.colors <- c("#66CDAA", "#836FFF")

## Metaplot of percent bins methylated by spreading v non-spreading
pdf(paste(genotype, TEtype, context, "spreading_metaplot_1kb_Mean.pdf", sep="_"))
plot1 <- ggplot(p.1, aes(x = group, y = mean, group = class, colour = class)) + geom_line(size=1) + theme_classic() + theme(axis.title.x = element_blank(), axis.title.y = element_blank()) + scale_color_manual(values=group.colors)
print(plot1)
dev.off()
}

```

```{r eval=F}

# TIR
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K27me3_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "11March",
  context = "H3K27me3"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K9me2_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "11March",
  context = "H3K9me2"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K56ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "11March",
  context = "H3K56ac"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_H3K9ac_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "11March",
  context = "H3K9ac"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_tir_DNase_orientation_metaplot.txt",
  fam_file= "B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "tir",
  date = "11March",
  context = "DNase"
)



# LTR
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K27me3_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "11March",
  context = "H3K27me3"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K9me2_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "11March",
  context = "H3K9me2"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K56ac_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "11March",
  context = "H3K56ac"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_H3K9ac_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "11March",
  context = "H3K9ac"
)
metaplot_chromatin_spreading(
  input_dir = "/scratch.global/nosha003/methylation_spreading",
  file = "B73_ltr_DNase_orientation_metaplot.txt",
  fam_file= "B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt",
  spreading = "all_low_b_flank_category_cluster_20Dec.txt",
  genotype = "B73",
  TEtype = "ltr",
  date = "11March",
  context = "DNase"
)
```


### Spreading v Non-spreading (Revisions)
- Need to re-create file: all_low_b_flank_category_cluster_20Dec.txt

- DT (1780 spreading elements, 6179 non-spreading elements)
    - elements > 1kb (35 spreading elements, 130 non-spreading elements)
- RL (1185 spreading elements, 2429 non-spreading elements)

#### CG and CHG counts
```{r eval=F}
library(dplyr)
library(ggplot2)
library(reshape2)

setwd("/home/springer/nosha003/methylation_B73v4/")
count <- read.delim("AGPv4_cg_sites_012017.bed", header=T, sep="\t")

setwd("/scratch.global/nosha003/methylation_spreading")
#b_tir_fam <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018_individualTEs.txt", sep="\t", header=T)
#b_ltr_fam <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018_individualTEs.txt", sep="\t", header=T)

setwd("/home/springer/nosha003/methylation_spreading/")
class <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", sep="\t", header=T)
spreading <- subset(class, class$mC_state == "mC")
non.spreading <- subset(class, class$mC_state == "C")

setwd("/home/springer/nosha003/methylation_spreading/")
tir <- read.delim("B73_tir_closestTEtoBin_20Dec2018.txt", header=F, sep="\t")
ltr <- read.delim("B73_ltr_closestTEtoBin_20Dec2018.txt", header=F, sep="\t")
tir$chr <- as.factor(tir$V1)
tir$start <- tir$V2
tir$end <- tir$V3
ltr$chr <- as.factor(ltr$V1)
ltr$start <- ltr$V2
ltr$end <- ltr$V3

tir_bins <- subset(tir, tir$V16 != 0 & abs(tir$V16) < 200)
ltr_bins <- subset(ltr, ltr$V16 != 0 & abs(ltr$V16) < 200)

tir_bins_count <- left_join(tir_bins, count, by=c("chr", "start", "end"))
ltr_bins_count <- left_join(ltr_bins, count, by=c("chr", "start", "end"))
tir_bins_count$TE <- substr(tir_bins_count$V15, 4, 24)
ltr_bins_count$TE <- substr(ltr_bins_count$V15, 4, 24)

tir_bins_count2 <- tir_bins_count %>% group_by(TE) %>% mutate(mean_cg = mean(X.cg.sites), mean_chg = mean(X.chg.sites), mean_chh = mean(X.chh.sites))
ltr_bins_count2 <- ltr_bins_count %>% group_by(TE) %>% mutate(mean_cg = mean(X.cg.sites), mean_chg = mean(X.chg.sites), mean_chh = mean(X.chh.sites))

tir_bins_count_unique <- (unique(tir_bins_count2[,c(23:26)]))
ltr_bins_count_unique<- (unique(ltr_bins_count2[,c(23:26)]))

tir_spreading <- subset(tir_bins_count_unique, tir_bins_count_unique$TE %in% spreading$TE)
tir_nonspreading <- subset(tir_bins_count_unique, tir_bins_count_unique$TE %in% non.spreading$TE)

ltr_spreading <- subset(ltr_bins_count_unique, ltr_bins_count_unique$TE %in% spreading$TE)
ltr_nonspreading <- subset(ltr_bins_count_unique, ltr_bins_count_unique$TE %in% non.spreading$TE)

tir_spreading_melt <- melt(tir_spreading)
tir_nonspreading_melt <- melt(tir_nonspreading)
ltr_spreading_melt <- melt(ltr_spreading)
ltr_nonspreading_melt <- melt(ltr_nonspreading)

tir_spreading_melt["class"] <- "Spreading"
tir_nonspreading_melt["class"] <- "Non.Spreading"
ltr_spreading_melt["class"] <- "Spreading"
ltr_nonspreading_melt["class"] <- "Non.Spreading"

b_tir_class <- rbind(tir_spreading_melt, tir_nonspreading_melt)
b_ltr_class <- rbind(ltr_spreading_melt, ltr_nonspreading_melt)

group.colors <- c("#66CDAA", "#836FFF")

pdf("tir_spreading_counts.pdf")
ggplot(b_tir_class, aes(x=class, y=value, fill=class)) + geom_violin() + facet_grid(. ~ variable) + scale_y_continuous(name = "C Counts") + scale_fill_manual(values=group.colors) + theme_classic() + xlab("")
dev.off()

pdf("ltr_spreading_counts.pdf")
ggplot(b_ltr_class, aes(x=class, y=value, fill=class)) + geom_violin() + facet_grid(. ~ variable) + scale_y_continuous(name = "C Counts") + scale_fill_manual(values=group.colors) + theme_classic() + xlab("")
dev.off()





# counts in TE body
tir_bins <- subset(tir, tir$V16 == 0)
ltr_bins <- subset(ltr, ltr$V16 == 0)

tir_bins_count <- left_join(tir_bins, count, by=c("chr", "start", "end"))
ltr_bins_count <- left_join(ltr_bins, count, by=c("chr", "start", "end"))
tir_bins_count$TE <- substr(tir_bins_count$V15, 4, 24)
ltr_bins_count$TE <- substr(ltr_bins_count$V15, 4, 24)

tir_bins_count2 <- tir_bins_count %>% group_by(TE) %>% mutate(mean_cg = mean(X.cg.sites), mean_chg = mean(X.chg.sites), mean_chh = mean(X.chh.sites))
ltr_bins_count2 <- ltr_bins_count %>% group_by(TE) %>% mutate(mean_cg = mean(X.cg.sites), mean_chg = mean(X.chg.sites), mean_chh = mean(X.chh.sites))

tir_bins_count_unique <- (unique(tir_bins_count2[,c(23:26)]))
ltr_bins_count_unique<- (unique(ltr_bins_count2[,c(23:26)]))

tir_spreading <- subset(tir_bins_count_unique, tir_bins_count_unique$TE %in% spreading$TE)
tir_nonspreading <- subset(tir_bins_count_unique, tir_bins_count_unique$TE %in% non.spreading$TE)

ltr_spreading <- subset(ltr_bins_count_unique, ltr_bins_count_unique$TE %in% spreading$TE)
ltr_nonspreading <- subset(ltr_bins_count_unique, ltr_bins_count_unique$TE %in% non.spreading$TE)

tir_spreading_melt <- melt(tir_spreading)
tir_nonspreading_melt <- melt(tir_nonspreading)
ltr_spreading_melt <- melt(ltr_spreading)
ltr_nonspreading_melt <- melt(ltr_nonspreading)

tir_spreading_melt["class"] <- "Spreading"
tir_nonspreading_melt["class"] <- "Non.Spreading"
ltr_spreading_melt["class"] <- "Spreading"
ltr_nonspreading_melt["class"] <- "Non.Spreading"

b_tir_class <- rbind(tir_spreading_melt, tir_nonspreading_melt)
b_ltr_class <- rbind(ltr_spreading_melt, ltr_nonspreading_melt)

group.colors <- c("#66CDAA", "#836FFF")

pdf("tir_spreading_counts_TEbody.pdf")
ggplot(b_tir_class, aes(x=class, y=value, fill=class)) + geom_violin() + facet_grid(. ~ variable) + scale_y_continuous(name = "C Counts") + scale_fill_manual(values=group.colors) + theme_classic() + xlab("")
dev.off()

pdf("ltr_spreading_counts_TEbody.pdf")
ggplot(b_ltr_class, aes(x=class, y=value, fill=class)) + geom_violin() + facet_grid(. ~ variable) + scale_y_continuous(name = "C Counts") + scale_fill_manual(values=group.colors) + theme_classic() + xlab("")
dev.off()

```


#### Attributes (Revisions)

##### Distance to Gene
- LTR
```{r, eval=F}
library(fields)
library(ggplot2)
setwd("/home/springer/nosha003/methylation_spreading/")
b_ltr <- read.delim("B73_ltr_closestGene_closestTEtoBin_20Dec2018.txt", sep="\t", header=F)
b_ltr$TE <- substr(b_ltr$V15, 4, 24)
b_ltr$fam <- substr(b_ltr$TE, 1, 8)

setwd("/scratch.global/nosha003/methylation_spreading")
class <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", sep="\t", header=T)
spreading <- subset(class, class$mC_state == "mC")
non.spreading <- subset(class, class$mC_state == "C")

b_ltr_class1 <- subset(b_ltr, b_ltr$TE %in% spreading$TE)
b_ltr_class2 <- subset(b_ltr, b_ltr$TE %in% non.spreading$TE)

b_ltr_class1["class"] <- "Spreading"
b_ltr_class2["class"] <- "Non.Spreading"

b_ltr_class <- rbind(b_ltr_class1, b_ltr_class2)

group.colors <- c("#66CDAA", "#836FFF")

library(dplyr)
library(reshape2)
b_ltr_class_bin <- b_ltr_class %>% mutate(category = ifelse(V24 == 0, "A", ifelse(V24 < 1000, "B", ifelse(V24 < 5000, "C", ifelse(V24 > 5000, "D", "NA")))))
b_ltr_class_bin_table <- with(b_ltr_class_bin, table(class, category))
b_ltr_class_bin_table2 <- b_ltr_class_bin_table / rowSums(b_ltr_class_bin_table)
b_ltr_melt <- melt(b_ltr_class_bin_table2)

pdf("ltr_spreading_dist_gene.pdf")
ggplot(b_ltr_melt, aes(x=class, y=value, fill=category)) + geom_bar(colour="black", stat = "identity") + scale_fill_brewer(palette="YlGnBu") + theme_classic()
dev.off()
```

- TIR
```{r, eval=F}
library(fields)
library(ggplot2)
setwd("/home/springer/nosha003/methylation_spreading/")
b_tir <- read.delim("B73_tir_closestGene_closestTEtoBin_20Dec2018.txt", sep="\t", header=F)
b_tir$TE <- substr(b_tir$V15, 4, 24)
b_tir$fam <- substr(b_tir$TE, 1, 8)

setwd("/scratch.global/nosha003/methylation_spreading")
class <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", sep="\t", header=T)
spreading <- subset(class, class$mC_state == "mC")
non.spreading <- subset(class, class$mC_state == "C")

b_tir_class1 <- subset(b_tir, b_tir$TE %in% spreading$TE)
b_tir_class2 <- subset(b_tir, b_tir$TE %in% non.spreading$TE)

b_tir_class1["class"] <- "Spreading"
b_tir_class2["class"] <- "Non.Spreading"

b_tir_class <- rbind(b_tir_class1, b_tir_class2)

group.colors <- c("#66CDAA", "#836FFF")

library(dplyr)
library(reshape2)
b_tir_class_bin <- b_tir_class %>% mutate(category = ifelse(V24 == 0, "A", ifelse(V24 < 1000, "B", ifelse(V24 < 5000, "C", ifelse(V24 > 5000, "D", "NA")))))
b_tir_class_bin_table <- with(b_tir_class_bin, table(class, category))
b_tir_class_bin_table2 <- b_tir_class_bin_table / rowSums(b_tir_class_bin_table)
b_tir_melt <- melt(b_tir_class_bin_table2)

pdf("tir_spreading_dist_gene.pdf")
ggplot(b_tir_melt, aes(x=class, y=value, fill=category)) + geom_bar(colour="black", stat = "identity") + scale_fill_brewer(palette="YlGnBu") + theme_classic()
dev.off()
```


##### Superfamily classification
```{r eval=F}
setwd("/scratch.global/nosha003/methylation_spreading")
class <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", sep="\t", header=T)
class$fam <- substr(class$TE, 1, 8)
class$superfam <- substr(class$fam, 1, 3)
class$order <- substr(class$fam, 1, 2)

class_spreading <- subset(class, class$mC_state == "mC")
class_nonspreading <- subset(class, class$mC_state == "C")
class_spreading$cluster <- "Spreading"
class_nonspreading$cluster <- "Non.Spreading"

class_cluster <- rbind(class_spreading, class_nonspreading)

df <- class_cluster[,c(2,7,8,3)]
colnames(df) <- c("cluster", "fam", "superfam", "order")

library(ggplot2)
library(reshape2)
library(RColorBrewer)

df_ltr <- subset(df, df$order == "RL")
df_tir <- subset(df, df$order == "DT")

df_ltr_count <- with(df_ltr, table(cluster, superfam))
df_ltr_prop <- df_ltr_count/rowSums(df_ltr_count)
df_ltr_melt <- melt(df_ltr_prop)

pdf("ltr_spreading_superfam.pdf")
ggplot(df_ltr_melt, aes(x=cluster, y=value, fill=superfam)) + geom_bar(stat = "identity") + scale_fill_brewer(palette="RdYlBu") + theme_classic()
dev.off()

df_tir_count <- with(df_tir, table(cluster, superfam))
df_tir_prop <- df_tir_count/rowSums(df_tir_count)
df_tir_melt <- melt(df_tir_prop)

pdf("tir_spreading_superfam.pdf")
ggplot(df_tir_melt, aes(x=cluster, y=value, fill=superfam)) + geom_bar(stat = "identity") + scale_fill_brewer(palette="RdYlBu") + theme_classic()
dev.off()
```


##### Copy Number classificiation
- LTR
```{r eval=F}
setwd("/scratch.global/nosha003/methylation_spreading")
class <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", sep="\t", header=T)
class$fam <- substr(class$TE, 1, 8)

class_spreading <- subset(class, class$mC_state == "mC")
class_nonspreading <- subset(class, class$mC_state == "C")
class_spreading$cluster <- "Spreading"
class_nonspreading$cluster <- "Non.Spreading"

cluster <- rbind(class_spreading, class_nonspreading)

setwd("/home/springer/nosha003/methylation_spreading/")
copy <- read.delim("B73_ltr_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
colnames(copy) <- c("count", "fam")

cluster_copy <- merge(cluster, copy, by="fam")

library(ggplot2)
library(reshape2)

cluster_copy_20 <- subset(cluster_copy, cluster_copy$count >= 20)
cluster_copy_bin <- cluster_copy_20 %>% mutate(category = ifelse(count >= 20 & count < 50, "A", ifelse(count >= 50 & count < 100, "B", ifelse(count >= 100 & count < 200, "C", ifelse(count >= 200, "D", "NA")))))
cluster_copy_bin_table <- with(cluster_copy_bin, table(cluster, category))
cluster_copy_bin_table2 <- cluster_copy_bin_table / rowSums(cluster_copy_bin_table)

df_melt <- melt(cluster_copy_bin_table2)

pdf("ltr_spreading_copynumber.pdf")
ggplot(df_melt, aes(x=cluster, y=value, fill=category)) + geom_bar(colour="black", stat = "identity") + scale_fill_brewer(palette="BuGn") + theme_classic()
dev.off()
```

- TIR
```{r eval=F}
setwd("/scratch.global/nosha003/methylation_spreading")
class <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", sep="\t", header=T)
class$fam <- substr(class$TE, 1, 8)

class_spreading <- subset(class, class$mC_state == "mC")
class_nonspreading <- subset(class, class$mC_state == "C")
class_spreading$cluster <- "Spreading"
class_nonspreading$cluster <- "Non.Spreading"

cluster <- rbind(class_spreading, class_nonspreading)

setwd("/home/springer/nosha003/methylation_spreading/")
copy <- read.delim("B73_tir_fam_count20_nonnested_20Dec2018.txt", header=F, sep="\t")
colnames(copy) <- c("count", "fam")

cluster_copy <- merge(cluster, copy, by="fam")

library(ggplot2)
library(reshape2)

cluster_copy_20 <- subset(cluster_copy, cluster_copy$count >= 20)
cluster_copy_bin <- cluster_copy_20 %>% mutate(category = ifelse(count >= 20 & count < 50, "A", ifelse(count >= 50 & count < 100, "B", ifelse(count >= 100 & count < 200, "C", ifelse(count >= 200, "D", "NA")))))
cluster_copy_bin_table <- with(cluster_copy_bin, table(cluster, category))
cluster_copy_bin_table2 <- cluster_copy_bin_table / rowSums(cluster_copy_bin_table)

df_melt <- melt(cluster_copy_bin_table2)

pdf("tir_spreading_copynumber.pdf")
ggplot(df_melt, aes(x=cluster, y=value, fill=category)) + geom_bar(colour="black", stat = "identity") + scale_fill_brewer(palette="BuGn") + theme_classic()
dev.off()
```

##### Length
- LTR
```{r eval=F}
library(fields)
library(ggplot2)
setwd("/home/springer/nosha003/database/B73v4/te/")
length <- read.delim("TE_length_disjoined_20Dec2018.txt", sep="\t", header=T, stringsAsFactors = F)

setwd("/scratch.global/nosha003/methylation_spreading")
class <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", sep="\t", header=T)
class$fam <- substr(class, class$TE, 1, 8)
spreading <- subset(class, class$mC_state == "mC")
non.spreading <- subset(class, class$mC_state == "C")

length$fam <- substr(length$te, 1, 8)

length2 <- as.data.frame(length)
b_tir_class1 <- subset(length2, length2$fam %in% spreading$fam)
b_tir_class2 <- subset(length2, length2$fam %in% non.spreading$fam)

b_tir_class1["class"] <- "Spreading"
b_tir_class2["class"] <- "Non.Spreading"

b_tir_class <- rbind(b_tir_class1, b_tir_class2)
b_tir_class$order <- substr(b_tir_class$fam, 1, 2)
tir <- subset(b_tir_class, b_tir_class$order == "DT")
ltr <- subset(b_tir_class, b_tir_class$order == "RL")

group.colors <- c("#66CDAA", "#836FFF")

pdf("tir_spreading_length.pdf")
ggplot(tir, aes(x=class, y=length, fill=class)) + geom_violin() + scale_y_continuous(name = "TE length") + scale_fill_manual(values=group.colors) + theme_classic() + xlab("") + ylim(0,10000)
dev.off()
pdf("ltr_spreading_length.pdf")
ggplot(ltr, aes(x=class, y=length, fill=class)) + geom_violin() + scale_y_continuous(name = "TE length") + scale_fill_manual(values=group.colors) + theme_classic() + xlab("") + ylim(0,30000)
dev.off()
```


##### Age (LTR only)
```{r eval=F}
library(fields)
library(ggplot2)
setwd("/home/springer/nosha003/database/B73v4/te/")
age <- read.delim("LTR_ages_1June.txt", sep="\t", header=T)
colnames(age) <- c("tename", "tbl")

setwd("/scratch.global/nosha003/methylation_spreading")
class <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", sep="\t", header=T)
class$fam <- substr(class, class$X, 1, 8)
spreading <- subset(class, class$mC_state == "mC")
non.spreading <- subset(class, class$mC_state == "C")

age$fam <- substr(age$tename, 1, 8)
b_tir_class1 <- subset(age, age$fam %in% spreading$fam)
b_tir_class2 <- subset(age, age$fam %in% non.spreading$fam)

b_tir_class1["class"] <- "Spreading"
b_tir_class2["class"] <- "Non.Spreading"

b_tir_class <- rbind(b_tir_class1, b_tir_class2)

group.colors <- c("#66CDAA", "#836FFF")

# violin plot
pdf("ltr_spreading_age.pdf")
ggplot(b_tir_class, aes(x=factor(class), y=abs(NA.2), fill=factor(class))) + geom_violin() + scale_y_continuous(name = "TE age") + scale_fill_manual(values=group.colors) + theme_classic() + xlab("")
dev.off()
```



### small RNA data at TE boundaries
/home/springer/shared/pcrisp/sRNA/24nt_CP5M_B_P_M.tsv

```{r eval=F}
# overlap small RNA data with spreading v non-spreading TE boundaries

# module load R/3.6.0

library(dplyr)
library(reshape2)
library(ggplot2)

setwd("/home/springer/nosha003/methylation_spreading/")
class <- read.delim("all_low_b_flank_category_cluster_20Dec.txt", sep="\t", header=T)

setwd("/home/springer/nosha003/database/B73v4/te")
te <- read.delim("B73.structuralTEv2.2018.12.20.filteredTE.disjoined.noctg.sort.gff3", header=F, sep="\t")
te_coord <- te[3:529113,]
te_coord$TE <- substr(te_coord$V9, 4, 24)
te_coord_class <- left_join(class, te_coord, by="TE")
te_coord_class$chr <- as.numeric(te_coord_class$V1)
te_coord_class$start <- as.numeric(te_coord_class$V4)
te_coord_class$end <- as.numeric(te_coord_class$V5)
te <- unique(te_coord_class[,c(1,6,7,10:11,16:18)])

spreading <- subset(te, te$mC_state == "mC")
non.spreading <- subset(te, te$mC_state == "C")

setwd("/home/springer/shared/pcrisp/sRNA")
srna <- read.delim("24nt_CP5M_B_P_M.tsv", sep="\t", header=T)
srna$end <- as.numeric(srna$start + srna$bin_size)

library(tidygenomics)
spreading_srna <- genome_intersect(spreading, srna, by=c("chr", "start", "end"), mode="both")
non.spreading_srna <- genome_intersect(non.spreading, srna, by=c("chr", "start", "end"), mode="both")

te_start <- te
te_end <- te
te_start$end <- te_start$start + 1
te_end$start <- te_end$end - 1
spreading_start <- subset(te_start, te_start$mC_state == "mC")
non.spreading_start <- subset(te_start, te_start$mC_state == "C")
spreading_end <- subset(te_end, te_end$mC_state == "mC")
non.spreading_end <- subset(te_end, te_end$mC_state == "C")

spreading_start_srna <- genome_intersect(spreading_start, srna, by=c("chr", "start", "end"), mode="both")
# 47 --> B (157.6609), M (129.1693), P (75.68502)
non.spreading_start_srna <- genome_intersect(non.spreading_start, srna, by=c("chr", "start", "end"), mode="both")
# 110 --> B (325.0208), M (199.4157), P (169.5559)
spreading_end_srna <- genome_intersect(spreading_end, srna, by=c("chr", "start", "end"), mode="both")
# 40 --> B (104.7977), M (69.98191), P (54.64573)
non.spreading_end_srna <- genome_intersect(non.spreading_end, srna, by=c("chr", "start", "end"), mode="both")
# 98 --> B (235.2604), M (166.6997), P (186.1849)

####### non.spreading seems to have greater 24nt sRNA at edges of TEs than spreading
## spreading: B73 = 262.4586, M = 199.1512, P = 130.3307
## non.spreading: B73 = 560.2812, M = 366.1154, P = 355.7408

# what about proportion of TEs that have small RNAs (> 0, > 1 ??)
spreading_srna_edges <- full_join(spreading_start_srna, spreading_end_srna, by="TE")
spreading_srna_edges[is.na(spreading_srna_edges)] <- 0
# 78 
non.spreading_srna_edges <- full_join(non.spreading_start_srna, non.spreading_end_srna, by="TE")
# 183
non.spreading_srna_edges[is.na(non.spreading_srna_edges)] <- 0

spreading_srna_edges2 <- spreading_srna_edges %>% mutate(B_start = ifelse(Zm_B_L_sum_24.x > 1, 1, 0), M_start = ifelse(Zm_Mo17_L_sum_24.x > 1, 1, 0), P_start = ifelse(Zm_P_L_sum_24.x > 1, 1, 0)) %>% mutate(B_end = ifelse(Zm_B_L_sum_24.y > 1, 1, 0), M_end = ifelse(Zm_Mo17_L_sum_24.y > 1, 1, 0), P_end = ifelse(Zm_P_L_sum_24.y > 1, 1, 0))
non.spreading_srna_edges2 <- non.spreading_srna_edges %>% mutate(B_start = ifelse(Zm_B_L_sum_24.x > 1, 1, 0), M_start = ifelse(Zm_Mo17_L_sum_24.x > 1, 1, 0), P_start = ifelse(Zm_P_L_sum_24.x > 1, 1, 0)) %>% mutate(B_end = ifelse(Zm_B_L_sum_24.y > 1, 1, 0), M_end = ifelse(Zm_Mo17_L_sum_24.y > 1, 1, 0), P_end = ifelse(Zm_P_L_sum_24.y > 1, 1, 0))

table(spreading_srna_edges2[,c(26,29)])
#        B_end
# B_start  0  1
#       0 36 14
#       1 23  5
### 42/78=0.54 
table(non.spreading_srna_edges2[,c(26,29)])
#        B_end
# B_start  0  1
#       0 85 38
#       1 50 10
### 98/183=0.54
######### No difference in proportion of spreading v non-spreading TEs that have small RNA count of > 1 at their edges

# heatmap
spreading_heatmap <- as.matrix(spreading_srna_edges2[,9:11])
non.spreading_heatmap <- as.matrix(non.spreading_srna_edges2[,9:11])

library(pheatmap)
library(viridis)
setwd("/scratch.global/nosha003")
pdf("spreading_sRNA_heatmap.pdf")
pheatmap(spreading_heatmap)
dev.off()
pdf("non.spreading_sRNA_heatmap.pdf")
pheatmap(non.spreading_heatmap)
dev.off()

quantile_breaks <- function(xs, n = 10) {
  breaks <- quantile(xs, probs = seq(0, 1, length.out = n))
  breaks[!duplicated(breaks)]
}
spreading_breaks <- quantile_breaks(spreading_heatmap)
non.spreading_breaks <- quantile_breaks(non.spreading_heatmap)

pdf("spreading_sRNA_heatmap2.pdf")
pheatmap(spreading_heatmap, breaks=spreading_breaks, color=inferno(length(spreading_breaks) - 1))
dev.off()
pdf("non.spreading_sRNA_heatmap2.pdf")
pheatmap(non.spreading_heatmap, breaks=non.spreading_breaks, color=inferno(length(non.spreading_breaks) - 1))
dev.off()

# density plot
library(reshape2)
spreading_melt <- melt(spreading_heatmap)
non.spreading_melt <- melt(non.spreading_heatmap)

setwd("/scratch.global/nosha003")
pdf("spreading_sRNA_density.pdf")
ggplot(spreading_melt) + geom_density(aes(x=value, color=Var2))
dev.off()
pdf("non.spreading_sRNA_density.pdf")
ggplot(non.spreading_melt) + geom_density(aes(x=value, color=Var2))
dev.off()

spreading_melt$class <- "spreading"
non.spreading_melt$class <- "non.spreading"
melt <- rbind(spreading_melt, non.spreading_melt)
pdf("sRNA_density.pdf")
ggplot(melt) + geom_density(aes(x=value, color=class))
dev.off()
pdf("sRNA_density2.pdf")
ggplot(melt) + geom_density(aes(x=value, color=class)) + xlim(0,10)
dev.off()
pdf("sRNA_histogram.pdf")
ggplot(melt) + geom_histogram(aes(x=value, color=class)) + xlim(0,10)
dev.off()
pdf("sRNA_violin.pdf")
ggplot(melt) + geom_violin(aes(y=value, x=Var2, color=class))
dev.off()
pdf("sRNA_violin2.pdf")
ggplot(melt) + geom_violin(aes(y=value, x=Var2, color=class)) + ylim(0,10)
dev.off()




```











